{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**\"Effects of cycloheximide on the interpretation of ribosome profiling experiments in *Schizosaccharomyces pombe*\"** Duncan & Mata, 2017 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**From Raw data to BAM files**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Downloading .fastq files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-07T20:10:28.248628Z",
     "start_time": "2018-03-07T15:39:44.794139Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "downloading reads to  /projects_eg/projects/simone/Duncan_Mata/fastq_files\n",
      "Read 26425976 spots for ERR1994959\n",
      "Written 26425976 spots for ERR1994959\n",
      "Read 36240950 spots for ERR1994960\n",
      "Written 36240950 spots for ERR1994960\n",
      "Read 63059496 spots for ERR1994961\n",
      "Written 63059496 spots for ERR1994961\n",
      "Read 26742065 spots for ERR1994962\n",
      "Written 26742065 spots for ERR1994962\n",
      "Read 16170457 spots for ERR1994963\n",
      "Written 16170457 spots for ERR1994963\n",
      "Read 23972947 spots for ERR1994964\n",
      "Written 23972947 spots for ERR1994964\n",
      "Read 40531552 spots for ERR1994965\n",
      "Written 40531552 spots for ERR1994965\n",
      "Read 32968604 spots for ERR1994966\n",
      "Written 32968604 spots for ERR1994966\n",
      "Read 21546857 spots for ERR1994967\n",
      "Written 21546857 spots for ERR1994967\n",
      "Read 33734733 spots for ERR1994968\n",
      "Written 33734733 spots for ERR1994968\n",
      "Read 83235717 spots for ERR1994969\n",
      "Written 83235717 spots for ERR1994969\n",
      "Read 31175890 spots for ERR1994970\n",
      "Written 31175890 spots for ERR1994970\n",
      "Read 18195992 spots for ERR1994971\n",
      "Written 18195992 spots for ERR1994971\n",
      "Read 23231253 spots for ERR1994972\n",
      "Written 23231253 spots for ERR1994972\n",
      "Read 40848270 spots for ERR1994973\n",
      "Written 40848270 spots for ERR1994973\n",
      "Read 31871091 spots for ERR1994974\n",
      "Written 31871091 spots for ERR1994974\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/fastq_files’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "fastqdump=\"/home/simone/Downloads/sratoolkit.2.8.2-1-centos_linux64/bin/fastq-dump\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/fastq_files\"\n",
    "mkdir $output_directory\n",
    "\n",
    "accession_base=\"ERR19949\"\n",
    "\n",
    "echo 'downloading reads to  '$output_directory\n",
    "END=74\n",
    "for ((i=59;i<=END;i++))\n",
    "do\n",
    "$fastqdump --split-files -I -O $output_directory $accession_base$i\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ERR1994959\twt.N.CHX.mrna.1\n",
    "\n",
    "ERR1994960\twt.N.CHX.mrna.3\n",
    "\n",
    "ERR1994961\twt.N.CHX.ribo.1\n",
    "\n",
    "ERR1994962\twt.N.CHX.ribo.3\n",
    "\n",
    "ERR1994963\twt.N.noCHX.mrna.1\n",
    "\n",
    "ERR1994964\twt.N.noCHX.mrna.3\n",
    "\n",
    "ERR1994965\twt.N.noCHX.ribo.1\n",
    "\n",
    "ERR1994966\twt.N.noCHX.ribo.3\n",
    "\n",
    "ERR1994967\twt.noN.CHX.mrna.1\n",
    "\n",
    "ERR1994968\twt.noN.CHX.mrna.3\n",
    "\n",
    "ERR1994969\twt.noN.CHX.ribo.1\n",
    "\n",
    "ERR1994970\twt.noN.CHX.ribo.3\n",
    "\n",
    "ERR1994971\twt.noN.noCHX.mrna.1\n",
    "\n",
    "ERR1994972\twt.noN.noCHX.mrna.3\n",
    "\n",
    "ERR1994973\twt.noN.noCHX.ribo.1\n",
    "\n",
    "ERR1994974\twt.noN.noCHX.ribo.3\n",
    "\n",
    "\n",
    "_____________________________________________________________________________________________\n",
    "\n",
    "\t\t\t\t\t\tCHX\n",
    "\n",
    "\n",
    "Ribo-seq\n",
    "\n",
    "\n",
    "ERR1994961\twt.N.CHX.ribo.1\n",
    "ERR1994962\twt.N.CHX.ribo.3\n",
    "\n",
    "ERR1994969\twt.noN.CHX.ribo.1\n",
    "ERR1994970\twt.noN.CHX.ribo.3\n",
    "\n",
    "\n",
    "---------------------------------------------------------------------------------------------\n",
    "\n",
    "\n",
    "RNA-seq\n",
    "\n",
    "\n",
    "ERR1994959\twt.N.CHX.mrna.1\n",
    "ERR1994960\twt.N.CHX.mrna.3\n",
    "\n",
    "ERR1994967\twt.noN.CHX.mrna.1\n",
    "ERR1994968\twt.noN.CHX.mrna.3\n",
    "\n",
    "\n",
    "_____________________________________________________________________________________________\n",
    "\n",
    "\t\t\t\t\t\tno CHX\n",
    "\n",
    "\n",
    "Ribo-seq\n",
    "\n",
    "\n",
    "ERR1994965\twt.N.noCHX.ribo.1\n",
    "ERR1994966\twt.N.noCHX.ribo.3\n",
    "\n",
    "ERR1994973\twt.noN.noCHX.ribo.1\n",
    "ERR1994974\twt.noN.noCHX.ribo.3\n",
    "\n",
    "\n",
    "---------------------------------------------------------------------------------------------\n",
    "\n",
    "\n",
    "RNA-seq\n",
    "\n",
    "\n",
    "ERR1994963\twt.N.noCHX.mrna.1\n",
    "ERR1994964\twt.N.noCHX.mrna.3\n",
    "\n",
    "ERR1994971\twt.noN.noCHX.mrna.1\n",
    "ERR1994972\twt.noN.noCHX.mrna.3\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Getting FastQC quality reports of raw data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-08T13:04:34.093283Z",
     "start_time": "2018-03-08T12:19:38.460312Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get fastqc reports to  /projects_eg/projects/simone/Duncan_Mata/fastqc_reports/\n",
      "Analysis complete for ERR1994959_1.fastq\n",
      "Analysis complete for ERR1994960_1.fastq\n",
      "Analysis complete for ERR1994961_1.fastq\n",
      "Analysis complete for ERR1994962_1.fastq\n",
      "Analysis complete for ERR1994963_1.fastq\n",
      "Analysis complete for ERR1994964_1.fastq\n",
      "Analysis complete for ERR1994965_1.fastq\n",
      "Analysis complete for ERR1994966_1.fastq\n",
      "Analysis complete for ERR1994967_1.fastq\n",
      "Analysis complete for ERR1994968_1.fastq\n",
      "Analysis complete for ERR1994969_1.fastq\n",
      "Analysis complete for ERR1994970_1.fastq\n",
      "Analysis complete for ERR1994971_1.fastq\n",
      "Analysis complete for ERR1994972_1.fastq\n",
      "Analysis complete for ERR1994973_1.fastq\n",
      "Analysis complete for ERR1994974_1.fastq\n",
      "Searching 32 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of ERR1994959_1.fastq\n",
      "Approx 5% complete for ERR1994959_1.fastq\n",
      "Approx 10% complete for ERR1994959_1.fastq\n",
      "Approx 15% complete for ERR1994959_1.fastq\n",
      "Approx 20% complete for ERR1994959_1.fastq\n",
      "Approx 25% complete for ERR1994959_1.fastq\n",
      "Approx 30% complete for ERR1994959_1.fastq\n",
      "Approx 35% complete for ERR1994959_1.fastq\n",
      "Approx 40% complete for ERR1994959_1.fastq\n",
      "Approx 45% complete for ERR1994959_1.fastq\n",
      "Approx 50% complete for ERR1994959_1.fastq\n",
      "Approx 55% complete for ERR1994959_1.fastq\n",
      "Approx 60% complete for ERR1994959_1.fastq\n",
      "Approx 65% complete for ERR1994959_1.fastq\n",
      "Approx 70% complete for ERR1994959_1.fastq\n",
      "Approx 75% complete for ERR1994959_1.fastq\n",
      "Approx 80% complete for ERR1994959_1.fastq\n",
      "Approx 85% complete for ERR1994959_1.fastq\n",
      "Approx 90% complete for ERR1994959_1.fastq\n",
      "Approx 95% complete for ERR1994959_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994960_1.fastq\n",
      "Approx 5% complete for ERR1994960_1.fastq\n",
      "Approx 10% complete for ERR1994960_1.fastq\n",
      "Approx 15% complete for ERR1994960_1.fastq\n",
      "Approx 20% complete for ERR1994960_1.fastq\n",
      "Approx 25% complete for ERR1994960_1.fastq\n",
      "Approx 30% complete for ERR1994960_1.fastq\n",
      "Approx 35% complete for ERR1994960_1.fastq\n",
      "Approx 40% complete for ERR1994960_1.fastq\n",
      "Approx 45% complete for ERR1994960_1.fastq\n",
      "Approx 50% complete for ERR1994960_1.fastq\n",
      "Approx 55% complete for ERR1994960_1.fastq\n",
      "Approx 60% complete for ERR1994960_1.fastq\n",
      "Approx 65% complete for ERR1994960_1.fastq\n",
      "Approx 70% complete for ERR1994960_1.fastq\n",
      "Approx 75% complete for ERR1994960_1.fastq\n",
      "Approx 80% complete for ERR1994960_1.fastq\n",
      "Approx 85% complete for ERR1994960_1.fastq\n",
      "Approx 90% complete for ERR1994960_1.fastq\n",
      "Approx 95% complete for ERR1994960_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994961_1.fastq\n",
      "Approx 5% complete for ERR1994961_1.fastq\n",
      "Approx 10% complete for ERR1994961_1.fastq\n",
      "Approx 15% complete for ERR1994961_1.fastq\n",
      "Approx 20% complete for ERR1994961_1.fastq\n",
      "Approx 25% complete for ERR1994961_1.fastq\n",
      "Approx 30% complete for ERR1994961_1.fastq\n",
      "Approx 35% complete for ERR1994961_1.fastq\n",
      "Approx 40% complete for ERR1994961_1.fastq\n",
      "Approx 45% complete for ERR1994961_1.fastq\n",
      "Approx 50% complete for ERR1994961_1.fastq\n",
      "Approx 55% complete for ERR1994961_1.fastq\n",
      "Approx 60% complete for ERR1994961_1.fastq\n",
      "Approx 65% complete for ERR1994961_1.fastq\n",
      "Approx 70% complete for ERR1994961_1.fastq\n",
      "Approx 75% complete for ERR1994961_1.fastq\n",
      "Approx 80% complete for ERR1994961_1.fastq\n",
      "Approx 85% complete for ERR1994961_1.fastq\n",
      "Approx 90% complete for ERR1994961_1.fastq\n",
      "Approx 95% complete for ERR1994961_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994962_1.fastq\n",
      "Approx 5% complete for ERR1994962_1.fastq\n",
      "Approx 10% complete for ERR1994962_1.fastq\n",
      "Approx 15% complete for ERR1994962_1.fastq\n",
      "Approx 20% complete for ERR1994962_1.fastq\n",
      "Approx 25% complete for ERR1994962_1.fastq\n",
      "Approx 30% complete for ERR1994962_1.fastq\n",
      "Approx 35% complete for ERR1994962_1.fastq\n",
      "Approx 40% complete for ERR1994962_1.fastq\n",
      "Approx 45% complete for ERR1994962_1.fastq\n",
      "Approx 50% complete for ERR1994962_1.fastq\n",
      "Approx 55% complete for ERR1994962_1.fastq\n",
      "Approx 60% complete for ERR1994962_1.fastq\n",
      "Approx 65% complete for ERR1994962_1.fastq\n",
      "Approx 70% complete for ERR1994962_1.fastq\n",
      "Approx 75% complete for ERR1994962_1.fastq\n",
      "Approx 80% complete for ERR1994962_1.fastq\n",
      "Approx 85% complete for ERR1994962_1.fastq\n",
      "Approx 90% complete for ERR1994962_1.fastq\n",
      "Approx 95% complete for ERR1994962_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994963_1.fastq\n",
      "Approx 5% complete for ERR1994963_1.fastq\n",
      "Approx 10% complete for ERR1994963_1.fastq\n",
      "Approx 15% complete for ERR1994963_1.fastq\n",
      "Approx 20% complete for ERR1994963_1.fastq\n",
      "Approx 25% complete for ERR1994963_1.fastq\n",
      "Approx 30% complete for ERR1994963_1.fastq\n",
      "Approx 35% complete for ERR1994963_1.fastq\n",
      "Approx 40% complete for ERR1994963_1.fastq\n",
      "Approx 45% complete for ERR1994963_1.fastq\n",
      "Approx 50% complete for ERR1994963_1.fastq\n",
      "Approx 55% complete for ERR1994963_1.fastq\n",
      "Approx 60% complete for ERR1994963_1.fastq\n",
      "Approx 65% complete for ERR1994963_1.fastq\n",
      "Approx 70% complete for ERR1994963_1.fastq\n",
      "Approx 75% complete for ERR1994963_1.fastq\n",
      "Approx 80% complete for ERR1994963_1.fastq\n",
      "Approx 85% complete for ERR1994963_1.fastq\n",
      "Approx 90% complete for ERR1994963_1.fastq\n",
      "Approx 95% complete for ERR1994963_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994964_1.fastq\n",
      "Approx 5% complete for ERR1994964_1.fastq\n",
      "Approx 10% complete for ERR1994964_1.fastq\n",
      "Approx 15% complete for ERR1994964_1.fastq\n",
      "Approx 20% complete for ERR1994964_1.fastq\n",
      "Approx 25% complete for ERR1994964_1.fastq\n",
      "Approx 30% complete for ERR1994964_1.fastq\n",
      "Approx 35% complete for ERR1994964_1.fastq\n",
      "Approx 40% complete for ERR1994964_1.fastq\n",
      "Approx 45% complete for ERR1994964_1.fastq\n",
      "Approx 50% complete for ERR1994964_1.fastq\n",
      "Approx 55% complete for ERR1994964_1.fastq\n",
      "Approx 60% complete for ERR1994964_1.fastq\n",
      "Approx 65% complete for ERR1994964_1.fastq\n",
      "Approx 70% complete for ERR1994964_1.fastq\n",
      "Approx 75% complete for ERR1994964_1.fastq\n",
      "Approx 80% complete for ERR1994964_1.fastq\n",
      "Approx 85% complete for ERR1994964_1.fastq\n",
      "Approx 90% complete for ERR1994964_1.fastq\n",
      "Approx 95% complete for ERR1994964_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994965_1.fastq\n",
      "Approx 5% complete for ERR1994965_1.fastq\n",
      "Approx 10% complete for ERR1994965_1.fastq\n",
      "Approx 15% complete for ERR1994965_1.fastq\n",
      "Approx 20% complete for ERR1994965_1.fastq\n",
      "Approx 25% complete for ERR1994965_1.fastq\n",
      "Approx 30% complete for ERR1994965_1.fastq\n",
      "Approx 35% complete for ERR1994965_1.fastq\n",
      "Approx 40% complete for ERR1994965_1.fastq\n",
      "Approx 45% complete for ERR1994965_1.fastq\n",
      "Approx 50% complete for ERR1994965_1.fastq\n",
      "Approx 55% complete for ERR1994965_1.fastq\n",
      "Approx 60% complete for ERR1994965_1.fastq\n",
      "Approx 65% complete for ERR1994965_1.fastq\n",
      "Approx 70% complete for ERR1994965_1.fastq\n",
      "Approx 75% complete for ERR1994965_1.fastq\n",
      "Approx 80% complete for ERR1994965_1.fastq\n",
      "Approx 85% complete for ERR1994965_1.fastq\n",
      "Approx 90% complete for ERR1994965_1.fastq\n",
      "Approx 95% complete for ERR1994965_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994966_1.fastq\n",
      "Approx 5% complete for ERR1994966_1.fastq\n",
      "Approx 10% complete for ERR1994966_1.fastq\n",
      "Approx 15% complete for ERR1994966_1.fastq\n",
      "Approx 20% complete for ERR1994966_1.fastq\n",
      "Approx 25% complete for ERR1994966_1.fastq\n",
      "Approx 30% complete for ERR1994966_1.fastq\n",
      "Approx 35% complete for ERR1994966_1.fastq\n",
      "Approx 40% complete for ERR1994966_1.fastq\n",
      "Approx 45% complete for ERR1994966_1.fastq\n",
      "Approx 50% complete for ERR1994966_1.fastq\n",
      "Approx 55% complete for ERR1994966_1.fastq\n",
      "Approx 60% complete for ERR1994966_1.fastq\n",
      "Approx 65% complete for ERR1994966_1.fastq\n",
      "Approx 70% complete for ERR1994966_1.fastq\n",
      "Approx 75% complete for ERR1994966_1.fastq\n",
      "Approx 80% complete for ERR1994966_1.fastq\n",
      "Approx 85% complete for ERR1994966_1.fastq\n",
      "Approx 90% complete for ERR1994966_1.fastq\n",
      "Approx 95% complete for ERR1994966_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994967_1.fastq\n",
      "Approx 5% complete for ERR1994967_1.fastq\n",
      "Approx 10% complete for ERR1994967_1.fastq\n",
      "Approx 15% complete for ERR1994967_1.fastq\n",
      "Approx 20% complete for ERR1994967_1.fastq\n",
      "Approx 25% complete for ERR1994967_1.fastq\n",
      "Approx 30% complete for ERR1994967_1.fastq\n",
      "Approx 35% complete for ERR1994967_1.fastq\n",
      "Approx 40% complete for ERR1994967_1.fastq\n",
      "Approx 45% complete for ERR1994967_1.fastq\n",
      "Approx 50% complete for ERR1994967_1.fastq\n",
      "Approx 55% complete for ERR1994967_1.fastq\n",
      "Approx 60% complete for ERR1994967_1.fastq\n",
      "Approx 65% complete for ERR1994967_1.fastq\n",
      "Approx 70% complete for ERR1994967_1.fastq\n",
      "Approx 75% complete for ERR1994967_1.fastq\n",
      "Approx 80% complete for ERR1994967_1.fastq\n",
      "Approx 85% complete for ERR1994967_1.fastq\n",
      "Approx 90% complete for ERR1994967_1.fastq\n",
      "Approx 95% complete for ERR1994967_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994968_1.fastq\n",
      "Approx 5% complete for ERR1994968_1.fastq\n",
      "Approx 10% complete for ERR1994968_1.fastq\n",
      "Approx 15% complete for ERR1994968_1.fastq\n",
      "Approx 20% complete for ERR1994968_1.fastq\n",
      "Approx 25% complete for ERR1994968_1.fastq\n",
      "Approx 30% complete for ERR1994968_1.fastq\n",
      "Approx 35% complete for ERR1994968_1.fastq\n",
      "Approx 40% complete for ERR1994968_1.fastq\n",
      "Approx 45% complete for ERR1994968_1.fastq\n",
      "Approx 50% complete for ERR1994968_1.fastq\n",
      "Approx 55% complete for ERR1994968_1.fastq\n",
      "Approx 60% complete for ERR1994968_1.fastq\n",
      "Approx 65% complete for ERR1994968_1.fastq\n",
      "Approx 70% complete for ERR1994968_1.fastq\n",
      "Approx 75% complete for ERR1994968_1.fastq\n",
      "Approx 80% complete for ERR1994968_1.fastq\n",
      "Approx 85% complete for ERR1994968_1.fastq\n",
      "Approx 90% complete for ERR1994968_1.fastq\n",
      "Approx 95% complete for ERR1994968_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994969_1.fastq\n",
      "Approx 5% complete for ERR1994969_1.fastq\n",
      "Approx 10% complete for ERR1994969_1.fastq\n",
      "Approx 15% complete for ERR1994969_1.fastq\n",
      "Approx 20% complete for ERR1994969_1.fastq\n",
      "Approx 25% complete for ERR1994969_1.fastq\n",
      "Approx 30% complete for ERR1994969_1.fastq\n",
      "Approx 35% complete for ERR1994969_1.fastq\n",
      "Approx 40% complete for ERR1994969_1.fastq\n",
      "Approx 45% complete for ERR1994969_1.fastq\n",
      "Approx 50% complete for ERR1994969_1.fastq\n",
      "Approx 55% complete for ERR1994969_1.fastq\n",
      "Approx 60% complete for ERR1994969_1.fastq\n",
      "Approx 65% complete for ERR1994969_1.fastq\n",
      "Approx 70% complete for ERR1994969_1.fastq\n",
      "Approx 75% complete for ERR1994969_1.fastq\n",
      "Approx 80% complete for ERR1994969_1.fastq\n",
      "Approx 85% complete for ERR1994969_1.fastq\n",
      "Approx 90% complete for ERR1994969_1.fastq\n",
      "Approx 95% complete for ERR1994969_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994970_1.fastq\n",
      "Approx 5% complete for ERR1994970_1.fastq\n",
      "Approx 10% complete for ERR1994970_1.fastq\n",
      "Approx 15% complete for ERR1994970_1.fastq\n",
      "Approx 20% complete for ERR1994970_1.fastq\n",
      "Approx 25% complete for ERR1994970_1.fastq\n",
      "Approx 30% complete for ERR1994970_1.fastq\n",
      "Approx 35% complete for ERR1994970_1.fastq\n",
      "Approx 40% complete for ERR1994970_1.fastq\n",
      "Approx 45% complete for ERR1994970_1.fastq\n",
      "Approx 50% complete for ERR1994970_1.fastq\n",
      "Approx 55% complete for ERR1994970_1.fastq\n",
      "Approx 60% complete for ERR1994970_1.fastq\n",
      "Approx 65% complete for ERR1994970_1.fastq\n",
      "Approx 70% complete for ERR1994970_1.fastq\n",
      "Approx 75% complete for ERR1994970_1.fastq\n",
      "Approx 80% complete for ERR1994970_1.fastq\n",
      "Approx 85% complete for ERR1994970_1.fastq\n",
      "Approx 90% complete for ERR1994970_1.fastq\n",
      "Approx 95% complete for ERR1994970_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994971_1.fastq\n",
      "Approx 5% complete for ERR1994971_1.fastq\n",
      "Approx 10% complete for ERR1994971_1.fastq\n",
      "Approx 15% complete for ERR1994971_1.fastq\n",
      "Approx 20% complete for ERR1994971_1.fastq\n",
      "Approx 25% complete for ERR1994971_1.fastq\n",
      "Approx 30% complete for ERR1994971_1.fastq\n",
      "Approx 35% complete for ERR1994971_1.fastq\n",
      "Approx 40% complete for ERR1994971_1.fastq\n",
      "Approx 45% complete for ERR1994971_1.fastq\n",
      "Approx 50% complete for ERR1994971_1.fastq\n",
      "Approx 55% complete for ERR1994971_1.fastq\n",
      "Approx 60% complete for ERR1994971_1.fastq\n",
      "Approx 65% complete for ERR1994971_1.fastq\n",
      "Approx 70% complete for ERR1994971_1.fastq\n",
      "Approx 75% complete for ERR1994971_1.fastq\n",
      "Approx 80% complete for ERR1994971_1.fastq\n",
      "Approx 85% complete for ERR1994971_1.fastq\n",
      "Approx 90% complete for ERR1994971_1.fastq\n",
      "Approx 95% complete for ERR1994971_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994972_1.fastq\n",
      "Approx 5% complete for ERR1994972_1.fastq\n",
      "Approx 10% complete for ERR1994972_1.fastq\n",
      "Approx 15% complete for ERR1994972_1.fastq\n",
      "Approx 20% complete for ERR1994972_1.fastq\n",
      "Approx 25% complete for ERR1994972_1.fastq\n",
      "Approx 30% complete for ERR1994972_1.fastq\n",
      "Approx 35% complete for ERR1994972_1.fastq\n",
      "Approx 40% complete for ERR1994972_1.fastq\n",
      "Approx 45% complete for ERR1994972_1.fastq\n",
      "Approx 50% complete for ERR1994972_1.fastq\n",
      "Approx 55% complete for ERR1994972_1.fastq\n",
      "Approx 60% complete for ERR1994972_1.fastq\n",
      "Approx 65% complete for ERR1994972_1.fastq\n",
      "Approx 70% complete for ERR1994972_1.fastq\n",
      "Approx 75% complete for ERR1994972_1.fastq\n",
      "Approx 80% complete for ERR1994972_1.fastq\n",
      "Approx 85% complete for ERR1994972_1.fastq\n",
      "Approx 90% complete for ERR1994972_1.fastq\n",
      "Approx 95% complete for ERR1994972_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994973_1.fastq\n",
      "Approx 5% complete for ERR1994973_1.fastq\n",
      "Approx 10% complete for ERR1994973_1.fastq\n",
      "Approx 15% complete for ERR1994973_1.fastq\n",
      "Approx 20% complete for ERR1994973_1.fastq\n",
      "Approx 25% complete for ERR1994973_1.fastq\n",
      "Approx 30% complete for ERR1994973_1.fastq\n",
      "Approx 35% complete for ERR1994973_1.fastq\n",
      "Approx 40% complete for ERR1994973_1.fastq\n",
      "Approx 45% complete for ERR1994973_1.fastq\n",
      "Approx 50% complete for ERR1994973_1.fastq\n",
      "Approx 55% complete for ERR1994973_1.fastq\n",
      "Approx 60% complete for ERR1994973_1.fastq\n",
      "Approx 65% complete for ERR1994973_1.fastq\n",
      "Approx 70% complete for ERR1994973_1.fastq\n",
      "Approx 75% complete for ERR1994973_1.fastq\n",
      "Approx 80% complete for ERR1994973_1.fastq\n",
      "Approx 85% complete for ERR1994973_1.fastq\n",
      "Approx 90% complete for ERR1994973_1.fastq\n",
      "Approx 95% complete for ERR1994973_1.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994974_1.fastq\n",
      "Approx 5% complete for ERR1994974_1.fastq\n",
      "Approx 10% complete for ERR1994974_1.fastq\n",
      "Approx 15% complete for ERR1994974_1.fastq\n",
      "Approx 20% complete for ERR1994974_1.fastq\n",
      "Approx 25% complete for ERR1994974_1.fastq\n",
      "Approx 30% complete for ERR1994974_1.fastq\n",
      "Approx 35% complete for ERR1994974_1.fastq\n",
      "Approx 40% complete for ERR1994974_1.fastq\n",
      "Approx 45% complete for ERR1994974_1.fastq\n",
      "Approx 50% complete for ERR1994974_1.fastq\n",
      "Approx 55% complete for ERR1994974_1.fastq\n",
      "Approx 60% complete for ERR1994974_1.fastq\n",
      "Approx 65% complete for ERR1994974_1.fastq\n",
      "Approx 70% complete for ERR1994974_1.fastq\n",
      "Approx 75% complete for ERR1994974_1.fastq\n",
      "Approx 80% complete for ERR1994974_1.fastq\n",
      "Approx 85% complete for ERR1994974_1.fastq\n",
      "Approx 90% complete for ERR1994974_1.fastq\n",
      "Approx 95% complete for ERR1994974_1.fastq\n",
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 16 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "addition=\"_1\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/fastqc_reports/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "accession_base=\"ERR19949\"\n",
    "echo 'Get fastqc reports to  '$output_directory\n",
    "END=74\n",
    "for ((i=59;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base$i$addition$extention\n",
    "done\n",
    "\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/fastqc_reports/\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Removing adapters using cutadapt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Riboseq data\n",
    "\n",
    "\n",
    "**Reads structure**:     RRRR(NNNN...NNNN)RRRR-adaptor-\n",
    "\n",
    "**R** represents random nucleotides. Random nucleotides serve as unique molecular identifiers (UMIs), **N** corresponds to the sequence of the RNA protected fragment, and the **adaptor** sequence is TGGAATTCTCGGGTGCCAAGG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First step: Remove initial N and adaptor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-08T19:15:03.092900Z",
     "start_time": "2018-03-08T15:24:54.360132Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994961_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994961_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 2506.22 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              63,059,496\n",
      "Reads with adapters:                62,498,503 (99.1%)\n",
      "Reads written (passing filters):    63,059,496 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,152,974,800 bp\n",
      "Total written (filtered):  2,371,900,197 bp (75.2%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 62498503 times.\n",
      "55453 times, it overlapped the 5' end of a read\n",
      "62443050 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t22889\t985304.6\t0\t22889\n",
      "4\t3715\t246326.2\t0\t3715\n",
      "5\t362\t61581.5\t0\t362\n",
      "6\t48\t15395.4\t0\t48\n",
      "7\t62\t3848.8\t0\t62\n",
      "8\t12\t962.2\t0\t12\n",
      "9\t38\t240.6\t0\t11 27\n",
      "10\t20\t60.1\t1\t1 19\n",
      "11\t262\t15.0\t1\t112 150\n",
      "12\t26970\t3.8\t1\t2051 24919\n",
      "13\t646\t0.9\t1\t50 596\n",
      "14\t21\t0.2\t1\t0 21\n",
      "15\t3\t0.1\t1\t0 3\n",
      "19\t5\t0.0\t1\t4 0 1\n",
      "20\t17\t0.0\t2\t3 13 1\n",
      "21\t383\t0.0\t2\t355 26 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t55245\t985304.6\t0\t55245\n",
      "4\t71375\t246326.2\t0\t71375\n",
      "5\t119821\t61581.5\t0\t119821\n",
      "6\t240305\t15395.4\t0\t240305\n",
      "7\t585916\t3848.8\t0\t585916\n",
      "8\t2290071\t962.2\t0\t2290071\n",
      "9\t3333592\t240.6\t0\t3323888 9704\n",
      "10\t5284386\t60.1\t1\t5235448 48938\n",
      "11\t11922066\t15.0\t1\t11786473 135593\n",
      "12\t11919744\t3.8\t1\t11766064 153680\n",
      "13\t8711949\t0.9\t1\t8566683 145266\n",
      "14\t8096498\t0.2\t1\t7950790 145708\n",
      "15\t4030677\t0.1\t1\t3936398 94279\n",
      "16\t2403173\t0.0\t1\t2331608 71565\n",
      "17\t1020448\t0.0\t1\t986274 34174\n",
      "18\t184954\t0.0\t1\t177361 7432 161\n",
      "19\t178639\t0.0\t1\t170902 7528 209\n",
      "20\t225421\t0.0\t2\t213162 11406 853\n",
      "21\t178662\t0.0\t2\t168546 9399 717\n",
      "22\t224476\t0.0\t2\t211826 11905 745\n",
      "23\t245242\t0.0\t2\t233210 11314 718\n",
      "24\t134516\t0.0\t2\t128180 5964 372\n",
      "25\t166230\t0.0\t2\t157917 7833 480\n",
      "26\t234920\t0.0\t2\t222895 11260 765\n",
      "27\t135848\t0.0\t2\t128954 6526 368\n",
      "28\t256516\t0.0\t2\t243316 12160 1040\n",
      "29\t96182\t0.0\t2\t91153 4656 373\n",
      "30\t40620\t0.0\t2\t38238 2199 183\n",
      "31\t27709\t0.0\t2\t26012 1558 139\n",
      "32\t9165\t0.0\t2\t8445 647 73\n",
      "33\t4507\t0.0\t2\t4130 337 40\n",
      "34\t2333\t0.0\t2\t2146 162 25\n",
      "35\t1502\t0.0\t2\t1338 138 26\n",
      "36\t1111\t0.0\t2\t1023 72 16\n",
      "37\t451\t0.0\t2\t418 24 9\n",
      "38\t423\t0.0\t2\t381 37 5\n",
      "39\t3280\t0.0\t2\t2998 229 53\n",
      "40\t656\t0.0\t2\t591 46 19\n",
      "41\t301\t0.0\t2\t263 34 4\n",
      "42\t3317\t0.0\t2\t2981 267 69\n",
      "43\t67\t0.0\t2\t62 3 2\n",
      "44\t29\t0.0\t2\t27 2\n",
      "45\t53\t0.0\t2\t48 5\n",
      "46\t40\t0.0\t2\t28 9 3\n",
      "47\t17\t0.0\t2\t11 3 3\n",
      "48\t119\t0.0\t2\t106 11 2\n",
      "49\t478\t0.0\t2\t415 57 6\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994962_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994962_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1057.16 s (40 us/read; 1.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              26,742,065\n",
      "Reads with adapters:                25,924,087 (96.9%)\n",
      "Reads written (passing filters):    26,742,065 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,337,103,250 bp\n",
      "Total written (filtered):    957,062,449 bp (71.6%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 25924087 times.\n",
      "29042 times, it overlapped the 5' end of a read\n",
      "25895045 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16779\t417844.8\t0\t16779\n",
      "4\t4627\t104461.2\t0\t4627\n",
      "5\t847\t26115.3\t0\t847\n",
      "6\t14\t6528.8\t0\t14\n",
      "7\t11\t1632.2\t0\t11\n",
      "8\t6\t408.1\t0\t6\n",
      "9\t14\t102.0\t0\t3 11\n",
      "10\t5\t25.5\t1\t0 5\n",
      "11\t71\t6.4\t1\t19 52\n",
      "12\t4498\t1.6\t1\t328 4170\n",
      "13\t309\t0.4\t1\t45 264\n",
      "14\t15\t0.1\t1\t0 15\n",
      "19\t9\t0.0\t1\t0 1 8\n",
      "20\t42\t0.0\t2\t13 28 1\n",
      "21\t1794\t0.0\t2\t1745 49\n",
      "22\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18417\t417844.8\t0\t18417\n",
      "4\t18808\t104461.2\t0\t18808\n",
      "5\t28322\t26115.3\t0\t28322\n",
      "6\t36273\t6528.8\t0\t36273\n",
      "7\t65116\t1632.2\t0\t65116\n",
      "8\t733227\t408.1\t0\t733227\n",
      "9\t1663203\t102.0\t0\t1659776 3427\n",
      "10\t2392604\t25.5\t1\t2361799 30805\n",
      "11\t4359408\t6.4\t1\t4296320 63088\n",
      "12\t3282213\t1.6\t1\t3234902 47311\n",
      "13\t2123405\t0.4\t1\t2082644 40761\n",
      "14\t2224448\t0.1\t1\t2176596 47852\n",
      "15\t1246755\t0.0\t1\t1218169 28586\n",
      "16\t1372652\t0.0\t1\t1340855 31797\n",
      "17\t858459\t0.0\t1\t836911 21548\n",
      "18\t1216719\t0.0\t1\t1186412 29425 882\n",
      "19\t1081641\t0.0\t1\t1050468 30298 875\n",
      "20\t860375\t0.0\t2\t833054 25771 1550\n",
      "21\t409302\t0.0\t2\t395278 13104 920\n",
      "22\t111357\t0.0\t2\t107321 3761 275\n",
      "23\t84207\t0.0\t2\t81192 2833 182\n",
      "24\t80827\t0.0\t2\t78225 2414 188\n",
      "25\t87639\t0.0\t2\t84990 2413 236\n",
      "26\t270464\t0.0\t2\t262659 7203 602\n",
      "27\t56508\t0.0\t2\t54779 1597 132\n",
      "28\t57457\t0.0\t2\t55701 1622 134\n",
      "29\t69488\t0.0\t2\t67549 1809 130\n",
      "30\t42108\t0.0\t2\t40897 1131 80\n",
      "31\t39489\t0.0\t2\t38467 929 93\n",
      "32\t23550\t0.0\t2\t22861 641 48\n",
      "33\t18219\t0.0\t2\t17753 440 26\n",
      "34\t11661\t0.0\t2\t11351 281 29\n",
      "35\t8137\t0.0\t2\t7936 188 13\n",
      "36\t7179\t0.0\t2\t6990 178 11\n",
      "37\t6019\t0.0\t2\t5832 179 8\n",
      "38\t7739\t0.0\t2\t7463 253 23\n",
      "39\t90883\t0.0\t2\t87516 3154 213\n",
      "40\t14769\t0.0\t2\t14170 572 27\n",
      "41\t11544\t0.0\t2\t11073 435 36\n",
      "42\t778480\t0.0\t2\t744950 30607 2923\n",
      "43\t10214\t0.0\t2\t9803 377 34\n",
      "44\t2795\t0.0\t2\t2686 99 10\n",
      "45\t7329\t0.0\t2\t7046 242 41\n",
      "46\t2356\t0.0\t2\t2269 77 10\n",
      "47\t435\t0.0\t2\t417 16 2\n",
      "48\t750\t0.0\t2\t704 37 9\n",
      "49\t2095\t0.0\t2\t2007 81 7\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994965_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994965_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1606.82 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              40,531,552\n",
      "Reads with adapters:                39,100,895 (96.5%)\n",
      "Reads written (passing filters):    40,531,552 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,026,577,600 bp\n",
      "Total written (filtered):  1,561,345,549 bp (77.0%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 39100895 times.\n",
      "81649 times, it overlapped the 5' end of a read\n",
      "39019246 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t42297\t633305.5\t0\t42297\n",
      "4\t10522\t158326.4\t0\t10522\n",
      "5\t823\t39581.6\t0\t823\n",
      "6\t387\t9895.4\t0\t387\n",
      "7\t179\t2473.8\t0\t179\n",
      "8\t39\t618.5\t0\t39\n",
      "9\t18\t154.6\t0\t11 7\n",
      "10\t10\t38.7\t1\t2 8\n",
      "11\t201\t9.7\t1\t82 119\n",
      "12\t26182\t2.4\t1\t1161 25021\n",
      "13\t635\t0.6\t1\t32 603\n",
      "14\t21\t0.2\t1\t0 21\n",
      "15\t1\t0.0\t1\t0 1\n",
      "20\t7\t0.0\t2\t0 6 1\n",
      "21\t327\t0.0\t2\t304 17 6\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t56107\t633305.5\t0\t56107\n",
      "4\t121043\t158326.4\t0\t121043\n",
      "5\t1152597\t39581.6\t0\t1152597\n",
      "6\t4390710\t9895.4\t0\t4390710\n",
      "7\t2749568\t2473.8\t0\t2749568\n",
      "8\t2330142\t618.5\t0\t2330142\n",
      "9\t1821607\t154.6\t0\t1815103 6504\n",
      "10\t1727022\t38.7\t1\t1670303 56719\n",
      "11\t4029790\t9.7\t1\t3900115 129675\n",
      "12\t4609680\t2.4\t1\t4456850 152830\n",
      "13\t3657050\t0.6\t1\t3523550 133500\n",
      "14\t3714185\t0.2\t1\t3585375 128810\n",
      "15\t1946433\t0.0\t1\t1868818 77615\n",
      "16\t1734090\t0.0\t1\t1656014 78076\n",
      "17\t1136486\t0.0\t1\t1077042 59444\n",
      "18\t534000\t0.0\t1\t505810 27988 202\n",
      "19\t560011\t0.0\t1\t528581 31060 370\n",
      "20\t1006192\t0.0\t2\t932466 57267 16459\n",
      "21\t202152\t0.0\t2\t186852 11849 3451\n",
      "22\t224603\t0.0\t2\t206922 13713 3968\n",
      "23\t208073\t0.0\t2\t191694 12668 3711\n",
      "24\t126343\t0.0\t2\t117209 7164 1970\n",
      "25\t137233\t0.0\t2\t127670 7504 2059\n",
      "26\t229851\t0.0\t2\t214121 12316 3414\n",
      "27\t135985\t0.0\t2\t126135 7733 2117\n",
      "28\t204788\t0.0\t2\t189489 11940 3359\n",
      "29\t93325\t0.0\t2\t86023 5676 1626\n",
      "30\t50978\t0.0\t2\t46820 3201 957\n",
      "31\t37831\t0.0\t2\t34831 2317 683\n",
      "32\t19294\t0.0\t2\t17635 1304 355\n",
      "33\t11306\t0.0\t2\t10233 829 244\n",
      "34\t8149\t0.0\t2\t7431 548 170\n",
      "35\t4378\t0.0\t2\t3999 306 73\n",
      "36\t3558\t0.0\t2\t3251 255 52\n",
      "37\t2186\t0.0\t2\t1996 149 41\n",
      "38\t4934\t0.0\t2\t4554 311 69\n",
      "39\t15808\t0.0\t2\t14455 1102 251\n",
      "40\t7218\t0.0\t2\t6500 572 146\n",
      "41\t1120\t0.0\t2\t1008 92 20\n",
      "42\t12266\t0.0\t2\t11120 899 247\n",
      "43\t248\t0.0\t2\t221 20 7\n",
      "44\t187\t0.0\t2\t174 11 2\n",
      "45\t275\t0.0\t2\t251 17 7\n",
      "46\t158\t0.0\t2\t147 10 1\n",
      "47\t31\t0.0\t2\t30 1\n",
      "48\t77\t0.0\t2\t72 4 1\n",
      "49\t178\t0.0\t2\t166 11 1\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994966_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994966_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1271.03 s (39 us/read; 1.56 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              32,968,604\n",
      "Reads with adapters:                32,233,799 (97.8%)\n",
      "Reads written (passing filters):    32,968,604 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,648,430,200 bp\n",
      "Total written (filtered):  1,131,414,306 bp (68.6%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 32233799 times.\n",
      "37657 times, it overlapped the 5' end of a read\n",
      "32196142 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16571\t515134.4\t0\t16571\n",
      "4\t4221\t128783.6\t0\t4221\n",
      "5\t635\t32195.9\t0\t635\n",
      "6\t17\t8049.0\t0\t17\n",
      "7\t59\t2012.2\t0\t59\n",
      "8\t23\t503.1\t0\t23\n",
      "9\t7\t125.8\t0\t5 2\n",
      "10\t6\t31.4\t1\t0 6\n",
      "11\t205\t7.9\t1\t126 79\n",
      "12\t14101\t2.0\t1\t657 13444\n",
      "13\t161\t0.5\t1\t21 140\n",
      "14\t10\t0.1\t1\t0 10\n",
      "20\t33\t0.0\t2\t12 21\n",
      "21\t1607\t0.0\t2\t1563 44\n",
      "22\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29125\t515134.4\t0\t29125\n",
      "4\t50478\t128783.6\t0\t50478\n",
      "5\t53266\t32195.9\t0\t53266\n",
      "6\t59758\t8049.0\t0\t59758\n",
      "7\t35868\t2012.2\t0\t35868\n",
      "8\t91108\t503.1\t0\t91108\n",
      "9\t511998\t125.8\t0\t510924 1074\n",
      "10\t855679\t31.4\t1\t844539 11140\n",
      "11\t2354573\t7.9\t1\t2320250 34323\n",
      "12\t2416286\t2.0\t1\t2380842 35444\n",
      "13\t4282591\t0.5\t1\t4199275 83316\n",
      "14\t2421939\t0.1\t1\t2369045 52894\n",
      "15\t1661801\t0.0\t1\t1622302 39499\n",
      "16\t4687036\t0.0\t1\t4576243 110793\n",
      "17\t4438235\t0.0\t1\t4326925 111310\n",
      "18\t2059020\t0.0\t1\t2005495 51511 2014\n",
      "19\t1455022\t0.0\t1\t1412914 40637 1471\n",
      "20\t1123984\t0.0\t2\t1089517 32392 2075\n",
      "21\t922268\t0.0\t2\t892570 27708 1990\n",
      "22\t306801\t0.0\t2\t296539 9600 662\n",
      "23\t289791\t0.0\t2\t280084 9088 619\n",
      "24\t246986\t0.0\t2\t239439 7003 544\n",
      "25\t278395\t0.0\t2\t270595 7179 621\n",
      "26\t377778\t0.0\t2\t366827 10140 811\n",
      "27\t195862\t0.0\t2\t190062 5367 433\n",
      "28\t144282\t0.0\t2\t139993 3957 332\n",
      "29\t140452\t0.0\t2\t136452 3750 250\n",
      "30\t94952\t0.0\t2\t92379 2393 180\n",
      "31\t90396\t0.0\t2\t88007 2214 175\n",
      "32\t55463\t0.0\t2\t53905 1451 107\n",
      "33\t28760\t0.0\t2\t27956 756 48\n",
      "34\t12300\t0.0\t2\t11916 358 26\n",
      "35\t3547\t0.0\t2\t3425 115 7\n",
      "36\t2154\t0.0\t2\t2086 66 2\n",
      "37\t1715\t0.0\t2\t1651 62 2\n",
      "38\t4971\t0.0\t2\t4801 160 10\n",
      "39\t265762\t0.0\t2\t256639 8578 545\n",
      "40\t81987\t0.0\t2\t79110 2675 202\n",
      "41\t3846\t0.0\t2\t3709 126 11\n",
      "42\t41248\t0.0\t2\t39675 1471 102\n",
      "43\t1881\t0.0\t2\t1809 66 6\n",
      "44\t3315\t0.0\t2\t3181 125 9\n",
      "45\t9645\t0.0\t2\t9308 318 19\n",
      "46\t1320\t0.0\t2\t1261 53 6\n",
      "47\t281\t0.0\t2\t271 9 1\n",
      "48\t457\t0.0\t2\t436 19 2\n",
      "49\t1760\t0.0\t2\t1705 48 7\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994969_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994969_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 3315.32 s (40 us/read; 1.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              83,235,717\n",
      "Reads with adapters:                81,883,053 (98.4%)\n",
      "Reads written (passing filters):    83,235,717 (100.0%)\n",
      "\n",
      "Total basepairs processed: 4,161,785,850 bp\n",
      "Total written (filtered):  3,025,242,026 bp (72.7%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 81883053 times.\n",
      "117999 times, it overlapped the 5' end of a read\n",
      "81765054 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t52070\t1300558.1\t0\t52070\n",
      "4\t8287\t325139.5\t0\t8287\n",
      "5\t739\t81284.9\t0\t739\n",
      "6\t74\t20321.2\t0\t74\n",
      "7\t60\t5080.3\t0\t60\n",
      "8\t25\t1270.1\t0\t25\n",
      "9\t38\t317.5\t0\t10 28\n",
      "10\t23\t79.4\t1\t1 22\n",
      "11\t444\t19.8\t1\t218 226\n",
      "12\t55727\t5.0\t1\t3739 51988\n",
      "13\t167\t1.2\t1\t9 158\n",
      "14\t5\t0.3\t1\t0 5\n",
      "19\t1\t0.0\t1\t0 1\n",
      "20\t4\t0.0\t2\t0 3 1\n",
      "21\t334\t0.0\t2\t313 21\n",
      "22\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t116387\t1300558.1\t0\t116387\n",
      "4\t98830\t325139.5\t0\t98830\n",
      "5\t138509\t81284.9\t0\t138509\n",
      "6\t173016\t20321.2\t0\t173016\n",
      "7\t126423\t5080.3\t0\t126423\n",
      "8\t477896\t1270.1\t0\t477896\n",
      "9\t1286960\t317.5\t0\t1283682 3278\n",
      "10\t2660786\t79.4\t1\t2610819 49967\n",
      "11\t8046001\t19.8\t1\t7880169 165832\n",
      "12\t12120989\t5.0\t1\t11839789 281200\n",
      "13\t12217602\t1.2\t1\t11863760 353842\n",
      "14\t17319786\t0.3\t1\t16813602 506184\n",
      "15\t11782618\t0.1\t1\t11397809 384809\n",
      "16\t5915431\t0.0\t1\t5725982 189449\n",
      "17\t4851144\t0.0\t1\t4680004 171140\n",
      "18\t1238539\t0.0\t1\t1181185 56264 1090\n",
      "19\t272624\t0.0\t1\t257647 14605 372\n",
      "20\t254645\t0.0\t2\t235825 17319 1501\n",
      "21\t277180\t0.0\t2\t255560 19705 1915\n",
      "22\t277524\t0.0\t2\t256560 19563 1401\n",
      "23\t311947\t0.0\t2\t289638 20788 1521\n",
      "24\t232726\t0.0\t2\t216632 15068 1026\n",
      "25\t299625\t0.0\t2\t278438 19832 1355\n",
      "26\t320912\t0.0\t2\t296045 22995 1872\n",
      "27\t173822\t0.0\t2\t159978 12786 1058\n",
      "28\t416248\t0.0\t2\t381744 31729 2775\n",
      "29\t119953\t0.0\t2\t110274 8906 773\n",
      "30\t59173\t0.0\t2\t54327 4425 421\n",
      "31\t47011\t0.0\t2\t42981 3672 358\n",
      "32\t17300\t0.0\t2\t15682 1479 139\n",
      "33\t10016\t0.0\t2\t8955 892 169\n",
      "34\t6351\t0.0\t2\t5612 621 118\n",
      "35\t3952\t0.0\t2\t3561 329 62\n",
      "36\t3606\t0.0\t2\t3265 280 61\n",
      "37\t1612\t0.0\t2\t1462 131 19\n",
      "38\t2357\t0.0\t2\t2138 182 37\n",
      "39\t36816\t0.0\t2\t33273 2988 555\n",
      "40\t14165\t0.0\t2\t12789 1176 200\n",
      "41\t1730\t0.0\t2\t1610 94 26\n",
      "42\t31382\t0.0\t2\t28786 2243 353\n",
      "43\t486\t0.0\t2\t455 28 3\n",
      "44\t210\t0.0\t2\t195 13 2\n",
      "45\t177\t0.0\t2\t158 17 2\n",
      "46\t106\t0.0\t2\t102 4\n",
      "47\t39\t0.0\t2\t35 4\n",
      "48\t112\t0.0\t2\t103 8 1\n",
      "49\t330\t0.0\t2\t299 30 1\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994970_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994970_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1223.98 s (39 us/read; 1.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              31,175,890\n",
      "Reads with adapters:                30,172,831 (96.8%)\n",
      "Reads written (passing filters):    31,175,890 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,558,794,500 bp\n",
      "Total written (filtered):  1,082,210,352 bp (69.4%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 30172831 times.\n",
      "50740 times, it overlapped the 5' end of a read\n",
      "30122091 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t26070\t487123.3\t0\t26070\n",
      "4\t6076\t121780.8\t0\t6076\n",
      "5\t850\t30445.2\t0\t850\n",
      "6\t34\t7611.3\t0\t34\n",
      "7\t65\t1902.8\t0\t65\n",
      "8\t62\t475.7\t0\t62\n",
      "9\t13\t118.9\t0\t4 9\n",
      "10\t2\t29.7\t1\t1 1\n",
      "11\t135\t7.4\t1\t79 56\n",
      "12\t16564\t1.9\t1\t906 15658\n",
      "13\t312\t0.5\t1\t77 235\n",
      "14\t9\t0.1\t1\t0 9\n",
      "19\t3\t0.0\t1\t0 0 3\n",
      "20\t10\t0.0\t2\t2 8\n",
      "21\t535\t0.0\t2\t518 16 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t46006\t487123.3\t0\t46006\n",
      "4\t124755\t121780.8\t0\t124755\n",
      "5\t117197\t30445.2\t0\t117197\n",
      "6\t91762\t7611.3\t0\t91762\n",
      "7\t50200\t1902.8\t0\t50200\n",
      "8\t99694\t475.7\t0\t99694\n",
      "9\t369109\t118.9\t0\t368135 974\n",
      "10\t1175159\t29.7\t1\t1157247 17912\n",
      "11\t2794651\t7.4\t1\t2743697 50954\n",
      "12\t2685143\t1.9\t1\t2630392 54751\n",
      "13\t3505082\t0.5\t1\t3418673 86409\n",
      "14\t2438404\t0.1\t1\t2375537 62867\n",
      "15\t2022352\t0.0\t1\t1967074 55278\n",
      "16\t2617985\t0.0\t1\t2546496 71489\n",
      "17\t2603349\t0.0\t1\t2523091 80258\n",
      "18\t1817706\t0.0\t1\t1759683 56234 1789\n",
      "19\t2192019\t0.0\t1\t2115072 74441 2506\n",
      "20\t1533064\t0.0\t2\t1477027 52922 3115\n",
      "21\t1233636\t0.0\t2\t1186405 44344 2887\n",
      "22\t378379\t0.0\t2\t363663 13807 909\n",
      "23\t351818\t0.0\t2\t339104 11981 733\n",
      "24\t418859\t0.0\t2\t403693 14270 896\n",
      "25\t318850\t0.0\t2\t307675 10520 655\n",
      "26\t309124\t0.0\t2\t298336 10199 589\n",
      "27\t160293\t0.0\t2\t154562 5392 339\n",
      "28\t156976\t0.0\t2\t151090 5595 291\n",
      "29\t144561\t0.0\t2\t139755 4519 287\n",
      "30\t87209\t0.0\t2\t84225 2806 178\n",
      "31\t55136\t0.0\t2\t53311 1725 100\n",
      "32\t30664\t0.0\t2\t29731 870 63\n",
      "33\t28273\t0.0\t2\t27355 863 55\n",
      "34\t11208\t0.0\t2\t10864 326 18\n",
      "35\t3989\t0.0\t2\t3867 116 6\n",
      "36\t2118\t0.0\t2\t2045 66 7\n",
      "37\t1428\t0.0\t2\t1388 38 2\n",
      "38\t3175\t0.0\t2\t3052 121 2\n",
      "39\t103482\t0.0\t2\t99149 4029 304\n",
      "40\t26305\t0.0\t2\t25025 1185 95\n",
      "41\t957\t0.0\t2\t906 46 5\n",
      "42\t7643\t0.0\t2\t7230 384 29\n",
      "43\t382\t0.0\t2\t358 23 1\n",
      "44\t398\t0.0\t2\t376 22\n",
      "45\t2496\t0.0\t2\t2371 113 12\n",
      "46\t236\t0.0\t2\t225 11\n",
      "47\t70\t0.0\t2\t67 3\n",
      "48\t130\t0.0\t2\t121 8 1\n",
      "49\t659\t0.0\t2\t620 36 3\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994973_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994973_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1604.71 s (39 us/read; 1.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              40,848,270\n",
      "Reads with adapters:                39,024,959 (95.5%)\n",
      "Reads written (passing filters):    40,848,270 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,042,413,500 bp\n",
      "Total written (filtered):  1,492,767,994 bp (73.1%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 39024959 times.\n",
      "80555 times, it overlapped the 5' end of a read\n",
      "38944404 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t44873\t638254.2\t0\t44873\n",
      "4\t10765\t159563.6\t0\t10765\n",
      "5\t837\t39890.9\t0\t837\n",
      "6\t152\t9972.7\t0\t152\n",
      "7\t90\t2493.2\t0\t90\n",
      "8\t34\t623.3\t0\t34\n",
      "9\t6\t155.8\t0\t4 2\n",
      "10\t13\t39.0\t1\t2 11\n",
      "11\t157\t9.7\t1\t75 82\n",
      "12\t23023\t2.4\t1\t1046 21977\n",
      "13\t455\t0.6\t1\t21 434\n",
      "14\t8\t0.2\t1\t0 8\n",
      "20\t1\t0.0\t2\t0 1\n",
      "21\t141\t0.0\t2\t126 14 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t42137\t638254.2\t0\t42137\n",
      "4\t47334\t159563.6\t0\t47334\n",
      "5\t52454\t39890.9\t0\t52454\n",
      "6\t94507\t9972.7\t0\t94507\n",
      "7\t324827\t2493.2\t0\t324827\n",
      "8\t900437\t623.3\t0\t900437\n",
      "9\t960122\t155.8\t0\t955580 4542\n",
      "10\t1148745\t39.0\t1\t1106394 42351\n",
      "11\t3028217\t9.7\t1\t2919900 108317\n",
      "12\t4653587\t2.4\t1\t4476133 177454\n",
      "13\t7259928\t0.6\t1\t6956085 303843\n",
      "14\t5912082\t0.2\t1\t5682274 229808\n",
      "15\t3689050\t0.0\t1\t3532329 156721\n",
      "16\t3208530\t0.0\t1\t3058384 150146\n",
      "17\t4031200\t0.0\t1\t3791324 239876\n",
      "18\t1661790\t0.0\t1\t1567781 93363 646\n",
      "19\t511233\t0.0\t1\t480107 30748 378\n",
      "20\t179564\t0.0\t2\t165751 10858 2955\n",
      "21\t185153\t0.0\t2\t169926 11904 3323\n",
      "22\t102799\t0.0\t2\t94221 6659 1919\n",
      "23\t99874\t0.0\t2\t91344 6680 1850\n",
      "24\t71248\t0.0\t2\t65507 4506 1235\n",
      "25\t77436\t0.0\t2\t71726 4515 1195\n",
      "26\t103588\t0.0\t2\t96376 5768 1444\n",
      "27\t107755\t0.0\t2\t99611 6515 1629\n",
      "28\t146723\t0.0\t2\t135577 8868 2278\n",
      "29\t60422\t0.0\t2\t55614 3795 1013\n",
      "30\t42536\t0.0\t2\t39014 2782 740\n",
      "31\t31588\t0.0\t2\t29002 2067 519\n",
      "32\t22419\t0.0\t2\t20487 1534 398\n",
      "33\t14935\t0.0\t2\t13618 1026 291\n",
      "34\t12053\t0.0\t2\t11051 805 197\n",
      "35\t7164\t0.0\t2\t6553 478 133\n",
      "36\t5563\t0.0\t2\t5136 351 76\n",
      "37\t4352\t0.0\t2\t4050 240 62\n",
      "38\t16884\t0.0\t2\t15646 1013 225\n",
      "39\t85540\t0.0\t2\t78112 6217 1211\n",
      "40\t26599\t0.0\t2\t23813 2190 596\n",
      "41\t1176\t0.0\t2\t1052 96 28\n",
      "42\t11861\t0.0\t2\t10393 1190 278\n",
      "43\t311\t0.0\t2\t271 33 7\n",
      "44\t146\t0.0\t2\t133 10 3\n",
      "45\t151\t0.0\t2\t140 11\n",
      "46\t102\t0.0\t2\t90 8 4\n",
      "47\t37\t0.0\t2\t37\n",
      "48\t88\t0.0\t2\t79 7 2\n",
      "49\t157\t0.0\t2\t141 14 2\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/ERR1994974_N_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994974_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1218.75 s (38 us/read; 1.57 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              31,871,091\n",
      "Reads with adapters:                30,904,102 (97.0%)\n",
      "Reads written (passing filters):    31,871,091 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,593,554,550 bp\n",
      "Total written (filtered):  1,076,637,622 bp (67.6%)\n",
      "\n",
      "=== Adapter Ribo_adaptor ===\n",
      "\n",
      "Sequence: TGGAATTCTCGGGTGCCAAGG; Type: variable 5'/3'; Length: 21; Trimmed: 30904102 times.\n",
      "61360 times, it overlapped the 5' end of a read\n",
      "30842742 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t25480\t497985.8\t0\t25480\n",
      "4\t5906\t124496.4\t0\t5906\n",
      "5\t861\t31124.1\t0\t861\n",
      "6\t35\t7781.0\t0\t35\n",
      "7\t71\t1945.3\t0\t71\n",
      "8\t45\t486.3\t0\t45\n",
      "9\t12\t121.6\t0\t7 5\n",
      "10\t4\t30.4\t1\t0 4\n",
      "11\t198\t7.6\t1\t111 87\n",
      "12\t26998\t1.9\t1\t1557 25441\n",
      "13\t725\t0.5\t1\t98 627\n",
      "14\t9\t0.1\t1\t0 9\n",
      "19\t1\t0.0\t1\t0 0 1\n",
      "20\t16\t0.0\t2\t4 12\n",
      "21\t994\t0.0\t2\t959 33 2\n",
      "22\t3\t0.0\t2\t0 3\n",
      "23\t2\t0.0\t2\t0 0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t47959\t497985.8\t0\t47959\n",
      "4\t124411\t124496.4\t0\t124411\n",
      "5\t100938\t31124.1\t0\t100938\n",
      "6\t72612\t7781.0\t0\t72612\n",
      "7\t83488\t1945.3\t0\t83488\n",
      "8\t276668\t486.3\t0\t276668\n",
      "9\t695032\t121.6\t0\t692918 2114\n",
      "10\t1393012\t30.4\t1\t1371133 21879\n",
      "11\t2247142\t7.6\t1\t2204522 42620\n",
      "12\t2092542\t1.9\t1\t2048724 43818\n",
      "13\t3042612\t0.5\t1\t2965790 76822\n",
      "14\t1970259\t0.1\t1\t1919583 50676\n",
      "15\t1714910\t0.0\t1\t1668039 46871\n",
      "16\t2545188\t0.0\t1\t2473846 71342\n",
      "17\t2238743\t0.0\t1\t2168629 70114\n",
      "18\t1602058\t0.0\t1\t1550354 50109 1595\n",
      "19\t1823757\t0.0\t1\t1760393 61463 1901\n",
      "20\t1762206\t0.0\t2\t1697971 60734 3501\n",
      "21\t2117814\t0.0\t2\t2036612 76260 4942\n",
      "22\t1497981\t0.0\t2\t1440739 53839 3403\n",
      "23\t761121\t0.0\t2\t732724 26601 1796\n",
      "24\t577933\t0.0\t2\t556652 20045 1236\n",
      "25\t355041\t0.0\t2\t342430 11815 796\n",
      "26\t324265\t0.0\t2\t312654 10857 754\n",
      "27\t176386\t0.0\t2\t169888 6127 371\n",
      "28\t196207\t0.0\t2\t189003 6787 417\n",
      "29\t167937\t0.0\t2\t162363 5264 310\n",
      "30\t119625\t0.0\t2\t115648 3703 274\n",
      "31\t80958\t0.0\t2\t78485 2336 137\n",
      "32\t49576\t0.0\t2\t48187 1306 83\n",
      "33\t44237\t0.0\t2\t42862 1319 56\n",
      "34\t19267\t0.0\t2\t18663 575 29\n",
      "35\t6149\t0.0\t2\t5967 181 1\n",
      "36\t3943\t0.0\t2\t3839 98 6\n",
      "37\t3411\t0.0\t2\t3320 89 2\n",
      "38\t7924\t0.0\t2\t7684 226 14\n",
      "39\t374258\t0.0\t2\t361207 12385 666\n",
      "40\t84448\t0.0\t2\t80990 3220 238\n",
      "41\t2949\t0.0\t2\t2796 150 3\n",
      "42\t24699\t0.0\t2\t23594 1028 77\n",
      "43\t1121\t0.0\t2\t1070 51\n",
      "44\t1099\t0.0\t2\t1047 51 1\n",
      "45\t8418\t0.0\t2\t8044 346 28\n",
      "46\t554\t0.0\t2\t532 17 5\n",
      "47\t224\t0.0\t2\t215 8 1\n",
      "48\t309\t0.0\t2\t293 12 4\n",
      "49\t1351\t0.0\t2\t1275 66 10\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "addition=\"_1\"\n",
    "label=\"_N_adaptor\"\n",
    "directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/\"\n",
    "mkdir $directory\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "cutadapt --trim-n -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_Ribo_adaptor.fa -o $output_directory$accession_base$i$label$extention $path_fastq_files$accession_base$i$addition$extention\n",
    "done\n",
    "\n",
    "\n",
    "# (-O) Minimum number of nucleotides of the adaptor to trim. By default is 3."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Second step: Remove PCR duplicate to generate a non-redundant dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-09T14:14:48.685963Z",
     "start_time": "2018-03-09T11:07:00.850258Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/’: File exists\n",
      "Input and filter stats:\n",
      "\tInput sequences: 63,059,496\n",
      "\tInput bases: 2,371,900,197\n",
      "\tInput mean length: 37.61\n",
      "\tGood sequences: 27,573,308 (43.73%)\n",
      "\tGood bases: 1,038,957,856\n",
      "\tGood mean length: 37.68\n",
      "\tBad sequences: 35,486,188 (56.27%)\n",
      "\tBad bases: 1,332,942,341\n",
      "\tBad mean length: 37.56\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 35486188\n",
      "Input and filter stats:\n",
      "\tInput sequences: 26,742,060\n",
      "\tInput bases: 957,062,449\n",
      "\tInput mean length: 35.79\n",
      "\tGood sequences: 12,134,571 (45.38%)\n",
      "\tGood bases: 445,811,712\n",
      "\tGood mean length: 36.74\n",
      "\tBad sequences: 14,607,489 (54.62%)\n",
      "\tBad bases: 511,250,737\n",
      "\tBad mean length: 35.00\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 14607489\n",
      "Input and filter stats:\n",
      "\tInput sequences: 40,531,552\n",
      "\tInput bases: 1,561,345,549\n",
      "\tInput mean length: 38.52\n",
      "\tGood sequences: 22,940,056 (56.60%)\n",
      "\tGood bases: 873,927,862\n",
      "\tGood mean length: 38.10\n",
      "\tBad sequences: 17,591,496 (43.40%)\n",
      "\tBad bases: 687,417,687\n",
      "\tBad mean length: 39.08\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 17591496\n",
      "Input and filter stats:\n",
      "\tInput sequences: 32,968,596\n",
      "\tInput bases: 1,131,414,306\n",
      "\tInput mean length: 34.32\n",
      "\tGood sequences: 12,189,919 (36.97%)\n",
      "\tGood bases: 422,019,514\n",
      "\tGood mean length: 34.62\n",
      "\tBad sequences: 20,778,677 (63.03%)\n",
      "\tBad bases: 709,394,792\n",
      "\tBad mean length: 34.14\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 20778677\n",
      "Input and filter stats:\n",
      "\tInput sequences: 83,235,717\n",
      "\tInput bases: 3,025,242,026\n",
      "\tInput mean length: 36.35\n",
      "\tGood sequences: 39,643,455 (47.63%)\n",
      "\tGood bases: 1,452,792,964\n",
      "\tGood mean length: 36.65\n",
      "\tBad sequences: 43,592,262 (52.37%)\n",
      "\tBad bases: 1,572,449,062\n",
      "\tBad mean length: 36.07\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 43592262\n",
      "Input and filter stats:\n",
      "\tInput sequences: 31,175,889\n",
      "\tInput bases: 1,082,210,352\n",
      "\tInput mean length: 34.71\n",
      "\tGood sequences: 12,199,061 (39.13%)\n",
      "\tGood bases: 427,243,789\n",
      "\tGood mean length: 35.02\n",
      "\tBad sequences: 18,976,828 (60.87%)\n",
      "\tBad bases: 654,966,563\n",
      "\tBad mean length: 34.51\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 18976828\n",
      "Input and filter stats:\n",
      "\tInput sequences: 40,848,270\n",
      "\tInput bases: 1,492,767,994\n",
      "\tInput mean length: 36.54\n",
      "\tGood sequences: 19,818,519 (48.52%)\n",
      "\tGood bases: 736,050,398\n",
      "\tGood mean length: 37.14\n",
      "\tBad sequences: 21,029,751 (51.48%)\n",
      "\tBad bases: 756,717,596\n",
      "\tBad mean length: 35.98\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 21029751\n",
      "Input and filter stats:\n",
      "\tInput sequences: 31,871,085\n",
      "\tInput bases: 1,076,637,622\n",
      "\tInput mean length: 33.78\n",
      "\tGood sequences: 11,729,082 (36.80%)\n",
      "\tGood bases: 401,100,553\n",
      "\tGood mean length: 34.20\n",
      "\tBad sequences: 20,142,003 (63.20%)\n",
      "\tBad bases: 675,537,069\n",
      "\tBad mean length: 33.54\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 20142003\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_N_adaptor/\"\n",
    "extention=\".fastq\"    # no extention for output\n",
    "label=\"_N_adaptor\"\n",
    "label2=\"_no_dup\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "perl /home/simone/Downloads/prinseq-lite-0.20.4/prinseq-lite.pl -out_good $output_directory$accession_base$i$label2 -fastq $path_fastq_files$accession_base$i$label$extention -derep 1 -no_qual_header\n",
    "done\n",
    "\n",
    "# -derep 1  --> remove exact duplicate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Third step: Remove first four and last four random nucleotides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-09T15:01:08.371023Z",
     "start_time": "2018-03-09T14:33:06.299360Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994961_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994961_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 304.51 s (11 us/read; 5.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              27,573,308\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           1,632,580 (5.9%)\n",
      "Reads written (passing filters):    25,940,728 (94.1%)\n",
      "\n",
      "Total basepairs processed: 1,038,957,856 bp\n",
      "Total written (filtered):    785,779,286 bp (75.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994962_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994962_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 128.80 s (11 us/read; 5.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              12,134,571\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           2,255,970 (18.6%)\n",
      "Reads written (passing filters):     9,878,601 (81.4%)\n",
      "\n",
      "Total basepairs processed:   445,811,712 bp\n",
      "Total written (filtered):    301,003,650 bp (67.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994965_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994965_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 252.74 s (11 us/read; 5.45 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              22,940,056\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           2,742,952 (12.0%)\n",
      "Reads written (passing filters):    20,197,104 (88.0%)\n",
      "\n",
      "Total basepairs processed:   873,927,862 bp\n",
      "Total written (filtered):    634,362,422 bp (72.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994966_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994966_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 122.38 s (10 us/read; 5.98 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              12,189,919\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           4,488,751 (36.8%)\n",
      "Reads written (passing filters):     7,701,168 (63.2%)\n",
      "\n",
      "Total basepairs processed:   422,019,514 bp\n",
      "Total written (filtered):    230,652,281 bp (54.7%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994969_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994969_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 434.07 s (11 us/read; 5.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              39,643,455\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           4,484,948 (11.3%)\n",
      "Reads written (passing filters):    35,158,507 (88.7%)\n",
      "\n",
      "Total basepairs processed: 1,452,792,964 bp\n",
      "Total written (filtered):  1,039,793,306 bp (71.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994970_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994970_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 116.11 s (10 us/read; 6.30 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              12,199,061\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           4,324,703 (35.5%)\n",
      "Reads written (passing filters):     7,874,358 (64.5%)\n",
      "\n",
      "Total basepairs processed:   427,243,789 bp\n",
      "Total written (filtered):    238,041,076 bp (55.7%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994973_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994973_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 208.87 s (11 us/read; 5.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,818,519\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           3,615,696 (18.2%)\n",
      "Reads written (passing filters):    16,202,823 (81.8%)\n",
      "\n",
      "Total basepairs processed:   736,050,398 bp\n",
      "Total written (filtered):    497,508,573 bp (67.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 4 -u -4 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/ERR1994974_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/ERR1994974_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 110.93 s (9 us/read; 6.34 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,729,082\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:           5,072,792 (43.2%)\n",
      "Reads written (passing filters):     6,656,290 (56.8%)\n",
      "\n",
      "Total basepairs processed:   401,100,553 bp\n",
      "Total written (filtered):    202,677,487 bp (50.5%)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_no_dup/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_no_dup\"\n",
    "label2=\"_trimmed\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "# remove first four and last four nucleotides\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "cutadapt -u 4 -u -4 -m 26 -o $output_directory$accession_base$i$label2$extention $path_fastq_files$accession_base$i$label$extention\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-13T10:03:25.682432Z",
     "start_time": "2018-03-13T09:54:16.300284Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR1994961_trimmed.fastq\n",
      "Analysis complete for ERR1994962_trimmed.fastq\n",
      "Analysis complete for ERR1994965_trimmed.fastq\n",
      "Analysis complete for ERR1994966_trimmed.fastq\n",
      "Analysis complete for ERR1994969_trimmed.fastq\n",
      "Analysis complete for ERR1994970_trimmed.fastq\n",
      "Analysis complete for ERR1994973_trimmed.fastq\n",
      "Analysis complete for ERR1994974_trimmed.fastq\n",
      "Searching 16 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of ERR1994961_trimmed.fastq\n",
      "Approx 5% complete for ERR1994961_trimmed.fastq\n",
      "Approx 10% complete for ERR1994961_trimmed.fastq\n",
      "Approx 15% complete for ERR1994961_trimmed.fastq\n",
      "Approx 20% complete for ERR1994961_trimmed.fastq\n",
      "Approx 25% complete for ERR1994961_trimmed.fastq\n",
      "Approx 30% complete for ERR1994961_trimmed.fastq\n",
      "Approx 35% complete for ERR1994961_trimmed.fastq\n",
      "Approx 40% complete for ERR1994961_trimmed.fastq\n",
      "Approx 45% complete for ERR1994961_trimmed.fastq\n",
      "Approx 50% complete for ERR1994961_trimmed.fastq\n",
      "Approx 55% complete for ERR1994961_trimmed.fastq\n",
      "Approx 60% complete for ERR1994961_trimmed.fastq\n",
      "Approx 65% complete for ERR1994961_trimmed.fastq\n",
      "Approx 70% complete for ERR1994961_trimmed.fastq\n",
      "Approx 75% complete for ERR1994961_trimmed.fastq\n",
      "Approx 80% complete for ERR1994961_trimmed.fastq\n",
      "Approx 85% complete for ERR1994961_trimmed.fastq\n",
      "Approx 90% complete for ERR1994961_trimmed.fastq\n",
      "Approx 95% complete for ERR1994961_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994962_trimmed.fastq\n",
      "Approx 5% complete for ERR1994962_trimmed.fastq\n",
      "Approx 10% complete for ERR1994962_trimmed.fastq\n",
      "Approx 15% complete for ERR1994962_trimmed.fastq\n",
      "Approx 20% complete for ERR1994962_trimmed.fastq\n",
      "Approx 25% complete for ERR1994962_trimmed.fastq\n",
      "Approx 30% complete for ERR1994962_trimmed.fastq\n",
      "Approx 35% complete for ERR1994962_trimmed.fastq\n",
      "Approx 40% complete for ERR1994962_trimmed.fastq\n",
      "Approx 45% complete for ERR1994962_trimmed.fastq\n",
      "Approx 50% complete for ERR1994962_trimmed.fastq\n",
      "Approx 55% complete for ERR1994962_trimmed.fastq\n",
      "Approx 60% complete for ERR1994962_trimmed.fastq\n",
      "Approx 65% complete for ERR1994962_trimmed.fastq\n",
      "Approx 70% complete for ERR1994962_trimmed.fastq\n",
      "Approx 75% complete for ERR1994962_trimmed.fastq\n",
      "Approx 80% complete for ERR1994962_trimmed.fastq\n",
      "Approx 85% complete for ERR1994962_trimmed.fastq\n",
      "Approx 90% complete for ERR1994962_trimmed.fastq\n",
      "Approx 95% complete for ERR1994962_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994965_trimmed.fastq\n",
      "Approx 5% complete for ERR1994965_trimmed.fastq\n",
      "Approx 10% complete for ERR1994965_trimmed.fastq\n",
      "Approx 15% complete for ERR1994965_trimmed.fastq\n",
      "Approx 20% complete for ERR1994965_trimmed.fastq\n",
      "Approx 25% complete for ERR1994965_trimmed.fastq\n",
      "Approx 30% complete for ERR1994965_trimmed.fastq\n",
      "Approx 35% complete for ERR1994965_trimmed.fastq\n",
      "Approx 40% complete for ERR1994965_trimmed.fastq\n",
      "Approx 45% complete for ERR1994965_trimmed.fastq\n",
      "Approx 50% complete for ERR1994965_trimmed.fastq\n",
      "Approx 55% complete for ERR1994965_trimmed.fastq\n",
      "Approx 60% complete for ERR1994965_trimmed.fastq\n",
      "Approx 65% complete for ERR1994965_trimmed.fastq\n",
      "Approx 70% complete for ERR1994965_trimmed.fastq\n",
      "Approx 75% complete for ERR1994965_trimmed.fastq\n",
      "Approx 80% complete for ERR1994965_trimmed.fastq\n",
      "Approx 85% complete for ERR1994965_trimmed.fastq\n",
      "Approx 90% complete for ERR1994965_trimmed.fastq\n",
      "Approx 95% complete for ERR1994965_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994966_trimmed.fastq\n",
      "Approx 5% complete for ERR1994966_trimmed.fastq\n",
      "Approx 10% complete for ERR1994966_trimmed.fastq\n",
      "Approx 15% complete for ERR1994966_trimmed.fastq\n",
      "Approx 20% complete for ERR1994966_trimmed.fastq\n",
      "Approx 25% complete for ERR1994966_trimmed.fastq\n",
      "Approx 30% complete for ERR1994966_trimmed.fastq\n",
      "Approx 35% complete for ERR1994966_trimmed.fastq\n",
      "Approx 40% complete for ERR1994966_trimmed.fastq\n",
      "Approx 45% complete for ERR1994966_trimmed.fastq\n",
      "Approx 50% complete for ERR1994966_trimmed.fastq\n",
      "Approx 55% complete for ERR1994966_trimmed.fastq\n",
      "Approx 60% complete for ERR1994966_trimmed.fastq\n",
      "Approx 65% complete for ERR1994966_trimmed.fastq\n",
      "Approx 70% complete for ERR1994966_trimmed.fastq\n",
      "Approx 75% complete for ERR1994966_trimmed.fastq\n",
      "Approx 80% complete for ERR1994966_trimmed.fastq\n",
      "Approx 85% complete for ERR1994966_trimmed.fastq\n",
      "Approx 90% complete for ERR1994966_trimmed.fastq\n",
      "Approx 95% complete for ERR1994966_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994969_trimmed.fastq\n",
      "Approx 5% complete for ERR1994969_trimmed.fastq\n",
      "Approx 10% complete for ERR1994969_trimmed.fastq\n",
      "Approx 15% complete for ERR1994969_trimmed.fastq\n",
      "Approx 20% complete for ERR1994969_trimmed.fastq\n",
      "Approx 25% complete for ERR1994969_trimmed.fastq\n",
      "Approx 30% complete for ERR1994969_trimmed.fastq\n",
      "Approx 35% complete for ERR1994969_trimmed.fastq\n",
      "Approx 40% complete for ERR1994969_trimmed.fastq\n",
      "Approx 45% complete for ERR1994969_trimmed.fastq\n",
      "Approx 50% complete for ERR1994969_trimmed.fastq\n",
      "Approx 55% complete for ERR1994969_trimmed.fastq\n",
      "Approx 60% complete for ERR1994969_trimmed.fastq\n",
      "Approx 65% complete for ERR1994969_trimmed.fastq\n",
      "Approx 70% complete for ERR1994969_trimmed.fastq\n",
      "Approx 75% complete for ERR1994969_trimmed.fastq\n",
      "Approx 80% complete for ERR1994969_trimmed.fastq\n",
      "Approx 85% complete for ERR1994969_trimmed.fastq\n",
      "Approx 90% complete for ERR1994969_trimmed.fastq\n",
      "Approx 95% complete for ERR1994969_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994970_trimmed.fastq\n",
      "Approx 5% complete for ERR1994970_trimmed.fastq\n",
      "Approx 10% complete for ERR1994970_trimmed.fastq\n",
      "Approx 15% complete for ERR1994970_trimmed.fastq\n",
      "Approx 20% complete for ERR1994970_trimmed.fastq\n",
      "Approx 25% complete for ERR1994970_trimmed.fastq\n",
      "Approx 30% complete for ERR1994970_trimmed.fastq\n",
      "Approx 35% complete for ERR1994970_trimmed.fastq\n",
      "Approx 40% complete for ERR1994970_trimmed.fastq\n",
      "Approx 45% complete for ERR1994970_trimmed.fastq\n",
      "Approx 50% complete for ERR1994970_trimmed.fastq\n",
      "Approx 55% complete for ERR1994970_trimmed.fastq\n",
      "Approx 60% complete for ERR1994970_trimmed.fastq\n",
      "Approx 65% complete for ERR1994970_trimmed.fastq\n",
      "Approx 70% complete for ERR1994970_trimmed.fastq\n",
      "Approx 75% complete for ERR1994970_trimmed.fastq\n",
      "Approx 80% complete for ERR1994970_trimmed.fastq\n",
      "Approx 85% complete for ERR1994970_trimmed.fastq\n",
      "Approx 90% complete for ERR1994970_trimmed.fastq\n",
      "Approx 95% complete for ERR1994970_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994973_trimmed.fastq\n",
      "Approx 5% complete for ERR1994973_trimmed.fastq\n",
      "Approx 10% complete for ERR1994973_trimmed.fastq\n",
      "Approx 15% complete for ERR1994973_trimmed.fastq\n",
      "Approx 20% complete for ERR1994973_trimmed.fastq\n",
      "Approx 25% complete for ERR1994973_trimmed.fastq\n",
      "Approx 30% complete for ERR1994973_trimmed.fastq\n",
      "Approx 35% complete for ERR1994973_trimmed.fastq\n",
      "Approx 40% complete for ERR1994973_trimmed.fastq\n",
      "Approx 45% complete for ERR1994973_trimmed.fastq\n",
      "Approx 50% complete for ERR1994973_trimmed.fastq\n",
      "Approx 55% complete for ERR1994973_trimmed.fastq\n",
      "Approx 60% complete for ERR1994973_trimmed.fastq\n",
      "Approx 65% complete for ERR1994973_trimmed.fastq\n",
      "Approx 70% complete for ERR1994973_trimmed.fastq\n",
      "Approx 75% complete for ERR1994973_trimmed.fastq\n",
      "Approx 80% complete for ERR1994973_trimmed.fastq\n",
      "Approx 85% complete for ERR1994973_trimmed.fastq\n",
      "Approx 90% complete for ERR1994973_trimmed.fastq\n",
      "Approx 95% complete for ERR1994973_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994974_trimmed.fastq\n",
      "Approx 5% complete for ERR1994974_trimmed.fastq\n",
      "Approx 10% complete for ERR1994974_trimmed.fastq\n",
      "Approx 15% complete for ERR1994974_trimmed.fastq\n",
      "Approx 20% complete for ERR1994974_trimmed.fastq\n",
      "Approx 25% complete for ERR1994974_trimmed.fastq\n",
      "Approx 30% complete for ERR1994974_trimmed.fastq\n",
      "Approx 35% complete for ERR1994974_trimmed.fastq\n",
      "Approx 40% complete for ERR1994974_trimmed.fastq\n",
      "Approx 45% complete for ERR1994974_trimmed.fastq\n",
      "Approx 50% complete for ERR1994974_trimmed.fastq\n",
      "Approx 55% complete for ERR1994974_trimmed.fastq\n",
      "Approx 60% complete for ERR1994974_trimmed.fastq\n",
      "Approx 65% complete for ERR1994974_trimmed.fastq\n",
      "Approx 70% complete for ERR1994974_trimmed.fastq\n",
      "Approx 75% complete for ERR1994974_trimmed.fastq\n",
      "Approx 80% complete for ERR1994974_trimmed.fastq\n",
      "Approx 85% complete for ERR1994974_trimmed.fastq\n",
      "Approx 90% complete for ERR1994974_trimmed.fastq\n",
      "Approx 95% complete for ERR1994974_trimmed.fastq\n",
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 8 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/fastqc_reports/Ribo_final/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base$i$label$extention\n",
    "done\n",
    "\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/fastqc_reports/Ribo_final/\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RNAseq data\n",
    "\n",
    "**Reads structure**:     RRRRRRRRR(NNNN....NNNN)\n",
    "\n",
    "**R(9)** corresponds to a UMI and **N** to the sequence of fragmented RNA molecules"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First step: Remove sequencing adapters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-08T22:32:49.828477Z",
     "start_time": "2018-03-08T19:15:03.095939Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994959_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994959_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1563.41 s (59 us/read; 1.01 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              26,425,976\n",
      "Reads with adapters:                 2,727,528 (10.3%)\n",
      "Reads written (passing filters):    26,425,976 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,321,298,800 bp\n",
      "Total written (filtered):  1,309,473,729 bp (99.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 994679 times.\n",
      "356177 times, it overlapped the 5' end of a read\n",
      "638502 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3332\t412905.9\t0\t3332\n",
      "4\t289\t103226.5\t0\t289\n",
      "5\t386\t25806.6\t0\t386\n",
      "6\t230516\t6451.7\t0\t230516\n",
      "7\t12\t1612.9\t0\t12\n",
      "8\t11\t403.2\t0\t11\n",
      "9\t56\t100.8\t0\t36 20\n",
      "10\t121557\t25.2\t1\t10 121547\n",
      "11\t7\t6.3\t1\t0 7\n",
      "12\t1\t1.6\t1\t0 1\n",
      "13\t6\t0.4\t1\t0 6\n",
      "20\t1\t0.0\t2\t1\n",
      "21\t1\t0.0\t2\t1\n",
      "24\t1\t0.0\t2\t0 1\n",
      "25\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t499482\t412905.9\t0\t499482\n",
      "4\t112931\t103226.5\t0\t112931\n",
      "5\t20156\t25806.6\t0\t20156\n",
      "6\t3409\t6451.7\t0\t3409\n",
      "7\t403\t1612.9\t0\t403\n",
      "8\t56\t403.2\t0\t56\n",
      "9\t419\t100.8\t0\t10 409\n",
      "10\t1205\t25.2\t1\t3 1202\n",
      "11\t256\t6.3\t1\t13 243\n",
      "12\t54\t1.6\t1\t3 51\n",
      "13\t128\t0.4\t1\t3 125\n",
      "14\t2\t0.1\t1\t0 2\n",
      "28\t1\t0.0\t2\t0 0 0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 378875 times.\n",
      "378875 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t330816\t412905.9\t0\t330816\n",
      "4\t693\t103226.5\t0\t693\n",
      "5\t571\t25806.6\t0\t571\n",
      "6\t235\t6451.7\t0\t235\n",
      "7\t6\t1612.9\t0\t6\n",
      "8\t4\t403.2\t0\t4\n",
      "9\t16\t100.8\t0\t1 15\n",
      "10\t109\t25.2\t1\t13 96\n",
      "11\t46423\t6.3\t1\t0 46423\n",
      "12\t2\t1.6\t1\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 666403 times.\n",
      "666403 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t665335\t412905.9\t0\t665335\n",
      "4\t877\t103226.5\t0\t877\n",
      "5\t171\t25806.6\t0\t171\n",
      "6\t10\t6451.7\t0\t10\n",
      "7\t1\t1612.9\t0\t1\n",
      "9\t2\t100.8\t0\t0 2\n",
      "13\t5\t0.4\t1\t0 5\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 195678 times.\n",
      "179531 times, it overlapped the 5' end of a read\n",
      "16147 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t178614\t412905.9\t0\t178614\n",
      "4\t608\t103226.5\t0\t608\n",
      "5\t235\t25806.6\t0\t235\n",
      "6\t62\t6451.7\t0\t62\n",
      "11\t4\t6.3\t1\t0 4\n",
      "12\t4\t1.6\t1\t0 4\n",
      "13\t3\t0.4\t1\t3\n",
      "24\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t5\t0.1\t1\t3 2\n",
      "17\t3\t0.0\t1\t0 3\n",
      "19\t1\t0.0\t1\t1\n",
      "20\t2\t0.0\t2\t1 0 1\n",
      "28\t1\t0.0\t2\t1\n",
      "32\t57\t0.0\t3\t53 4\n",
      "33\t15368\t0.0\t3\t14432 860 69 7\n",
      "34\t616\t0.0\t3\t574 36 3 3\n",
      "35\t37\t0.0\t3\t30 5 2\n",
      "36\t16\t0.0\t3\t14 1 1\n",
      "37\t15\t0.0\t3\t13 2\n",
      "38\t7\t0.0\t3\t6 1\n",
      "39\t4\t0.0\t3\t4\n",
      "40\t1\t0.0\t3\t1\n",
      "41\t9\t0.0\t3\t7 2\n",
      "42\t5\t0.0\t3\t2 3\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 491893 times.\n",
      "491893 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2025\t412905.9\t0\t2025\n",
      "4\t523\t103226.5\t0\t523\n",
      "5\t478418\t25806.6\t0\t478418\n",
      "6\t3\t6451.7\t0\t3\n",
      "7\t2\t1612.9\t0\t2\n",
      "8\t6\t403.2\t0\t6\n",
      "10\t102\t25.2\t1\t0 102\n",
      "11\t30\t6.3\t1\t6 24\n",
      "12\t10783\t1.6\t1\t0 10783\n",
      "13\t1\t0.4\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994960_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994960_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 2155.68 s (59 us/read; 1.01 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              36,240,950\n",
      "Reads with adapters:                 3,776,991 (10.4%)\n",
      "Reads written (passing filters):    36,240,950 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,812,047,500 bp\n",
      "Total written (filtered):  1,793,681,116 bp (99.0%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 1339178 times.\n",
      "452333 times, it overlapped the 5' end of a read\n",
      "886845 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2513\t566264.8\t0\t2513\n",
      "4\t241\t141566.2\t0\t241\n",
      "5\t663\t35391.6\t0\t663\n",
      "6\t315600\t8847.9\t0\t315600\n",
      "7\t10\t2212.0\t0\t10\n",
      "8\t8\t553.0\t0\t8\n",
      "9\t46\t138.2\t0\t39 7\n",
      "10\t133171\t34.6\t1\t2 133169\n",
      "11\t11\t8.6\t1\t0 11\n",
      "13\t1\t0.5\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "20\t8\t0.0\t2\t7 1\n",
      "23\t1\t0.0\t2\t0 1\n",
      "24\t59\t0.0\t2\t2 57\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t697765\t566264.8\t0\t697765\n",
      "4\t155605\t141566.2\t0\t155605\n",
      "5\t25406\t35391.6\t0\t25406\n",
      "6\t4891\t8847.9\t0\t4891\n",
      "7\t521\t2212.0\t0\t521\n",
      "8\t98\t553.0\t0\t98\n",
      "9\t567\t138.2\t0\t15 552\n",
      "10\t1360\t34.6\t1\t15 1345\n",
      "11\t415\t8.6\t1\t20 395\n",
      "12\t89\t2.2\t1\t9 80\n",
      "13\t116\t0.5\t1\t5 111\n",
      "14\t11\t0.1\t1\t0 11\n",
      "15\t1\t0.0\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 480726 times.\n",
      "480722 times, it overlapped the 5' end of a read\n",
      "4 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t430439\t566264.8\t0\t430439\n",
      "4\t589\t141566.2\t0\t589\n",
      "5\t184\t35391.6\t0\t184\n",
      "6\t471\t8847.9\t0\t471\n",
      "7\t7\t2212.0\t0\t7\n",
      "9\t24\t138.2\t0\t0 24\n",
      "10\t137\t34.6\t1\t14 123\n",
      "11\t48868\t8.6\t1\t0 48868\n",
      "12\t3\t2.2\t1\t0 3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t4\t0.0\t1\t0 4\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 962859 times.\n",
      "962859 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t961971\t566264.8\t0\t961971\n",
      "4\t556\t141566.2\t0\t556\n",
      "5\t123\t35391.6\t0\t123\n",
      "6\t203\t8847.9\t0\t203\n",
      "13\t6\t0.5\t1\t0 6\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 346314 times.\n",
      "242159 times, it overlapped the 5' end of a read\n",
      "104155 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t241025\t566264.8\t0\t241025\n",
      "4\t304\t141566.2\t0\t304\n",
      "5\t306\t35391.6\t0\t306\n",
      "6\t19\t8847.9\t0\t19\n",
      "7\t2\t2212.0\t0\t2\n",
      "8\t160\t553.0\t0\t160\n",
      "9\t3\t138.2\t0\t0 3\n",
      "10\t1\t34.6\t1\t1\n",
      "11\t2\t8.6\t1\t0 2\n",
      "12\t73\t2.2\t1\t43 30\n",
      "13\t184\t0.5\t1\t181 3\n",
      "14\t1\t0.1\t1\t0 1\n",
      "16\t2\t0.0\t1\t0 2\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t5\t0.0\t1\t5\n",
      "21\t2\t0.0\t2\t1 0 1\n",
      "22\t48\t0.0\t2\t0 0 48\n",
      "23\t1\t0.0\t2\t0 1\n",
      "24\t14\t0.0\t2\t0 0 14\n",
      "28\t2\t0.0\t2\t2\n",
      "30\t3\t0.0\t3\t0 2 1\n",
      "33\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.5\t1\t0 1\n",
      "14\t9\t0.1\t1\t3 6\n",
      "15\t6\t0.0\t1\t3 3\n",
      "16\t5\t0.0\t1\t5\n",
      "17\t2\t0.0\t1\t2\n",
      "18\t2\t0.0\t1\t2\n",
      "19\t1\t0.0\t1\t1\n",
      "20\t5\t0.0\t2\t3 1 1\n",
      "22\t3\t0.0\t2\t3\n",
      "23\t2\t0.0\t2\t2\n",
      "24\t2\t0.0\t2\t2\n",
      "27\t2\t0.0\t2\t1 0 1\n",
      "28\t2\t0.0\t2\t2\n",
      "29\t3\t0.0\t2\t1 2\n",
      "30\t3\t0.0\t3\t2 0 1\n",
      "31\t13\t0.0\t3\t7 5 1\n",
      "32\t376\t0.0\t3\t352 22 1 1\n",
      "33\t94516\t0.0\t3\t90214 3932 295 75\n",
      "34\t8397\t0.0\t3\t8020 337 33 7\n",
      "35\t282\t0.0\t3\t254 15 10 3\n",
      "36\t103\t0.0\t3\t92 4 4 3\n",
      "37\t125\t0.0\t3\t119 3 1 2\n",
      "38\t38\t0.0\t3\t36 1 0 1\n",
      "39\t39\t0.0\t3\t37 1 0 1\n",
      "40\t22\t0.0\t3\t20 1 1\n",
      "41\t106\t0.0\t3\t94 10 2\n",
      "42\t55\t0.0\t3\t9 39 7\n",
      "43\t3\t0.0\t3\t0 0 3\n",
      "44\t9\t0.0\t3\t2 4 3\n",
      "45\t19\t0.0\t3\t3 13 2 1\n",
      "46\t2\t0.0\t3\t0 1 0 1\n",
      "47\t1\t0.0\t3\t0 0 0 1\n",
      "48\t1\t0.0\t3\t1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 647914 times.\n",
      "647914 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1360\t566264.8\t0\t1360\n",
      "4\t467\t141566.2\t0\t467\n",
      "5\t634532\t35391.6\t0\t634532\n",
      "6\t1\t8847.9\t0\t1\n",
      "8\t3\t553.0\t0\t3\n",
      "10\t27\t34.6\t1\t0 27\n",
      "11\t48\t8.6\t1\t5 43\n",
      "12\t11472\t2.2\t1\t0 11472\n",
      "13\t4\t0.5\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994963_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994963_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 956.57 s (59 us/read; 1.01 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,170,457\n",
      "Reads with adapters:                 1,657,222 (10.2%)\n",
      "Reads written (passing filters):    16,170,457 (100.0%)\n",
      "\n",
      "Total basepairs processed:   808,522,850 bp\n",
      "Total written (filtered):    801,394,787 bp (99.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 600547 times.\n",
      "217403 times, it overlapped the 5' end of a read\n",
      "383144 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2102\t252663.4\t0\t2102\n",
      "4\t200\t63165.8\t0\t200\n",
      "5\t192\t15791.5\t0\t192\n",
      "6\t142139\t3947.9\t0\t142139\n",
      "7\t8\t987.0\t0\t8\n",
      "8\t4\t246.7\t0\t4\n",
      "9\t60\t61.7\t0\t51 9\n",
      "10\t72683\t15.4\t1\t7 72676\n",
      "11\t9\t3.9\t1\t0 9\n",
      "13\t1\t0.2\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "21\t1\t0.0\t2\t1\n",
      "25\t3\t0.0\t2\t0 2 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t299254\t252663.4\t0\t299254\n",
      "4\t69001\t63165.8\t0\t69001\n",
      "5\t11342\t15791.5\t0\t11342\n",
      "6\t1947\t3947.9\t0\t1947\n",
      "7\t249\t987.0\t0\t249\n",
      "8\t27\t246.7\t0\t27\n",
      "9\t283\t61.7\t0\t3 280\n",
      "10\t715\t15.4\t1\t1 714\n",
      "11\t201\t3.9\t1\t17 184\n",
      "12\t37\t1.0\t1\t1 36\n",
      "13\t86\t0.2\t1\t1 85\n",
      "14\t2\t0.1\t1\t0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 230417 times.\n",
      "230415 times, it overlapped the 5' end of a read\n",
      "2 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t201770\t252663.4\t0\t201770\n",
      "4\t480\t63165.8\t0\t480\n",
      "5\t355\t15791.5\t0\t355\n",
      "6\t166\t3947.9\t0\t166\n",
      "7\t5\t987.0\t0\t5\n",
      "9\t13\t61.7\t0\t1 12\n",
      "10\t90\t15.4\t1\t24 66\n",
      "11\t27529\t3.9\t1\t0 27529\n",
      "12\t7\t1.0\t1\t0 7\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t1\t0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 407924 times.\n",
      "407923 times, it overlapped the 5' end of a read\n",
      "1 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t407310\t252663.4\t0\t407310\n",
      "4\t504\t63165.8\t0\t504\n",
      "5\t97\t15791.5\t0\t97\n",
      "6\t5\t3947.9\t0\t5\n",
      "10\t1\t15.4\t1\t0 1\n",
      "13\t4\t0.2\t1\t0 4\n",
      "33\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t1\t0.0\t1\t1\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 118770 times.\n",
      "110531 times, it overlapped the 5' end of a read\n",
      "8239 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t109992\t252663.4\t0\t109992\n",
      "4\t348\t63165.8\t0\t348\n",
      "5\t141\t15791.5\t0\t141\n",
      "6\t44\t3947.9\t0\t44\n",
      "12\t4\t1.0\t1\t0 4\n",
      "13\t2\t0.2\t1\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t4\t0.1\t1\t2 2\n",
      "15\t2\t0.0\t1\t2\n",
      "18\t2\t0.0\t1\t2\n",
      "20\t1\t0.0\t2\t1\n",
      "29\t1\t0.0\t2\t0 1\n",
      "32\t36\t0.0\t3\t33 3\n",
      "33\t7833\t0.0\t3\t7341 437 47 8\n",
      "34\t304\t0.0\t3\t284 17 3\n",
      "35\t23\t0.0\t3\t20 3\n",
      "36\t8\t0.0\t3\t8\n",
      "37\t2\t0.0\t3\t2\n",
      "39\t4\t0.0\t3\t3 1\n",
      "40\t6\t0.0\t3\t6\n",
      "41\t8\t0.0\t3\t7 1\n",
      "42\t2\t0.0\t3\t1 1\n",
      "45\t3\t0.0\t3\t0 0 2 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 299564 times.\n",
      "299564 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1235\t252663.4\t0\t1235\n",
      "4\t314\t63165.8\t0\t314\n",
      "5\t291770\t15791.5\t0\t291770\n",
      "6\t4\t3947.9\t0\t4\n",
      "8\t4\t246.7\t0\t4\n",
      "10\t40\t15.4\t1\t0 40\n",
      "11\t24\t3.9\t1\t1 23\n",
      "12\t6173\t1.0\t1\t0 6173\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994964_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994964_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1419.25 s (59 us/read; 1.01 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              23,972,947\n",
      "Reads with adapters:                 2,465,203 (10.3%)\n",
      "Reads written (passing filters):    23,972,947 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,198,647,350 bp\n",
      "Total written (filtered):  1,187,004,939 bp (99.0%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 880661 times.\n",
      "295845 times, it overlapped the 5' end of a read\n",
      "584816 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1558\t374577.3\t0\t1558\n",
      "4\t173\t93644.3\t0\t173\n",
      "5\t433\t23411.1\t0\t433\n",
      "6\t208121\t5852.8\t0\t208121\n",
      "7\t1\t1463.2\t0\t1\n",
      "8\t14\t365.8\t0\t14\n",
      "9\t28\t91.4\t0\t24 4\n",
      "10\t85473\t22.9\t1\t0 85473\n",
      "11\t10\t5.7\t1\t0 10\n",
      "13\t1\t0.4\t1\t0 1\n",
      "20\t3\t0.0\t2\t3\n",
      "24\t29\t0.0\t2\t1 27 1\n",
      "25\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t462885\t374577.3\t0\t462885\n",
      "4\t100401\t93644.3\t0\t100401\n",
      "5\t16466\t23411.1\t0\t16466\n",
      "6\t3105\t5852.8\t0\t3105\n",
      "7\t309\t1463.2\t0\t309\n",
      "8\t46\t365.8\t0\t46\n",
      "9\t352\t91.4\t0\t10 342\n",
      "10\t845\t22.9\t1\t8 837\n",
      "11\t281\t5.7\t1\t10 271\n",
      "12\t56\t1.4\t1\t5 51\n",
      "13\t65\t0.4\t1\t3 62\n",
      "14\t4\t0.1\t1\t0 4\n",
      "15\t1\t0.0\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 312409 times.\n",
      "312407 times, it overlapped the 5' end of a read\n",
      "2 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t280626\t374577.3\t0\t280626\n",
      "4\t418\t93644.3\t0\t418\n",
      "5\t108\t23411.1\t0\t108\n",
      "6\t286\t5852.8\t0\t286\n",
      "7\t15\t1463.2\t0\t15\n",
      "9\t19\t91.4\t0\t0 19\n",
      "10\t83\t22.9\t1\t7 76\n",
      "11\t30850\t5.7\t1\t0 30850\n",
      "12\t2\t1.4\t1\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t1\t0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 630548 times.\n",
      "630548 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t629902\t374577.3\t0\t629902\n",
      "4\t394\t93644.3\t0\t394\n",
      "5\t87\t23411.1\t0\t87\n",
      "6\t160\t5852.8\t0\t160\n",
      "13\t5\t0.4\t1\t0 5\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 218449 times.\n",
      "161574 times, it overlapped the 5' end of a read\n",
      "56875 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t160294\t374577.3\t0\t160294\n",
      "4\t221\t93644.3\t0\t221\n",
      "5\t197\t23411.1\t0\t197\n",
      "6\t15\t5852.8\t0\t15\n",
      "8\t78\t365.8\t0\t78\n",
      "9\t1\t91.4\t0\t0 1\n",
      "10\t1\t22.9\t1\t0 1\n",
      "11\t1\t5.7\t1\t0 1\n",
      "12\t36\t1.4\t1\t19 17\n",
      "13\t627\t0.4\t1\t625 2\n",
      "14\t1\t0.1\t1\t0 1\n",
      "16\t2\t0.0\t1\t0 2\n",
      "17\t8\t0.0\t1\t8\n",
      "18\t6\t0.0\t1\t0 6\n",
      "19\t49\t0.0\t1\t45 4\n",
      "20\t1\t0.0\t2\t0 1\n",
      "21\t3\t0.0\t2\t1 1 1\n",
      "22\t24\t0.0\t2\t1 0 23\n",
      "24\t1\t0.0\t2\t0 0 1\n",
      "25\t2\t0.0\t2\t2\n",
      "28\t2\t0.0\t2\t2\n",
      "30\t2\t0.0\t3\t1 1\n",
      "31\t1\t0.0\t3\t0 1\n",
      "32\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.4\t1\t0 1\n",
      "14\t7\t0.1\t1\t4 3\n",
      "15\t1\t0.0\t1\t1\n",
      "16\t1\t0.0\t1\t1\n",
      "17\t2\t0.0\t1\t1 1\n",
      "18\t3\t0.0\t1\t2 1\n",
      "20\t2\t0.0\t2\t1 0 1\n",
      "21\t1\t0.0\t2\t1\n",
      "26\t2\t0.0\t2\t2\n",
      "28\t1\t0.0\t2\t1\n",
      "29\t1\t0.0\t2\t1\n",
      "31\t3\t0.0\t3\t1 2\n",
      "32\t290\t0.0\t3\t265 23 1 1\n",
      "33\t50789\t0.0\t3\t48416 2139 190 44\n",
      "34\t5294\t0.0\t3\t5051 231 10 2\n",
      "35\t185\t0.0\t3\t164 12 8 1\n",
      "36\t56\t0.0\t3\t50 2 2 2\n",
      "37\t74\t0.0\t3\t72 2\n",
      "38\t12\t0.0\t3\t10 2\n",
      "39\t18\t0.0\t3\t16 2\n",
      "40\t13\t0.0\t3\t13\n",
      "41\t61\t0.0\t3\t57 3 1\n",
      "42\t36\t0.0\t3\t11 22 2 1\n",
      "44\t5\t0.0\t3\t2 3\n",
      "45\t12\t0.0\t3\t1 7 4\n",
      "46\t2\t0.0\t3\t1 1\n",
      "47\t2\t0.0\t3\t1 1\n",
      "49\t1\t0.0\t3\t0 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 423136 times.\n",
      "423136 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t838\t374577.3\t0\t838\n",
      "4\t268\t93644.3\t0\t268\n",
      "5\t414959\t23411.1\t0\t414959\n",
      "6\t1\t5852.8\t0\t1\n",
      "8\t1\t365.8\t0\t1\n",
      "9\t1\t91.4\t0\t0 1\n",
      "10\t19\t22.9\t1\t0 19\n",
      "11\t23\t5.7\t1\t3 20\n",
      "12\t7025\t1.4\t1\t0 7025\n",
      "13\t1\t0.4\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994967_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994967_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1261.60 s (59 us/read; 1.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,546,857\n",
      "Reads with adapters:                 2,187,135 (10.2%)\n",
      "Reads written (passing filters):    21,546,857 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,077,342,850 bp\n",
      "Total written (filtered):  1,067,997,564 bp (99.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 795458 times.\n",
      "284501 times, it overlapped the 5' end of a read\n",
      "510957 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2735\t336669.6\t0\t2735\n",
      "4\t263\t84167.4\t0\t263\n",
      "5\t303\t21041.9\t0\t303\n",
      "6\t192032\t5260.5\t0\t192032\n",
      "7\t4\t1315.1\t0\t4\n",
      "8\t9\t328.8\t0\t9\n",
      "9\t44\t82.2\t0\t30 14\n",
      "10\t89103\t20.5\t1\t10 89093\n",
      "11\t5\t5.1\t1\t0 5\n",
      "20\t2\t0.0\t2\t1 1\n",
      "23\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t398145\t336669.6\t0\t398145\n",
      "4\t94101\t84167.4\t0\t94101\n",
      "5\t12818\t21041.9\t0\t12818\n",
      "6\t2028\t5260.5\t0\t2028\n",
      "7\t285\t1315.1\t0\t285\n",
      "8\t25\t328.8\t0\t25\n",
      "9\t365\t82.2\t0\t10 355\n",
      "10\t2783\t20.5\t1\t1 2782\n",
      "11\t284\t5.1\t1\t13 271\n",
      "12\t49\t1.3\t1\t0 49\n",
      "13\t67\t0.3\t1\t0 67\n",
      "14\t5\t0.1\t1\t0 5\n",
      "28\t2\t0.0\t2\t0 0 0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 303178 times.\n",
      "303173 times, it overlapped the 5' end of a read\n",
      "5 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t265253\t336669.6\t0\t265253\n",
      "4\t583\t84167.4\t0\t583\n",
      "5\t459\t21041.9\t0\t459\n",
      "6\t228\t5260.5\t0\t228\n",
      "7\t9\t1315.1\t0\t9\n",
      "8\t5\t328.8\t0\t5\n",
      "9\t17\t82.2\t0\t3 14\n",
      "10\t120\t20.5\t1\t18 102\n",
      "11\t36495\t5.1\t1\t0 36495\n",
      "12\t4\t1.3\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t5\t0.0\t1\t0 5\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 536797 times.\n",
      "536797 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t535914\t336669.6\t0\t535914\n",
      "4\t707\t84167.4\t0\t707\n",
      "5\t151\t21041.9\t0\t151\n",
      "6\t18\t5260.5\t0\t18\n",
      "7\t4\t1315.1\t0\t4\n",
      "11\t1\t5.1\t1\t0 1\n",
      "33\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 158052 times.\n",
      "148156 times, it overlapped the 5' end of a read\n",
      "9896 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t147424\t336669.6\t0\t147424\n",
      "4\t473\t84167.4\t0\t473\n",
      "5\t192\t21041.9\t0\t192\n",
      "6\t58\t5260.5\t0\t58\n",
      "9\t1\t82.2\t0\t0 1\n",
      "10\t2\t20.5\t1\t0 2\n",
      "12\t5\t1.3\t1\t0 5\n",
      "22\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.3\t1\t0 1\n",
      "14\t1\t0.1\t1\t1\n",
      "15\t1\t0.0\t1\t0 1\n",
      "16\t2\t0.0\t1\t1 1\n",
      "17\t1\t0.0\t1\t1\n",
      "31\t1\t0.0\t3\t0 0 1\n",
      "32\t23\t0.0\t3\t21 2\n",
      "33\t9425\t0.0\t3\t8856 504 55 10\n",
      "34\t389\t0.0\t3\t363 21 4 1\n",
      "35\t17\t0.0\t3\t14 2 1\n",
      "36\t2\t0.0\t3\t1 0 1\n",
      "37\t8\t0.0\t3\t8\n",
      "38\t3\t0.0\t3\t3\n",
      "39\t4\t0.0\t3\t4\n",
      "40\t5\t0.0\t3\t5\n",
      "41\t8\t0.0\t3\t6 2\n",
      "42\t5\t0.0\t3\t1 4\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 393650 times.\n",
      "393650 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1612\t336669.6\t0\t1612\n",
      "4\t444\t84167.4\t0\t444\n",
      "5\t384141\t21041.9\t0\t384141\n",
      "6\t4\t5260.5\t0\t4\n",
      "10\t59\t20.5\t1\t0 59\n",
      "11\t29\t5.1\t1\t6 23\n",
      "12\t7360\t1.3\t1\t0 7360\n",
      "13\t1\t0.3\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994968_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994968_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 2024.58 s (60 us/read; 1.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              33,734,733\n",
      "Reads with adapters:                 3,488,453 (10.3%)\n",
      "Reads written (passing filters):    33,734,733 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,686,736,650 bp\n",
      "Total written (filtered):  1,670,644,198 bp (99.0%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 1264785 times.\n",
      "418425 times, it overlapped the 5' end of a read\n",
      "846360 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2317\t527105.2\t0\t2317\n",
      "4\t250\t131776.3\t0\t250\n",
      "5\t566\t32944.1\t0\t566\n",
      "6\t294083\t8236.0\t0\t294083\n",
      "7\t5\t2059.0\t0\t5\n",
      "8\t6\t514.8\t0\t6\n",
      "9\t40\t128.7\t0\t33 7\n",
      "10\t121096\t32.2\t1\t3 121093\n",
      "11\t12\t8.0\t1\t0 12\n",
      "13\t1\t0.5\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "20\t10\t0.0\t2\t10\n",
      "24\t38\t0.0\t2\t1 36 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t660446\t527105.2\t0\t660446\n",
      "4\t156943\t131776.3\t0\t156943\n",
      "5\t20527\t32944.1\t0\t20527\n",
      "6\t4093\t8236.0\t0\t4093\n",
      "7\t496\t2059.0\t0\t496\n",
      "8\t84\t514.8\t0\t84\n",
      "9\t450\t128.7\t0\t18 432\n",
      "10\t2657\t32.2\t1\t12 2645\n",
      "11\t413\t8.0\t1\t19 394\n",
      "12\t94\t2.0\t1\t7 87\n",
      "13\t131\t0.5\t1\t8 123\n",
      "14\t26\t0.1\t1\t0 26\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 443989 times.\n",
      "443985 times, it overlapped the 5' end of a read\n",
      "4 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t397802\t527105.2\t0\t397802\n",
      "4\t606\t131776.3\t0\t606\n",
      "5\t154\t32944.1\t0\t154\n",
      "6\t436\t8236.0\t0\t436\n",
      "7\t19\t2059.0\t0\t19\n",
      "9\t22\t128.7\t0\t0 22\n",
      "10\t111\t32.2\t1\t17 94\n",
      "11\t44835\t8.0\t1\t0 44835\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.1\t1\t0 1\n",
      "15\t3\t0.0\t1\t0 3\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 890752 times.\n",
      "890752 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t889875\t527105.2\t0\t889875\n",
      "4\t558\t131776.3\t0\t558\n",
      "5\t102\t32944.1\t0\t102\n",
      "6\t207\t8236.0\t0\t207\n",
      "13\t9\t0.5\t1\t0 9\n",
      "33\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 292056 times.\n",
      "225048 times, it overlapped the 5' end of a read\n",
      "67008 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t223468\t527105.2\t0\t223468\n",
      "4\t297\t131776.3\t0\t297\n",
      "5\t298\t32944.1\t0\t298\n",
      "6\t27\t8236.0\t0\t27\n",
      "7\t1\t2059.0\t0\t1\n",
      "8\t128\t514.8\t0\t128\n",
      "9\t3\t128.7\t0\t0 3\n",
      "10\t4\t32.2\t1\t0 4\n",
      "11\t1\t8.0\t1\t0 1\n",
      "12\t50\t2.0\t1\t26 24\n",
      "13\t664\t0.5\t1\t659 5\n",
      "16\t1\t0.0\t1\t1\n",
      "17\t6\t0.0\t1\t5 1\n",
      "18\t4\t0.0\t1\t0 4\n",
      "19\t62\t0.0\t1\t58 4\n",
      "20\t1\t0.0\t2\t0 1\n",
      "21\t2\t0.0\t2\t0 2\n",
      "22\t19\t0.0\t2\t0 0 19\n",
      "23\t1\t0.0\t2\t0 1\n",
      "24\t7\t0.0\t2\t0 1 6\n",
      "27\t1\t0.0\t2\t0 0 1\n",
      "30\t2\t0.0\t3\t1 1\n",
      "33\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.5\t1\t0 1\n",
      "14\t6\t0.1\t1\t1 5\n",
      "15\t3\t0.0\t1\t3\n",
      "16\t6\t0.0\t1\t4 2\n",
      "17\t3\t0.0\t1\t3\n",
      "18\t3\t0.0\t1\t3\n",
      "19\t1\t0.0\t1\t1\n",
      "21\t2\t0.0\t2\t2\n",
      "22\t1\t0.0\t2\t1\n",
      "23\t1\t0.0\t2\t0 1\n",
      "25\t1\t0.0\t2\t1\n",
      "26\t2\t0.0\t2\t1 0 1\n",
      "28\t2\t0.0\t2\t2\n",
      "29\t1\t0.0\t2\t1\n",
      "30\t1\t0.0\t3\t0 1\n",
      "31\t8\t0.0\t3\t3 5\n",
      "32\t275\t0.0\t3\t259 14 2\n",
      "33\t60249\t0.0\t3\t57340 2643 216 50\n",
      "34\t5918\t0.0\t3\t5641 248 22 7\n",
      "35\t180\t0.0\t3\t161 15 3 1\n",
      "36\t70\t0.0\t3\t59 3 8\n",
      "37\t81\t0.0\t3\t75 6\n",
      "38\t16\t0.0\t3\t16\n",
      "39\t20\t0.0\t3\t19 1\n",
      "40\t10\t0.0\t3\t10\n",
      "41\t81\t0.0\t3\t65 14 1 1\n",
      "42\t36\t0.0\t3\t6 25 5\n",
      "43\t2\t0.0\t3\t0 1 1\n",
      "44\t9\t0.0\t3\t2 4 3\n",
      "45\t12\t0.0\t3\t5 4 3\n",
      "46\t6\t0.0\t3\t2 2 0 2\n",
      "48\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 596871 times.\n",
      "596871 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1231\t527105.2\t0\t1231\n",
      "4\t434\t131776.3\t0\t434\n",
      "5\t584917\t32944.1\t0\t584917\n",
      "8\t1\t514.8\t0\t1\n",
      "10\t31\t32.2\t1\t0 31\n",
      "11\t31\t8.0\t1\t6 25\n",
      "12\t10223\t2.0\t1\t0 10223\n",
      "13\t3\t0.5\t1\t0 3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994971_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994971_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1082.33 s (59 us/read; 1.01 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,195,992\n",
      "Reads with adapters:                 1,904,225 (10.5%)\n",
      "Reads written (passing filters):    18,195,992 (100.0%)\n",
      "\n",
      "Total basepairs processed:   909,799,600 bp\n",
      "Total written (filtered):    901,415,712 bp (99.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 706410 times.\n",
      "248823 times, it overlapped the 5' end of a read\n",
      "457587 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2422\t284312.4\t0\t2422\n",
      "4\t221\t71078.1\t0\t221\n",
      "5\t260\t17769.5\t0\t260\n",
      "6\t155309\t4442.4\t0\t155309\n",
      "7\t5\t1110.6\t0\t5\n",
      "8\t5\t277.6\t0\t5\n",
      "9\t34\t69.4\t0\t25 9\n",
      "10\t90541\t17.4\t1\t7 90534\n",
      "11\t17\t4.3\t1\t0 17\n",
      "12\t1\t1.1\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "21\t2\t0.0\t2\t2\n",
      "24\t1\t0.0\t2\t0 1\n",
      "25\t3\t0.0\t2\t0 2 1\n",
      "30\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t360974\t284312.4\t0\t360974\n",
      "4\t80988\t71078.1\t0\t80988\n",
      "5\t10598\t17769.5\t0\t10598\n",
      "6\t1777\t4442.4\t0\t1777\n",
      "7\t247\t1110.6\t0\t247\n",
      "8\t28\t277.6\t0\t28\n",
      "9\t356\t69.4\t0\t7 349\n",
      "10\t2184\t17.4\t1\t1 2183\n",
      "11\t343\t4.3\t1\t13 330\n",
      "12\t37\t1.1\t1\t0 37\n",
      "13\t51\t0.3\t1\t1 50\n",
      "14\t2\t0.1\t1\t0 2\n",
      "28\t2\t0.0\t2\t0 0 0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 258572 times.\n",
      "258571 times, it overlapped the 5' end of a read\n",
      "1 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t225194\t284312.4\t0\t225194\n",
      "4\t479\t71078.1\t0\t479\n",
      "5\t399\t17769.5\t0\t399\n",
      "6\t145\t4442.4\t0\t145\n",
      "7\t4\t1110.6\t0\t4\n",
      "9\t18\t69.4\t0\t0 18\n",
      "10\t70\t17.4\t1\t6 64\n",
      "11\t32262\t4.3\t1\t0 32262\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.1\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 466894 times.\n",
      "466894 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t466146\t284312.4\t0\t466146\n",
      "4\t604\t71078.1\t0\t604\n",
      "5\t127\t17769.5\t0\t127\n",
      "6\t9\t4442.4\t0\t9\n",
      "7\t2\t1110.6\t0\t2\n",
      "9\t3\t69.4\t0\t3\n",
      "10\t1\t17.4\t1\t0 1\n",
      "13\t2\t0.3\t1\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 135491 times.\n",
      "120451 times, it overlapped the 5' end of a read\n",
      "15040 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t119825\t284312.4\t0\t119825\n",
      "4\t408\t71078.1\t0\t408\n",
      "5\t161\t17769.5\t0\t161\n",
      "6\t41\t4442.4\t0\t41\n",
      "10\t4\t17.4\t1\t0 4\n",
      "12\t8\t1.1\t1\t0 8\n",
      "13\t4\t0.3\t1\t2 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t4\t0.1\t1\t2 2\n",
      "17\t1\t0.0\t1\t1\n",
      "29\t1\t0.0\t2\t0 0 1\n",
      "31\t1\t0.0\t3\t0 1\n",
      "32\t49\t0.0\t3\t42 6 1\n",
      "33\t14211\t0.0\t3\t13320 810 67 14\n",
      "34\t678\t0.0\t3\t623 49 5 1\n",
      "35\t41\t0.0\t3\t36 3 1 1\n",
      "36\t10\t0.0\t3\t10\n",
      "37\t15\t0.0\t3\t14 1\n",
      "38\t4\t0.0\t3\t4\n",
      "39\t6\t0.0\t3\t6\n",
      "40\t2\t0.0\t3\t2\n",
      "41\t5\t0.0\t3\t3 2\n",
      "42\t5\t0.0\t3\t5\n",
      "44\t2\t0.0\t3\t2\n",
      "45\t5\t0.0\t3\t1 0 3 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 336858 times.\n",
      "336858 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1445\t284312.4\t0\t1445\n",
      "4\t376\t71078.1\t0\t376\n",
      "5\t327077\t17769.5\t0\t327077\n",
      "6\t5\t4442.4\t0\t5\n",
      "7\t2\t1110.6\t0\t2\n",
      "8\t1\t277.6\t0\t1\n",
      "10\t60\t17.4\t1\t0 60\n",
      "11\t34\t4.3\t1\t4 30\n",
      "12\t7857\t1.1\t1\t0 7857\n",
      "13\t1\t0.3\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/ERR1994972_adaptor.fastq /projects_eg/projects/simone/Duncan_Mata/fastq_files/ERR1994972_1.fastq\n",
      "Running on 1 core\n",
      "Trimming 5 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1399.38 s (60 us/read; 1.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              23,231,253\n",
      "Reads with adapters:                 2,381,476 (10.3%)\n",
      "Reads written (passing filters):    23,231,253 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,161,562,650 bp\n",
      "Total written (filtered):  1,150,991,799 bp (99.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 864711 times.\n",
      "283172 times, it overlapped the 5' end of a read\n",
      "581539 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1646\t362988.3\t0\t1646\n",
      "4\t153\t90747.1\t0\t153\n",
      "5\t427\t22686.8\t0\t427\n",
      "6\t198881\t5671.7\t0\t198881\n",
      "7\t7\t1417.9\t0\t7\n",
      "8\t4\t354.5\t0\t4\n",
      "9\t27\t88.6\t0\t23 4\n",
      "10\t81988\t22.2\t1\t3 81985\n",
      "11\t8\t5.5\t1\t0 8\n",
      "12\t1\t1.4\t1\t0 1\n",
      "13\t1\t0.3\t1\t0 1\n",
      "20\t6\t0.0\t2\t6\n",
      "24\t23\t0.0\t2\t0 22 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t453049\t362988.3\t0\t453049\n",
      "4\t109026\t90747.1\t0\t109026\n",
      "5\t14143\t22686.8\t0\t14143\n",
      "6\t2489\t5671.7\t0\t2489\n",
      "7\t355\t1417.9\t0\t355\n",
      "8\t51\t354.5\t0\t51\n",
      "9\t306\t88.6\t0\t14 292\n",
      "10\t1629\t22.2\t1\t4 1625\n",
      "11\t289\t5.5\t1\t12 277\n",
      "12\t85\t1.4\t1\t4 81\n",
      "13\t101\t0.3\t1\t6 95\n",
      "14\t14\t0.1\t1\t0 14\n",
      "15\t2\t0.0\t1\t0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 301768 times.\n",
      "301766 times, it overlapped the 5' end of a read\n",
      "2 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t269548\t362988.3\t0\t269548\n",
      "4\t411\t90747.1\t0\t411\n",
      "5\t130\t22686.8\t0\t130\n",
      "6\t284\t5671.7\t0\t284\n",
      "7\t5\t1417.9\t0\t5\n",
      "9\t19\t88.6\t0\t0 19\n",
      "10\t101\t22.2\t1\t15 86\n",
      "11\t31267\t5.5\t1\t0 31267\n",
      "12\t1\t1.4\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t1\t0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 621908 times.\n",
      "621908 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t621326\t362988.3\t0\t621326\n",
      "4\t373\t90747.1\t0\t373\n",
      "5\t76\t22686.8\t0\t76\n",
      "6\t128\t5671.7\t0\t128\n",
      "10\t1\t22.2\t1\t0 1\n",
      "13\t4\t0.3\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 184707 times.\n",
      "152854 times, it overlapped the 5' end of a read\n",
      "31853 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t151578\t362988.3\t0\t151578\n",
      "4\t193\t90747.1\t0\t193\n",
      "5\t194\t22686.8\t0\t194\n",
      "6\t16\t5671.7\t0\t16\n",
      "8\t86\t354.5\t0\t86\n",
      "9\t2\t88.6\t0\t1 1\n",
      "11\t1\t5.5\t1\t1\n",
      "12\t31\t1.4\t1\t21 10\n",
      "13\t643\t0.3\t1\t642 1\n",
      "14\t6\t0.1\t1\t1 5\n",
      "15\t1\t0.0\t1\t0 1\n",
      "16\t2\t0.0\t1\t2\n",
      "17\t7\t0.0\t1\t7\n",
      "18\t3\t0.0\t1\t0 3\n",
      "19\t55\t0.0\t1\t52 3\n",
      "20\t2\t0.0\t2\t0 1 1\n",
      "21\t2\t0.0\t2\t0 0 2\n",
      "22\t25\t0.0\t2\t0 1 24\n",
      "24\t6\t0.0\t2\t0 0 6\n",
      "28\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t4\t0.1\t1\t1 3\n",
      "15\t2\t0.0\t1\t2\n",
      "16\t7\t0.0\t1\t5 2\n",
      "17\t4\t0.0\t1\t3 1\n",
      "18\t1\t0.0\t1\t1\n",
      "19\t2\t0.0\t1\t2\n",
      "20\t2\t0.0\t2\t2\n",
      "22\t1\t0.0\t2\t0 1\n",
      "24\t2\t0.0\t2\t2\n",
      "25\t1\t0.0\t2\t1\n",
      "26\t2\t0.0\t2\t2\n",
      "27\t2\t0.0\t2\t2\n",
      "30\t1\t0.0\t3\t0 1\n",
      "31\t2\t0.0\t3\t0 1 1\n",
      "32\t120\t0.0\t3\t110 9 0 1\n",
      "33\t28582\t0.0\t3\t27076 1322 143 41\n",
      "34\t2834\t0.0\t3\t2708 111 14 1\n",
      "35\t103\t0.0\t3\t85 12 4 2\n",
      "36\t38\t0.0\t3\t34 1 2 1\n",
      "37\t32\t0.0\t3\t29 2 1\n",
      "38\t9\t0.0\t3\t9\n",
      "39\t8\t0.0\t3\t8\n",
      "40\t6\t0.0\t3\t6\n",
      "41\t37\t0.0\t3\t32 5\n",
      "42\t29\t0.0\t3\t1 26 2\n",
      "43\t1\t0.0\t3\t0 0 0 1\n",
      "44\t2\t0.0\t3\t1 1\n",
      "45\t15\t0.0\t3\t2 9 2 2\n",
      "46\t4\t0.0\t3\t0 2 1 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 408382 times.\n",
      "408382 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t924\t362988.3\t0\t924\n",
      "4\t304\t90747.1\t0\t304\n",
      "5\t399564\t22686.8\t0\t399564\n",
      "6\t1\t5671.7\t0\t1\n",
      "8\t2\t354.5\t0\t2\n",
      "10\t22\t22.2\t1\t0 22\n",
      "11\t37\t5.5\t1\t7 30\n",
      "12\t7528\t1.4\t1\t0 7528\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "addition=\"_1\"\n",
    "label=\"_adaptor\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "cutadapt -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/Duncan_RNA_adaptors.fa -o $output_directory$accession_base$i$label$extention $path_fastq_files$accession_base$i$addition$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Second step: Remove PCR duplicate to generate a non-redundant dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-09T11:07:00.847492Z",
     "start_time": "2018-03-09T09:17:23.130122Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Input and filter stats:\n",
      "\tInput sequences: 26,425,976\n",
      "\tInput bases: 1,309,473,729\n",
      "\tInput mean length: 49.55\n",
      "\tGood sequences: 14,209,410 (53.77%)\n",
      "\tGood bases: 704,248,450\n",
      "\tGood mean length: 49.56\n",
      "\tBad sequences: 12,216,566 (46.23%)\n",
      "\tBad bases: 605,225,279\n",
      "\tBad mean length: 49.54\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 12216566\n",
      "Input and filter stats:\n",
      "\tInput sequences: 36,240,950\n",
      "\tInput bases: 1,793,681,116\n",
      "\tInput mean length: 49.49\n",
      "\tGood sequences: 21,865,162 (60.33%)\n",
      "\tGood bases: 1,083,737,143\n",
      "\tGood mean length: 49.56\n",
      "\tBad sequences: 14,375,788 (39.67%)\n",
      "\tBad bases: 709,943,973\n",
      "\tBad mean length: 49.38\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 14375788\n",
      "Input and filter stats:\n",
      "\tInput sequences: 16,170,457\n",
      "\tInput bases: 801,394,787\n",
      "\tInput mean length: 49.56\n",
      "\tGood sequences: 7,080,648 (43.79%)\n",
      "\tGood bases: 350,905,234\n",
      "\tGood mean length: 49.56\n",
      "\tBad sequences: 9,089,809 (56.21%)\n",
      "\tBad bases: 450,489,553\n",
      "\tBad mean length: 49.56\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 9089809\n",
      "Input and filter stats:\n",
      "\tInput sequences: 23,972,947\n",
      "\tInput bases: 1,187,004,939\n",
      "\tInput mean length: 49.51\n",
      "\tGood sequences: 14,881,527 (62.08%)\n",
      "\tGood bases: 737,593,208\n",
      "\tGood mean length: 49.56\n",
      "\tBad sequences: 9,091,420 (37.92%)\n",
      "\tBad bases: 449,411,731\n",
      "\tBad mean length: 49.43\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 9091420\n",
      "Input and filter stats:\n",
      "\tInput sequences: 21,546,857\n",
      "\tInput bases: 1,067,997,564\n",
      "\tInput mean length: 49.57\n",
      "\tGood sequences: 9,688,112 (44.96%)\n",
      "\tGood bases: 480,220,292\n",
      "\tGood mean length: 49.57\n",
      "\tBad sequences: 11,858,745 (55.04%)\n",
      "\tBad bases: 587,777,272\n",
      "\tBad mean length: 49.56\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 11858745\n",
      "Input and filter stats:\n",
      "\tInput sequences: 33,734,733\n",
      "\tInput bases: 1,670,644,198\n",
      "\tInput mean length: 49.52\n",
      "\tGood sequences: 21,387,910 (63.40%)\n",
      "\tGood bases: 1,060,167,128\n",
      "\tGood mean length: 49.57\n",
      "\tBad sequences: 12,346,823 (36.60%)\n",
      "\tBad bases: 610,477,070\n",
      "\tBad mean length: 49.44\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 12346823\n",
      "Input and filter stats:\n",
      "\tInput sequences: 18,195,992\n",
      "\tInput bases: 901,415,712\n",
      "\tInput mean length: 49.54\n",
      "\tGood sequences: 8,465,112 (46.52%)\n",
      "\tGood bases: 419,442,614\n",
      "\tGood mean length: 49.55\n",
      "\tBad sequences: 9,730,880 (53.48%)\n",
      "\tBad bases: 481,973,098\n",
      "\tBad mean length: 49.53\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 9730880\n",
      "Input and filter stats:\n",
      "\tInput sequences: 23,231,253\n",
      "\tInput bases: 1,150,991,799\n",
      "\tInput mean length: 49.54\n",
      "\tGood sequences: 15,550,945 (66.94%)\n",
      "\tGood bases: 770,868,634\n",
      "\tGood mean length: 49.57\n",
      "\tBad sequences: 7,680,308 (33.06%)\n",
      "\tBad bases: 380,123,165\n",
      "\tBad mean length: 49.49\n",
      "\tSequences filtered by specified parameters:\n",
      "\tderep: 7680308\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_adaptor/\"\n",
    "extention=\".fastq\"  # no extention for output\n",
    "label=\"_adaptor\"\n",
    "label2=\"_no_dup\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "perl /home/simone/Downloads/prinseq-lite-0.20.4/prinseq-lite.pl -out_good $output_directory$accession_base$i$label2 -fastq $path_fastq_files$accession_base$i$label$extention -derep 1 -no_qual_header\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Third step: Remove first nine random nucleotides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-09T15:25:46.273465Z",
     "start_time": "2018-03-09T15:01:08.373878Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994959_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994959_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 185.23 s (13 us/read; 4.60 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,209,410\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:             147,772 (1.0%)\n",
      "Reads written (passing filters):    14,061,638 (99.0%)\n",
      "\n",
      "Total basepairs processed:   704,248,450 bp\n",
      "Quality-trimmed:               5,873,882 bp (0.8%)\n",
      "Total written (filtered):    569,134,636 bp (80.8%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994960_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994960_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 282.68 s (13 us/read; 4.64 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,865,162\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:             247,823 (1.1%)\n",
      "Reads written (passing filters):    21,617,339 (98.9%)\n",
      "\n",
      "Total basepairs processed: 1,083,737,143 bp\n",
      "Quality-trimmed:              10,074,188 bp (0.9%)\n",
      "Total written (filtered):    873,399,556 bp (80.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994963_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994963_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 91.85 s (13 us/read; 4.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,080,648\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:              82,281 (1.2%)\n",
      "Reads written (passing filters):     6,998,367 (98.8%)\n",
      "\n",
      "Total basepairs processed:   350,905,234 bp\n",
      "Quality-trimmed:               3,212,918 bp (0.9%)\n",
      "Total written (filtered):    283,230,179 bp (80.7%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994964_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994964_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 193.22 s (13 us/read; 4.62 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,881,527\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:             176,177 (1.2%)\n",
      "Reads written (passing filters):    14,705,350 (98.8%)\n",
      "\n",
      "Total basepairs processed:   737,593,208 bp\n",
      "Quality-trimmed:               7,043,614 bp (1.0%)\n",
      "Total written (filtered):    594,165,047 bp (80.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994967_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994967_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 127.12 s (13 us/read; 4.57 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               9,688,112\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:             108,592 (1.1%)\n",
      "Reads written (passing filters):     9,579,520 (98.9%)\n",
      "\n",
      "Total basepairs processed:   480,220,292 bp\n",
      "Quality-trimmed:               4,284,039 bp (0.9%)\n",
      "Total written (filtered):    387,756,983 bp (80.7%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994968_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994968_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 283.17 s (13 us/read; 4.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,387,910\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:             234,401 (1.1%)\n",
      "Reads written (passing filters):    21,153,509 (98.9%)\n",
      "\n",
      "Total basepairs processed: 1,060,167,128 bp\n",
      "Quality-trimmed:               9,622,616 bp (0.9%)\n",
      "Total written (filtered):    854,749,287 bp (80.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994971_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994971_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 108.49 s (13 us/read; 4.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,465,112\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:              92,812 (1.1%)\n",
      "Reads written (passing filters):     8,372,300 (98.9%)\n",
      "\n",
      "Total basepairs processed:   419,442,614 bp\n",
      "Quality-trimmed:               3,602,995 bp (0.9%)\n",
      "Total written (filtered):    338,828,303 bp (80.8%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -u 9 -q 30 -m 26 -o /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/ERR1994972_trimmed.fastq /projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/ERR1994972_no_dup.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 201.77 s (13 us/read; 4.62 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              15,550,945\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads that were too short:             184,110 (1.2%)\n",
      "Reads written (passing filters):    15,366,835 (98.8%)\n",
      "\n",
      "Total basepairs processed:   770,868,634 bp\n",
      "Quality-trimmed:               7,473,075 bp (1.0%)\n",
      "Total written (filtered):    620,839,266 bp (80.5%)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_no_dup/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_no_dup\"\n",
    "label2=\"_trimmed\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "# remove first nine nucleotides\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "cutadapt -u 9 -q 30 -m 26 -o $output_directory$accession_base$i$label2$extention $path_fastq_files$accession_base$i$label$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fourth step: Reverse complement reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-12T10:23:10.215917Z",
     "start_time": "2018-03-12T10:08:10.839786Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load FASTX-Toolkit/0.0.14\n",
    "\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_trimmed/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "label2=\"_revcomp\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_revcomp/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "fastx_reverse_complement -i $path_fastq_files$accession_base$i$label$extention -o $output_directory$accession_base$i$label2$extention\n",
    "done\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-13T10:11:35.196500Z",
     "start_time": "2018-03-13T10:03:25.685271Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR1994959_revcomp.fastq\n",
      "Analysis complete for ERR1994960_revcomp.fastq\n",
      "Analysis complete for ERR1994963_revcomp.fastq\n",
      "Analysis complete for ERR1994964_revcomp.fastq\n",
      "Analysis complete for ERR1994967_revcomp.fastq\n",
      "Analysis complete for ERR1994968_revcomp.fastq\n",
      "Analysis complete for ERR1994971_revcomp.fastq\n",
      "Analysis complete for ERR1994972_revcomp.fastq\n",
      "Searching 16 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of ERR1994959_revcomp.fastq\n",
      "Approx 5% complete for ERR1994959_revcomp.fastq\n",
      "Approx 10% complete for ERR1994959_revcomp.fastq\n",
      "Approx 15% complete for ERR1994959_revcomp.fastq\n",
      "Approx 20% complete for ERR1994959_revcomp.fastq\n",
      "Approx 25% complete for ERR1994959_revcomp.fastq\n",
      "Approx 30% complete for ERR1994959_revcomp.fastq\n",
      "Approx 35% complete for ERR1994959_revcomp.fastq\n",
      "Approx 40% complete for ERR1994959_revcomp.fastq\n",
      "Approx 45% complete for ERR1994959_revcomp.fastq\n",
      "Approx 50% complete for ERR1994959_revcomp.fastq\n",
      "Approx 55% complete for ERR1994959_revcomp.fastq\n",
      "Approx 60% complete for ERR1994959_revcomp.fastq\n",
      "Approx 65% complete for ERR1994959_revcomp.fastq\n",
      "Approx 70% complete for ERR1994959_revcomp.fastq\n",
      "Approx 75% complete for ERR1994959_revcomp.fastq\n",
      "Approx 80% complete for ERR1994959_revcomp.fastq\n",
      "Approx 85% complete for ERR1994959_revcomp.fastq\n",
      "Approx 90% complete for ERR1994959_revcomp.fastq\n",
      "Approx 95% complete for ERR1994959_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994960_revcomp.fastq\n",
      "Approx 5% complete for ERR1994960_revcomp.fastq\n",
      "Approx 10% complete for ERR1994960_revcomp.fastq\n",
      "Approx 15% complete for ERR1994960_revcomp.fastq\n",
      "Approx 20% complete for ERR1994960_revcomp.fastq\n",
      "Approx 25% complete for ERR1994960_revcomp.fastq\n",
      "Approx 30% complete for ERR1994960_revcomp.fastq\n",
      "Approx 35% complete for ERR1994960_revcomp.fastq\n",
      "Approx 40% complete for ERR1994960_revcomp.fastq\n",
      "Approx 45% complete for ERR1994960_revcomp.fastq\n",
      "Approx 50% complete for ERR1994960_revcomp.fastq\n",
      "Approx 55% complete for ERR1994960_revcomp.fastq\n",
      "Approx 60% complete for ERR1994960_revcomp.fastq\n",
      "Approx 65% complete for ERR1994960_revcomp.fastq\n",
      "Approx 70% complete for ERR1994960_revcomp.fastq\n",
      "Approx 75% complete for ERR1994960_revcomp.fastq\n",
      "Approx 80% complete for ERR1994960_revcomp.fastq\n",
      "Approx 85% complete for ERR1994960_revcomp.fastq\n",
      "Approx 90% complete for ERR1994960_revcomp.fastq\n",
      "Approx 95% complete for ERR1994960_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994963_revcomp.fastq\n",
      "Approx 5% complete for ERR1994963_revcomp.fastq\n",
      "Approx 10% complete for ERR1994963_revcomp.fastq\n",
      "Approx 15% complete for ERR1994963_revcomp.fastq\n",
      "Approx 20% complete for ERR1994963_revcomp.fastq\n",
      "Approx 25% complete for ERR1994963_revcomp.fastq\n",
      "Approx 30% complete for ERR1994963_revcomp.fastq\n",
      "Approx 35% complete for ERR1994963_revcomp.fastq\n",
      "Approx 40% complete for ERR1994963_revcomp.fastq\n",
      "Approx 45% complete for ERR1994963_revcomp.fastq\n",
      "Approx 50% complete for ERR1994963_revcomp.fastq\n",
      "Approx 55% complete for ERR1994963_revcomp.fastq\n",
      "Approx 60% complete for ERR1994963_revcomp.fastq\n",
      "Approx 65% complete for ERR1994963_revcomp.fastq\n",
      "Approx 70% complete for ERR1994963_revcomp.fastq\n",
      "Approx 75% complete for ERR1994963_revcomp.fastq\n",
      "Approx 80% complete for ERR1994963_revcomp.fastq\n",
      "Approx 85% complete for ERR1994963_revcomp.fastq\n",
      "Approx 90% complete for ERR1994963_revcomp.fastq\n",
      "Approx 95% complete for ERR1994963_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994964_revcomp.fastq\n",
      "Approx 5% complete for ERR1994964_revcomp.fastq\n",
      "Approx 10% complete for ERR1994964_revcomp.fastq\n",
      "Approx 15% complete for ERR1994964_revcomp.fastq\n",
      "Approx 20% complete for ERR1994964_revcomp.fastq\n",
      "Approx 25% complete for ERR1994964_revcomp.fastq\n",
      "Approx 30% complete for ERR1994964_revcomp.fastq\n",
      "Approx 35% complete for ERR1994964_revcomp.fastq\n",
      "Approx 40% complete for ERR1994964_revcomp.fastq\n",
      "Approx 45% complete for ERR1994964_revcomp.fastq\n",
      "Approx 50% complete for ERR1994964_revcomp.fastq\n",
      "Approx 55% complete for ERR1994964_revcomp.fastq\n",
      "Approx 60% complete for ERR1994964_revcomp.fastq\n",
      "Approx 65% complete for ERR1994964_revcomp.fastq\n",
      "Approx 70% complete for ERR1994964_revcomp.fastq\n",
      "Approx 75% complete for ERR1994964_revcomp.fastq\n",
      "Approx 80% complete for ERR1994964_revcomp.fastq\n",
      "Approx 85% complete for ERR1994964_revcomp.fastq\n",
      "Approx 90% complete for ERR1994964_revcomp.fastq\n",
      "Approx 95% complete for ERR1994964_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994967_revcomp.fastq\n",
      "Approx 5% complete for ERR1994967_revcomp.fastq\n",
      "Approx 10% complete for ERR1994967_revcomp.fastq\n",
      "Approx 15% complete for ERR1994967_revcomp.fastq\n",
      "Approx 20% complete for ERR1994967_revcomp.fastq\n",
      "Approx 25% complete for ERR1994967_revcomp.fastq\n",
      "Approx 30% complete for ERR1994967_revcomp.fastq\n",
      "Approx 35% complete for ERR1994967_revcomp.fastq\n",
      "Approx 40% complete for ERR1994967_revcomp.fastq\n",
      "Approx 45% complete for ERR1994967_revcomp.fastq\n",
      "Approx 50% complete for ERR1994967_revcomp.fastq\n",
      "Approx 55% complete for ERR1994967_revcomp.fastq\n",
      "Approx 60% complete for ERR1994967_revcomp.fastq\n",
      "Approx 65% complete for ERR1994967_revcomp.fastq\n",
      "Approx 70% complete for ERR1994967_revcomp.fastq\n",
      "Approx 75% complete for ERR1994967_revcomp.fastq\n",
      "Approx 80% complete for ERR1994967_revcomp.fastq\n",
      "Approx 85% complete for ERR1994967_revcomp.fastq\n",
      "Approx 90% complete for ERR1994967_revcomp.fastq\n",
      "Approx 95% complete for ERR1994967_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994968_revcomp.fastq\n",
      "Approx 5% complete for ERR1994968_revcomp.fastq\n",
      "Approx 10% complete for ERR1994968_revcomp.fastq\n",
      "Approx 15% complete for ERR1994968_revcomp.fastq\n",
      "Approx 20% complete for ERR1994968_revcomp.fastq\n",
      "Approx 25% complete for ERR1994968_revcomp.fastq\n",
      "Approx 30% complete for ERR1994968_revcomp.fastq\n",
      "Approx 35% complete for ERR1994968_revcomp.fastq\n",
      "Approx 40% complete for ERR1994968_revcomp.fastq\n",
      "Approx 45% complete for ERR1994968_revcomp.fastq\n",
      "Approx 50% complete for ERR1994968_revcomp.fastq\n",
      "Approx 55% complete for ERR1994968_revcomp.fastq\n",
      "Approx 60% complete for ERR1994968_revcomp.fastq\n",
      "Approx 65% complete for ERR1994968_revcomp.fastq\n",
      "Approx 70% complete for ERR1994968_revcomp.fastq\n",
      "Approx 75% complete for ERR1994968_revcomp.fastq\n",
      "Approx 80% complete for ERR1994968_revcomp.fastq\n",
      "Approx 85% complete for ERR1994968_revcomp.fastq\n",
      "Approx 90% complete for ERR1994968_revcomp.fastq\n",
      "Approx 95% complete for ERR1994968_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994971_revcomp.fastq\n",
      "Approx 5% complete for ERR1994971_revcomp.fastq\n",
      "Approx 10% complete for ERR1994971_revcomp.fastq\n",
      "Approx 15% complete for ERR1994971_revcomp.fastq\n",
      "Approx 20% complete for ERR1994971_revcomp.fastq\n",
      "Approx 25% complete for ERR1994971_revcomp.fastq\n",
      "Approx 30% complete for ERR1994971_revcomp.fastq\n",
      "Approx 35% complete for ERR1994971_revcomp.fastq\n",
      "Approx 40% complete for ERR1994971_revcomp.fastq\n",
      "Approx 45% complete for ERR1994971_revcomp.fastq\n",
      "Approx 50% complete for ERR1994971_revcomp.fastq\n",
      "Approx 55% complete for ERR1994971_revcomp.fastq\n",
      "Approx 60% complete for ERR1994971_revcomp.fastq\n",
      "Approx 65% complete for ERR1994971_revcomp.fastq\n",
      "Approx 70% complete for ERR1994971_revcomp.fastq\n",
      "Approx 75% complete for ERR1994971_revcomp.fastq\n",
      "Approx 80% complete for ERR1994971_revcomp.fastq\n",
      "Approx 85% complete for ERR1994971_revcomp.fastq\n",
      "Approx 90% complete for ERR1994971_revcomp.fastq\n",
      "Approx 95% complete for ERR1994971_revcomp.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994972_revcomp.fastq\n",
      "Approx 5% complete for ERR1994972_revcomp.fastq\n",
      "Approx 10% complete for ERR1994972_revcomp.fastq\n",
      "Approx 15% complete for ERR1994972_revcomp.fastq\n",
      "Approx 20% complete for ERR1994972_revcomp.fastq\n",
      "Approx 25% complete for ERR1994972_revcomp.fastq\n",
      "Approx 30% complete for ERR1994972_revcomp.fastq\n",
      "Approx 35% complete for ERR1994972_revcomp.fastq\n",
      "Approx 40% complete for ERR1994972_revcomp.fastq\n",
      "Approx 45% complete for ERR1994972_revcomp.fastq\n",
      "Approx 50% complete for ERR1994972_revcomp.fastq\n",
      "Approx 55% complete for ERR1994972_revcomp.fastq\n",
      "Approx 60% complete for ERR1994972_revcomp.fastq\n",
      "Approx 65% complete for ERR1994972_revcomp.fastq\n",
      "Approx 70% complete for ERR1994972_revcomp.fastq\n",
      "Approx 75% complete for ERR1994972_revcomp.fastq\n",
      "Approx 80% complete for ERR1994972_revcomp.fastq\n",
      "Approx 85% complete for ERR1994972_revcomp.fastq\n",
      "Approx 90% complete for ERR1994972_revcomp.fastq\n",
      "Approx 95% complete for ERR1994972_revcomp.fastq\n",
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 8 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_revcomp/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_revcomp\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/fastqc_reports/RNA_final/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base$i$label$extention\n",
    "done\n",
    "\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/fastqc_reports/RNA_final/\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download *Schizosaccharomyces pombe* genome (.fa) and annotation (.gff3) files from: https://www.pombase.org/downloads/genome-datasets\n",
    "using GFF utilities: http://ccb.jhu.edu/software/stringtie/gff.shtml "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Riboseq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mapping Riboseq reads to *S. pombe* rRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-12T13:19:34.478683Z",
     "start_time": "2018-03-12T13:19:23.387399Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'gclib'...\n",
      "Cloning into 'gffread'...\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/GBase.cpp -o ../gclib/GBase.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/GArgs.cpp -o ../gclib/GArgs.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/GFaSeqGet.cpp -o ../gclib/GFaSeqGet.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/gdna.cpp -o ../gclib/gdna.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/codons.cpp -o ../gclib/codons.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/gff.cpp -o ../gclib/gff.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/GStr.cpp -o ../gclib/GStr.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c ../gclib/GFastaIndex.cpp -o ../gclib/GFastaIndex.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c gff_utils.cpp -o gff_utils.o\n",
      "g++ -g -DDEBUG -Wall -Wextra -I. -I../gclib  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_REENTRANT -fno-strict-aliasing -fno-exceptions -fno-rtti -c gffread.cpp -o gffread.o\n",
      "g++ -g   -o gffread ../gclib/GBase.o ../gclib/GArgs.o ../gclib/GFaSeqGet.o ../gclib/gdna.o ../gclib/codons.o ../gclib/gff.o ../gclib/GStr.o ../gclib/GFastaIndex.o gff_utils.o gffread.o \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# installing gffread\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref\n",
    "git clone https://github.com/gpertea/gclib\n",
    "git clone https://github.com/gpertea/gffread\n",
    "cd gffread\n",
    "make\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "sed 's/III\\t/chromosome_3\\t/g' schizosaccharomyces_pombe.genome.gff3 > SChizosaccharomyces_pombe.genome.gff3\n",
    "sed 's/II\\t/chromosome_2\\t/g' SChizosaccharomyces_pombe.genome.gff3 > Schizosaccharomyces_pombe.genome.gff3\n",
    "sed 's/I\\t/chromosome_1\\t/g' Schizosaccharomyces_pombe.genome.gff3 > SCHizosaccharomyces_pombe.genome.gff3\n",
    "sed 's/MT\\t/mating_type_region\\t/g' SCHizosaccharomyces_pombe.genome.gff3 > S_pombe.genome.gff3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "http://bioconductor.org/packages/3.7/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-16T14:58:47.308092Z",
     "start_time": "2018-03-16T14:58:47.303123Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Cell magic `%%R` not found.\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "source(\"https://bioconductor.org/biocLite.R\")\n",
    "biocLite(\"rtracklayer\")\n",
    "library(rtracklayer)\n",
    "\n",
    "gff <- import(\"/projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/S_pombe.genome.gff3\")\n",
    "\n",
    "head(gff$type)\n",
    "rRNA_genes <- which(gff$type == \"rRNA_gene\")\n",
    "rRNA <- which(gff$type == \"rRNA\")\n",
    "\n",
    "total_rRNA <- sort(c(rRNA_genes,rRNA))\n",
    "gff_rRNA <- gff[total_rRNA]\n",
    "\n",
    "\n",
    "export(gff_rRNA,\"/projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/S_pombe_rRNA.gff3\", format = \"gff3\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-16T14:58:54.684131Z",
     "start_time": "2018-03-16T14:58:53.893343Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref\n",
    "\n",
    "#create a fasta index (.fai) file\n",
    "module load SAMtools\n",
    "samtools faidx Schizosaccharomyces_pombe_all_chromosomes.fa\n",
    "\n",
    "\n",
    "/projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/gffread/gffread -w schizosaccharomyces_pombe_rRNA.fa -g Schizosaccharomyces_pombe_all_chromosomes.fa S_pombe_rRNA.gff3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-12T16:29:52.691783Z",
     "start_time": "2018-03-12T15:47:03.748031Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"schizosaccharomyces_pombe_rRNA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/schizosaccharomyces_pombe_rRNA.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 7259\n",
      "Using parameters --bmax 5445 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 5445 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 29037 (target: 5444)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 29037 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 29038 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 8043\n",
      "fchr[G]: 13690\n",
      "fchr[T]: 20718\n",
      "fchr[$]: 29037\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 4206100 bytes to primary EBWT file: schizosaccharomyces_pombe_rRNA.1.bt2\n",
      "Wrote 7264 bytes to secondary EBWT file: schizosaccharomyces_pombe_rRNA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 29037\n",
      "    bwtLen: 29038\n",
      "    sz: 7260\n",
      "    bwtSz: 7260\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 1815\n",
      "    offsSz: 7260\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 152\n",
      "    numLines: 152\n",
      "    ebwtTotLen: 9728\n",
      "    ebwtTotSz: 9728\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:01\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 7259\n",
      "Using parameters --bmax 5445 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 5445 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 29037 (target: 5444)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 29037 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 29038 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 8043\n",
      "fchr[G]: 13690\n",
      "fchr[T]: 20718\n",
      "fchr[$]: 29037\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 4206100 bytes to primary EBWT file: schizosaccharomyces_pombe_rRNA.rev.1.bt2\n",
      "Wrote 7264 bytes to secondary EBWT file: schizosaccharomyces_pombe_rRNA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 29037\n",
      "    bwtLen: 29038\n",
      "    sz: 7260\n",
      "    bwtSz: 7260\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 1815\n",
      "    offsSz: 7260\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 152\n",
      "    numLines: 152\n",
      "    ebwtTotLen: 9728\n",
      "    ebwtTotSz: 9728\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:00\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building a SMALL index\n",
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_rRNA_mapping/’: File exists\n",
      "25940728 reads; of these:\n",
      "  25940728 (100.00%) were unpaired; of these:\n",
      "    20145928 (77.66%) aligned 0 times\n",
      "    1689 (0.01%) aligned exactly 1 time\n",
      "    5793111 (22.33%) aligned >1 times\n",
      "22.34% overall alignment rate\n",
      "9878601 reads; of these:\n",
      "  9878601 (100.00%) were unpaired; of these:\n",
      "    7264243 (73.54%) aligned 0 times\n",
      "    131 (0.00%) aligned exactly 1 time\n",
      "    2614227 (26.46%) aligned >1 times\n",
      "26.46% overall alignment rate\n",
      "20197104 reads; of these:\n",
      "  20197104 (100.00%) were unpaired; of these:\n",
      "    12897703 (63.86%) aligned 0 times\n",
      "    1700 (0.01%) aligned exactly 1 time\n",
      "    7297701 (36.13%) aligned >1 times\n",
      "36.14% overall alignment rate\n",
      "7701168 reads; of these:\n",
      "  7701168 (100.00%) were unpaired; of these:\n",
      "    4009350 (52.06%) aligned 0 times\n",
      "    246 (0.00%) aligned exactly 1 time\n",
      "    3691572 (47.94%) aligned >1 times\n",
      "47.94% overall alignment rate\n",
      "35158507 reads; of these:\n",
      "  35158507 (100.00%) were unpaired; of these:\n",
      "    24382521 (69.35%) aligned 0 times\n",
      "    8813 (0.03%) aligned exactly 1 time\n",
      "    10767173 (30.62%) aligned >1 times\n",
      "30.65% overall alignment rate\n",
      "7874358 reads; of these:\n",
      "  7874358 (100.00%) were unpaired; of these:\n",
      "    3950288 (50.17%) aligned 0 times\n",
      "    445 (0.01%) aligned exactly 1 time\n",
      "    3923625 (49.83%) aligned >1 times\n",
      "49.83% overall alignment rate\n",
      "16202823 reads; of these:\n",
      "  16202823 (100.00%) were unpaired; of these:\n",
      "    8030614 (49.56%) aligned 0 times\n",
      "    7291 (0.04%) aligned exactly 1 time\n",
      "    8164918 (50.39%) aligned >1 times\n",
      "50.44% overall alignment rate\n",
      "6656290 reads; of these:\n",
      "  6656290 (100.00%) were unpaired; of these:\n",
      "    2862946 (43.01%) aligned 0 times\n",
      "    434 (0.01%) aligned exactly 1 time\n",
      "    3792910 (56.98%) aligned >1 times\n",
      "56.99% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# create bowtie2 index file\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/\n",
    "bowtie2-build /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/schizosaccharomyces_pombe_rRNA.fa  schizosaccharomyces_pombe_rRNA\n",
    "export BOWTIE2_INDEXES='.'\n",
    "\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_trimmed/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "label2=\"_filtered\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_rRNA_mapping/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "# remove first four and last four nucleotides\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base$i$label2$extention -x /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/schizosaccharomyces_pombe_rRNA $path_fastq_files$accession_base$i$label$extention>/dev/null \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-13T10:17:15.265123Z",
     "start_time": "2018-03-13T10:11:35.199448Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for ERR1994961_filtered.fastq\n",
      "Analysis complete for ERR1994962_filtered.fastq\n",
      "Analysis complete for ERR1994965_filtered.fastq\n",
      "Analysis complete for ERR1994966_filtered.fastq\n",
      "Analysis complete for ERR1994969_filtered.fastq\n",
      "Analysis complete for ERR1994970_filtered.fastq\n",
      "Analysis complete for ERR1994973_filtered.fastq\n",
      "Analysis complete for ERR1994974_filtered.fastq\n",
      "Searching 16 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of ERR1994961_filtered.fastq\n",
      "Approx 5% complete for ERR1994961_filtered.fastq\n",
      "Approx 10% complete for ERR1994961_filtered.fastq\n",
      "Approx 15% complete for ERR1994961_filtered.fastq\n",
      "Approx 20% complete for ERR1994961_filtered.fastq\n",
      "Approx 25% complete for ERR1994961_filtered.fastq\n",
      "Approx 30% complete for ERR1994961_filtered.fastq\n",
      "Approx 35% complete for ERR1994961_filtered.fastq\n",
      "Approx 40% complete for ERR1994961_filtered.fastq\n",
      "Approx 45% complete for ERR1994961_filtered.fastq\n",
      "Approx 50% complete for ERR1994961_filtered.fastq\n",
      "Approx 55% complete for ERR1994961_filtered.fastq\n",
      "Approx 60% complete for ERR1994961_filtered.fastq\n",
      "Approx 65% complete for ERR1994961_filtered.fastq\n",
      "Approx 70% complete for ERR1994961_filtered.fastq\n",
      "Approx 75% complete for ERR1994961_filtered.fastq\n",
      "Approx 80% complete for ERR1994961_filtered.fastq\n",
      "Approx 85% complete for ERR1994961_filtered.fastq\n",
      "Approx 90% complete for ERR1994961_filtered.fastq\n",
      "Approx 95% complete for ERR1994961_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994962_filtered.fastq\n",
      "Approx 5% complete for ERR1994962_filtered.fastq\n",
      "Approx 10% complete for ERR1994962_filtered.fastq\n",
      "Approx 15% complete for ERR1994962_filtered.fastq\n",
      "Approx 20% complete for ERR1994962_filtered.fastq\n",
      "Approx 25% complete for ERR1994962_filtered.fastq\n",
      "Approx 30% complete for ERR1994962_filtered.fastq\n",
      "Approx 35% complete for ERR1994962_filtered.fastq\n",
      "Approx 40% complete for ERR1994962_filtered.fastq\n",
      "Approx 45% complete for ERR1994962_filtered.fastq\n",
      "Approx 50% complete for ERR1994962_filtered.fastq\n",
      "Approx 55% complete for ERR1994962_filtered.fastq\n",
      "Approx 60% complete for ERR1994962_filtered.fastq\n",
      "Approx 65% complete for ERR1994962_filtered.fastq\n",
      "Approx 70% complete for ERR1994962_filtered.fastq\n",
      "Approx 75% complete for ERR1994962_filtered.fastq\n",
      "Approx 80% complete for ERR1994962_filtered.fastq\n",
      "Approx 85% complete for ERR1994962_filtered.fastq\n",
      "Approx 90% complete for ERR1994962_filtered.fastq\n",
      "Approx 95% complete for ERR1994962_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994965_filtered.fastq\n",
      "Approx 5% complete for ERR1994965_filtered.fastq\n",
      "Approx 10% complete for ERR1994965_filtered.fastq\n",
      "Approx 15% complete for ERR1994965_filtered.fastq\n",
      "Approx 20% complete for ERR1994965_filtered.fastq\n",
      "Approx 25% complete for ERR1994965_filtered.fastq\n",
      "Approx 30% complete for ERR1994965_filtered.fastq\n",
      "Approx 35% complete for ERR1994965_filtered.fastq\n",
      "Approx 40% complete for ERR1994965_filtered.fastq\n",
      "Approx 45% complete for ERR1994965_filtered.fastq\n",
      "Approx 50% complete for ERR1994965_filtered.fastq\n",
      "Approx 55% complete for ERR1994965_filtered.fastq\n",
      "Approx 60% complete for ERR1994965_filtered.fastq\n",
      "Approx 65% complete for ERR1994965_filtered.fastq\n",
      "Approx 70% complete for ERR1994965_filtered.fastq\n",
      "Approx 75% complete for ERR1994965_filtered.fastq\n",
      "Approx 80% complete for ERR1994965_filtered.fastq\n",
      "Approx 85% complete for ERR1994965_filtered.fastq\n",
      "Approx 90% complete for ERR1994965_filtered.fastq\n",
      "Approx 95% complete for ERR1994965_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994966_filtered.fastq\n",
      "Approx 5% complete for ERR1994966_filtered.fastq\n",
      "Approx 10% complete for ERR1994966_filtered.fastq\n",
      "Approx 15% complete for ERR1994966_filtered.fastq\n",
      "Approx 20% complete for ERR1994966_filtered.fastq\n",
      "Approx 25% complete for ERR1994966_filtered.fastq\n",
      "Approx 30% complete for ERR1994966_filtered.fastq\n",
      "Approx 35% complete for ERR1994966_filtered.fastq\n",
      "Approx 40% complete for ERR1994966_filtered.fastq\n",
      "Approx 45% complete for ERR1994966_filtered.fastq\n",
      "Approx 50% complete for ERR1994966_filtered.fastq\n",
      "Approx 55% complete for ERR1994966_filtered.fastq\n",
      "Approx 60% complete for ERR1994966_filtered.fastq\n",
      "Approx 65% complete for ERR1994966_filtered.fastq\n",
      "Approx 70% complete for ERR1994966_filtered.fastq\n",
      "Approx 75% complete for ERR1994966_filtered.fastq\n",
      "Approx 80% complete for ERR1994966_filtered.fastq\n",
      "Approx 85% complete for ERR1994966_filtered.fastq\n",
      "Approx 90% complete for ERR1994966_filtered.fastq\n",
      "Approx 95% complete for ERR1994966_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994969_filtered.fastq\n",
      "Approx 5% complete for ERR1994969_filtered.fastq\n",
      "Approx 10% complete for ERR1994969_filtered.fastq\n",
      "Approx 15% complete for ERR1994969_filtered.fastq\n",
      "Approx 20% complete for ERR1994969_filtered.fastq\n",
      "Approx 25% complete for ERR1994969_filtered.fastq\n",
      "Approx 30% complete for ERR1994969_filtered.fastq\n",
      "Approx 35% complete for ERR1994969_filtered.fastq\n",
      "Approx 40% complete for ERR1994969_filtered.fastq\n",
      "Approx 45% complete for ERR1994969_filtered.fastq\n",
      "Approx 50% complete for ERR1994969_filtered.fastq\n",
      "Approx 55% complete for ERR1994969_filtered.fastq\n",
      "Approx 60% complete for ERR1994969_filtered.fastq\n",
      "Approx 65% complete for ERR1994969_filtered.fastq\n",
      "Approx 70% complete for ERR1994969_filtered.fastq\n",
      "Approx 75% complete for ERR1994969_filtered.fastq\n",
      "Approx 80% complete for ERR1994969_filtered.fastq\n",
      "Approx 85% complete for ERR1994969_filtered.fastq\n",
      "Approx 90% complete for ERR1994969_filtered.fastq\n",
      "Approx 95% complete for ERR1994969_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994970_filtered.fastq\n",
      "Approx 5% complete for ERR1994970_filtered.fastq\n",
      "Approx 10% complete for ERR1994970_filtered.fastq\n",
      "Approx 15% complete for ERR1994970_filtered.fastq\n",
      "Approx 20% complete for ERR1994970_filtered.fastq\n",
      "Approx 25% complete for ERR1994970_filtered.fastq\n",
      "Approx 30% complete for ERR1994970_filtered.fastq\n",
      "Approx 35% complete for ERR1994970_filtered.fastq\n",
      "Approx 40% complete for ERR1994970_filtered.fastq\n",
      "Approx 45% complete for ERR1994970_filtered.fastq\n",
      "Approx 50% complete for ERR1994970_filtered.fastq\n",
      "Approx 55% complete for ERR1994970_filtered.fastq\n",
      "Approx 60% complete for ERR1994970_filtered.fastq\n",
      "Approx 65% complete for ERR1994970_filtered.fastq\n",
      "Approx 70% complete for ERR1994970_filtered.fastq\n",
      "Approx 75% complete for ERR1994970_filtered.fastq\n",
      "Approx 80% complete for ERR1994970_filtered.fastq\n",
      "Approx 85% complete for ERR1994970_filtered.fastq\n",
      "Approx 90% complete for ERR1994970_filtered.fastq\n",
      "Approx 95% complete for ERR1994970_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994973_filtered.fastq\n",
      "Approx 5% complete for ERR1994973_filtered.fastq\n",
      "Approx 10% complete for ERR1994973_filtered.fastq\n",
      "Approx 15% complete for ERR1994973_filtered.fastq\n",
      "Approx 20% complete for ERR1994973_filtered.fastq\n",
      "Approx 25% complete for ERR1994973_filtered.fastq\n",
      "Approx 30% complete for ERR1994973_filtered.fastq\n",
      "Approx 35% complete for ERR1994973_filtered.fastq\n",
      "Approx 40% complete for ERR1994973_filtered.fastq\n",
      "Approx 45% complete for ERR1994973_filtered.fastq\n",
      "Approx 50% complete for ERR1994973_filtered.fastq\n",
      "Approx 55% complete for ERR1994973_filtered.fastq\n",
      "Approx 60% complete for ERR1994973_filtered.fastq\n",
      "Approx 65% complete for ERR1994973_filtered.fastq\n",
      "Approx 70% complete for ERR1994973_filtered.fastq\n",
      "Approx 75% complete for ERR1994973_filtered.fastq\n",
      "Approx 80% complete for ERR1994973_filtered.fastq\n",
      "Approx 85% complete for ERR1994973_filtered.fastq\n",
      "Approx 90% complete for ERR1994973_filtered.fastq\n",
      "Approx 95% complete for ERR1994973_filtered.fastq\n",
      "Unknown option: i\n",
      "Started analysis of ERR1994974_filtered.fastq\n",
      "Approx 5% complete for ERR1994974_filtered.fastq\n",
      "Approx 10% complete for ERR1994974_filtered.fastq\n",
      "Approx 15% complete for ERR1994974_filtered.fastq\n",
      "Approx 20% complete for ERR1994974_filtered.fastq\n",
      "Approx 25% complete for ERR1994974_filtered.fastq\n",
      "Approx 30% complete for ERR1994974_filtered.fastq\n",
      "Approx 35% complete for ERR1994974_filtered.fastq\n",
      "Approx 40% complete for ERR1994974_filtered.fastq\n",
      "Approx 45% complete for ERR1994974_filtered.fastq\n",
      "Approx 50% complete for ERR1994974_filtered.fastq\n",
      "Approx 55% complete for ERR1994974_filtered.fastq\n",
      "Approx 60% complete for ERR1994974_filtered.fastq\n",
      "Approx 65% complete for ERR1994974_filtered.fastq\n",
      "Approx 70% complete for ERR1994974_filtered.fastq\n",
      "Approx 75% complete for ERR1994974_filtered.fastq\n",
      "Approx 80% complete for ERR1994974_filtered.fastq\n",
      "Approx 85% complete for ERR1994974_filtered.fastq\n",
      "Approx 90% complete for ERR1994974_filtered.fastq\n",
      "Approx 95% complete for ERR1994974_filtered.fastq\n",
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 8 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_rRNA_mapping/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_filtered\"\n",
    "output_directory=\"/projects_eg/projects/simone/Duncan_Mata/fastqc_reports/Ribo_unmapped/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base$i$label$extention\n",
    "done\n",
    "\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/fastqc_reports/Ribo_unmapped/\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mapping unmapped Riboseq reads to *S. pombe* genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-13T11:50:52.344596Z",
     "start_time": "2018-03-13T10:32:26.456996Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "20145928 reads; of these:\n",
      "  20145928 (100.00%) were unpaired; of these:\n",
      "    1174379 (5.83%) aligned 0 times\n",
      "    16921401 (83.99%) aligned exactly 1 time\n",
      "    2050148 (10.18%) aligned >1 times\n",
      "94.17% overall alignment rate\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
      "7264243 reads; of these:\n",
      "  7264243 (100.00%) were unpaired; of these:\n",
      "    326412 (4.49%) aligned 0 times\n",
      "    6115990 (84.19%) aligned exactly 1 time\n",
      "    821841 (11.31%) aligned >1 times\n",
      "95.51% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "12897703 reads; of these:\n",
      "  12897703 (100.00%) were unpaired; of these:\n",
      "    2513172 (19.49%) aligned 0 times\n",
      "    8865973 (68.74%) aligned exactly 1 time\n",
      "    1518558 (11.77%) aligned >1 times\n",
      "80.51% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "4009350 reads; of these:\n",
      "  4009350 (100.00%) were unpaired; of these:\n",
      "    368099 (9.18%) aligned 0 times\n",
      "    3134299 (78.17%) aligned exactly 1 time\n",
      "    506952 (12.64%) aligned >1 times\n",
      "90.82% overall alignment rate\n",
      "24382521 reads; of these:\n",
      "  24382521 (100.00%) were unpaired; of these:\n",
      "    2546277 (10.44%) aligned 0 times\n",
      "    20358615 (83.50%) aligned exactly 1 time\n",
      "    1477629 (6.06%) aligned >1 times\n",
      "89.56% overall alignment rate\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
      "3950288 reads; of these:\n",
      "  3950288 (100.00%) were unpaired; of these:\n",
      "    441468 (11.18%) aligned 0 times\n",
      "    3211793 (81.31%) aligned exactly 1 time\n",
      "    297027 (7.52%) aligned >1 times\n",
      "88.82% overall alignment rate\n",
      "8030614 reads; of these:\n",
      "  8030614 (100.00%) were unpaired; of these:\n",
      "    3047344 (37.95%) aligned 0 times\n",
      "    4164966 (51.86%) aligned exactly 1 time\n",
      "    818304 (10.19%) aligned >1 times\n",
      "62.05% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "2862946 reads; of these:\n",
      "  2862946 (100.00%) were unpaired; of these:\n",
      "    406837 (14.21%) aligned 0 times\n",
      "    2205299 (77.03%) aligned exactly 1 time\n",
      "    250810 (8.76%) aligned >1 times\n",
      "85.79% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# create bowtie2 index file\n",
    "#cd /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref\n",
    "#bowtie2-build /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/Schizosaccharomyces_pombe_all_chromosomes.fa  Schizosaccharomyces_pombe_all_chromosomes\n",
    "#export BOWTIE2_INDEXES='.'\n",
    "\n",
    "\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/Ribo_rRNA_mapping/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_filtered\"\n",
    "\n",
    "\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "#mkdir $sam_output_directory\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "#mkdir $bam_output_directory\n",
    "\n",
    "Ribo_bam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/bam_files/Riboseq/\"\n",
    "mkdir $Ribo_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "sorted_bam_bai=\"_sorted.bam.bai\"\n",
    "\n",
    "module load SAMtools\n",
    "list='61 62 65 66 69 70 73 74'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/Schizosaccharomyces_pombe_all_chromosomes -U $path_fastq_files$accession_base$i$label$extention -S $sam_output_directory$accession_base$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$accession_base$i$sam_ext > $Ribo_bam_output_directory$accession_base$i$bam_ext\n",
    "samtools sort $Ribo_bam_output_directory$accession_base$i$bam_ext -o $Ribo_bam_output_directory$accession_base$i$sorted_bam_ext -O BAM\n",
    "samtools index -b $Ribo_bam_output_directory$accession_base$i$sorted_bam_ext  $Ribo_bam_output_directory$accession_base$i$sorted_bam_bai\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RNAseq data "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mapping RNAseq data to *S. pombe* genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-12T18:15:44.700778Z",
     "start_time": "2018-03-12T16:37:48.367009Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"Schizosaccharomyces_pombe_all_chromosomes.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/Schizosaccharomyces_pombe_all_chromosomes.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 3157744\n",
      "Using parameters --bmax 2368308 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 2368308 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:01\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.2631e+07 (target: 2368307)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 12630977 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:04\n",
      "Returning block of 12630978 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 4036348\n",
      "fchr[G]: 6311338\n",
      "fchr[T]: 8588523\n",
      "fchr[$]: 12630977\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 8405215 bytes to primary EBWT file: Schizosaccharomyces_pombe_all_chromosomes.1.bt2\n",
      "Wrote 3157752 bytes to secondary EBWT file: Schizosaccharomyces_pombe_all_chromosomes.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 12630977\n",
      "    bwtLen: 12630978\n",
      "    sz: 3157745\n",
      "    bwtSz: 3157745\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 789437\n",
      "    offsSz: 3157748\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 65787\n",
      "    numLines: 65787\n",
      "    ebwtTotLen: 4210368\n",
      "    ebwtTotSz: 4210368\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:06\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 3157744\n",
      "Using parameters --bmax 2368308 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 2368308 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:01\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.2631e+07 (target: 2368307)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 12630977 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:04\n",
      "Returning block of 12630978 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 4036348\n",
      "fchr[G]: 6311338\n",
      "fchr[T]: 8588523\n",
      "fchr[$]: 12630977\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 8405215 bytes to primary EBWT file: Schizosaccharomyces_pombe_all_chromosomes.rev.1.bt2\n",
      "Wrote 3157752 bytes to secondary EBWT file: Schizosaccharomyces_pombe_all_chromosomes.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 12630977\n",
      "    bwtLen: 12630978\n",
      "    sz: 3157745\n",
      "    bwtSz: 3157745\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 789437\n",
      "    offsSz: 3157748\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 65787\n",
      "    numLines: 65787\n",
      "    ebwtTotLen: 4210368\n",
      "    ebwtTotSz: 4210368\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:06\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building a SMALL index\n",
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/sam_files/’: File exists\n",
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/Duncan_Mata/bam_files/’: File exists\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "14061638 reads; of these:\n",
      "  14061638 (100.00%) were unpaired; of these:\n",
      "    218467 (1.55%) aligned 0 times\n",
      "    13383489 (95.18%) aligned exactly 1 time\n",
      "    459682 (3.27%) aligned >1 times\n",
      "98.45% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994959.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994959.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994959.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994959_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994959_sorted.bam\": No such file or directory\n",
      "21617339 reads; of these:\n",
      "  21617339 (100.00%) were unpaired; of these:\n",
      "    356558 (1.65%) aligned 0 times\n",
      "    20441190 (94.56%) aligned exactly 1 time\n",
      "    819591 (3.79%) aligned >1 times\n",
      "98.35% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994960.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994960.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994960.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994960_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994960_sorted.bam\": No such file or directory\n",
      "6998367 reads; of these:\n",
      "  6998367 (100.00%) were unpaired; of these:\n",
      "    118170 (1.69%) aligned 0 times\n",
      "    6609876 (94.45%) aligned exactly 1 time\n",
      "    270321 (3.86%) aligned >1 times\n",
      "98.31% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994963.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994963.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994963.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994963_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994963_sorted.bam\": No such file or directory\n",
      "14705350 reads; of these:\n",
      "  14705350 (100.00%) were unpaired; of these:\n",
      "    265363 (1.80%) aligned 0 times\n",
      "    13741613 (93.45%) aligned exactly 1 time\n",
      "    698374 (4.75%) aligned >1 times\n",
      "98.20% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994964.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994964.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994964.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994964_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994964_sorted.bam\": No such file or directory\n",
      "9579520 reads; of these:\n",
      "  9579520 (100.00%) were unpaired; of these:\n",
      "    150126 (1.57%) aligned 0 times\n",
      "    9161687 (95.64%) aligned exactly 1 time\n",
      "    267707 (2.79%) aligned >1 times\n",
      "98.43% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994967.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994967.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994967.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994967_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994967_sorted.bam\": No such file or directory\n",
      "21153509 reads; of these:\n",
      "  21153509 (100.00%) were unpaired; of these:\n",
      "    389982 (1.84%) aligned 0 times\n",
      "    20082722 (94.94%) aligned exactly 1 time\n",
      "    680805 (3.22%) aligned >1 times\n",
      "98.16% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994968.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994968.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994968.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994968_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994968_sorted.bam\": No such file or directory\n",
      "8372300 reads; of these:\n",
      "  8372300 (100.00%) were unpaired; of these:\n",
      "    162277 (1.94%) aligned 0 times\n",
      "    7912147 (94.50%) aligned exactly 1 time\n",
      "    297876 (3.56%) aligned >1 times\n",
      "98.06% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994971.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994971.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994971.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994971_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994971_sorted.bam\": No such file or directory\n",
      "15366835 reads; of these:\n",
      "  15366835 (100.00%) were unpaired; of these:\n",
      "    256218 (1.67%) aligned 0 times\n",
      "    14602500 (95.03%) aligned exactly 1 time\n",
      "    508117 (3.31%) aligned >1 times\n",
      "98.33% overall alignment rate\n",
      "bash: line 33: /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994972.bam: No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994972.bam\n",
      "samtools sort: can't open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994972.bam\": No such file or directory\n",
      "[E::hts_open_format] Failed to open file /projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994972_sorted.bam\n",
      "samtools index: failed to open \"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/ERR1994972_sorted.bam\": No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# create bowtie2 index file\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref\n",
    "bowtie2-build /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/Schizosaccharomyces_pombe_all_chromosomes.fa  Schizosaccharomyces_pombe_all_chromosomes\n",
    "export BOWTIE2_INDEXES='.'\n",
    "\n",
    "\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_revcomp/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_revcomp\"\n",
    "\n",
    "\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "mkdir $sam_output_directory\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "mkdir $bam_output_directory\n",
    "\n",
    "RNA_bam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/\"\n",
    "mkdir $RNA_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "sorted_bam_bai=\"_sorted.bam.bai\"\n",
    "\n",
    "module load SAMtools\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/Schizosaccharomyces_pombe_all_chromosomes -U $path_fastq_files$accession_base$i$label$extention -S $sam_output_directory$accession_base$i$sam_ext\n",
    "#samtools view -S -b $sam_output_directory$accession_base$i$sam_ext > $RNA_bam_output_directory$accession_base$i$bam_ext\n",
    "#samtools sort $RNA_bam_output_directory$accession_base$i$bam_ext -o $RNA_bam_output_directory$accession_base$i$sorted_bam_ext -O BAM\n",
    "#samtools index -b $RNA_bam_output_directory$accession_base$i$sorted_bam_ext  $RNA_bam_output_directory$accession_base$i$sorted_bam_bai\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-13T09:52:08.383119Z",
     "start_time": "2018-03-13T09:12:00.343549Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/Duncan_Mata/trimming/RNA_revcomp/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_revcomp\"\n",
    "\n",
    "\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "#mkdir $sam_output_directory\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "#mkdir $bam_output_directory\n",
    "\n",
    "RNA_bam_output_directory=\"/projects_eg/projects/simone/Duncan_Mata/bam_files/RNAseq/\"\n",
    "mkdir $RNA_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "sorted_bam_bai=\"_sorted.bam.bai\"\n",
    "\n",
    "module load SAMtools\n",
    "list='59 60 63 64 67 68 71 72'\n",
    "accession_base=\"ERR19949\"\n",
    "for i in $list\n",
    "do \n",
    "#bowtie2 --no-unal -x /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/Schizosaccharomyces_pombe_all_chromosomes -U $path_fastq_files$accession_base$i$label$extention -S $sam_output_directory$accession_base$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$accession_base$i$sam_ext > $RNA_bam_output_directory$accession_base$i$bam_ext\n",
    "samtools sort $RNA_bam_output_directory$accession_base$i$bam_ext -o $RNA_bam_output_directory$accession_base$i$sorted_bam_ext -O BAM\n",
    "samtools index -b $RNA_bam_output_directory$accession_base$i$sorted_bam_ext  $RNA_bam_output_directory$accession_base$i$sorted_bam_bai\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DGE analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-19T12:59:51.824154Z",
     "start_time": "2018-03-19T12:59:51.100091Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /projects_eg/projects/simone/Duncan_Mata/S_pombe_ref\n",
    "\n",
    "# convert .gff3 file to .gtf for DGE analysis in R\n",
    "\n",
    "\n",
    "/projects_eg/projects/simone/Duncan_Mata/S_pombe_ref/gffread/gffread S_pombe.genome.gff3 -T -o S_pombe.genome.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
