{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**From paper \"Promoter sequences direct cytoplasmic localization and translation of mRNAs during starvation in yeast\"** Zid, O'Shea, 2014\n",
    "        Project SRP041039 with SRA accession numbers SRR1223686-SRR1223706  and SRR1288391-SRR1288408"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# From Raw data to BAM files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Downloading reads from project SRP041039\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Downloading  fastq files from the SRA (paper raw data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-09T17:23:54.191447Z",
     "start_time": "2018-02-09T12:03:09.905031Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "downloading first group of reads to  /home/simone/iPython_notebooks/fastq_files/zid-oshea-2014\n",
      "Read 32658689 spots for SRR1223690\n",
      "Written 32658689 spots for SRR1223690\n",
      "Read 19491861 spots for SRR1223691\n",
      "Written 19491861 spots for SRR1223691\n",
      "Read 22641779 spots for SRR1223692\n",
      "Written 22641779 spots for SRR1223692\n",
      "Read 13592624 spots for SRR1223693\n",
      "Written 13592624 spots for SRR1223693\n",
      "Read 31977655 spots for SRR1223694\n",
      "Written 31977655 spots for SRR1223694\n",
      "Read 13992051 spots for SRR1223695\n",
      "Written 13992051 spots for SRR1223695\n",
      "Read 21075312 spots for SRR1223696\n",
      "Written 21075312 spots for SRR1223696\n",
      "Read 19360675 spots for SRR1223697\n",
      "Written 19360675 spots for SRR1223697\n",
      "Read 25664498 spots for SRR1223698\n",
      "Written 25664498 spots for SRR1223698\n",
      "Read 46043180 spots for SRR1223699\n",
      "Written 46043180 spots for SRR1223699\n",
      "Read 77123881 spots for SRR1223700\n",
      "Written 77123881 spots for SRR1223700\n",
      "Read 16536415 spots for SRR1223701\n",
      "Written 16536415 spots for SRR1223701\n",
      "Read 24871285 spots for SRR1223702\n",
      "Written 24871285 spots for SRR1223702\n",
      "Read 11331341 spots for SRR1223703\n",
      "Written 11331341 spots for SRR1223703\n",
      "Read 18566575 spots for SRR1223704\n",
      "Written 18566575 spots for SRR1223704\n",
      "Read 16112188 spots for SRR1223705\n",
      "Written 16112188 spots for SRR1223705\n",
      "Read 12734357 spots for SRR1223706\n",
      "Written 12734357 spots for SRR1223706\n",
      "downloading second group of reads to /home/simone/iPython_notebooks/fastq_files/zid-oshea-2014\n",
      "Read 17778707 spots for SRR1288391\n",
      "Written 17778707 spots for SRR1288391\n",
      "Read 4000000 spots for SRR1288392\n",
      "Written 4000000 spots for SRR1288392\n",
      "Read 4000000 spots for SRR1288393\n",
      "Written 4000000 spots for SRR1288393\n",
      "Read 4000000 spots for SRR1288394\n",
      "Written 4000000 spots for SRR1288394\n",
      "Read 4000000 spots for SRR1288395\n",
      "Written 4000000 spots for SRR1288395\n",
      "Read 2300685 spots for SRR1288396\n",
      "Written 2300685 spots for SRR1288396\n",
      "Read 17569655 spots for SRR1288397\n",
      "Written 17569655 spots for SRR1288397\n",
      "Read 4000000 spots for SRR1288398\n",
      "Written 4000000 spots for SRR1288398\n",
      "Read 3067230 spots for SRR1288399\n",
      "Written 3067230 spots for SRR1288399\n",
      "Read 20351396 spots for SRR1288400\n",
      "Written 20351396 spots for SRR1288400\n",
      "Read 4000000 spots for SRR1288401\n",
      "Written 4000000 spots for SRR1288401\n",
      "Read 4000000 spots for SRR1288402\n",
      "Written 4000000 spots for SRR1288402\n",
      "Read 4000000 spots for SRR1288403\n",
      "Written 4000000 spots for SRR1288403\n",
      "Read 1637998 spots for SRR1288404\n",
      "Written 1637998 spots for SRR1288404\n",
      "Read 15079613 spots for SRR1288405\n",
      "Written 15079613 spots for SRR1288405\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/simone/iPython_notebooks/fastq_files/zid-oshea-2014’: File exists\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='5'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - failed SRR1288406\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='5'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted while writing file within file system module - system bad file descriptor error fd='6'\n",
      "2018-02-09T17:23:31 fastq-dump.2.8.2 err: storage exhausted whil"
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 303199 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%bash\n",
    "fastqdump=\"/home/simone/Downloads/sratoolkit.2.8.2-1-centos_linux64/bin/fastq-dump\" \n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/fastq_files\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# First we will download the group of reads that goes from SRR1223686 to SRR1223706\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'downloading first group of reads to  '$output_directory\n",
    "END=706\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "$fastqdump -I -O $output_directory $accession_base1$i\n",
    "done\n",
    "\n",
    "# Now we will download the second group of reads that go from SRR1288391 to SRR1288408\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'downloading second group of reads to '$output_directory\n",
    "END=408\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "$fastqdump -I -O $output_directory $accession_base2$i\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We accidentally canceled the last program to download the reads before it had finished. Now we will get the last ones.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-19T13:15:38.320133Z",
     "start_time": "2018-02-19T13:15:38.269183Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 3: ll: command not found\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "working_directory=\"/projects_eg/projects/simone/glu_paper/fastq_files\"\n",
    "\n",
    "ll $working_directory"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Organizing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Double click on cell below to see organization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These are the reads organized by yeast species (BY4741, EY0690), sequencing tool (RNASeq, RiboSeq), and RNASeq method for RNAseq samples (total RNA, polyA selection and depleted rRNA)\n",
    "\n",
    "_____________________________________________________________________________________________\n",
    "\n",
    "\t\t\t\t\t\tBY4741\n",
    "\n",
    "\n",
    "\tRibo-seq\n",
    "\n",
    "SRR1223686\tBY-Ribo_logphase_Replicate1\n",
    "SRR1288391\tBY-Ribo_logphase_Replicate2\n",
    "SRR1223697\tBY-Ribo_-Glu15_Replicate1\n",
    "SRR1288400\tBY-Ribo_-Glu15_Replicate2\n",
    "\n",
    "---------------------------------------------------------------------------------------------\n",
    "\n",
    "\tRNA-seq\n",
    "\n",
    "Total\n",
    "\n",
    "SRR1223689\tBY-RNA_logphase_Total_Replicate1\n",
    "SRR1223690\tBY-RNA_logphase_Total_Replicate1\n",
    "SRR1223700\tBY-RNA_-Glu15_Total_Replicate1\n",
    "\n",
    "polyA\n",
    "\n",
    "SRR1223691\tBY-RNA_logphase_polyA_Replicate1\n",
    "SRR1288397\tBY-RNA_logphase_polyA_Replicate2\n",
    "SRR1223701\tBY-RNA_-Glu15_polyA_Replicate1\n",
    "SRR1288405\tBY-RNA_-Glu15_polyA_Replicate2\n",
    "\n",
    "\n",
    "_____________________________________________________________________________________________\n",
    "\n",
    "\t\t\t\t\t\tEY0690\n",
    "\n",
    "\n",
    "\tRibo-seq\n",
    "\n",
    "SRR1223687\tEY-Ribo_logphase_Replicate1\n",
    "SRR1223688\tEY-Ribo_logphase_Replicate1\n",
    "SRR1288392\tEY-Ribo_logphase_Replicate2\n",
    "SRR1288393\tEY-Ribo_logphase_Replicate2\n",
    "SRR1288394\tEY-Ribo_logphase_Replicate2\n",
    "SRR1288395\tEY-Ribo_logphase_Replicate2\n",
    "SRR1288396\tEY-Ribo_logphase_Replicate2\n",
    "SRR1223698\tEY-Ribo_-Glu15_Replicate1\n",
    "SRR1223699\tEY-Ribo_-Glu15_Replicate1\n",
    "SRR1288401\tEY-Ribo_-Glu15_Replicate2\n",
    "SRR1288402\tEY-Ribo_-Glu15_Replicate2\n",
    "SRR1288403\tEY-Ribo_-Glu15_Replicate2\n",
    "SRR1288404\tEY-Ribo_-Glu15_Replicate2\n",
    "\n",
    "---------------------------------------------------------------------------------------------\n",
    "\n",
    "\tRNA-seq\n",
    "\n",
    "Total\n",
    "\n",
    "SRR1223692\tEY-RNA_logphase_Total_Replicate1\n",
    "SRR1223693\tEY-RNA_logphase_Total_Replicate1\n",
    "SRR1223702\tEY-RNA_-Glu15_Total_Replicate1\n",
    "SRR1223703\tEY-RNA_-Glu15_Total_Replicate1\n",
    "\n",
    "polyA\n",
    "\n",
    "SRR1223696\tEY-RNA_logphase_polyA_Replicate1\n",
    "SRR1288398\tEY-RNA_logphase_polyA_Replicate2\n",
    "SRR1288399\tEY-RNA_logphase_polyA_Replicate2\n",
    "SRR1223706\tEY-RNA_-Glu15_polyA_Replicate1\n",
    "SRR1288406\tEY-RNA_-Glu15_polyA_Replicate2\n",
    "SRR1288407\tEY-RNA_-Glu15_polyA_Replicate2\n",
    "SRR1288408\tEY-RNA_-Glu15_polyA_Replicate2\n",
    "\n",
    "RiboZero\n",
    "\n",
    "SRR1223694\tEY-RNA_logphase_RiboZero_Replicate1\n",
    "SRR1223695\tEY-RNA_logphase_RiboZero_Replicate1\n",
    "SRR1223704\tEY-RNA_-Glu15_RiboZero_Replicate1\n",
    "SRR1223705\tEY-RNA_-Glu15_RiboZero_Replicate1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Checking the quality of the reads using FastQC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting fastQC reports of our two groups of reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-19T19:01:08.642001Z",
     "start_time": "2018-02-19T15:11:45.359880Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get fastQC report of first group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results\n",
      "Analysis complete for SRR1223687.fastq\n",
      "Analysis complete for SRR1223688.fastq\n",
      "Analysis complete for SRR1223689.fastq\n",
      "Analysis complete for SRR1223690.fastq\n",
      "Analysis complete for SRR1223691.fastq\n",
      "Analysis complete for SRR1223692.fastq\n",
      "Analysis complete for SRR1223693.fastq\n",
      "Analysis complete for SRR1223694.fastq\n",
      "Analysis complete for SRR1223695.fastq\n",
      "Analysis complete for SRR1223696.fastq\n",
      "Analysis complete for SRR1223697.fastq\n",
      "Analysis complete for SRR1223698.fastq\n",
      "Analysis complete for SRR1223699.fastq\n",
      "Analysis complete for SRR1223700.fastq\n",
      "Analysis complete for SRR1223701.fastq\n",
      "Analysis complete for SRR1223702.fastq\n",
      "Analysis complete for SRR1223703.fastq\n",
      "Analysis complete for SRR1223704.fastq\n",
      "Analysis complete for SRR1223705.fastq\n",
      "Analysis complete for SRR1223706.fastq\n",
      "get fastQC report of second group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results\n",
      "Analysis complete for SRR1288391.fastq\n",
      "Analysis complete for SRR1288392.fastq\n",
      "Analysis complete for SRR1288393.fastq\n",
      "Analysis complete for SRR1288394.fastq\n",
      "Analysis complete for SRR1288395.fastq\n",
      "Analysis complete for SRR1288396.fastq\n",
      "Analysis complete for SRR1288397.fastq\n",
      "Analysis complete for SRR1288398.fastq\n",
      "Analysis complete for SRR1288399.fastq\n",
      "Analysis complete for SRR1288400.fastq\n",
      "Analysis complete for SRR1288401.fastq\n",
      "Analysis complete for SRR1288402.fastq\n",
      "Analysis complete for SRR1288403.fastq\n",
      "Analysis complete for SRR1288404.fastq\n",
      "Analysis complete for SRR1288405.fastq\n",
      "Analysis complete for SRR1288406.fastq\n",
      "Analysis complete for SRR1288407.fastq\n",
      "Analysis complete for SRR1288408.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/fastqc_results’: File exists\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223687.fastq\n",
      "Approx 5% complete for SRR1223687.fastq\n",
      "Approx 10% complete for SRR1223687.fastq\n",
      "Approx 15% complete for SRR1223687.fastq\n",
      "Approx 20% complete for SRR1223687.fastq\n",
      "Approx 25% complete for SRR1223687.fastq\n",
      "Approx 30% complete for SRR1223687.fastq\n",
      "Approx 35% complete for SRR1223687.fastq\n",
      "Approx 40% complete for SRR1223687.fastq\n",
      "Approx 45% complete for SRR1223687.fastq\n",
      "Approx 50% complete for SRR1223687.fastq\n",
      "Approx 55% complete for SRR1223687.fastq\n",
      "Approx 60% complete for SRR1223687.fastq\n",
      "Approx 65% complete for SRR1223687.fastq\n",
      "Approx 70% complete for SRR1223687.fastq\n",
      "Approx 75% complete for SRR1223687.fastq\n",
      "Approx 80% complete for SRR1223687.fastq\n",
      "Approx 85% complete for SRR1223687.fastq\n",
      "Approx 90% complete for SRR1223687.fastq\n",
      "Approx 95% complete for SRR1223687.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223688.fastq\n",
      "Approx 5% complete for SRR1223688.fastq\n",
      "Approx 10% complete for SRR1223688.fastq\n",
      "Approx 15% complete for SRR1223688.fastq\n",
      "Approx 20% complete for SRR1223688.fastq\n",
      "Approx 25% complete for SRR1223688.fastq\n",
      "Approx 30% complete for SRR1223688.fastq\n",
      "Approx 35% complete for SRR1223688.fastq\n",
      "Approx 40% complete for SRR1223688.fastq\n",
      "Approx 45% complete for SRR1223688.fastq\n",
      "Approx 50% complete for SRR1223688.fastq\n",
      "Approx 55% complete for SRR1223688.fastq\n",
      "Approx 60% complete for SRR1223688.fastq\n",
      "Approx 65% complete for SRR1223688.fastq\n",
      "Approx 70% complete for SRR1223688.fastq\n",
      "Approx 75% complete for SRR1223688.fastq\n",
      "Approx 80% complete for SRR1223688.fastq\n",
      "Approx 85% complete for SRR1223688.fastq\n",
      "Approx 90% complete for SRR1223688.fastq\n",
      "Approx 95% complete for SRR1223688.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223689.fastq\n",
      "Approx 5% complete for SRR1223689.fastq\n",
      "Approx 10% complete for SRR1223689.fastq\n",
      "Approx 15% complete for SRR1223689.fastq\n",
      "Approx 20% complete for SRR1223689.fastq\n",
      "Approx 25% complete for SRR1223689.fastq\n",
      "Approx 30% complete for SRR1223689.fastq\n",
      "Approx 35% complete for SRR1223689.fastq\n",
      "Approx 40% complete for SRR1223689.fastq\n",
      "Approx 45% complete for SRR1223689.fastq\n",
      "Approx 50% complete for SRR1223689.fastq\n",
      "Approx 55% complete for SRR1223689.fastq\n",
      "Approx 60% complete for SRR1223689.fastq\n",
      "Approx 65% complete for SRR1223689.fastq\n",
      "Approx 70% complete for SRR1223689.fastq\n",
      "Approx 75% complete for SRR1223689.fastq\n",
      "Approx 80% complete for SRR1223689.fastq\n",
      "Approx 85% complete for SRR1223689.fastq\n",
      "Approx 90% complete for SRR1223689.fastq\n",
      "Approx 95% complete for SRR1223689.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223690.fastq\n",
      "Approx 5% complete for SRR1223690.fastq\n",
      "Approx 10% complete for SRR1223690.fastq\n",
      "Approx 15% complete for SRR1223690.fastq\n",
      "Approx 20% complete for SRR1223690.fastq\n",
      "Approx 25% complete for SRR1223690.fastq\n",
      "Approx 30% complete for SRR1223690.fastq\n",
      "Approx 35% complete for SRR1223690.fastq\n",
      "Approx 40% complete for SRR1223690.fastq\n",
      "Approx 45% complete for SRR1223690.fastq\n",
      "Approx 50% complete for SRR1223690.fastq\n",
      "Approx 55% complete for SRR1223690.fastq\n",
      "Approx 60% complete for SRR1223690.fastq\n",
      "Approx 65% complete for SRR1223690.fastq\n",
      "Approx 70% complete for SRR1223690.fastq\n",
      "Approx 75% complete for SRR1223690.fastq\n",
      "Approx 80% complete for SRR1223690.fastq\n",
      "Approx 85% complete for SRR1223690.fastq\n",
      "Approx 90% complete for SRR1223690.fastq\n",
      "Approx 95% complete for SRR1223690.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223691.fastq\n",
      "Approx 5% complete for SRR1223691.fastq\n",
      "Approx 10% complete for SRR1223691.fastq\n",
      "Approx 15% complete for SRR1223691.fastq\n",
      "Approx 20% complete for SRR1223691.fastq\n",
      "Approx 25% complete for SRR1223691.fastq\n",
      "Approx 30% complete for SRR1223691.fastq\n",
      "Approx 35% complete for SRR1223691.fastq\n",
      "Approx 40% complete for SRR1223691.fastq\n",
      "Approx 45% complete for SRR1223691.fastq\n",
      "Approx 50% complete for SRR1223691.fastq\n",
      "Approx 55% complete for SRR1223691.fastq\n",
      "Approx 60% complete for SRR1223691.fastq\n",
      "Approx 65% complete for SRR1223691.fastq\n",
      "Approx 70% complete for SRR1223691.fastq\n",
      "Approx 75% complete for SRR1223691.fastq\n",
      "Approx 80% complete for SRR1223691.fastq\n",
      "Approx 85% complete for SRR1223691.fastq\n",
      "Approx 90% complete for SRR1223691.fastq\n",
      "Approx 95% complete for SRR1223691.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223692.fastq\n",
      "Approx 5% complete for SRR1223692.fastq\n",
      "Approx 10% complete for SRR1223692.fastq\n",
      "Approx 15% complete for SRR1223692.fastq\n",
      "Approx 20% complete for SRR1223692.fastq\n",
      "Approx 25% complete for SRR1223692.fastq\n",
      "Approx 30% complete for SRR1223692.fastq\n",
      "Approx 35% complete for SRR1223692.fastq\n",
      "Approx 40% complete for SRR1223692.fastq\n",
      "Approx 45% complete for SRR1223692.fastq\n",
      "Approx 50% complete for SRR1223692.fastq\n",
      "Approx 55% complete for SRR1223692.fastq\n",
      "Approx 60% complete for SRR1223692.fastq\n",
      "Approx 65% complete for SRR1223692.fastq\n",
      "Approx 70% complete for SRR1223692.fastq\n",
      "Approx 75% complete for SRR1223692.fastq\n",
      "Approx 80% complete for SRR1223692.fastq\n",
      "Approx 85% complete for SRR1223692.fastq\n",
      "Approx 90% complete for SRR1223692.fastq\n",
      "Approx 95% complete for SRR1223692.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223693.fastq\n",
      "Approx 5% complete for SRR1223693.fastq\n",
      "Approx 10% complete for SRR1223693.fastq\n",
      "Approx 15% complete for SRR1223693.fastq\n",
      "Approx 20% complete for SRR1223693.fastq\n",
      "Approx 25% complete for SRR1223693.fastq\n",
      "Approx 30% complete for SRR1223693.fastq\n",
      "Approx 35% complete for SRR1223693.fastq\n",
      "Approx 40% complete for SRR1223693.fastq\n",
      "Approx 45% complete for SRR1223693.fastq\n",
      "Approx 50% complete for SRR1223693.fastq\n",
      "Approx 55% complete for SRR1223693.fastq\n",
      "Approx 60% complete for SRR1223693.fastq\n",
      "Approx 65% complete for SRR1223693.fastq\n",
      "Approx 70% complete for SRR1223693.fastq\n",
      "Approx 75% complete for SRR1223693.fastq\n",
      "Approx 80% complete for SRR1223693.fastq\n",
      "Approx 85% complete for SRR1223693.fastq\n",
      "Approx 90% complete for SRR1223693.fastq\n",
      "Approx 95% complete for SRR1223693.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223694.fastq\n",
      "Approx 5% complete for SRR1223694.fastq\n",
      "Approx 10% complete for SRR1223694.fastq\n",
      "Approx 15% complete for SRR1223694.fastq\n",
      "Approx 20% complete for SRR1223694.fastq\n",
      "Approx 25% complete for SRR1223694.fastq\n",
      "Approx 30% complete for SRR1223694.fastq\n",
      "Approx 35% complete for SRR1223694.fastq\n",
      "Approx 40% complete for SRR1223694.fastq\n",
      "Approx 45% complete for SRR1223694.fastq\n",
      "Approx 50% complete for SRR1223694.fastq\n",
      "Approx 55% complete for SRR1223694.fastq\n",
      "Approx 60% complete for SRR1223694.fastq\n",
      "Approx 65% complete for SRR1223694.fastq\n",
      "Approx 70% complete for SRR1223694.fastq\n",
      "Approx 75% complete for SRR1223694.fastq\n",
      "Approx 80% complete for SRR1223694.fastq\n",
      "Approx 85% complete for SRR1223694.fastq\n",
      "Approx 90% complete for SRR1223694.fastq\n",
      "Approx 95% complete for SRR1223694.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223695.fastq\n",
      "Approx 5% complete for SRR1223695.fastq\n",
      "Approx 10% complete for SRR1223695.fastq\n",
      "Approx 15% complete for SRR1223695.fastq\n",
      "Approx 20% complete for SRR1223695.fastq\n",
      "Approx 25% complete for SRR1223695.fastq\n",
      "Approx 30% complete for SRR1223695.fastq\n",
      "Approx 35% complete for SRR1223695.fastq\n",
      "Approx 40% complete for SRR1223695.fastq\n",
      "Approx 45% complete for SRR1223695.fastq\n",
      "Approx 50% complete for SRR1223695.fastq\n",
      "Approx 55% complete for SRR1223695.fastq\n",
      "Approx 60% complete for SRR1223695.fastq\n",
      "Approx 65% complete for SRR1223695.fastq\n",
      "Approx 70% complete for SRR1223695.fastq\n",
      "Approx 75% complete for SRR1223695.fastq\n",
      "Approx 80% complete for SRR1223695.fastq\n",
      "Approx 85% complete for SRR1223695.fastq\n",
      "Approx 90% complete for SRR1223695.fastq\n",
      "Approx 95% complete for SRR1223695.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223696.fastq\n",
      "Approx 5% complete for SRR1223696.fastq\n",
      "Approx 10% complete for SRR1223696.fastq\n",
      "Approx 15% complete for SRR1223696.fastq\n",
      "Approx 20% complete for SRR1223696.fastq\n",
      "Approx 25% complete for SRR1223696.fastq\n",
      "Approx 30% complete for SRR1223696.fastq\n",
      "Approx 35% complete for SRR1223696.fastq\n",
      "Approx 40% complete for SRR1223696.fastq\n",
      "Approx 45% complete for SRR1223696.fastq\n",
      "Approx 50% complete for SRR1223696.fastq\n",
      "Approx 55% complete for SRR1223696.fastq\n",
      "Approx 60% complete for SRR1223696.fastq\n",
      "Approx 65% complete for SRR1223696.fastq\n",
      "Approx 70% complete for SRR1223696.fastq\n",
      "Approx 75% complete for SRR1223696.fastq\n",
      "Approx 80% complete for SRR1223696.fastq\n",
      "Approx 85% "
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 33856 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/fastqc_results\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# First we analyse the group of reads that goes from SRR1223686 to SRR1223706\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get fastQC report of first group of reads to  '$output_directory\n",
    "END=706\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "# Now we analyse the second group of reads that go from SRR1288391 to SRR1288408\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get fastQC report of second group of reads to '$output_directory\n",
    "END=408\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting a summary of all 39 fastQC reports with multiQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-20T14:28:34.257054Z",
     "start_time": "2018-02-20T14:28:28.230299Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Searching 79 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 39 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[WARNING]         multiqc : Previous MultiQC output found! Adjusting filenames..\n",
      "[WARNING]         multiqc : Use -f or --force to overwrite existing reports instead\n",
      "[INFO   ]         multiqc : Report      : multiqc_report_1.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data_1\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /projects_eg/projects/simone/glu_paper/fastqc_results\n",
    "pip install multiqc\n",
    "multiqc .\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-20T15:06:20.328830Z",
     "start_time": "2018-02-20T15:06:20.277894Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "<!--\n",
       "  __  __       _ _   _  ___   ____\n",
       " |  \\/  |_   _| | |_(_)/ _ \\ / ___|\n",
       " | |\\/| | | | | | __| | | | | |\n",
       " | |  | | |_| | | |_| | |_| | |___\n",
       " |_|  |_|\\__,_|_|\\__|_|\\__\\_\\\\____|\n",
       "\n",
       "Abandon hope all ye who enter here..\n",
       "http://multiqc.info\n",
       "-->\n",
       "\n",
       "  \n",
       "\n",
       "<meta charset=\"utf-8\">\n",
       "<meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n",
       "<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n",
       "\n",
       "<meta name=\"description\" content=\"MultiQC: A modular tool to aggregate results from bioinformatics analyses across many samples into a single report\">\n",
       "<meta name=\"author\" content=\"MultiQC\">\n",
       "<title>MultiQC Report</title>\n",
       "\n",
       "<!-- JSON plot data -->\n",
       "<script type=\"text/javascript\">\n",
       "mqc_compressed_plotdata = 'N4IgZghgzgLgjgYwPoAcCmAnJAjaalRpwCuaAdgviRADYCWMAnqjQPYwgBcABKCmzCRN0XbiAAezemTQgANGIAmEGHhhRRAbU2gyEALayeIAMoAlMwEYATNYDMANgAcD+UpUQtmywrtOAdNYA7JYALA6WEaFB4UEAugqa1r5B/gCc1gAMDmk5dpZO1k6hoQncmnYp/jE2TkEArGn1Mfl2ZZqhVQ52mWlp2X2WlvWh9e31XfUOQTmRw9mWbYkOVdY5U9EzYWFp7UFV9UPFwYXBmU5jiU5d1sONTi6hlkGZpYlpVSHPmXZrQcF2eKJSyZLqvEK3fpTP7tIafByhTKZKZhJy/Uaw5LcQH+aaOPrTaz1c4Y4GVbGpKJTepOXKhJxIpblMIHabZEaZIajSKXZkTCn+Q7PZxrRprOq87wrAXNOyI8LZBmhOxNWH7GXnIZs2UlN7M64y5UlBwOMUIxoOWEfAXhZw0tKhazhNam9pZVZ9Yq0l5Pex6pI+AWOBkFabKtHdf22Kp2er1HprYYPaaS+wxxXdIJpPzI4ZuzoCyx9PoNB6RJGqxJE3wOdIxIINJFBX5Fk1u6U4kF9fIFZsjYpu9WdhppCEghoxS1Vg2x9IZLLx+mAyK7KvWxz+ApFNFxommtHtHo1/x+Q7/BFZ/6HwMbkb1P2xvrxzn+37HmnhgoZUdZpkVckbrSJRxnGzxBA8f5yu+DR+g8WQuJYh78reMz/DSYTnHiU7lI4x6jNEyozBkiI5IeQ61sSfqRLSmQzL8h4zrW/QqsiIHhGEkHrrWFznA80RyiCDKSoix45AsRZIvBwTtE8x6ZNYUIZNCLqIYkjq+PU/jREMryHP2ClOHEZQgAgrBsBgoggAAxPUCDYBcCAgAAvgougGEYYjmFYtjBkEbggMoqheDedhabYSb2DkiJ1G6vihTSPRFi8vYuNhFRxYKDo9C4TS/LSaUFhupojNu0zgWkDzjKJ5UvL0IL4e0Ha1s4/S0VMPwMtMeyiaEuWEax9hAuUjG4kSNiLEUcoNe8vgBBEuVZkMQyPrCoLYnNBQhtRhRUrCN5zca2wIpCTiqcyWJ+Lipogn4cqFLk1iwgBc0mn4WbdKazhpSy3CIiee72Jy9jZuEsL8n9vxyiaDSIhkA7AtKEPEtDprAfSq7MuqEMSWJPSHD8Q3eAaf0KT0yK/K8yK0laCgQ3Y6INk0NgPW6a0k/T9hhMqvScnGbqBiTNJoiEuT/ImbpYiTMy3JyyJrLYDpuuSJOA+cuTPpC+a05kgSRYqjpTA2UbgzrnMMtkTo6Zh7ba4ENgbCE9NsmdSRYzrzwooC80vIUbrE+7WUhOBtG0tJa629kjoRLc/zgREh5szrBOLeEFVOg2kr5L4aT+Kdo4jCUjjXQ0h4XTnhsMqetKB4eAE57mfYKY6Py9WlUHrSeLhZIX5ySfDOHIak9K2s2veAo0h5NSe2TyfniK6RjFRDqE6TzNmClrCqcoMRlZNe+FWrb4kKq+JYJ4ZBOrzdP00MyWtCmbtmyL8YSdTRDJgZrLn0RIr0RT3pyGSWInSbnrMqeaaJbD9w6OSW4m5eixnNs0BEYQZIFgKP4WiDRHDPAvv2QmowFDDHPhNC4RIXTEkXuEIh1hBTp3AjEIoLwVpqXVDnZU6dYjFFeEWQyakDSpCRK9GIvUHTdHpjJa0mlNQVXpoCU6Ul/TEl8NUZwo9epDCJF8P8hwVFFFyICKE/w0jDBiEZBQJkzKsAssYKyig+pgGcq5EAehDCWS8jYSKDx/KBU8DwbQN5aEqm6B+I2ZNoEXVoQ4GO0Swj6PxK+XwUT7wmLNqMGYfDyiFVrM/HhNhjGEiqtiTSo5ATd1pMMcImTNAdhKbRQ+9JI6nm6sU3OWRbCNEiAiEJ7QZyaSVMSOODZ1h/nXKke8JpXjcNuL1R6wI1o4nRgJHcJQwjfRvOMp00TnC9Feu/YEF1xmRkmiEXoo4noHCdIcJoqEQzfUKgEYo9MGy9FOkIokYNZq5wqhETxyomhzOZB2R5UVmYZN/v6Z4XzTqAkVFMB0pxYQzkeTLEk4i+KLyLNCtEjh3qNkRITd0HdTpPCyrdRo9J+ZfIzk8I4fhgjdAltC84thbkMKzFGZ6udeZQIiPTKEWtiW3AKPTJ4rwoFumQiC5oJjfg1RtsS14Tw+J1CaPGQcXyKomPvJA3Iiw/bZ0wVmD8oxigIieG6dcARszNizBcBwUzHAJ0NfU5aRJcXVKztieuuQHWNFyC+GwpdDXjjjqxJ4xVXxcoyIlCIxIlULkPA8sKiI7zRHXk7JCXQb6h3XsMSIk9RJonycSDWQbj5Dk0gXc44Fnz9FOjvbEK9hhrGyE7J5qbDzrloYcOoWpLyMKoQsnW9pGmEVbfYYSn8AiUQKAbcCvo8xqWAbWEVKTolEjRBwmSsDQqxORCEdqyp1VqQLFkdIvwQm0UijYc4Ml+QYKVQbQojgykyWlMQpKPlfhZCfhPVhNCTy/HuCY/ihs/z0gUDnU0DRNrZjjLGSRChKRNGVMtMpDQ0rKO4LWZ5hw9wmLlkI8xYhTLmUsnYhxTjeAuPcu4iwnjgxpB8R4YKp9qhZg6b1OoE4w7lEiVpZsG9wiGNJYeJJ1R5KvCyhkFugKOjHgUhVL8IILzgSKRuTmTx6RZHkrGSUU9bjHR6PnXi/pyKbhcDqnIm97zyZGl+UcJpnjBETTNAUosbA6YuH21aVRcroSVDEQZe1/OXpGH4Zm+qDlhdOHUOUJYTEXI85QiS7U/V/h+iqLSow4xPCmtkZwnzvWdx7FsQSzZMsdhzj2H8asPqK2BEOHO+bx4RBrfBJFIacrDFSfiSqwJ1w1d6/TAupiOms0NXKc48YeimoaKmG8NWIjCxVMY5sS7+NlxPNsrccotEGyVlNu8PRYyTVGH+J0U21WAhEUMf1krDVRymMuBs8XQZVmq5Z16DJMNIhcBqkrcwmxXOoha6cXzf4jD6M3CmhKxltIeqBkxmZF5HgFAo+89IX0lU9Rs3O0TzgmfvPeAtx9gFzVHNkBsS0fjlpwrAgI2xAT0iLKY19Sb2NcmbBVC4LFYhZuxKFTqtgigbwPSUQtTa6HBIGLg44ZFfAr0ph0h4OLXTHxnLQkIX0X7BEvF29jwH8g0nAjYKI/oRLcAfoULBHVOT9AbB/BQwR0gXD6K8i4nGIPAM0it2wuYVRFl7Tul3tC86chNLppohQCFnvdj8AxLw3sDAIQ+piJKi4LnmvsrJH74rZhhYDf4cxhLqhBJ3U5v9unteqZB7g4zvxU7oiae6SHuD9Jjxw0cqqdXjDWtIsCxRSV5QdYZYyZHrEUfsYuRxLkaOuI8qYBjPkxIsaCv47wXOCidNsDkRm9OkgSfkmcNO7PsjiexEEsaVmvapyoXJEJhRaTXNjxpnJM8RhGxpCyxqGk5y9AZLIIlD2AtKzijipz0xJTaRpR9J1gXDea9ZZAzDtAI5qI/Cxi/gcx+YCgiiOhzak6RCQr47OY/6JhQK9SYhdCDrrz7gPTfQASUgwT5AmKohzawiFT1wv6kyLD5AtSSjDCap86DAtTRKIYIyarBDczP5ijIhqjQohCQLIhsichOrAjIqrxi7ND/A/B/DfRWoXoxCO6Wy3ZpREqXRqqnSmKNDySHBUolYO6ciFDhBTDFqpjbbip5DGgThO5Vh1yYIJhKoyoXyLxXYOF+p9g0gPjVLVhA6DSYF+CFxhCphfY3SuHCrdy0SA7ZYYEcixiOhaqEozj1wKQtS9TQ6iGWpfIgRFy6iTioLHwLIBAAge4jjBJezXiQ6KHDIgSsGviHLpALiR4uH9Bo6MFaQGJNC6EPBE6vjZLVAuEqjoxYK54VDIT9IsGMKKEqiihS5yi4gmpixUyZxDihSODHqowKwl6Nr5BtK7gAolAvCZzWgPzEQOrDBR7PCbR3wu6pBCbKoRjPjSzO7W6aSLDxiogXxfQFTAK0LdwjiNiVLWxqTkgYIgziLxgwZRGW7oJ+7RARDRKdL3Rp6AYjDbKR4VTKgm7vpQbfI7g6raq6guzRDIaCh5T0hj5bqDQyQGh+45SFzcJvKMpqTWgrybC0prC+jySrHYahTGrdIlRiJKokaWLka2Iz5yhz7OKL70beSRQmLr5+LlBb4dwco5BlSaLESxQdyE7gjwSOhZCbbpQdxCx6YnB6YfgyTVGyZQGuHgkaYzrraboJQVR/42m0RoyIKVKLxDgbTnE052gVSEwaGyjwTPj8T5CoHVGNAco6S87PA4GXQgT3B1DunVJwjhnyRfC9xFCH42DVEZy9pzYcrJZFlB6YEPRgQNHMjJqYkFEoJy7fRSpaS0gPCLhOi/yYrApsnIjPxbALDfSxmCgTR2g5gRga76hfLhAiZbwghkkuxYodwFGeFxKNKEpNFaRbjD58HygbkBhbk9Bxr9qIhrBModyPiuYRr5A9FHbvnbDgSFDNDliXbJr/xnhDAmIvAZCPYdxEgOhvG6ZbAKqXRyrRKtTLgWjZEBB8ELrBHXlmEzipDF6bqsEMiNb8YI7kIXhvLnbQLo6LLppCijzQh46XLHDRKzIOh2Hk6rB1DFDIxnhvy1wxhd5Cb5BVIfLHyFREUOgLhKT2CdbHyDwHGzJqyRTdAX7HwdjjIIQl4/ibSQRDiCIgxixNDwXNCviEWSbegtSUzwqvgI5MzcirmyIQYLKpB+CFCCTxiNgEI3i4axh4xvTIHcVZIXRgmPmx7xr5y7AT5WI2JiA2R2QOTUZuRuLGAeKr4ZCGlsbYhnw+YOhUyyWyLWm/B0Lxju5iS8Ro4SYxCAVfj5CvBHxZLHjPJIhhC1rcbkWaDIS1jBD4ybrmwNiEyGYk50jIj0HyYWYKzMztb5F+C9KKZFAvmjD9BDDnhZkdzyTm7cavLwaFkBBIipwkg8iRDyYVk4gVQfjPD0jUjqYxYeZ87bjaT/CZbjEw6oZEQPiQqcEjkmjwUl4WWS7AjITsL2zOB3DxigESElYUwtpzqVJxzyFA5JRnZVnu6EwFAhr0z3TVwcQnrMhDaBD3iVYKUmg2aTYlZ7i15oZhj2HZZEiR48isE5AxHbYLiODbhOjCypj+E2D6ZSGVYaKCo5HP69RBw05mJVig2YKs4ZB7kqjwVRjTnxjUi7iOi+iEpLncZAUSXNAA4Q6FiGychCat7RJVGFhwXVxhjD6vjuWXmqFKh8pcydEY4uCnQpim5PCExvh5WYK7imjihqJCzCXYg6yvSeggF8UjCc5C4niwx4xRxBGC6zjJj9VrLox3B7FMTyjXrASBonHHhdw8bBzPyQQjQn7AzbUTWvnHzrgryiwlBkWEjNzCQLKhToVXyYlQJToSbKamgQGUTihALsY3zbKnQ0h2biFZIAQjqmhAwg4nD3VZJnrWrBiaijlCgFT8hfwbqs3/ASiuY0nW5nEcgFBnUralgyTl7lw1qbSmK1p3n17EJnAXDEikpah17Wg2CBCMwQjTC9i8T95EI6w7Hc2WwcWbAqmT4JXWQamhBakL50YZUr76l2A5Wb43hnwdIchioxA8KpjH7rx5xq68LyYARBKQWqr/xHDybzEQVJ1iKAyLy9WYLmnQFeyzD+hTxTLU62E3UHiJAWaUwIgqigRyIuwV3yjDIqowSbU4ioRxhQWGxfiFl/HvEQhnh6a5ChYCg1D2Cc0AVY6QoDHBYx6sFOiNIMGfAlDfil5CSY2/WVK8KbQmgC4g0uos5i7yghGwjTlqoEqPiRBGzI2XSpQk3OAgEIjdVY3Epv0e7NDuqQoGEQQk2OyihuOU1hPoE8gzBTKhH7TpC0SRijj1pWbuHYqlnRwdbxx+HUo95zaegdki0BB9h1UPQnWkj8bDk1BE57nXpFDIVzRMJCZ0gfjNWuxbm2b66l3k0GoumlQPDERQKXYGFVlarzA0jdCer22eiXj/n/LT0VCFNgTDzkz3YxQ8UumPH1S85PF3n0zUrBBTHSwQW2Cx2XSiN/yx652vjDmAhrbglvRpp7F/Hu4IZ/ypIlwVpVBVnyiuFojIk4RwE943zGocwE0VD13VC3ShjmjZh+AFTt0TF+CyJE4zBthqSBKgLNDtW6j5D9Uj1h10IH0AO0RKQFSM47aji2BoYwryRoIu5VobYB4VZ8T3ph6BCIhbh0TP63TH0YIH0iPbObRl5EKD7oVVKxhdI8lEJnzOZE5bwiZXXt4taLR9p1TyzKjAMN6CYaUdTPDnGUpxVqmJXwOINpVL6ZX6mhAYPGlYPVA3xcirIESCP8YSbsT4xLPzxS04QSY2aRDzVA3MmtUdQvC3CY7QvlCsOtorWdJknVKGaijRT4T1iSjTUWyLBoitSxyLXYh9W7JCZkn3hHNrMcREFyhrYvDfS7O5AgxklMxRC6OXXFDDKWM+VUGPXsjVwXZ86tmpDFjy0jK3CZnAi/XrqWwhzTYfZ8hPb9i2FbRPACFfawr9tdK4KZbNaBDFNvxzZjmY3FHXv4O3WnTNiEk0yw3lgbxrYZBmELI1b/bUj1WzmLaGpFC/KdKEQ4oVNU2tDmzOCvrnG/nZZebbCeImLFChG/XSaiLgRo0hAwU5EWyGIiozJDOSbm6yJktQxYUHFKT3Z2a0SXZWV6QRaWMFCAGpgI6OAog+SoQYF20iURhN1NysGu04hxgOqHqFz0zsjBrW4tazw7Cyixhyyh1FCCiPGxILrRP0RSXsZRPPBeVoSAUp3No3UqiYfGpDB7EN3Q71qMfvFtzLy4jRD5S27OVTA3GhS5aPFPJfhEjPERttqgZHVMxOg/F+3CqUQiziK6ggk24WyOxdKOimfLq/G5z7YGwfkhwEKwJgmUkSUrjaQQZnqUOtg/zli4ISvcBFj+0uACSPkYQFlqQfqaTOiqfRBTA7KXNZLl60L83yLTa1kpi6vcCKcdWFaYamoFTWhEWc3EgfF5bdBk7ZtrSroVT/YfmTgfIutT7qlUbz6eu6mMbrB+sBJc7v25jjQ5mYsCaV34j3Y7A+0SZ5Q/CMgIjP4FRyQwrFq4oVUIjv7+3IF0ibPSlhkbgO7B6RNGEG3lDCNsGOh2qqd/gV1bgOjzYOo3zyPqP3bKZkLVtqPVD4zNv2CqoXDDt/Gku9QOmKgKSTsKOWdVLvHzjWN6NCgugZt75HNZY5ztKtSjsTX3jFai0RpfH/KsR+MurjgOmOioi+GYyap6RcXljaGSWbkdxNDTDfjTBuP/wfsWFq8H2EsEhRgXlwyTj3Tn0xGFNmW0oQQZt8xVgXTWryxwQNWczIePK8KeWiIGSUv8bJrniROmpJSALS3UrURGx8o6oxHTn7iHo/5oS9C0cgR/2FZ3QEXVFR73HNix4xEGFWwc1MwmJo721wRnjGreOsV6NogzEKR07RI+2mO+pARZBkzdn/gHBPKlo66/po7SVGp3ZRUWyaUDytXgQKTd2LCAF7GaSK28HprcwWWK7S6kybCFEMmWUZRAxU6DLAZJZ13saUStBZDbKJhuUu45wtA2reiNXg5ZKfxDzdAB6B2/vhhMsgJY/D79DRRzB0OwJsGQiDqGBSZjCR0EKGAuMmEaCNUqED6MUk0lTR8E5QBHVrnq0FBClUKtLFUNfVpI8ppWEJX0ICDG5MQgK8afqiRGIEikFAVac2DalbTNhU4NrFeDrlawoDn8BHPbrA0oyz5UqtGdKp5FQb4hXAFiXxLlTKq5BMOSmBlo8FKqhREa6SJqtoU9QZRLGzCPKFBWgpqR3w73KIE4ShAuxWGhwUYApAmhdckQBmd8A321T4NLiaUCzPhjHI2YrqkUOthuA0TXRFgEkTdPIwCDSssEh6NVHIXmRfIcwcYBAsE1UbAhCmOYInA1zliowGezREGPykkhQhIUXKM7Drimi1AnSP0P6E4RBCTQYco7EXjnCczksfQ6ISFF9jua1kfMxIX2E1kNQ/J3a8KchOcnUItDOMi4ZGGhHOpE1PQlSFbIVjh5JB/2buPJroU6TOAbeLQ4eNgJ2j5E3y2WUWEs3Vhi4zC/hecKWmxzJhUBgfQ1A8EvCjALwH9QjjnG9CthnC3tQ/GsENQNg78JQGumMOCAPCaefbVOBoidJFBDUJodCBTBwQIoLa2WRQu8UaFixzajRQ1NWhczBhiQSxcTgBzEjtJEWX0TOC71zgAVTw1wnMqHUujt8PaMqH9CgX04CgFIPoMnozDc4p0ckcwFAbuAGi2cRyEJWPGzmzCr8yqIwEEIyGbj3Zt+zLSEIsCMR5kORR/a3JSEoh842QCHNui7gbpoU7greXmM2AS4DcGEdfRKAQyZZ1d/4eIQLkmTGHKgiEfVatj8H3y+RwBRCUKOLizBKpmIU0GrhXj7D7hUQm8R2MfWSE3xJBb8b9rgJwzwIN4rKVlHRDG4rx7gToFiLalhRf0VEUnHZJIOFHygbWoUH4PuwMYpDl6PVT+LnG2CnR4sJIbHPEC4HT5Du2pZBoIL1L4g/Iog1jJgy+Q5A1qGBJSAfUIYulQIqLSrpnGqKthdaFgn9N9yPKmVMIxqJGokH+Yw5SUZyDkCNS3I1ptkMAnMvYNCHMRbCbUVoMmVCEfRDETeQKr4KAwKxHgXMHXAdV/r9jocawAlsO2wpHtziQeJ8HeXrJ/kC4p4HQnpnnZFMeOXlf5DAI4JfIrkjgYpqp3NQi8vRmoYwexAD5Sh/xbOHjFJ1lAHklymoFzLQ0tjlkNCrEdCFIW/BXhBs/mBAuaiBJ4wsmC7VsM5gBB8sowhTMLrkjqpFh5M0YDzI8MXALYWcyHP4kDFpbU4QYLsMIjiAdQKhxQTVCZrERxCBUkoZFTYAxO0pAZoQ69fEgKyrBGUgMW4PlOeAdxRgRopfXfviRZy8SuIBOZPPOHuwsFnU9bXEMwgJi84Rx4nYyu1hmD4ZyYwNHCAMXhqxhPoSPPjF3wFAtxo4BRZBJ0JwjzFnMKORIhpQKC0jie34SMcVygm4RWkP4V+s8jHqci/cxyUYrzB7A3Eq0RA8qELAdA+1sWdEqKMgVWp0NiWKSXFLVG5rBDH+vgHWAyAAmqELwoVDoMAnYRGxVUGiMkoDxLG2JbI9kWyHwJ1IoMqxrNc7iaXphaRtmx0YfA3wYkZR3iuCPlAE0giLSM+bjbpJPy9LmSSpOkTDpJCUTHhzw/EI0JOjHIQ9awPOH5L+zUTLjzJpyBEfWCpFuDmo3bW1OyH2zyYDC+4bXkdTomrdvAF5aGF9xyj64vO0QhsWOU0QeSHcSQyTJo1uipQKeZIalMJNGzNBGhB5AsBDH+peVIw8YQKd4BN"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 1521022 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import HTML\n",
    "\n",
    "HTML(filename=\"/projects_eg/projects/simone/glu_paper/fastqc_results/multiqc_report.html\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Removing adapters using cutadapt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Creating fasta file with adapters to trim"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As we have SINGLE end reads we create a fasta file with adapters that we find in ~/Downloads/Trimmomatic-0.36/adapters:\n",
    "\n",
    "TruSeq2-SE.fa\n",
    "TruSeq3-SE.fa\n",
    "\n",
    "and add the polyA tail adapter: \n",
    ">polyA_adapter\n",
    "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-21T13:38:26.323123Z",
     "start_time": "2018-02-21T13:38:26.273469Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd ~/Downloads/Trimmomatic-0.36/adapters\n",
    "cat TruSeq2-SE.fa TruSeq3-SE.fa > adapters.fa\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trimming differently RiboSeq and RNAseq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Riboseq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-22T14:53:20.659068Z",
     "start_time": "2018-02-22T09:24:04.094060Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get trimmed reads of first group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/SRR1223686_trimmed.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223686.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1662.32 s (72 us/read; 0.84 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              23,135,864\n",
      "Reads with adapters:                22,916,601 (99.1%)\n",
      "Reads that were too short:          14,159,805 (61.2%)\n",
      "Reads written (passing filters):     8,976,059 (38.8%)\n",
      "\n",
      "Total basepairs processed:   971,706,288 bp\n",
      "Total written (filtered):    246,794,987 bp (25.4%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 694 times.\n",
      "676 times, it overlapped the 5' end of a read\n",
      "18 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t31\t22593.6\t0\t31\n",
      "6\t19\t5648.4\t0\t19\n",
      "9\t1\t88.3\t0\t1\n",
      "10\t1\t22.1\t1\t0 1\n",
      "12\t2\t1.4\t1\t2\n",
      "14\t1\t0.1\t1\t1\n",
      "15\t7\t0.0\t1\t7\n",
      "16\t9\t0.0\t1\t6 3\n",
      "17\t17\t0.0\t1\t10 7\n",
      "18\t34\t0.0\t1\t21 13\n",
      "19\t208\t0.0\t1\t197 11\n",
      "20\t9\t0.0\t2\t1 0 8\n",
      "21\t46\t0.0\t2\t28 6 12\n",
      "22\t55\t0.0\t2\t4 32 19\n",
      "23\t78\t0.0\t2\t52 15 11\n",
      "24\t70\t0.0\t2\t0 12 58\n",
      "25\t46\t0.0\t2\t0 27 19\n",
      "26\t28\t0.0\t2\t0 14 14\n",
      "27\t9\t0.0\t2\t0 1 8\n",
      "28\t5\t0.0\t2\t0 0 5\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t17\t22593.6\t0\t17\n",
      "6\t1\t5648.4\t0\t1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 625 times.\n",
      "625 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t86\t22593.6\t0\t86\n",
      "6\t24\t5648.4\t0\t24\n",
      "7\t508\t1412.1\t0\t508\n",
      "8\t2\t353.0\t0\t2\n",
      "9\t1\t88.3\t0\t0 1\n",
      "10\t4\t22.1\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 23 times.\n",
      "23 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t18\t22593.6\t0\t18\n",
      "6\t1\t5648.4\t0\t1\n",
      "7\t2\t1412.1\t0\t2\n",
      "11\t1\t5.5\t1\t0 1\n",
      "12\t1\t1.4\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 66 times.\n",
      "66 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t49\t22593.6\t0\t49\n",
      "6\t8\t5648.4\t0\t8\n",
      "7\t1\t1412.1\t0\t1\n",
      "8\t1\t353.0\t0\t1\n",
      "10\t1\t22.1\t1\t0 1\n",
      "11\t4\t5.5\t1\t0 4\n",
      "12\t1\t1.4\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 65 times.\n",
      "65 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t41\t22593.6\t0\t41\n",
      "6\t10\t5648.4\t0\t10\n",
      "9\t6\t88.3\t0\t0 6\n",
      "10\t6\t22.1\t1\t0 6\n",
      "11\t1\t5.5\t1\t0 1\n",
      "12\t1\t1.4\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 22915128 times.\n",
      "21730 times, it overlapped the 5' end of a read\n",
      "22893398 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1291\t22593.6\t0\t1291\n",
      "6\t372\t5648.4\t0\t372\n",
      "7\t634\t1412.1\t0\t634\n",
      "8\t220\t353.0\t0\t220\n",
      "9\t49\t88.3\t0\t49\n",
      "10\t96\t22.1\t1\t36 60\n",
      "11\t37\t5.5\t1\t16 21\n",
      "12\t48\t1.4\t1\t19 29\n",
      "13\t49\t0.3\t1\t21 28\n",
      "14\t61\t0.1\t1\t28 33\n",
      "15\t38\t0.0\t1\t16 22\n",
      "16\t64\t0.0\t1\t16 48\n",
      "17\t24\t0.0\t1\t15 9\n",
      "18\t20\t0.0\t1\t10 10\n",
      "19\t73\t0.0\t1\t20 53\n",
      "20\t710\t0.0\t2\t42 188 480\n",
      "21\t2552\t0.0\t2\t81 794 1677\n",
      "22\t4567\t0.0\t2\t39 1513 3015\n",
      "23\t2153\t0.0\t2\t33 750 1370\n",
      "24\t1195\t0.0\t2\t80 389 726\n",
      "25\t426\t0.0\t2\t12 155 259\n",
      "26\t296\t0.0\t2\t44 77 175\n",
      "27\t123\t0.0\t2\t26 23 74\n",
      "28\t49\t0.0\t2\t8 12 29\n",
      "29\t67\t0.0\t2\t14 19 34\n",
      "30\t6516\t0.0\t3\t5479 585 252 200\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2610\t22593.6\t0\t2610\n",
      "6\t2415\t5648.4\t0\t2415\n",
      "7\t1762\t1412.1\t0\t1762\n",
      "8\t2029\t353.0\t0\t2029\n",
      "9\t7499\t88.3\t0\t7499\n",
      "10\t28199\t22.1\t1\t23939 4260\n",
      "11\t107369\t5.5\t1\t102004 5365\n",
      "12\t397856\t1.4\t1\t382049 15807\n",
      "13\t917872\t0.3\t1\t881125 36747\n",
      "14\t1440008\t0.1\t1\t1329682 110326\n",
      "15\t1805613\t0.0\t1\t1546686 258927\n",
      "16\t4039737\t0.0\t1\t3604992 434745\n",
      "17\t5522781\t0.0\t1\t5024630 498151\n",
      "18\t3470855\t0.0\t1\t3060186 410669\n",
      "19\t1738139\t0.0\t1\t1307987 430152\n",
      "20\t1336213\t0.0\t2\t482606 480961 372646\n",
      "21\t844801\t0.0\t2\t110664 349317 384820\n",
      "22\t520228\t0.0\t2\t25636 182726 311866\n",
      "23\t260171\t0.0\t2\t4170 59985 196016\n",
      "24\t181418\t0.0\t2\t2279 34996 144143\n",
      "25\t77712\t0.0\t2\t2564 21504 53644\n",
      "26\t41139\t0.0\t2\t2127 14728 24284\n",
      "27\t23539\t0.0\t2\t2265 11094 10180\n",
      "28\t12538\t0.0\t2\t1725 6721 4092\n",
      "29\t13903\t0.0\t2\t2588 7633 3682\n",
      "30\t17161\t0.0\t3\t2723 6888 2708 4842\n",
      "31\t13673\t0.0\t3\t2925 5881 2370 2497\n",
      "32\t13865\t0.0\t3\t4294 6912 1400 1259\n",
      "33\t7836\t0.0\t3\t3052 3491 713 580\n",
      "34\t7165\t0.0\t3\t3417 2894 524 330\n",
      "35\t6572\t0.0\t3\t3667 2163 460 282\n",
      "36\t10914\t0.0\t3\t7080 2985 598 251\n",
      "37\t5605\t0.0\t3\t3894 1238 299 174\n",
      "38\t5587\t0.0\t3\t4141 992 310 144\n",
      "39\t4262\t0.0\t3\t3249 621 229 163\n",
      "40\t2763\t0.0\t3\t2057 346 184 176\n",
      "41\t1589\t0.0\t3\t1323 129 57 80\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/SRR1223687_trimmed.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223687.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1500.33 s (93 us/read; 0.64 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,101,967\n",
      "Reads with adapters:                 8,082,132 (50.2%)\n",
      "Reads that were too short:             350,581 (2.2%)\n",
      "Reads written (passing filters):    15,751,386 (97.8%)\n",
      "\n",
      "Total basepairs processed:   821,200,317 bp\n",
      "Total written (filtered):    683,156,660 bp (83.2%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 3681670 times.\n",
      "4061 times, it overlapped the 5' end of a read\n",
      "3677609 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2356\t15724.6\t0\t2356\n",
      "6\t1532\t3931.1\t0\t1532\n",
      "7\t4\t982.8\t0\t4\n",
      "8\t4\t245.7\t0\t4\n",
      "9\t4\t61.4\t0\t0 4\n",
      "10\t9\t15.4\t1\t0 9\n",
      "11\t3\t3.8\t1\t0 3\n",
      "12\t5\t1.0\t1\t0 5\n",
      "16\t4\t0.0\t1\t4\n",
      "17\t47\t0.0\t1\t33 14\n",
      "18\t2\t0.0\t1\t0 2\n",
      "19\t8\t0.0\t1\t0 8\n",
      "20\t22\t0.0\t2\t17 0 5\n",
      "21\t8\t0.0\t2\t3 2 3\n",
      "22\t9\t0.0\t2\t3 3 3\n",
      "23\t4\t0.0\t2\t1 0 3\n",
      "24\t12\t0.0\t2\t0 7 5\n",
      "25\t10\t0.0\t2\t0 2 8\n",
      "26\t9\t0.0\t2\t0 2 7\n",
      "27\t4\t0.0\t2\t0 0 4\n",
      "28\t2\t0.0\t2\t0 0 2\n",
      "29\t2\t0.0\t2\t0 0 0 2\n",
      "30\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t390070\t15724.6\t0\t390070\n",
      "6\t424770\t3931.1\t0\t424770\n",
      "7\t364806\t982.8\t0\t364806\n",
      "8\t394527\t245.7\t0\t394527\n",
      "9\t353317\t61.4\t0\t349958 3359\n",
      "10\t675913\t15.4\t1\t328084 347829\n",
      "11\t476625\t3.8\t1\t251612 225013\n",
      "12\t337531\t1.0\t1\t151497 186034\n",
      "13\t256660\t0.2\t1\t119597 137063\n",
      "14\t2399\t0.1\t1\t33 2366\n",
      "15\t931\t0.0\t1\t0 931\n",
      "16\t59\t0.0\t1\t0 59\n",
      "17\t1\t0.0\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 8758 times.\n",
      "8246 times, it overlapped the 5' end of a read\n",
      "512 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1785\t15724.6\t0\t1785\n",
      "6\t345\t3931.1\t0\t345\n",
      "7\t6055\t982.8\t0\t6055\n",
      "8\t55\t245.7\t0\t55\n",
      "9\t3\t61.4\t0\t2 1\n",
      "10\t2\t15.4\t1\t0 2\n",
      "11\t1\t3.8\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t173\t0.2\t1\t0 173\n",
      "14\t252\t0.1\t1\t127 125\n",
      "15\t87\t0.0\t1\t0 87\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 1348 times.\n",
      "1348 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1245\t15724.6\t0\t1245\n",
      "6\t13\t3931.1\t0\t13\n",
      "7\t2\t982.8\t0\t2\n",
      "8\t85\t245.7\t0\t85\n",
      "9\t1\t61.4\t0\t0 1\n",
      "11\t2\t3.8\t1\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 524337 times.\n",
      "888 times, it overlapped the 5' end of a read\n",
      "523449 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t331\t15724.6\t0\t331\n",
      "6\t20\t3931.1\t0\t20\n",
      "7\t2\t982.8\t0\t2\n",
      "8\t1\t245.7\t0\t1\n",
      "9\t4\t61.4\t0\t1 3\n",
      "10\t26\t15.4\t1\t0 26\n",
      "11\t26\t3.8\t1\t0 26\n",
      "12\t10\t1.0\t1\t0 10\n",
      "13\t1\t0.2\t1\t0 1\n",
      "22\t1\t0.0\t2\t1\n",
      "24\t12\t0.0\t2\t12\n",
      "25\t1\t0.0\t2\t1\n",
      "26\t9\t0.0\t2\t7 2\n",
      "27\t24\t0.0\t2\t24\n",
      "28\t22\t0.0\t2\t21 1\n",
      "29\t22\t0.0\t2\t20 1 0 1\n",
      "30"
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 120509 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/cutadapt/\"\n",
    "\n",
    "\n",
    "# First we trim the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get trimmed reads of first group of Riboseq reads to  '$output_directory\n",
    "END=688\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the second group of reads that go from SRR1223697 to SRR1223699\n",
    "echo 'get trimmed reads of second group of Riboseq reads to  '$output_directory\n",
    "END=699\n",
    "for ((i=697;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the third group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get trimmed reads of third group of Riboseq reads to  '$output_directory\n",
    "END=396\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the fourth group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get trimmed reads of fourth group of Riboseq reads to  '$output_directory\n",
    "END=404\n",
    "for ((i=400;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### RNAseq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-22T23:53:15.078423Z",
     "start_time": "2018-02-22T15:08:19.901642Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get trimmed reads of first group of RNAseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223689_trimmed.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223689.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223690_trimmed.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223690.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 2316.77 s (71 us/read; 0.85 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              32,658,689\n",
      "Reads with adapters:                32,334,387 (99.0%)\n",
      "Reads that were too short:          25,263,728 (77.4%)\n",
      "Reads written (passing filters):     7,394,961 (22.6%)\n",
      "\n",
      "Total basepairs processed: 1,632,934,450 bp\n",
      "Quality-trimmed:             258,823,625 bp (15.9%)\n",
      "Total written (filtered):    207,064,240 bp (12.7%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 41935 times.\n",
      "863 times, it overlapped the 5' end of a read\n",
      "41072 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t534\t31893.3\t0\t534\n",
      "6\t202\t7973.3\t0\t202\n",
      "7\t12\t1993.3\t0\t12\n",
      "8\t2\t498.3\t0\t2\n",
      "9\t14\t124.6\t0\t0 14\n",
      "10\t10\t31.1\t1\t0 10\n",
      "17\t3\t0.0\t1\t3\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t3\t0.0\t1\t0 3\n",
      "20\t7\t0.0\t2\t6 0 1\n",
      "21\t9\t0.0\t2\t8 0 1\n",
      "22\t12\t0.0\t2\t2 9 1\n",
      "23\t8\t0.0\t2\t1 3 4\n",
      "24\t9\t0.0\t2\t1 1 7\n",
      "25\t26\t0.0\t2\t0 2 24\n",
      "26\t4\t0.0\t2\t0 1 3\n",
      "27\t2\t0.0\t2\t0 1 1\n",
      "28\t1\t0.0\t2\t0 0 1\n",
      "29\t1\t0.0\t2\t0 0 1\n",
      "30\t2\t0.0\t3\t0 0 1 1\n",
      "31\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t8997\t31893.3\t0\t8997\n",
      "6\t17961\t7973.3\t0\t17961\n",
      "7\t1207\t1993.3\t0\t1207\n",
      "8\t3011\t498.3\t0\t3011\n",
      "9\t7706\t124.6\t0\t7705 1\n",
      "10\t508\t31.1\t1\t404 104\n",
      "11\t995\t7.8\t1\t878 117\n",
      "12\t486\t1.9\t1\t457 29\n",
      "13\t201\t0.5\t1\t191 10\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 234 times.\n",
      "234 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t172\t31893.3\t0\t172\n",
      "6\t30\t7973.3\t0\t30\n",
      "7\t32\t1993.3\t0\t32\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 237 times.\n",
      "229 times, it overlapped the 5' end of a read\n",
      "8 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t131\t31893.3\t0\t131\n",
      "6\t41\t7973.3\t0\t41\n",
      "7\t33\t1993.3\t0\t33\n",
      "9\t3\t124.6\t0\t0 3\n",
      "10\t13\t31.1\t1\t0 13\n",
      "27\t1\t0.0\t2\t1\n",
      "29\t1\t0.0\t2\t0 1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t4\t0.0\t3\t4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.1\t1\t0 1\n",
      "16\t1\t0.0\t1\t0 1\n",
      "17\t1\t0.0\t1\t1\n",
      "22\t1\t0.0\t2\t1\n",
      "26\t1\t0.0\t2\t1\n",
      "27\t2\t0.0\t2\t2\n",
      "47\t1\t0.0\t3\t1\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 5003 times.\n",
      "587 times, it overlapped the 5' end of a read\n",
      "4416 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t246\t31893.3\t0\t246\n",
      "6\t58\t7973.3\t0\t58\n",
      "7\t2\t1993.3\t0\t2\n",
      "8\t1\t498.3\t0\t1\n",
      "9\t1\t124.6\t0\t0 1\n",
      "10\t13\t31.1\t1\t0 13\n",
      "12\t1\t1.9\t1\t1\n",
      "16\t1\t0.0\t1\t1\n",
      "17\t1\t0.0\t1\t1\n",
      "22\t4\t0.0\t2\t4\n",
      "23\t4\t0.0\t2\t4\n",
      "24\t10\t0.0\t2\t10\n",
      "25\t2\t0.0\t2\t2\n",
      "26\t93\t0.0\t2\t86 6 1\n",
      "27\t29\t0.0\t2\t27 2\n",
      "28\t19\t0.0\t2\t19\n",
      "29\t18\t0.0\t2\t18\n",
      "30\t10\t0.0\t3\t9 1\n",
      "31\t30\t0.0\t3\t27 3\n",
      "32\t29\t0.0\t3\t24 5\n",
      "33\t6\t0.0\t3\t6\n",
      "34\t9\t0.0\t3\t7 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.5\t1\t0 2\n",
      "14\t638\t0.1\t1\t501 137\n",
      "15\t495\t0.0\t1\t407 88\n",
      "16\t484\t0.0\t1\t387 97\n",
      "17\t1002\t0.0\t1\t816 186\n",
      "18\t60\t0.0\t1\t51 9\n",
      "19\t58\t0.0\t1\t49 8 1\n",
      "20\t74\t0.0\t2\t63 9 2\n",
      "21\t146\t0.0\t2\t115 30 1\n",
      "22\t24\t0.0\t2\t21 2 1\n",
      "23\t139\t0.0\t2\t115 22 2\n",
      "24\t138\t0.0\t2\t114 18 6\n",
      "25\t188\t0.0\t2\t149 31 8\n",
      "26\t45\t0.0\t2\t37 7 1\n",
      "27\t71\t0.0\t2\t55 11 5\n",
      "28\t146\t0.0\t2\t116 24 6\n",
      "29\t496\t0.0\t2\t397 77 20 2\n",
      "30\t12\t0.0\t3\t11 1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t6\t0.0\t3\t6\n",
      "34\t4\t0.0\t3\t4\n",
      "38\t5\t0.0\t3\t4 0 1\n",
      "39\t7\t0.0\t3\t4 3\n",
      "40\t7\t0.0\t3\t5 2\n",
      "41\t19\t0.0\t3\t18 0 1\n",
      "42\t35\t0.0\t3\t28 4 2 1\n",
      "43\t25\t0.0\t3\t20 5\n",
      "44\t21\t0.0\t3\t16 5\n",
      "45\t25\t0.0\t3\t20 2 2 1\n",
      "46\t11\t0.0\t3\t9 1 0 1\n",
      "47\t13\t0.0\t3\t12 1\n",
      "48\t8\t0.0\t3\t6 1 1\n",
      "49\t11\t0.0\t3\t10 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 84 times.\n",
      "84 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t75\t31893.3\t0\t75\n",
      "6\t7\t7973.3\t0\t7\n",
      "9\t1\t124.6\t0\t0 1\n",
      "11\t1\t7.8\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 32286894 times.\n",
      "18125 times, it overlapped the 5' end of a read\n",
      "32268769 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t614\t31893.3\t0\t614\n",
      "6\t86\t7973.3\t0\t86\n",
      "7\t58\t1993.3\t0\t58\n",
      "8\t12\t498.3\t0\t12\n",
      "9\t6\t124.6\t0\t6\n",
      "10\t18\t31.1\t1\t10 8\n",
      "11\t21\t7.8\t1\t10 11\n",
      "12\t34\t1.9\t1\t11 23\n",
      "13\t44\t0.5\t1\t20 24\n",
      "14\t39\t0.1\t1\t18 21\n",
      "15\t92\t0.0\t1\t28 64\n",
      "16\t94\t0.0\t1\t21 73\n",
      "17\t113\t0.0\t1\t26 87\n",
      "18\t212\t0.0\t1\t37 175\n",
      "19\t288\t0.0\t1\t36 252\n",
      "20\t632\t0.0\t2\t28 414 190\n",
      "21\t1635\t0.0\t2\t22 1106 507\n",
      "22\t4042\t0.0\t2\t45 2269 1728\n",
      "23\t4118\t0.0\t2\t21 1449 2648\n",
      "24\t2185\t0.0\t2\t9 419 1757\n",
      "25\t991\t0.0\t2\t11 200 780\n",
      "26\t417\t0.0\t2\t5 140 272\n",
      "27\t269\t0.0\t2\t4 116 149\n",
      "28\t121\t0.0\t2\t3 57 61\n",
      "29\t118\t0.0\t2\t3 63 52\n",
      "30\t1866\t0.0\t3\t975 315 161 415\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t8125\t31893.3\t0\t8125\n",
      "6\t8763\t7973.3\t0\t8763\n",
      "7\t9606\t1993.3\t0\t9606\n",
      "8\t10381\t498.3\t0\t10381\n",
      "9\t12852\t124.6\t0\t12852\n",
      "10\t18068\t31.1\t1\t16280 1788\n",
      "11\t23932\t7.8\t1\t21673 2259\n",
      "12\t32249\t1.9\t1\t29354 2895\n",
      "13\t49479\t0.5\t1\t45616 3863\n",
      "14\t92025\t0.1\t1\t87222 4803\n",
      "15\t192168\t0.0\t1\t184884 7284\n",
      "16\t449333\t0.0\t1\t436787 12546\n",
      "17\t1109953\t0.0\t1\t1085828 24125\n",
      "18\t3458331\t0.0\t1\t3410524 47807\n",
      "19\t7691576\t0.0\t1\t7550343 141233\n",
      "20\t6345557\t0.0\t2\t5582507 442437 320613\n",
      "21\t3269961\t0.0\t2\t1123792 1145508 1000661\n",
      "22\t3772792\t0.0\t2\t183383 1284816 2304593\n",
      "23\t3001734\t0.0\t2\t35114 661895 2304725\n",
      "24\t1516406\t0.0\t2\t11240 262533 1242633\n",
      "25\t602523\t0.0\t2\t4882 95168 502473\n",
      "26\t227004\t0.0\t2\t2149 44588 180267\n",
      "27\t93443\t0.0\t2\t1778 26065 65600\n",
      "28\t52041\t0.0\t2\t2021 27166 22854\n",
      "29\t44804\t0.0\t2\t2221 26727 15856\n",
      "30\t62602\t0.0\t3\t2455 21767 10209 28171\n",
      "31\t31170\t0.0\t3\t2791 17463 3760 7156\n",
      "32\t19991\t0.0\t3\t3600 12869 1515 2007\n",
      "33\t14582\t0.0\t3\t4111 8991 767 713\n",
      "34\t11240\t0.0\t3\t4395 6041 482 322\n",
      "35\t8793\t0.0\t3\t4500 3732 341 220\n",
      "36\t7011\t0.0\t3\t4312 2353 225 121\n",
      "37\t5275\t0.0\t3\t3763 1276 146 90\n",
      "38\t4056\t0.0\t3\t3246 665 97 48\n",
      "39\t3070\t0.0\t3\t2637 333 67 33\n",
      "40\t2307\t0.0\t3\t2049 201 37 20\n",
      "41\t1732\t0.0\t3\t1576 113 30 13\n",
      "42\t1312\t0.0\t3\t1214 56 30 12\n",
      "43\t880\t0.0\t3\t832 33 11 4\n",
      "44\t643\t0.0\t3\t603 20 14 6\n",
      "45\t451\t0.0\t3\t425 11 7 8\n",
      "46\t302\t0.0\t3\t286 8 6 2\n",
      "47\t163\t0.0\t3\t158 0 4 1\n",
      "48\t64\t0.0\t3\t60 2 2\n",
      "49\t19\t0.0\t3\t19\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223691_trimmed.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223691.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1382.94 s (71 us/read; 0.85 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,491,861\n",
      "Reads with adapters:                19,266,461 (98.8%)\n",
      "Reads that were too short:          13,496,675 (69.2%)\n",
      "Reads written (passing filters):     5,995,186 (30.8%)\n",
      "\n",
      "Total basepairs processed:   818,658,162 bp\n",
      "Quality-trimmed:              11,550,283 bp (1.4%)\n",
      "Total written (filtered):    164,498,660 bp (20.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 1948 times.\n",
      "1918 times, it overlapped the 5' end of a read\n",
      "30 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t225\t19035.0\t0\t225\n",
      "6\t58\t4758.8\t0\t58\n",
      "7\t5\t1189.7\t0\t5\n",
      "8\t3\t297.4\t0\t3\n",
      "9\t6\t74.4\t0\t4 2\n",
      "10\t9\t18.6\t1\t4 5\n",
      "11\t4\t4.6\t1\t1 3\n",
      "12\t1\t1.2\t1\t1\n",
      "13\t1\t0.3\t1\t0 1\n",
      "14\t2\t0.1\t1\t2\n",
      "15\t3\t0.0\t1\t2 1\n",
      "16\t3\t0.0\t1\t2 1\n",
      "17\t7\t0.0\t1\t4 3\n",
      "18\t24\t0.0\t1\t19 5\n",
      "19\t679\t0.0\t1\t667 12\n",
      "20\t29\t0.0\t2\t10 7 12\n",
      "21\t170\t0.0\t2\t104 28 38\n",
      "22\t161\t0.0\t2\t28 86 47\n",
      "23\t133\t0.0\t2\t11 44 78\n",
      "24\t189\t0.0\t2\t2 19 168\n",
      "25\t100\t0.0\t2\t0 63 37\n",
      "26\t58\t0.0\t2\t0 33 25\n",
      "27\t25\t0.0\t2\t0 5 20\n",
      "28\t11\t0.0\t2\t0 0 10 1\n",
      "29\t10\t0.0\t2\t0 0 8 2\n",
      "30\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t22\t19035.0\t0\t22\n",
      "6\t4\t4758.8\t0\t4\n",
      "7\t2\t1189.7\t0\t2\n",
      "8\t1\t297.4\t0\t1\n",
      "11\t1\t4.6\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 234 times.\n",
      "234 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors"
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 145630 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# First we trim the group of reads that goes from SRR1223689 to SRR1223696\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get trimmed reads of first group of RNAseq reads to  '$output_directory\n",
    "END=696\n",
    "for ((i=689;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the second group of reads that go from SRR1223700 to SRR1223706\n",
    "echo 'get trimmed reads of second group of RNAseq reads to  '$output_directory\n",
    "END=706\n",
    "for ((i=700;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the third group of reads that go from SRR1288397 to SRR1288399\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get trimmed reads of third group of RNAseq reads to  '$output_directory\n",
    "END=399\n",
    "for ((i=397;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the fourth group of reads that go from SRR1288405 to SRR1288408\n",
    "echo 'get trimmed reads of fourth group of RNAseq reads to  '$output_directory\n",
    "END=408\n",
    "for ((i=405;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -q 30 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T16:33:11.660341Z",
     "start_time": "2018-02-27T16:33:11.657269Z"
    }
   },
   "source": [
    "### Trim N at beginning of the reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T15:40:37.669813Z",
     "start_time": "2018-02-27T14:18:39.316037Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get trimmed reads of first group of RNAseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223689_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223689_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 385.27 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,169,626\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    11,169,626 (100.0%)\n",
      "\n",
      "Total basepairs processed:   309,898,164 bp\n",
      "Total written (filtered):    309,894,881 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223690_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223690_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 253.97 s (34 us/read; 1.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,394,961\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     7,394,961 (100.0%)\n",
      "\n",
      "Total basepairs processed:   207,064,240 bp\n",
      "Total written (filtered):    207,060,285 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223691_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223691_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 201.63 s (34 us/read; 1.78 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               5,995,186\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     5,995,186 (100.0%)\n",
      "\n",
      "Total basepairs processed:   164,498,660 bp\n",
      "Total written (filtered):    164,498,529 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223692_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223692_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 360.44 s (34 us/read; 1.76 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              10,596,801\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    10,596,801 (100.0%)\n",
      "\n",
      "Total basepairs processed:   299,782,240 bp\n",
      "Total written (filtered):    299,779,087 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223693_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223693_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 213.72 s (34 us/read; 1.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               6,374,697\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     6,374,697 (100.0%)\n",
      "\n",
      "Total basepairs processed:   180,612,468 bp\n",
      "Total written (filtered):    180,609,044 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223694_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223694_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 291.83 s (35 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,456,526\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     8,456,526 (100.0%)\n",
      "\n",
      "Total basepairs processed:   236,912,042 bp\n",
      "Total written (filtered):    236,909,569 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223695_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223695_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 127.74 s (34 us/read; 1.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,717,623\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     3,717,623 (100.0%)\n",
      "\n",
      "Total basepairs processed:   104,751,192 bp\n",
      "Total written (filtered):    104,749,140 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223696_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223696_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 134.64 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,842,147\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     3,842,147 (100.0%)\n",
      "\n",
      "Total basepairs processed:   106,204,959 bp\n",
      "Total written (filtered):    106,197,837 bp (100.0%)\n",
      "\n",
      "get trimmed reads of second group of RNAseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223700_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223700_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 966.69 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              28,081,555\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    28,081,555 (100.0%)\n",
      "\n",
      "Total basepairs processed:   798,979,172 bp\n",
      "Total written (filtered):    798,963,174 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223701_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223701_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 120.57 s (34 us/read; 1.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,593,755\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     3,593,755 (100.0%)\n",
      "\n",
      "Total basepairs processed:    99,969,428 bp\n",
      "Total written (filtered):     99,969,420 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223702_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223702_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 168.23 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,879,838\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,879,838 (100.0%)\n",
      "\n",
      "Total basepairs processed:   136,842,526 bp\n",
      "Total written (filtered):    136,841,117 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223703_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223703_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 76.97 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               2,236,755\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     2,236,755 (100.0%)\n",
      "\n",
      "Total basepairs processed:    63,268,142 bp\n",
      "Total written (filtered):     63,266,817 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223704_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223704_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 171.02 s (35 us/read; 1.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,922,213\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,922,213 (100.0%)\n",
      "\n",
      "Total basepairs processed:   138,588,005 bp\n",
      "Total written (filtered):    138,586,573 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223705_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223705_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 149.46 s (35 us/read; 1.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,303,430\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,303,430 (100.0%)\n",
      "\n",
      "Total basepairs processed:   122,045,085 bp\n",
      "Total written (filtered):    122,042,613 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1223706_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1223706_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 253.48 s (35 us/read; 1.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,173,036\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     7,173,036 (100.0%)\n",
      "\n",
      "Total basepairs processed:   209,649,984 bp\n",
      "Total written (filtered):    209,643,757 bp (100.0%)\n",
      "\n",
      "get trimmed reads of third group of RNAseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288397_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288397_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 274.19 s (33 us/read; 1.83 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,362,706\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     8,362,706 (100.0%)\n",
      "\n",
      "Total basepairs processed:   239,809,275 bp\n",
      "Total written (filtered):    239,803,071 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288398_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288398_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 109.05 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,099,545\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     3,099,545 (100.0%)\n",
      "\n",
      "Total basepairs processed:    92,197,313 bp\n",
      "Total written (filtered):     92,196,619 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288399_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288399_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 83.21 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               2,375,844\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     2,375,844 (100.0%)\n",
      "\n",
      "Total basepairs processed:    70,698,640 bp\n",
      "Total written (filtered):     70,696,910 bp (100.0%)\n",
      "\n",
      "get trimmed reads of fourth group of RNAseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288405_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288405_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 319.91 s (33 us/read; 1.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               9,629,781\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     9,629,781 (100.0%)\n",
      "\n",
      "Total basepairs processed:   293,349,535 bp\n",
      "Total written (filtered):    293,343,549 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288406_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288406_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 94.57 s (35 us/read; 1.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               2,686,751\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     2,686,751 (100.0%)\n",
      "\n",
      "Total basepairs processed:    80,759,676 bp\n",
      "Total written (filtered):     80,759,544 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288407_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288407_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 94.92 s (35 us/read; 1.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               2,687,435\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     2,687,435 (100.0%)\n",
      "\n",
      "Total basepairs processed:    80,813,868 bp\n",
      "Total written (filtered):     80,813,092 bp (100.0%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -o /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/SRR1288408_final_trim.fastq /projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/SRR1288408_trimmed.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 56.91 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               1,622,541\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     1,622,541 (100.0%)\n",
      "\n",
      "Total basepairs processed:    48,802,652 bp\n",
      "Total written (filtered):     48,800,516 bp (100.0%)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/RNAseq/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "label1=\"_final_trim\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# First we trim the group of reads that goes from SRR1223689 to SRR1223696\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get trimmed reads of first group of RNAseq reads to  '$output_directory\n",
    "END=696\n",
    "for ((i=689;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -o $output_directory$accession_base1$i$label1$extention $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the second group of reads that go from SRR1223700 to SRR1223706\n",
    "echo 'get trimmed reads of second group of RNAseq reads to  '$output_directory\n",
    "END=706\n",
    "for ((i=700;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -o $output_directory$accession_base1$i$label1$extention $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the third group of reads that go from SRR1288397 to SRR1288399\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get trimmed reads of third group of RNAseq reads to  '$output_directory\n",
    "END=399\n",
    "for ((i=397;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -o $output_directory$accession_base2$i$label1$extention $path_fastq_files$accession_base2$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the fourth group of reads that go from SRR1288405 to SRR1288408\n",
    "echo 'get trimmed reads of fourth group of RNAseq reads to  '$output_directory\n",
    "END=408\n",
    "for ((i=405;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -o $output_directory$accession_base2$i$label1$extention $path_fastq_files$accession_base2$i$label$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting fastQC reports of reads after adapters trimming"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Riboseq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-23T00:49:39.637038Z",
     "start_time": "2018-02-22T23:53:15.081158Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get fastQC report of first group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_trimmed/\n",
      "Analysis complete for SRR1223686_trimmed.fastq\n",
      "Analysis complete for SRR1223687_trimmed.fastq\n",
      "Analysis complete for SRR1223688_trimmed.fastq\n",
      "get fastQC report of second group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_trimmed/\n",
      "Analysis complete for SRR1223697_trimmed.fastq\n",
      "Analysis complete for SRR1223698_trimmed.fastq\n",
      "Analysis complete for SRR1223699_trimmed.fastq\n",
      "get fastQC report of third group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_trimmed/\n",
      "Analysis complete for SRR1288391_trimmed.fastq\n",
      "Analysis complete for SRR1288392_trimmed.fastq\n",
      "Analysis complete for SRR1288393_trimmed.fastq\n",
      "Analysis complete for SRR1288394_trimmed.fastq\n",
      "Analysis complete for SRR1288395_trimmed.fastq\n",
      "Analysis complete for SRR1288396_trimmed.fastq\n",
      "get fastQC report of third group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_trimmed/\n",
      "Analysis complete for SRR1288400_trimmed.fastq\n",
      "Analysis complete for SRR1288401_trimmed.fastq\n",
      "Analysis complete for SRR1288402_trimmed.fastq\n",
      "Analysis complete for SRR1288403_trimmed.fastq\n",
      "Analysis complete for SRR1288404_trimmed.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of SRR1223686_trimmed.fastq\n",
      "Approx 5% complete for SRR1223686_trimmed.fastq\n",
      "Approx 10% complete for SRR1223686_trimmed.fastq\n",
      "Approx 15% complete for SRR1223686_trimmed.fastq\n",
      "Approx 20% complete for SRR1223686_trimmed.fastq\n",
      "Approx 25% complete for SRR1223686_trimmed.fastq\n",
      "Approx 30% complete for SRR1223686_trimmed.fastq\n",
      "Approx 35% complete for SRR1223686_trimmed.fastq\n",
      "Approx 40% complete for SRR1223686_trimmed.fastq\n",
      "Approx 45% complete for SRR1223686_trimmed.fastq\n",
      "Approx 50% complete for SRR1223686_trimmed.fastq\n",
      "Approx 55% complete for SRR1223686_trimmed.fastq\n",
      "Approx 60% complete for SRR1223686_trimmed.fastq\n",
      "Approx 65% complete for SRR1223686_trimmed.fastq\n",
      "Approx 70% complete for SRR1223686_trimmed.fastq\n",
      "Approx 75% complete for SRR1223686_trimmed.fastq\n",
      "Approx 80% complete for SRR1223686_trimmed.fastq\n",
      "Approx 85% complete for SRR1223686_trimmed.fastq\n",
      "Approx 90% complete for SRR1223686_trimmed.fastq\n",
      "Approx 95% complete for SRR1223686_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223687_trimmed.fastq\n",
      "Approx 5% complete for SRR1223687_trimmed.fastq\n",
      "Approx 10% complete for SRR1223687_trimmed.fastq\n",
      "Approx 15% complete for SRR1223687_trimmed.fastq\n",
      "Approx 20% complete for SRR1223687_trimmed.fastq\n",
      "Approx 25% complete for SRR1223687_trimmed.fastq\n",
      "Approx 30% complete for SRR1223687_trimmed.fastq\n",
      "Approx 35% complete for SRR1223687_trimmed.fastq\n",
      "Approx 40% complete for SRR1223687_trimmed.fastq\n",
      "Approx 45% complete for SRR1223687_trimmed.fastq\n",
      "Approx 50% complete for SRR1223687_trimmed.fastq\n",
      "Approx 55% complete for SRR1223687_trimmed.fastq\n",
      "Approx 60% complete for SRR1223687_trimmed.fastq\n",
      "Approx 65% complete for SRR1223687_trimmed.fastq\n",
      "Approx 70% complete for SRR1223687_trimmed.fastq\n",
      "Approx 75% complete for SRR1223687_trimmed.fastq\n",
      "Approx 80% complete for SRR1223687_trimmed.fastq\n",
      "Approx 85% complete for SRR1223687_trimmed.fastq\n",
      "Approx 90% complete for SRR1223687_trimmed.fastq\n",
      "Approx 95% complete for SRR1223687_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223688_trimmed.fastq\n",
      "Approx 5% complete for SRR1223688_trimmed.fastq\n",
      "Approx 10% complete for SRR1223688_trimmed.fastq\n",
      "Approx 15% complete for SRR1223688_trimmed.fastq\n",
      "Approx 20% complete for SRR1223688_trimmed.fastq\n",
      "Approx 25% complete for SRR1223688_trimmed.fastq\n",
      "Approx 30% complete for SRR1223688_trimmed.fastq\n",
      "Approx 35% complete for SRR1223688_trimmed.fastq\n",
      "Approx 40% complete for SRR1223688_trimmed.fastq\n",
      "Approx 45% complete for SRR1223688_trimmed.fastq\n",
      "Approx 50% complete for SRR1223688_trimmed.fastq\n",
      "Approx 55% complete for SRR1223688_trimmed.fastq\n",
      "Approx 60% complete for SRR1223688_trimmed.fastq\n",
      "Approx 65% complete for SRR1223688_trimmed.fastq\n",
      "Approx 70% complete for SRR1223688_trimmed.fastq\n",
      "Approx 75% complete for SRR1223688_trimmed.fastq\n",
      "Approx 80% complete for SRR1223688_trimmed.fastq\n",
      "Approx 85% complete for SRR1223688_trimmed.fastq\n",
      "Approx 90% complete for SRR1223688_trimmed.fastq\n",
      "Approx 95% complete for SRR1223688_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223697_trimmed.fastq\n",
      "Approx 5% complete for SRR1223697_trimmed.fastq\n",
      "Approx 10% complete for SRR1223697_trimmed.fastq\n",
      "Approx 15% complete for SRR1223697_trimmed.fastq\n",
      "Approx 20% complete for SRR1223697_trimmed.fastq\n",
      "Approx 25% complete for SRR1223697_trimmed.fastq\n",
      "Approx 30% complete for SRR1223697_trimmed.fastq\n",
      "Approx 35% complete for SRR1223697_trimmed.fastq\n",
      "Approx 40% complete for SRR1223697_trimmed.fastq\n",
      "Approx 45% complete for SRR1223697_trimmed.fastq\n",
      "Approx 50% complete for SRR1223697_trimmed.fastq\n",
      "Approx 55% complete for SRR1223697_trimmed.fastq\n",
      "Approx 60% complete for SRR1223697_trimmed.fastq\n",
      "Approx 65% complete for SRR1223697_trimmed.fastq\n",
      "Approx 70% complete for SRR1223697_trimmed.fastq\n",
      "Approx 75% complete for SRR1223697_trimmed.fastq\n",
      "Approx 80% complete for SRR1223697_trimmed.fastq\n",
      "Approx 85% complete for SRR1223697_trimmed.fastq\n",
      "Approx 90% complete for SRR1223697_trimmed.fastq\n",
      "Approx 95% complete for SRR1223697_trimmed.fastq\n",
      "Approx 100% complete for SRR1223697_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223698_trimmed.fastq\n",
      "Approx 5% complete for SRR1223698_trimmed.fastq\n",
      "Approx 10% complete for SRR1223698_trimmed.fastq\n",
      "Approx 15% complete for SRR1223698_trimmed.fastq\n",
      "Approx 20% complete for SRR1223698_trimmed.fastq\n",
      "Approx 25% complete for SRR1223698_trimmed.fastq\n",
      "Approx 30% complete for SRR1223698_trimmed.fastq\n",
      "Approx 35% complete for SRR1223698_trimmed.fastq\n",
      "Approx 40% complete for SRR1223698_trimmed.fastq\n",
      "Approx 45% complete for SRR1223698_trimmed.fastq\n",
      "Approx 50% complete for SRR1223698_trimmed.fastq\n",
      "Approx 55% complete for SRR1223698_trimmed.fastq\n",
      "Approx 60% complete for SRR1223698_trimmed.fastq\n",
      "Approx 65% complete for SRR1223698_trimmed.fastq\n",
      "Approx 70% complete for SRR1223698_trimmed.fastq\n",
      "Approx 75% complete for SRR1223698_trimmed.fastq\n",
      "Approx 80% complete for SRR1223698_trimmed.fastq\n",
      "Approx 85% complete for SRR1223698_trimmed.fastq\n",
      "Approx 90% complete for SRR1223698_trimmed.fastq\n",
      "Approx 95% complete for SRR1223698_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223699_trimmed.fastq\n",
      "Approx 5% complete for SRR1223699_trimmed.fastq\n",
      "Approx 10% complete for SRR1223699_trimmed.fastq\n",
      "Approx 15% complete for SRR1223699_trimmed.fastq\n",
      "Approx 20% complete for SRR1223699_trimmed.fastq\n",
      "Approx 25% complete for SRR1223699_trimmed.fastq\n",
      "Approx 30% complete for SRR1223699_trimmed.fastq\n",
      "Approx 35% complete for SRR1223699_trimmed.fastq\n",
      "Approx 40% complete for SRR1223699_trimmed.fastq\n",
      "Approx 45% complete for SRR1223699_trimmed.fastq\n",
      "Approx 50% complete for SRR1223699_trimmed.fastq\n",
      "Approx 55% complete for SRR1223699_trimmed.fastq\n",
      "Approx 60% complete for SRR1223699_trimmed.fastq\n",
      "Approx 65% complete for SRR1223699_trimmed.fastq\n",
      "Approx 70% complete for SRR1223699_trimmed.fastq\n",
      "Approx 75% complete for SRR1223699_trimmed.fastq\n",
      "Approx 80% complete for SRR1223699_trimmed.fastq\n",
      "Approx 85% complete for SRR1223699_trimmed.fastq\n",
      "Approx 90% complete for SRR1223699_trimmed.fastq\n",
      "Approx 95% complete for SRR1223699_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288391_trimmed.fastq\n",
      "Approx 5% complete for SRR1288391_trimmed.fastq\n",
      "Approx 10% complete for SRR1288391_trimmed.fastq\n",
      "Approx 15% complete for SRR1288391_trimmed.fastq\n",
      "Approx 20% complete for SRR1288391_trimmed.fastq\n",
      "Approx 25% complete for SRR1288391_trimmed.fastq\n",
      "Approx 30% complete for SRR1288391_trimmed.fastq\n",
      "Approx 35% complete for SRR1288391_trimmed.fastq\n",
      "Approx 40% complete for SRR1288391_trimmed.fastq\n",
      "Approx 45% complete for SRR1288391_trimmed.fastq\n",
      "Approx 50% complete for SRR1288391_trimmed.fastq\n",
      "Approx 55% complete for SRR1288391_trimmed.fastq\n",
      "Approx 60% complete for SRR1288391_trimmed.fastq\n",
      "Approx 65% complete for SRR1288391_trimmed.fastq\n",
      "Approx 70% complete for SRR1288391_trimmed.fastq\n",
      "Approx 75% complete for SRR1288391_trimmed.fastq\n",
      "Approx 80% complete for SRR1288391_trimmed.fastq\n",
      "Approx 85% complete for SRR1288391_trimmed.fastq\n",
      "Approx 90% complete for SRR1288391_trimmed.fastq\n",
      "Approx 95% complete for SRR1288391_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288392_trimmed.fastq\n",
      "Approx 5% complete for SRR1288392_trimmed.fastq\n",
      "Approx 10% complete for SRR1288392_trimmed.fastq\n",
      "Approx 15% complete for SRR1288392_trimmed.fastq\n",
      "Approx 20% complete for SRR1288392_trimmed.fastq\n",
      "Approx 25% complete for SRR1288392_trimmed.fastq\n",
      "Approx 30% complete for SRR1288392_trimmed.fastq\n",
      "Approx 35% complete for SRR1288392_trimmed.fastq\n",
      "Approx 40% complete for SRR1288392_trimmed.fastq\n",
      "Approx 45% complete for SRR1288392_trimmed.fastq\n",
      "Approx 50% complete for SRR1288392_trimmed.fastq\n",
      "Approx 55% complete for SRR1288392_trimmed.fastq\n",
      "Approx 60% complete for SRR1288392_trimmed.fastq\n",
      "Approx 65% complete for SRR1288392_trimmed.fastq\n",
      "Approx 70% complete for SRR1288392_trimmed.fastq\n",
      "Approx 75% complete for SRR1288392_trimmed.fastq\n",
      "Approx 80% complete for SRR1288392_trimmed.fastq\n",
      "Approx 85% complete for SRR1288392_trimmed.fastq\n",
      "Approx 90% complete for SRR1288392_trimmed.fastq\n",
      "Approx 95% complete for SRR1288392_trimmed.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288393_trimmed.fastq\n",
      "Approx 5% complete for SRR1288393_trimmed.fastq\n",
      "Approx 10% complete for SRR1288393_trimmed.fastq\n",
      "Approx 15% complete for SRR1288393_trimmed.fastq\n",
      "Approx 20% complete for SRR1288393_trimmed.fastq\n",
      "Approx 25% complete for SRR1288393_trimmed.fastq\n",
      "Approx 30% complete for SRR1288393_trimmed.fastq\n",
      "Approx 35% complete for SRR1288393_trimmed.fastq\n",
      "Approx 40% complete for SRR1288393_trimmed.fastq\n",
      "Approx 45% complete for SRR1288393_trimmed.fastq\n",
      "Approx 50% complete for SRR1288393_trimmed.fastq\n",
      "Approx 55% complete for SRR1288393_trimmed.fastq\n",
      "Approx 60% complete for SRR1288393_trimmed.fastq\n",
      "Approx 65% complete for SRR1288393_trimmed.fastq\n",
      "Approx 70% complete for SRR1288393_trimme"
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 10000 exceeded with 18197 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/Riboseq/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_trimmed\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_trimmed/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get fastQC report of first group of reads to  '$output_directory\n",
    "END=688\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223697 to SRR1223699\n",
    "\n",
    "echo 'get fastQC report of second group of reads to  '$output_directory\n",
    "END=699\n",
    "for ((i=697;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get fastQC report of third group of reads to '$output_directory\n",
    "END=396\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get fastQC report of third group of reads to '$output_directory\n",
    "END=404\n",
    "for ((i=400;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$label$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-23T08:38:20.279273Z",
     "start_time": "2018-02-23T08:38:10.156854Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Searching 34 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 17 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_trimmed/\n",
    "\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### RNAseq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-28T10:28:15.331129Z",
     "start_time": "2018-02-28T09:43:34.522142Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get fastQC report of first group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results/RNAseq_final_trimming/\n",
      "Analysis complete for SRR1223689_final_trim.fastq\n",
      "Analysis complete for SRR1223690_final_trim.fastq\n",
      "Analysis complete for SRR1223691_final_trim.fastq\n",
      "Analysis complete for SRR1223692_final_trim.fastq\n",
      "Analysis complete for SRR1223693_final_trim.fastq\n",
      "Analysis complete for SRR1223694_final_trim.fastq\n",
      "Analysis complete for SRR1223695_final_trim.fastq\n",
      "Analysis complete for SRR1223696_final_trim.fastq\n",
      "get fastQC report of second group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results/RNAseq_final_trimming/\n",
      "Analysis complete for SRR1223700_final_trim.fastq\n",
      "Analysis complete for SRR1223701_final_trim.fastq\n",
      "Analysis complete for SRR1223702_final_trim.fastq\n",
      "Analysis complete for SRR1223703_final_trim.fastq\n",
      "Analysis complete for SRR1223704_final_trim.fastq\n",
      "Analysis complete for SRR1223705_final_trim.fastq\n",
      "Analysis complete for SRR1223706_final_trim.fastq\n",
      "get fastQC report of third group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results/RNAseq_final_trimming/\n",
      "Analysis complete for SRR1288397_final_trim.fastq\n",
      "Analysis complete for SRR1288398_final_trim.fastq\n",
      "Analysis complete for SRR1288399_final_trim.fastq\n",
      "get fastQC report of third group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results/RNAseq_final_trimming/\n",
      "Analysis complete for SRR1288405_final_trim.fastq\n",
      "Analysis complete for SRR1288406_final_trim.fastq\n",
      "Analysis complete for SRR1288407_final_trim.fastq\n",
      "Analysis complete for SRR1288408_final_trim.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of SRR1223689_final_trim.fastq\n",
      "Approx 5% complete for SRR1223689_final_trim.fastq\n",
      "Approx 10% complete for SRR1223689_final_trim.fastq\n",
      "Approx 15% complete for SRR1223689_final_trim.fastq\n",
      "Approx 20% complete for SRR1223689_final_trim.fastq\n",
      "Approx 25% complete for SRR1223689_final_trim.fastq\n",
      "Approx 30% complete for SRR1223689_final_trim.fastq\n",
      "Approx 35% complete for SRR1223689_final_trim.fastq\n",
      "Approx 40% complete for SRR1223689_final_trim.fastq\n",
      "Approx 45% complete for SRR1223689_final_trim.fastq\n",
      "Approx 50% complete for SRR1223689_final_trim.fastq\n",
      "Approx 55% complete for SRR1223689_final_trim.fastq\n",
      "Approx 60% complete for SRR1223689_final_trim.fastq\n",
      "Approx 65% complete for SRR1223689_final_trim.fastq\n",
      "Approx 70% complete for SRR1223689_final_trim.fastq\n",
      "Approx 75% complete for SRR1223689_final_trim.fastq\n",
      "Approx 80% complete for SRR1223689_final_trim.fastq\n",
      "Approx 85% complete for SRR1223689_final_trim.fastq\n",
      "Approx 90% complete for SRR1223689_final_trim.fastq\n",
      "Approx 95% complete for SRR1223689_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223690_final_trim.fastq\n",
      "Approx 5% complete for SRR1223690_final_trim.fastq\n",
      "Approx 10% complete for SRR1223690_final_trim.fastq\n",
      "Approx 15% complete for SRR1223690_final_trim.fastq\n",
      "Approx 20% complete for SRR1223690_final_trim.fastq\n",
      "Approx 25% complete for SRR1223690_final_trim.fastq\n",
      "Approx 30% complete for SRR1223690_final_trim.fastq\n",
      "Approx 35% complete for SRR1223690_final_trim.fastq\n",
      "Approx 40% complete for SRR1223690_final_trim.fastq\n",
      "Approx 45% complete for SRR1223690_final_trim.fastq\n",
      "Approx 50% complete for SRR1223690_final_trim.fastq\n",
      "Approx 55% complete for SRR1223690_final_trim.fastq\n",
      "Approx 60% complete for SRR1223690_final_trim.fastq\n",
      "Approx 65% complete for SRR1223690_final_trim.fastq\n",
      "Approx 70% complete for SRR1223690_final_trim.fastq\n",
      "Approx 75% complete for SRR1223690_final_trim.fastq\n",
      "Approx 80% complete for SRR1223690_final_trim.fastq\n",
      "Approx 85% complete for SRR1223690_final_trim.fastq\n",
      "Approx 90% complete for SRR1223690_final_trim.fastq\n",
      "Approx 95% complete for SRR1223690_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223691_final_trim.fastq\n",
      "Approx 5% complete for SRR1223691_final_trim.fastq\n",
      "Approx 10% complete for SRR1223691_final_trim.fastq\n",
      "Approx 15% complete for SRR1223691_final_trim.fastq\n",
      "Approx 20% complete for SRR1223691_final_trim.fastq\n",
      "Approx 25% complete for SRR1223691_final_trim.fastq\n",
      "Approx 30% complete for SRR1223691_final_trim.fastq\n",
      "Approx 35% complete for SRR1223691_final_trim.fastq\n",
      "Approx 40% complete for SRR1223691_final_trim.fastq\n",
      "Approx 45% complete for SRR1223691_final_trim.fastq\n",
      "Approx 50% complete for SRR1223691_final_trim.fastq\n",
      "Approx 55% complete for SRR1223691_final_trim.fastq\n",
      "Approx 60% complete for SRR1223691_final_trim.fastq\n",
      "Approx 65% complete for SRR1223691_final_trim.fastq\n",
      "Approx 70% complete for SRR1223691_final_trim.fastq\n",
      "Approx 75% complete for SRR1223691_final_trim.fastq\n",
      "Approx 80% complete for SRR1223691_final_trim.fastq\n",
      "Approx 85% complete for SRR1223691_final_trim.fastq\n",
      "Approx 90% complete for SRR1223691_final_trim.fastq\n",
      "Approx 95% complete for SRR1223691_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223692_final_trim.fastq\n",
      "Approx 5% complete for SRR1223692_final_trim.fastq\n",
      "Approx 10% complete for SRR1223692_final_trim.fastq\n",
      "Approx 15% complete for SRR1223692_final_trim.fastq\n",
      "Approx 20% complete for SRR1223692_final_trim.fastq\n",
      "Approx 25% complete for SRR1223692_final_trim.fastq\n",
      "Approx 30% complete for SRR1223692_final_trim.fastq\n",
      "Approx 35% complete for SRR1223692_final_trim.fastq\n",
      "Approx 40% complete for SRR1223692_final_trim.fastq\n",
      "Approx 45% complete for SRR1223692_final_trim.fastq\n",
      "Approx 50% complete for SRR1223692_final_trim.fastq\n",
      "Approx 55% complete for SRR1223692_final_trim.fastq\n",
      "Approx 60% complete for SRR1223692_final_trim.fastq\n",
      "Approx 65% complete for SRR1223692_final_trim.fastq\n",
      "Approx 70% complete for SRR1223692_final_trim.fastq\n",
      "Approx 75% complete for SRR1223692_final_trim.fastq\n",
      "Approx 80% complete for SRR1223692_final_trim.fastq\n",
      "Approx 85% complete for SRR1223692_final_trim.fastq\n",
      "Approx 90% complete for SRR1223692_final_trim.fastq\n",
      "Approx 95% complete for SRR1223692_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223693_final_trim.fastq\n",
      "Approx 5% complete for SRR1223693_final_trim.fastq\n",
      "Approx 10% complete for SRR1223693_final_trim.fastq\n",
      "Approx 15% complete for SRR1223693_final_trim.fastq\n",
      "Approx 20% complete for SRR1223693_final_trim.fastq\n",
      "Approx 25% complete for SRR1223693_final_trim.fastq\n",
      "Approx 30% complete for SRR1223693_final_trim.fastq\n",
      "Approx 35% complete for SRR1223693_final_trim.fastq\n",
      "Approx 40% complete for SRR1223693_final_trim.fastq\n",
      "Approx 45% complete for SRR1223693_final_trim.fastq\n",
      "Approx 50% complete for SRR1223693_final_trim.fastq\n",
      "Approx 55% complete for SRR1223693_final_trim.fastq\n",
      "Approx 60% complete for SRR1223693_final_trim.fastq\n",
      "Approx 65% complete for SRR1223693_final_trim.fastq\n",
      "Approx 70% complete for SRR1223693_final_trim.fastq\n",
      "Approx 75% complete for SRR1223693_final_trim.fastq\n",
      "Approx 80% complete for SRR1223693_final_trim.fastq\n",
      "Approx 85% complete for SRR1223693_final_trim.fastq\n",
      "Approx 90% complete for SRR1223693_final_trim.fastq\n",
      "Approx 95% complete for SRR1223693_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223694_final_trim.fastq\n",
      "Approx 5% complete for SRR1223694_final_trim.fastq\n",
      "Approx 10% complete for SRR1223694_final_trim.fastq\n",
      "Approx 15% complete for SRR1223694_final_trim.fastq\n",
      "Approx 20% complete for SRR1223694_final_trim.fastq\n",
      "Approx 25% complete for SRR1223694_final_trim.fastq\n",
      "Approx 30% complete for SRR1223694_final_trim.fastq\n",
      "Approx 35% complete for SRR1223694_final_trim.fastq\n",
      "Approx 40% complete for SRR1223694_final_trim.fastq\n",
      "Approx 45% complete for SRR1223694_final_trim.fastq\n",
      "Approx 50% complete for SRR1223694_final_trim.fastq\n",
      "Approx 55% complete for SRR1223694_final_trim.fastq\n",
      "Approx 60% complete for SRR1223694_final_trim.fastq\n",
      "Approx 65% complete for SRR1223694_final_trim.fastq\n",
      "Approx 70% complete for SRR1223694_final_trim.fastq\n",
      "Approx 75% complete for SRR1223694_final_trim.fastq\n",
      "Approx 80% complete for SRR1223694_final_trim.fastq\n",
      "Approx 85% complete for SRR1223694_final_trim.fastq\n",
      "Approx 90% complete for SRR1223694_final_trim.fastq\n",
      "Approx 95% complete for SRR1223694_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223695_final_trim.fastq\n",
      "Approx 5% complete for SRR1223695_final_trim.fastq\n",
      "Approx 10% complete for SRR1223695_final_trim.fastq\n",
      "Approx 15% complete for SRR1223695_final_trim.fastq\n",
      "Approx 20% complete for SRR1223695_final_trim.fastq\n",
      "Approx 25% complete for SRR1223695_final_trim.fastq\n",
      "Approx 30% complete for SRR1223695_final_trim.fastq\n",
      "Approx 35% complete for SRR1223695_final_trim.fastq\n",
      "Approx 40% complete for SRR1223695_final_trim.fastq\n",
      "Approx 45% complete for SRR1223695_final_trim.fastq\n",
      "Approx 50% complete for SRR1223695_final_trim.fastq\n",
      "Approx 55% complete for SRR1223695_final_trim.fastq\n",
      "Approx 60% complete for SRR1223695_final_trim.fastq\n",
      "Approx 65% complete for SRR1223695_final_trim.fastq\n",
      "Approx 70% complete for SRR1223695_final_trim.fastq\n",
      "Approx 75% complete for SRR1223695_final_trim.fastq\n",
      "Approx 80% complete for SRR1223695_final_trim.fastq\n",
      "Approx 85% complete for SRR1223695_final_trim.fastq\n",
      "Approx 90% complete for SRR1223695_final_trim.fastq\n",
      "Approx 95% complete for SRR1223695_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223696_final_trim.fastq\n",
      "Approx 5% complete for SRR1223696_final_trim.fastq\n",
      "Approx 10% complete for SRR1223696_final_trim.fastq\n",
      "Approx 15% complete for SRR1223696_final_trim.fastq\n",
      "Approx 20% complete for SRR1223696_final_trim.fastq\n",
      "Approx 25% complete for SRR1223696_final_trim.fastq\n",
      "Approx 30% complete for SRR1223696_final_trim.fastq\n",
      "Approx 35% complete for SRR1223696_final_trim.fastq\n",
      "Approx 40% complete for SRR1223696_final_trim.fastq\n",
      "Approx 45% complete for SRR1223696_final_trim.fastq\n",
      "Approx 50% complete for SRR1223696_final_trim.fastq\n",
      "Approx 55% complete for SRR1223696_final_trim.fastq\n",
      "Approx 60% complete for SRR1223696_final_trim.fastq\n",
      "Approx 65% complete for SRR1223696_final_trim.fastq\n",
      "Approx 70% complete for SRR1223696_final_trim.fastq\n",
      "Approx 75% complete for SRR1223696_final_trim.fastq\n",
      "Approx 80% complete for SRR1223696_final_trim.fastq\n",
      "Approx 85% complete for SRR1223696_final_trim.fastq\n",
      "Approx 90% complete for SRR1223696_final_trim.fastq\n",
      "Approx 95% complete for SRR1223696_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223700_final_trim.fastq\n",
      "Approx 5% complete for SRR1223700_final_trim.fastq\n",
      "Approx 10% complete for SRR1223700_final_trim.fastq\n",
      "Approx 15% complete for SRR1223700_final_trim.fastq\n",
      "Approx 20% complete for SRR1223700_final_trim.fastq\n",
      "Approx 25% complete for SRR1223700_final_trim.fastq\n",
      "Approx 30% complete for SRR1223700_final_trim.fastq\n",
      "Approx 35% complete for SRR1223700_final_trim.fastq\n",
      "Approx 40% complete for SRR1223700_final_trim.fastq\n",
      "Approx 45% complete for SRR1223700_final_trim.fastq\n",
      "Approx 50% complete for SRR1223700_final_trim.fastq\n",
      "Approx 55% complete for SRR1223700_final_trim.fastq\n",
      "Approx 60% complete for SRR1223700_final_trim.fastq\n",
      "Approx 65% complete for SRR1223700_final_trim.fastq\n",
      "Approx 70% complete for SRR1223700_final_trim.fastq\n",
      "Approx 75% complete for SRR1223700_final_trim.fastq\n",
      "Approx 80% complete for SRR1223700_final_trim.fastq\n",
      "Approx 85% complete for SRR1223700_final_trim.fastq\n",
      "Approx 90% complete for SRR1223700_final_trim.fastq\n",
      "Approx 95% complete for SRR1223700_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223701_final_trim.fastq\n",
      "Approx 5% complete for SRR1223701_final_trim.fastq\n",
      "Approx 10% complete for SRR1223701_final_trim.fastq\n",
      "Approx 15% complete for SRR1223701_final_trim.fastq\n",
      "Approx 20% complete for SRR1223701_final_trim.fastq\n",
      "Approx 25% complete for SRR1223701_final_trim.fastq\n",
      "Approx 30% complete for SRR1223701_final_trim.fastq\n",
      "Approx 35% complete for SRR1223701_final_trim.fastq\n",
      "Approx 40% complete for SRR1223701_final_trim.fastq\n",
      "Approx 45% complete for SRR1223701_final_trim.fastq\n",
      "Approx 50% complete for SRR1223701_final_trim.fastq\n",
      "Approx 55% complete for SRR1223701_final_trim.fastq\n",
      "Approx 60% complete for SRR1223701_final_trim.fastq\n",
      "Approx 65% complete for SRR1223701_final_trim.fastq\n",
      "Approx 70% complete for SRR1223701_final_trim.fastq\n",
      "Approx 75% complete for SRR1223701_final_trim.fastq\n",
      "Approx 80% complete for SRR1223701_final_trim.fastq\n",
      "Approx 85% complete for SRR1223701_final_trim.fastq\n",
      "Approx 90% complete for SRR1223701_final_trim.fastq\n",
      "Approx 95% complete for SRR1223701_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223702_final_trim.fastq\n",
      "Approx 5% complete for SRR1223702_final_trim.fastq\n",
      "Approx 10% complete for SRR1223702_final_trim.fastq\n",
      "Approx 15% complete for SRR1223702_final_trim.fastq\n",
      "Approx 20% complete for SRR1223702_final_trim.fastq\n",
      "Approx 25% complete for SRR1223702_final_trim.fastq\n",
      "Approx 30% complete for SRR1223702_final_trim.fastq\n",
      "Approx 35% complete for SRR1223702_final_trim.fastq\n",
      "Approx 40% complete for SRR1223702_final_trim.fastq\n",
      "Approx 45% complete for SRR1223702_final_trim.fastq\n",
      "Approx 50% complete for SRR1223702_final_trim.fastq\n",
      "Approx 55% complete for SRR1223702_final_trim.fastq\n",
      "Approx 60% complete for SRR1223702_final_trim.fastq\n",
      "Approx 65% complete for SRR1223702_final_trim.fastq\n",
      "Approx 70% complete for SRR1223702_final_trim.fastq\n",
      "Approx 75% complete for SRR1223702_final_trim.fastq\n",
      "Approx 80% complete for SRR1223702_final_trim.fastq\n",
      "Approx 85% complete for SRR1223702_final_trim.fastq\n",
      "Approx 90% complete for SRR1223702_final_trim.fastq\n",
      "Approx 95% complete for SRR1223702_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223703_final_trim.fastq\n",
      "Approx 5% complete for SRR1223703_final_trim.fastq\n",
      "Approx 10% complete for SRR1223703_final_trim.fastq\n",
      "Approx 15% complete for SRR1223703_final_trim.fastq\n",
      "Approx 20% complete for SRR1223703_final_trim.fastq\n",
      "Approx 25% complete for SRR1223703_final_trim.fastq\n",
      "Approx 30% complete for SRR1223703_final_trim.fastq\n",
      "Approx 35% complete for SRR1223703_final_trim.fastq\n",
      "Approx 40% complete for SRR1223703_final_trim.fastq\n",
      "Approx 45% complete for SRR1223703_final_trim.fastq\n",
      "Approx 50% complete for SRR1223703_final_trim.fastq\n",
      "Approx 55% complete for SRR1223703_final_trim.fastq\n",
      "Approx 60% complete for SRR1223703_final_trim.fastq\n",
      "Approx 65% complete for SRR1223703_final_trim.fastq\n",
      "Approx 70% complete for SRR1223703_final_trim.fastq\n",
      "Approx 75% complete for SRR1223703_final_trim.fastq\n",
      "Approx 80% complete for SRR1223703_final_trim.fastq\n",
      "Approx 85% complete for SRR1223703_final_trim.fastq\n",
      "Approx 90% complete for SRR1223703_final_trim.fastq\n",
      "Approx 95% complete for SRR1223703_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223704_final_trim.fastq\n",
      "Approx 5% complete for SRR1223704_final_trim.fastq\n",
      "Approx 10% complete for SRR1223704_final_trim.fastq\n",
      "Approx 15% complete for SRR1223704_final_trim.fastq\n",
      "Approx 20% complete for SRR1223704_final_trim.fastq\n",
      "Approx 25% complete for SRR1223704_final_trim.fastq\n",
      "Approx 30% complete for SRR1223704_final_trim.fastq\n",
      "Approx 35% complete for SRR1223704_final_trim.fastq\n",
      "Approx 40% complete for SRR1223704_final_trim.fastq\n",
      "Approx 45% complete for SRR1223704_final_trim.fastq\n",
      "Approx 50% complete for SRR1223704_final_trim.fastq\n",
      "Approx 55% complete for SRR1223704_final_trim.fastq\n",
      "Approx 60% complete for SRR1223704_final_trim.fastq\n",
      "Approx 65% complete for SRR1223704_final_trim.fastq\n",
      "Approx 70% complete for SRR1223704_final_trim.fastq\n",
      "Approx 75% complete for SRR1223704_final_trim.fastq\n",
      "Approx 80% complete for SRR1223704_final_trim.fastq\n",
      "Approx 85% complete for SRR1223704_final_trim.fastq\n",
      "Approx 90% complete for SRR1223704_final_trim.fastq\n",
      "Approx 95% complete for SRR1223704_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223705_final_trim.fastq\n",
      "Approx 5% complete for SRR1223705_final_trim.fastq\n",
      "Approx 10% complete for SRR1223705_final_trim.fastq\n",
      "Approx 15% complete for SRR1223705_final_trim.fastq\n",
      "Approx 20% complete for SRR1223705_final_trim.fastq\n",
      "Approx 25% complete for SRR1223705_final_trim.fastq\n",
      "Approx 30% complete for SRR1223705_final_trim.fastq\n",
      "Approx 35% complete for SRR1223705_final_trim.fastq\n",
      "Approx 40% complete for SRR1223705_final_trim.fastq\n",
      "Approx 45% complete for SRR1223705_final_trim.fastq\n",
      "Approx 50% complete for SRR1223705_final_trim.fastq\n",
      "Approx 55% complete for SRR1223705_final_trim.fastq\n",
      "Approx 60% complete for SRR1223705_final_trim.fastq\n",
      "Approx 65% complete for SRR1223705_final_trim.fastq\n",
      "Approx 70% complete for SRR1223705_final_trim.fastq\n",
      "Approx 75% complete for SRR1223705_final_trim.fastq\n",
      "Approx 80% complete for SRR1223705_final_trim.fastq\n",
      "Approx 85% complete for SRR1223705_final_trim.fastq\n",
      "Approx 90% complete for SRR1223705_final_trim.fastq\n",
      "Approx 95% complete for SRR1223705_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223706_final_trim.fastq\n",
      "Approx 5% complete for SRR1223706_final_trim.fastq\n",
      "Approx 10% complete for SRR1223706_final_trim.fastq\n",
      "Approx 15% complete for SRR1223706_final_trim.fastq\n",
      "Approx 20% complete for SRR1223706_final_trim.fastq\n",
      "Approx 25% complete for SRR1223706_final_trim.fastq\n",
      "Approx 30% complete for SRR1223706_final_trim.fastq\n",
      "Approx 35% complete for SRR1223706_final_trim.fastq\n",
      "Approx 40% complete for SRR1223706_final_trim.fastq\n",
      "Approx 45% complete for SRR1223706_final_trim.fastq\n",
      "Approx 50% complete for SRR1223706_final_trim.fastq\n",
      "Approx 55% complete for SRR1223706_final_trim.fastq\n",
      "Approx 60% complete for SRR1223706_final_trim.fastq\n",
      "Approx 65% complete for SRR1223706_final_trim.fastq\n",
      "Approx 70% complete for SRR1223706_final_trim.fastq\n",
      "Approx 75% complete for SRR1223706_final_trim.fastq\n",
      "Approx 80% complete for SRR1223706_final_trim.fastq\n",
      "Approx 85% complete for SRR1223706_final_trim.fastq\n",
      "Approx 90% complete for SRR1223706_final_trim.fastq\n",
      "Approx 95% complete for SRR1223706_final_trim.fastq\n",
      "Approx 100% complete for SRR1223706_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288397_final_trim.fastq\n",
      "Approx 5% complete for SRR1288397_final_trim.fastq\n",
      "Approx 10% complete for SRR1288397_final_trim.fastq\n",
      "Approx 15% complete for SRR1288397_final_trim.fastq\n",
      "Approx 20% complete for SRR1288397_final_trim.fastq\n",
      "Approx 25% complete for SRR1288397_final_trim.fastq\n",
      "Approx 30% complete for SRR1288397_final_trim.fastq\n",
      "Approx 35% complete for SRR1288397_final_trim.fastq\n",
      "Approx 40% complete for SRR1288397_final_trim.fastq\n",
      "Approx 45% complete for SRR1288397_final_trim.fastq\n",
      "Approx 50% complete for SRR1288397_final_trim.fastq\n",
      "Approx 55% complete for SRR1288397_final_trim.fastq\n",
      "Approx 60% complete for SRR1288397_final_trim.fastq\n",
      "Approx 65% complete for SRR1288397_final_trim.fastq\n",
      "Approx 70% complete for SRR1288397_final_trim.fastq\n",
      "Approx 75% complete for SRR1288397_final_trim.fastq\n",
      "Approx 80% complete for SRR1288397_final_trim.fastq\n",
      "Approx 85% complete for SRR1288397_final_trim.fastq\n",
      "Approx 90% complete for SRR1288397_final_trim.fastq\n",
      "Approx 95% complete for SRR1288397_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288398_final_trim.fastq\n",
      "Approx 5% complete for SRR1288398_final_trim.fastq\n",
      "Approx 10% complete for SRR1288398_final_trim.fastq\n",
      "Approx 15% complete for SRR1288398_final_trim.fastq\n",
      "Approx 20% complete for SRR1288398_final_trim.fastq\n",
      "Approx 25% complete for SRR1288398_final_trim.fastq\n",
      "Approx 30% complete for SRR1288398_final_trim.fastq\n",
      "Approx 35% complete for SRR1288398_final_trim.fastq\n",
      "Approx 40% complete for SRR1288398_final_trim.fastq\n",
      "Approx 45% complete for SRR1288398_final_trim.fastq\n",
      "Approx 50% complete for SRR1288398_final_trim.fastq\n",
      "Approx 55% complete for SRR1288398_final_trim.fastq\n",
      "Approx 60% complete for SRR1288398_final_trim.fastq\n",
      "Approx 65% complete for SRR1288398_final_trim.fastq\n",
      "Approx 70% complete for SRR1288398_final_trim.fastq\n",
      "Approx 75% complete for SRR1288398_final_trim.fastq\n",
      "Approx 80% complete for SRR1288398_final_trim.fastq\n",
      "Approx 85% complete for SRR1288398_final_trim.fastq\n",
      "Approx 90% complete for SRR1288398_final_trim.fastq\n",
      "Approx 95% complete for SRR1288398_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288399_final_trim.fastq\n",
      "Approx 5% complete for SRR1288399_final_trim.fastq\n",
      "Approx 10% complete for SRR1288399_final_trim.fastq\n",
      "Approx 15% complete for SRR1288399_final_trim.fastq\n",
      "Approx 20% complete for SRR1288399_final_trim.fastq\n",
      "Approx 25% complete for SRR1288399_final_trim.fastq\n",
      "Approx 30% complete for SRR1288399_final_trim.fastq\n",
      "Approx 35% complete for SRR1288399_final_trim.fastq\n",
      "Approx 40% complete for SRR1288399_final_trim.fastq\n",
      "Approx 45% complete for SRR1288399_final_trim.fastq\n",
      "Approx 50% complete for SRR1288399_final_trim.fastq\n",
      "Approx 55% complete for SRR1288399_final_trim.fastq\n",
      "Approx 60% complete for SRR1288399_final_trim.fastq\n",
      "Approx 65% complete for SRR1288399_final_trim.fastq\n",
      "Approx 70% complete for SRR1288399_final_trim.fastq\n",
      "Approx 75% complete for SRR1288399_final_trim.fastq\n",
      "Approx 80% complete for SRR1288399_final_trim.fastq\n",
      "Approx 85% complete for SRR1288399_final_trim.fastq\n",
      "Approx 90% complete for SRR1288399_final_trim.fastq\n",
      "Approx 95% complete for SRR1288399_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288405_final_trim.fastq\n",
      "Approx 5% complete for SRR1288405_final_trim.fastq\n",
      "Approx 10% complete for SRR1288405_final_trim.fastq\n",
      "Approx 15% complete for SRR1288405_final_trim.fastq\n",
      "Approx 20% complete for SRR1288405_final_trim.fastq\n",
      "Approx 25% complete for SRR1288405_final_trim.fastq\n",
      "Approx 30% complete for SRR1288405_final_trim.fastq\n",
      "Approx 35% complete for SRR1288405_final_trim.fastq\n",
      "Approx 40% complete for SRR1288405_final_trim.fastq\n",
      "Approx 45% complete for SRR1288405_final_trim.fastq\n",
      "Approx 50% complete for SRR1288405_final_trim.fastq\n",
      "Approx 55% complete for SRR1288405_final_trim.fastq\n",
      "Approx 60% complete for SRR1288405_final_trim.fastq\n",
      "Approx 65% complete for SRR1288405_final_trim.fastq\n",
      "Approx 70% complete for SRR1288405_final_trim.fastq\n",
      "Approx 75% complete for SRR1288405_final_trim.fastq\n",
      "Approx 80% complete for SRR1288405_final_trim.fastq\n",
      "Approx 85% complete for SRR1288405_final_trim.fastq\n",
      "Approx 90% complete for SRR1288405_final_trim.fastq\n",
      "Approx 95% complete for SRR1288405_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288406_final_trim.fastq\n",
      "Approx 5% complete for SRR1288406_final_trim.fastq\n",
      "Approx 10% complete for SRR1288406_final_trim.fastq\n",
      "Approx 15% complete for SRR1288406_final_trim.fastq\n",
      "Approx 20% complete for SRR1288406_final_trim.fastq\n",
      "Approx 25% complete for SRR1288406_final_trim.fastq\n",
      "Approx 30% complete for SRR1288406_final_trim.fastq\n",
      "Approx 35% complete for SRR1288406_final_trim.fastq\n",
      "Approx 40% complete for SRR1288406_final_trim.fastq\n",
      "Approx 45% complete for SRR1288406_final_trim.fastq\n",
      "Approx 50% complete for SRR1288406_final_trim.fastq\n",
      "Approx 55% complete for SRR1288406_final_trim.fastq\n",
      "Approx 60% complete for SRR1288406_final_trim.fastq\n",
      "Approx 65% complete for SRR1288406_final_trim.fastq\n",
      "Approx 70% complete for SRR1288406_final_trim.fastq\n",
      "Approx 75% complete for SRR1288406_final_trim.fastq\n",
      "Approx 80% complete for SRR1288406_final_trim.fastq\n",
      "Approx 85% complete for SRR1288406_final_trim.fastq\n",
      "Approx 90% complete for SRR1288406_final_trim.fastq\n",
      "Approx 95% complete for SRR1288406_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288407_final_trim.fastq\n",
      "Approx 5% complete for SRR1288407_final_trim.fastq\n",
      "Approx 10% complete for SRR1288407_final_trim.fastq\n",
      "Approx 15% complete for SRR1288407_final_trim.fastq\n",
      "Approx 20% complete for SRR1288407_final_trim.fastq\n",
      "Approx 25% complete for SRR1288407_final_trim.fastq\n",
      "Approx 30% complete for SRR1288407_final_trim.fastq\n",
      "Approx 35% complete for SRR1288407_final_trim.fastq\n",
      "Approx 40% complete for SRR1288407_final_trim.fastq\n",
      "Approx 45% complete for SRR1288407_final_trim.fastq\n",
      "Approx 50% complete for SRR1288407_final_trim.fastq\n",
      "Approx 55% complete for SRR1288407_final_trim.fastq\n",
      "Approx 60% complete for SRR1288407_final_trim.fastq\n",
      "Approx 65% complete for SRR1288407_final_trim.fastq\n",
      "Approx 70% complete for SRR1288407_final_trim.fastq\n",
      "Approx 75% complete for SRR1288407_final_trim.fastq\n",
      "Approx 80% complete for SRR1288407_final_trim.fastq\n",
      "Approx 85% complete for SRR1288407_final_trim.fastq\n",
      "Approx 90% complete for SRR1288407_final_trim.fastq\n",
      "Approx 95% complete for SRR1288407_final_trim.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288408_final_trim.fastq\n",
      "Approx 5% complete for SRR1288408_final_trim.fastq\n",
      "Approx 10% complete for SRR1288408_final_trim.fastq\n",
      "Approx 15% complete for SRR1288408_final_trim.fastq\n",
      "Approx 20% complete for SRR1288408_final_trim.fastq\n",
      "Approx 25% complete for SRR1288408_final_trim.fastq\n",
      "Approx 30% complete for SRR1288408_final_trim.fastq\n",
      "Approx 35% complete for SRR1288408_final_trim.fastq\n",
      "Approx 40% complete for SRR1288408_final_trim.fastq\n",
      "Approx 45% complete for SRR1288408_final_trim.fastq\n",
      "Approx 50% complete for SRR1288408_final_trim.fastq\n",
      "Approx 55% complete for SRR1288408_final_trim.fastq\n",
      "Approx 60% complete for SRR1288408_final_trim.fastq\n",
      "Approx 65% complete for SRR1288408_final_trim.fastq\n",
      "Approx 70% complete for SRR1288408_final_trim.fastq\n",
      "Approx 75% complete for SRR1288408_final_trim.fastq\n",
      "Approx 80% complete for SRR1288408_final_trim.fastq\n",
      "Approx 85% complete for SRR1288408_final_trim.fastq\n",
      "Approx 90% complete for SRR1288408_final_trim.fastq\n",
      "Approx 95% complete for SRR1288408_final_trim.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_final_trim\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/fastqc_results/RNAseq_final_trimming/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223689 to SRR1223696\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get fastQC report of first group of reads to  '$output_directory\n",
    "END=696\n",
    "for ((i=689;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223700 to SRR1223706\n",
    "\n",
    "echo 'get fastQC report of second group of reads to  '$output_directory\n",
    "END=706\n",
    "for ((i=700;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288397 to SRR1288399\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get fastQC report of third group of reads to '$output_directory\n",
    "END=399\n",
    "for ((i=397;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288405 to SRR1288408\n",
    "echo 'get fastQC report of third group of reads to '$output_directory\n",
    "END=408\n",
    "for ((i=405;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$label$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-23T08:39:02.325259Z",
     "start_time": "2018-02-23T08:38:57.109544Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Searching 44 files..\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.4\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '.'\n",
      "[INFO   ]          fastqc : Found 22 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /projects_eg/projects/simone/glu_paper/fastqc_results/RNAseq_trimmed/\n",
    "\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Redo cutadapt for Riboseq Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trim N at the beginning of the reads (if present) and last seven nucleotides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-26T19:33:59.296051Z",
     "start_time": "2018-02-26T16:55:14.121571Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get trimmed reads of first group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223686_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223686.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 826.07 s (36 us/read; 1.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              23,135,864\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    23,135,864 (100.0%)\n",
      "\n",
      "Total basepairs processed:   971,706,288 bp\n",
      "Total written (filtered):    809,732,303 bp (83.3%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223687_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223687.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 672.05 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,101,967\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    16,101,967 (100.0%)\n",
      "\n",
      "Total basepairs processed:   821,200,317 bp\n",
      "Total written (filtered):    708,480,362 bp (86.3%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223688_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223688.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1704.66 s (41 us/read; 1.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              41,451,837\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    41,451,837 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,072,591,850 bp\n",
      "Total written (filtered):  1,782,424,762 bp (86.0%)\n",
      "\n",
      "get trimmed reads of second group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223697_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223697.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 722.14 s (37 us/read; 1.61 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,360,675\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    19,360,675 (100.0%)\n",
      "\n",
      "Total basepairs processed:   813,148,350 bp\n",
      "Total written (filtered):    677,572,332 bp (83.3%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223698_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223698.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1070.13 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              25,664,498\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    25,664,498 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,308,889,398 bp\n",
      "Total written (filtered):  1,129,228,167 bp (86.3%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223699_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1223699.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1885.54 s (41 us/read; 1.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              46,043,180\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    46,043,180 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,302,159,000 bp\n",
      "Total written (filtered):  1,979,851,994 bp (86.0%)\n",
      "\n",
      "get trimmed reads of third group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288391_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288391.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 609.81 s (34 us/read; 1.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,778,707\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    17,778,707 (100.0%)\n",
      "\n",
      "Total basepairs processed:   640,033,452 bp\n",
      "Total written (filtered):    515,547,570 bp (80.6%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288392_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288392.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 166.42 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    180,000,000 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288393_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288393.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 166.29 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    179,998,947 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288394_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288394.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 167.19 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    179,998,517 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288395_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288395.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 166.80 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    179,998,753 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288396_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288396.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 95.47 s (41 us/read; 1.45 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               2,300,685\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     2,300,685 (100.0%)\n",
      "\n",
      "Total basepairs processed:   119,635,620 bp\n",
      "Total written (filtered):    103,526,659 bp (86.5%)\n",
      "\n",
      "get trimmed reads of fourth group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288400_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288400.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 697.72 s (34 us/read; 1.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              20,351,396\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    20,351,396 (100.0%)\n",
      "\n",
      "Total basepairs processed:   732,650,256 bp\n",
      "Total written (filtered):    589,971,891 bp (80.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288401_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288401.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 166.52 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    180,000,000 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288402_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288402.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 166.77 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    179,998,277 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288403_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288403.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 166.28 s (42 us/read; 1.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     4,000,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:   208,000,000 bp\n",
      "Total written (filtered):    179,998,802 bp (86.5%)\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: --trim-n -u -7 -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288404_first_trimming.fastq /projects_eg/projects/simone/glu_paper/fastq_files/SRR1288404.fastq\n",
      "Running on 1 core\n",
      "Trimming 0 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 67.74 s (41 us/read; 1.45 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               1,637,998\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):     1,637,998 (100.0%)\n",
      "\n",
      "Total basepairs processed:    85,175,896 bp\n",
      "Total written (filtered):     73,706,841 bp (86.5%)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/fastq_files/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_first_trimming\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# First we trim the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get trimmed reads of first group of Riboseq reads to  '$output_directory\n",
    "END=688\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -u -7 -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the second group of reads that go from SRR1223697 to SRR1223699\n",
    "echo 'get trimmed reads of second group of Riboseq reads to  '$output_directory\n",
    "END=699\n",
    "for ((i=697;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -u -7 -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the third group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get trimmed reads of third group of Riboseq reads to  '$output_directory\n",
    "END=396\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -u -7 -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the fourth group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get trimmed reads of fourth group of Riboseq reads to  '$output_directory\n",
    "END=404\n",
    "for ((i=400;i<=END;i++))\n",
    "do\n",
    "cutadapt --trim-n -u -7 -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T09:10:56.495268Z",
     "start_time": "2018-02-27T09:10:56.491967Z"
    }
   },
   "source": [
    "### Trim polyA adapters and adapters SE from Trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T14:18:39.312883Z",
     "start_time": "2018-02-27T09:20:23.089972Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get trimmed reads of first group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1223686_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223686_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1590.62 s (69 us/read; 0.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              23,135,864\n",
      "Reads with adapters:                22,835,242 (98.7%)\n",
      "Reads that were too short:          14,209,922 (61.4%)\n",
      "Reads written (passing filters):     8,925,942 (38.6%)\n",
      "\n",
      "Total basepairs processed:   809,732,303 bp\n",
      "Total written (filtered):    245,267,219 bp (30.3%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 2590 times.\n",
      "2580 times, it overlapped the 5' end of a read\n",
      "10 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t197\t22593.6\t0\t197\n",
      "6\t401\t5648.4\t0\t401\n",
      "7\t13\t1412.1\t0\t13\n",
      "8\t22\t353.0\t0\t22\n",
      "9\t55\t88.3\t0\t8 47\n",
      "10\t104\t22.1\t1\t0 104\n",
      "11\t44\t5.5\t1\t0 44\n",
      "12\t16\t1.4\t1\t0 16\n",
      "15\t2\t0.0\t1\t1 1\n",
      "17\t3\t0.0\t1\t3\n",
      "18\t186\t0.0\t1\t115 71\n",
      "19\t1085\t0.0\t1\t966 118 1\n",
      "20\t71\t0.0\t2\t4 15 52\n",
      "21\t90\t0.0\t2\t28 17 45\n",
      "22\t55\t0.0\t2\t4 32 19\n",
      "23\t78\t0.0\t2\t52 15 11\n",
      "24\t70\t0.0\t2\t0 12 58\n",
      "25\t46\t0.0\t2\t0 27 19\n",
      "26\t28\t0.0\t2\t0 14 14\n",
      "27\t9\t0.0\t2\t0 1 8\n",
      "28\t5\t0.0\t2\t0 0 5\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t4\t22593.6\t0\t4\n",
      "11\t6\t5.5\t1\t0 6\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 767 times.\n",
      "767 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t341\t22593.6\t0\t341\n",
      "6\t182\t5648.4\t0\t182\n",
      "7\t66\t1412.1\t0\t66\n",
      "8\t17\t353.0\t0\t17\n",
      "9\t83\t88.3\t0\t1 82\n",
      "10\t44\t22.1\t1\t2 42\n",
      "11\t26\t5.5\t1\t0 26\n",
      "12\t6\t1.4\t1\t0 6\n",
      "13\t2\t0.3\t1\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 199 times.\n",
      "199 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t50\t22593.6\t0\t50\n",
      "6\t31\t5648.4\t0\t31\n",
      "7\t3\t1412.1\t0\t3\n",
      "9\t68\t88.3\t0\t3 65\n",
      "10\t17\t22.1\t1\t1 16\n",
      "11\t13\t5.5\t1\t0 13\n",
      "12\t17\t1.4\t1\t0 17\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 1075 times.\n",
      "1075 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t465\t22593.6\t0\t465\n",
      "6\t102\t5648.4\t0\t102\n",
      "7\t39\t1412.1\t0\t39\n",
      "8\t38\t353.0\t0\t38\n",
      "9\t38\t88.3\t0\t19 19\n",
      "10\t67\t22.1\t1\t1 66\n",
      "11\t253\t5.5\t1\t1 252\n",
      "12\t71\t1.4\t1\t0 71\n",
      "13\t1\t0.3\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 824 times.\n",
      "824 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t163\t22593.6\t0\t163\n",
      "6\t74\t5648.4\t0\t74\n",
      "7\t16\t1412.1\t0\t16\n",
      "8\t8\t353.0\t0\t8\n",
      "9\t141\t88.3\t0\t7 134\n",
      "10\t304\t22.1\t1\t3 301\n",
      "11\t102\t5.5\t1\t0 102\n",
      "12\t12\t1.4\t1\t0 12\n",
      "13\t2\t0.3\t1\t0 2\n",
      "14\t1\t0.1\t1\t0 1\n",
      "16\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 22829787 times.\n",
      "51499 times, it overlapped the 5' end of a read\n",
      "22778288 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2787\t22593.6\t0\t2787\n",
      "6\t1405\t5648.4\t0\t1405\n",
      "7\t18529\t1412.1\t0\t18529\n",
      "8\t6083\t353.0\t0\t6083\n",
      "9\t945\t88.3\t0\t945\n",
      "10\t915\t22.1\t1\t281 634\n",
      "11\t685\t5.5\t1\t144 541\n",
      "12\t481\t1.4\t1\t89 392\n",
      "13\t396\t0.3\t1\t76 320\n",
      "14\t194\t0.1\t1\t53 141\n",
      "15\t62\t0.0\t1\t17 45\n",
      "16\t101\t0.0\t1\t26 75\n",
      "17\t39\t0.0\t1\t19 20\n",
      "18\t27\t0.0\t1\t12 15\n",
      "19\t77\t0.0\t1\t20 57\n",
      "20\t726\t0.0\t2\t42 191 493\n",
      "21\t2561\t0.0\t2\t81 796 1684\n",
      "22\t4568\t0.0\t2\t39 1513 3016\n",
      "23\t2158\t0.0\t2\t33 752 1373\n",
      "24\t1195\t0.0\t2\t80 389 726\n",
      "25\t428\t0.0\t2\t12 155 261\n",
      "26\t297\t0.0\t2\t44 78 175\n",
      "27\t124\t0.0\t2\t26 23 75\n",
      "28\t49\t0.0\t2\t8 12 29\n",
      "29\t69\t0.0\t2\t14 19 36\n",
      "30\t6598\t0.0\t3\t5490 607 265 236\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t515419\t22593.6\t0\t515419\n",
      "6\t1161038\t5648.4\t0\t1161038\n",
      "7\t1654971\t1412.1\t0\t1654971\n",
      "8\t1571355\t353.0\t0\t1571355\n",
      "9\t3693867\t88.3\t0\t3693867\n",
      "10\t5827811\t22.1\t1\t5295436 532375\n",
      "11\t3763851\t5.5\t1\t3334502 429349\n",
      "12\t2009481\t1.4\t1\t1601937 407544\n",
      "13\t1172402\t0.3\t1\t769858 402544\n",
      "14\t619957\t0.1\t1\t316398 303559\n",
      "15\t357766\t0.0\t1\t113152 244614\n",
      "16\t131753\t0.0\t1\t17686 114067\n",
      "17\t81791\t0.0\t1\t5108 76683\n",
      "18\t34933\t0.0\t1\t4160 30773\n",
      "19\t22416\t0.0\t1\t3275 19141\n",
      "20\t35566\t0.0\t2\t2778 14823 17965\n",
      "21\t17770\t0.0\t2\t1668 8065 8037\n",
      "22\t13296\t0.0\t2\t1718 7128 4450\n",
      "23\t11256\t0.0\t2\t1634 6015 3607\n",
      "24\t9934\t0.0\t2\t1715 4833 3386\n",
      "25\t13093\t0.0\t2\t3200 6593 3300\n",
      "26\t7184\t0.0\t2\t1559 3028 2597\n",
      "27\t5857\t0.0\t2\t1803 1986 2068\n",
      "28\t6134\t0.0\t2\t2853 2077 1204\n",
      "29\t13961\t0.0\t2\t7090 4236 2635\n",
      "30\t8886\t0.0\t3\t3894 1993 1558 1441\n",
      "31\t7190\t0.0\t3\t4141 1358 860 831\n",
      "32\t4792\t0.0\t3\t3249 737 430 376\n",
      "33\t2928\t0.0\t3\t2057 380 235 256\n",
      "34\t1630\t0.0\t3\t1312 137 81 100\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1223687_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223687_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1225.61 s (76 us/read; 0.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,101,967\n",
      "Reads with adapters:                13,961,856 (86.7%)\n",
      "Reads that were too short:           7,467,540 (46.4%)\n",
      "Reads written (passing filters):     8,634,427 (53.6%)\n",
      "\n",
      "Total basepairs processed:   708,480,362 bp\n",
      "Total written (filtered):    281,218,948 bp (39.7%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 660922 times.\n",
      "1524 times, it overlapped the 5' end of a read\n",
      "659398 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t967\t15724.6\t0\t967\n",
      "6\t410\t3931.1\t0\t410\n",
      "7\t2\t982.8\t0\t2\n",
      "9\t2\t61.4\t0\t0 2\n",
      "10\t5\t15.4\t1\t0 5\n",
      "12\t1\t1.0\t1\t0 1\n",
      "16\t4\t0.0\t1\t4\n",
      "17\t45\t0.0\t1\t32 13\n",
      "18\t2\t0.0\t1\t0 2\n",
      "19\t7\t0.0\t1\t0 7\n",
      "20\t17\t0.0\t2\t15 0 2\n",
      "21\t3\t0.0\t2\t2 0 1\n",
      "22\t7\t0.0\t2\t3 3 1\n",
      "23\t4\t0.0\t2\t1 0 3\n",
      "24\t14\t0.0\t2\t0 7 7\n",
      "25\t16\t0.0\t2\t0 3 13\n",
      "26\t9\t0.0\t2\t0 2 7\n",
      "27\t4\t0.0\t2\t0 0 4\n",
      "28\t2\t0.0\t2\t0 0 2\n",
      "29\t2\t0.0\t2\t0 0 0 2\n",
      "30\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t168165\t15724.6\t0\t168165\n",
      "6\t135899\t3931.1\t0\t135899\n",
      "7\t100226\t982.8\t0\t100226\n",
      "8\t72063\t245.7\t0\t72063\n",
      "9\t47126\t61.4\t0\t46805 321\n",
      "10\t66724\t15.4\t1\t28836 37888\n",
      "11\t33675\t3.8\t1\t13504 20171\n",
      "12\t21025\t1.0\t1\t8346 12679\n",
      "13\t14283\t0.2\t1\t5760 8523\n",
      "14\t105\t0.1\t1\t0 105\n",
      "15\t107\t0.0\t1\t0 107\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 7315 times.\n",
      "7312 times, it overlapped the 5' end of a read\n",
      "3 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t823\t15724.6\t0\t823\n",
      "6\t189\t3931.1\t0\t189\n",
      "7\t6268\t982.8\t0\t6268\n",
      "8\t28\t245.7\t0\t28\n",
      "10\t4\t15.4\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.2\t1\t0 1\n",
      "15\t2\t0.0\t1\t0 2\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 335 times.\n",
      "335 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t316\t15724.6\t0\t316\n",
      "6\t8\t3931.1\t0\t8\n",
      "8\t10\t245.7\t0\t10\n",
      "11\t1\t3.8\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 30636 times.\n",
      "558 times, it overlapped the 5' end of a read\n",
      "30078 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t64\t15724.6\t0\t64\n",
      "6\t8\t3931.1\t0\t8\n",
      "7\t2\t982.8\t0\t2\n",
      "8\t1\t245.7\t0\t1\n",
      "9\t1\t61.4\t0\t1\n",
      "10\t8\t15.4\t1\t0 8\n",
      "11\t3\t3.8\t1\t0 3\n",
      "12\t4\t1.0\t1\t0 4\n",
      "22\t1\t0.0\t2\t1\n",
      "24\t12\t0.0\t2\t12\n",
      "25\t1\t0.0\t2\t1\n",
      "26\t9\t0.0\t2\t7 2\n",
      "27\t24\t0.0\t2\t24\n",
      "28\t22\t0.0\t2\t21 1\n",
      "29\t22\t0.0\t2\t20 1 0 1\n",
      "30\t92\t0.0\t3\t86 5 1\n",
      "31\t38\t0.0\t3\t34 4\n",
      "32\t222\t0.0\t3\t215 7\n",
      "33\t7\t0.0\t3\t7\n",
      "34\t17\t0.0\t3\t14 2 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t144\t0.2\t1\t0 144\n",
      "14\t8069\t0.1\t1\t3345 4724\n",
      "15\t5148\t0.0\t1\t2036 3112\n",
      "16\t2596\t0.0\t1\t1087 1509\n",
      "17\t1660\t0.0\t1\t667 993\n",
      "18\t1156\t0.0\t1\t460 694 2\n",
      "19\t854\t0.0\t1\t343 491 20\n",
      "20\t1170\t0.0\t2\t292 392 486\n",
      "21\t1273\t0.0\t2\t309 454 510\n",
      "22\t1493\t0.0\t2\t504 603 386\n",
      "23\t4236\t0.0\t2\t1677 1790 769\n",
      "24\t1002\t0.0\t2\t390 371 241\n",
      "25\t636\t0.0\t2\t308 195 133\n",
      "26\t201\t0.0\t2\t88 68 45\n",
      "27\t64\t0.0\t2\t34 19 11\n",
      "28\t11\t0.0\t2\t9 2\n",
      "29\t10\t0.0\t2\t8 0 1 1\n",
      "30\t16\t0.0\t3\t10 2 2 2\n",
      "31\t13\t0.0\t3\t7 3 2 1\n",
      "32\t19\t0.0\t3\t14 4 0 1\n",
      "33\t26\t0.0\t3\t22 4\n",
      "34\t24\t0.0\t3\t24\n",
      "35\t34\t0.0\t3\t31 2 1\n",
      "36\t45\t0.0\t3\t42 1 2\n",
      "37\t31\t0.0\t3\t31\n",
      "38\t46\t0.0\t3\t38 3 1 4\n",
      "39\t38\t0.0\t3\t33 5\n",
      "40\t21\t0.0\t3\t21\n",
      "41\t22\t0.0\t3\t19 3\n",
      "42\t15\t0.0\t3\t14 1\n",
      "43\t5\t0.0\t3\t5\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 729 times.\n",
      "729 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t320\t15724.6\t0\t320\n",
      "6\t329\t3931.1\t0\t329\n",
      "7\t3\t982.8\t0\t3\n",
      "8\t1\t245.7\t0\t1\n",
      "9\t20\t61.4\t0\t1 19\n",
      "10\t50\t15.4\t1\t0 50\n",
      "11\t6\t3.8\t1\t0 6\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 13261919 times.\n",
      "22748 times, it overlapped the 5' end of a read\n",
      "13239171 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2703\t15724.6\t0\t2703\n",
      "6\t256\t3931.1\t0\t256\n",
      "7\t64\t982.8\t0\t64\n",
      "8\t21\t245.7\t0\t21\n",
      "9\t5\t61.4\t0\t5\n",
      "10\t13\t15.4\t1\t2 11\n",
      "11\t3\t3.8\t1\t2 1\n",
      "12\t1\t1.0\t1\t0 1\n",
      "13\t2\t0.2\t1\t2\n",
      "14\t1\t0.1\t1\t1\n",
      "15\t1\t0.0\t1\t0 1\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t4\t0.0\t1\t1 3\n",
      "20\t61\t0.0\t2\t1 5 55\n",
      "21\t209\t0.0\t2\t3 47 159\n",
      "22\t1309\t0.0\t2\t5 183 1121\n",
      "23\t1776\t0.0\t2\t5 371 1400\n",
      "24\t5764\t0.0\t2\t3 1931 3830\n",
      "25\t5791\t0.0\t2\t6 2077 3708\n",
      "26\t2377\t0.0\t2\t1 824 1552\n",
      "27\t772\t0.0\t2\t6 278 488\n",
      "28\t385\t0.0\t2\t14 157 214\n",
      "29\t247\t0.0\t2\t13 105 129\n",
      "30\t982\t0.0\t3\t323 230 180 249\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2181\t15724.6\t0\t2181\n",
      "6\t754\t3931.1\t0\t754\n",
      "7\t497\t982.8\t0\t497\n",
      "8\t562\t245.7\t0\t562\n",
      "9\t1415\t61.4\t0\t1415\n",
      "10\t4929\t15.4\t1\t4511 418\n",
      "11\t18133\t3.8\t1\t17436 697\n",
      "12\t68755\t1.0\t1\t67577 1178\n",
      "13\t264032\t0.2\t1\t258234 5798\n",
      "14\t674452\t0.1\t1\t658496 15956\n",
      "15\t792803\t0.0\t1\t755330 37473\n",
      "16\t1161704\t0.0\t1\t1052513 109191\n",
      "17\t1604188\t0.0\t1\t1275026 329162\n",
      "18\t1209236\t0.0\t1\t1040814 168422\n",
      "19\t1230144\t0.0\t1\t976526 253618\n",
      "20\t1715415\t0.0\t2\t933828 495324 286263\n",
      "21\t1398150\t0.0\t2\t475952 621338 300860\n",
      "22\t1222151\t0.0\t2\t259779 510604 451768\n",
      "23\t883330\t0.0\t2\t58011 276289 549030\n",
      "24\t565200\t0.0\t2\t13032 121253 430915\n",
      "25\t232073\t0.0\t2\t2723 36599 192751\n",
      "26\t80009\t0.0\t2\t1381 12227 66401\n",
      "27\t33869\t0.0\t2\t1433 6100 26336\n",
      "28\t11016\t0.0\t2\t987 3383 6646\n",
      "29\t7898\t0.0\t2\t759 2669 4470\n",
      "30\t11143\t0.0\t3\t608 1808 2185 6542\n",
      "31\t8713\t0.0\t3\t713 2070 1802 4128\n",
      "32\t7931\t0.0\t3\t546 1778 1750 3857\n",
      "33\t6280\t0.0\t3\t485 1468 1369 2958\n",
      "34\t5889\t0.0\t3\t544 1667 1054 2624\n",
      "35\t4194\t0.0\t3\t384 1234 821 1755\n",
      "36\t2822\t0.0\t3\t335 1062 504 921\n",
      "37\t2202\t0.0\t3\t252 793 320 837\n",
      "38\t1582\t0.0\t3\t288 698 229 367\n",
      "39\t1351\t0.0\t3\t267 602 185 297\n",
      "40\t1439\t0.0\t3\t279 625 214 321\n",
      "41\t759\t0.0\t3\t101 264 143 251\n",
      "42\t1647\t0.0\t3\t213 292 319 823\n",
      "43\t323\t0.0\t3\t56 82 64 121\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1223688_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223688_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 3095.15 s (75 us/read; 0.80 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              41,451,837\n",
      "Reads with adapters:                38,824,503 (93.7%)\n",
      "Reads that were too short:          21,888,230 (52.8%)\n",
      "Reads written (passing filters):    19,563,607 (47.2%)\n",
      "\n",
      "Total basepairs processed: 1,782,424,762 bp\n",
      "Total written (filtered):    588,711,622 bp (33.0%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 116791 times.\n",
      "1544 times, it overlapped the 5' end of a read\n",
      "115247 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t902\t40480.3\t0\t902\n",
      "6\t547\t10120.1\t0\t547\n",
      "7\t3\t2530.0\t0\t3\n",
      "9\t7\t158.1\t0\t0 7\n",
      "10\t11\t39.5\t1\t0 11\n",
      "11\t2\t9.9\t1\t0 2\n",
      "12\t2\t2.5\t1\t0 2\n",
      "17\t12\t0.0\t1\t11 1\n",
      "18\t3\t0.0\t1\t0 3\n",
      "19\t4\t0.0\t1\t0 4\n",
      "20\t1\t0.0\t2\t0 0 1\n",
      "21\t2\t0.0\t2\t2\n",
      "22\t3\t0.0\t2\t0 3\n",
      "23\t5\t0.0\t2\t5\n",
      "24\t7\t0.0\t2\t0 3 4\n",
      "25\t17\t0.0\t2\t0 4 13\n",
      "26\t8\t0.0\t2\t0 1 7\n",
      "27\t3\t0.0\t2\t0 0 3\n",
      "29\t4\t0.0\t2\t0 0 2 2\n",
      "31\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t31898\t40480.3\t0\t31898\n",
      "6\t22267\t10120.1\t0\t22267\n",
      "7\t13911\t2530.0\t0\t13911\n",
      "8\t9419\t632.5\t0\t9419\n",
      "9\t6873\t158.1\t0\t6818 55\n",
      "10\t15932\t39.5\t1\t3371 12561\n",
      "11\t10750\t9.9\t1\t2274 8476\n",
      "12\t2448\t2.5\t1\t395 2053\n",
      "13\t1653\t0.6\t1\t264 1389\n",
      "14\t31\t0.2\t1\t0 31\n",
      "15\t65\t0.0\t1\t0 65\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 6010 times.\n",
      "6009 times, it overlapped the 5' end of a read\n",
      "1 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1233\t40480.3\t0\t1233\n",
      "6\t259\t10120.1\t0\t259\n",
      "7\t4490\t2530.0\t0\t4490\n",
      "8\t22\t632.5\t0\t22\n",
      "10\t4\t39.5\t1\t0 4\n",
      "11\t1\t9.9\t1\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.2\t1\t1\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 488 times.\n",
      "482 times, it overlapped the 5' end of a read\n",
      "6 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t405\t40480.3\t0\t405\n",
      "6\t36\t10120.1\t0\t36\n",
      "7\t17\t2530.0\t0\t17\n",
      "8\t1\t632.5\t0\t1\n",
      "9\t4\t158.1\t0\t0 4\n",
      "10\t10\t39.5\t1\t0 10\n",
      "12\t1\t2.5\t1\t0 1\n",
      "29\t2\t0.0\t2\t2\n",
      "30\t1\t0.0\t3\t0 0 1\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t4\t0.0\t3\t4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t1\t1\n",
      "16\t1\t0.0\t1\t1\n",
      "24\t2\t0.0\t2\t2\n",
      "36\t1\t0.0\t3\t1\n",
      "37\t1\t0.0\t3\t1\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 4543 times.\n",
      "516 times, it overlapped the 5' end of a read\n",
      "4027 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t121\t40480.3\t0\t121\n",
      "6\t40\t10120.1\t0\t40\n",
      "8\t1\t632.5\t0\t1\n",
      "9\t5\t158.1\t0\t1 4\n",
      "10\t21\t39.5\t1\t0 21\n",
      "11\t9\t9.9\t1\t0 9\n",
      "12\t2\t2.5\t1\t0 2\n",
      "23\t7\t0.0\t2\t7\n",
      "24\t35\t0.0\t2\t35\n",
      "25\t5\t0.0\t2\t5\n",
      "26\t23\t0.0\t2\t22 1\n",
      "27\t44\t0.0\t2\t42 2\n",
      "28\t21\t0.0\t2\t20 1\n",
      "29\t18\t0.0\t2\t15 1 0 2\n",
      "30\t45\t0.0\t3\t44 1\n",
      "31\t20\t0.0\t3\t14 4 1 1\n",
      "32\t90\t0.0\t3\t85 5\n",
      "34\t9\t0.0\t3\t9\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.6\t1\t0 2\n",
      "14\t734\t0.2\t1\t141 593\n",
      "15\t849\t0.0\t1\t257 592\n",
      "16\t432\t0.0\t1\t147 285\n",
      "17\t315\t0.0\t1\t98 217\n",
      "18\t148\t0.0\t1\t62 81 5\n",
      "19\t114\t0.0\t1\t34 65 15\n",
      "20\t179\t0.0\t2\t25 44 110\n",
      "21\t141\t0.0\t2\t33 34 74\n",
      "22\t347\t0.0\t2\t143 100 104\n",
      "23\t228\t0.0\t2\t46 66 116\n",
      "24\t161\t0.0\t2\t25 50 86\n",
      "25\t99\t0.0\t2\t27 34 38\n",
      "26\t45\t0.0\t2\t10 19 16\n",
      "27\t27\t0.0\t2\t11 11 5\n",
      "28\t10\t0.0\t2\t3 5 2\n",
      "29\t5\t0.0\t2\t2 1 1 1\n",
      "30\t14\t0.0\t3\t7 5 2\n",
      "31\t9\t0.0\t3\t3 3 3\n",
      "32\t9\t0.0\t3\t5 3 1\n",
      "33\t9\t0.0\t3\t8 0 1\n",
      "34\t19\t0.0\t3\t12 7\n",
      "35\t28\t0.0\t3\t24 4\n",
      "36\t23\t0.0\t3\t18 5\n",
      "37\t25\t0.0\t3\t18 5 0 2\n",
      "38\t16\t0.0\t3\t14 2\n",
      "39\t17\t0.0\t3\t14 1 0 2\n",
      "40\t8\t0.0\t3\t6 2\n",
      "41\t7\t0.0\t3\t3 4\n",
      "42\t7\t0.0\t3\t4 2 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 667 times.\n",
      "667 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t429\t40480.3\t0\t429\n",
      "6\t181\t10120.1\t0\t181\n",
      "7\t3\t2530.0\t0\t3\n",
      "9\t6\t158.1\t0\t0 6\n",
      "10\t44\t39.5\t1\t0 44\n",
      "11\t4\t9.9\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 38696004 times.\n",
      "41791 times, it overlapped the 5' end of a read\n",
      "38654213 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t6554\t40480.3\t0\t6554\n",
      "6\t727\t10120.1\t0\t727\n",
      "7\t247\t2530.0\t0\t247\n",
      "8\t51\t632.5\t0\t51\n",
      "9\t14\t158.1\t0\t14\n",
      "10\t23\t39.5\t1\t4 19\n",
      "11\t8\t9.9\t1\t4 4\n",
      "12\t4\t2.5\t1\t2 2\n",
      "13\t3\t0.6\t1\t1 2\n",
      "14\t3\t0.2\t1\t0 3\n",
      "15\t1\t0.0\t1\t0 1\n",
      "16\t2\t0.0\t1\t0 2\n",
      "17\t1\t0.0\t1\t0 1\n",
      "18\t6\t0.0\t1\t3 3\n",
      "19\t5\t0.0\t1\t0 5\n",
      "20\t210\t0.0\t2\t1 19 190\n",
      "21\t541\t0.0\t2\t3 131 407\n",
      "22\t2956\t0.0\t2\t9 601 2346\n",
      "23\t3404\t0.0\t2\t11 1030 2363\n",
      "24\t9187\t0.0\t2\t2 3426 5759\n",
      "25\t6346\t0.0\t2\t7 2522 3817\n",
      "26\t2570\t0.0\t2\t9 980 1581\n",
      "27\t727\t0.0\t2\t11 250 466\n",
      "28\t568\t0.0\t2\t3 334 231\n",
      "29\t501\t0.0\t2\t3 259 239\n",
      "30\t7132\t0.0\t3\t5067 1220 420 425\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t39092\t40480.3\t0\t39092\n",
      "6\t7125\t10120.1\t0\t7125\n",
      "7\t8284\t2530.0\t0\t8284\n",
      "8\t7657\t632.5\t0\t7657\n",
      "9\t20838\t158.1\t0\t20838\n",
      "10\t71318\t39.5\t1\t69381 1937\n",
      "11\t254117\t9.9\t1\t248868 5249\n",
      "12\t901805\t2.5\t1\t880694 21111\n",
      "13\t2095884\t0.6\t1\t2040939 54945\n",
      "14\t2380485\t0.2\t1\t2256229 124256\n",
      "15\t3346389\t0.0\t1\t2989922 356467\n",
      "16\t4487224\t0.0\t1\t3609539 877685\n",
      "17\t3182308\t0.0\t1\t2720490 461818\n",
      "18\t3151856\t0.0\t1\t2702228 449628\n",
      "19\t3415048\t0.0\t1\t2802943 612105\n",
      "20\t3240965\t0.0\t2\t2140707 603803 496455\n",
      "21\t3006038\t0.0\t2\t1661069 750134 594835\n",
      "22\t2516733\t0.0\t2\t825175 889568 801990\n",
      "23\t2499551\t0.0\t2\t359105 957113 1183333\n",
      "24\t1572516\t0.0\t2\t147479 432475 992562\n",
      "25\t866542\t0.0\t2\t95570 216036 554936\n",
      "26\t537011\t0.0\t2\t73743 137694 325574\n",
      "27\t227564\t0.0\t2\t47073 55278 125213\n",
      "28\t155809\t0.0\t2\t36778 48053 70978\n",
      "29\t130752\t0.0\t2\t33308 39710 57734\n",
      "30\t161173\t0.0\t3\t25677 31095 32070 72331\n",
      "31\t105233\t0.0\t3\t20497 22268 19848 42620\n",
      "32\t64787\t0.0\t3\t14099 15652 11087 23949\n",
      "33\t55491\t0.0\t3\t13741 15875 7353 18522\n",
      "34\t33377\t0.0\t3\t8658 10203 4520 9996\n",
      "35\t23591\t0.0\t3\t6977 8311 2806 5497\n",
      "36\t17619\t0.0\t3\t5697 6430 1842 3650\n",
      "37\t15326\t0.0\t3\t5870 5787 1296 2373\n",
      "38\t13127\t0.0\t3\t5523 4585 1077 1942\n",
      "39\t15593\t0.0\t3\t7099 4973 1209 2312\n",
      "40\t7342\t0.0\t3\t2734 1892 909 1807\n",
      "41\t16351\t0.0\t3\t4130 2850 2937 6434\n",
      "42\t2292\t0.0\t3\t1319 503 204 266\n",
      "\n",
      "get trimmed reads of second group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1223697_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223697_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1282.15 s (66 us/read; 0.91 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,360,675\n",
      "Reads with adapters:                18,829,649 (97.3%)\n",
      "Reads that were too short:          14,381,247 (74.3%)\n",
      "Reads written (passing filters):     4,979,428 (25.7%)\n",
      "\n",
      "Total basepairs processed:   677,572,332 bp\n",
      "Total written (filtered):    142,020,916 bp (21.0%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 3565 times.\n",
      "3543 times, it overlapped the 5' end of a read\n",
      "22 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t158\t18906.9\t0\t158\n",
      "6\t182\t4726.7\t0\t182\n",
      "7\t7\t1181.7\t0\t7\n",
      "8\t4\t295.4\t0\t4\n",
      "9\t20\t73.9\t0\t0 20\n",
      "10\t82\t18.5\t1\t0 82\n",
      "11\t18\t4.6\t1\t0 18\n",
      "12\t7\t1.2\t1\t1 6\n",
      "14\t1\t0.1\t1\t1\n",
      "15\t3\t0.0\t1\t0 3\n",
      "16\t1\t0.0\t1\t1\n",
      "17\t19\t0.0\t1\t19\n",
      "18\t488\t0.0\t1\t383 105\n",
      "19\t1851\t0.0\t1\t1613 237 1\n",
      "20\t456\t0.0\t2\t12 149 295\n",
      "21\t117\t0.0\t2\t32 12 73\n",
      "22\t40\t0.0\t2\t4 26 10\n",
      "23\t26\t0.0\t2\t7 6 13\n",
      "24\t36\t0.0\t2\t0 8 28\n",
      "25\t10\t0.0\t2\t0 3 7\n",
      "26\t12\t0.0\t2\t0 6 6\n",
      "27\t4\t0.0\t2\t0 0 4\n",
      "28\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t7\t18906.9\t0\t7\n",
      "6\t5\t4726.7\t0\t5\n",
      "9\t1\t73.9\t0\t1\n",
      "10\t6\t18.5\t1\t0 6\n",
      "11\t3\t4.6\t1\t0 3\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 442 times.\n",
      "442 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t168\t18906.9\t0\t168\n",
      "6\t24\t4726.7\t0\t24\n",
      "7\t172\t1181.7\t0\t172\n",
      "8\t17\t295.4\t0\t17\n",
      "9\t23\t73.9\t0\t0 23\n",
      "10\t23\t18.5\t1\t4 19\n",
      "11\t15\t4.6\t1\t0 15\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 233 times.\n",
      "233 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t51\t18906.9\t0\t51\n",
      "6\t19\t4726.7\t0\t19\n",
      "7\t15\t1181.7\t0\t15\n",
      "8\t11\t295.4\t0\t11\n",
      "9\t40\t73.9\t0\t0 40\n",
      "10\t89\t18.5\t1\t0 89\n",
      "11\t4\t4.6\t1\t0 4\n",
      "12\t3\t1.2\t1\t0 3\n",
      "14\t1\t0.1\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 402 times.\n",
      "402 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t226\t18906.9\t0\t226\n",
      "6\t19\t4726.7\t0\t19\n",
      "7\t16\t1181.7\t0\t16\n",
      "9\t12\t73.9\t0\t4 8\n",
      "10\t36\t18.5\t1\t0 36\n",
      "11\t71\t4.6\t1\t0 71\n",
      "12\t18\t1.2\t1\t0 18\n",
      "13\t2\t0.3\t1\t0 2\n",
      "15\t1\t0.0\t1\t0 1\n",
      "16\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 318 times.\n",
      "318 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t132\t18906.9\t0\t132\n",
      "6\t41\t4726.7\t0\t41\n",
      "7\t4\t1181.7\t0\t4\n",
      "8\t2\t295.4\t0\t2\n",
      "9\t44\t73.9\t0\t7 37\n",
      "10\t74\t18.5\t1\t0 74\n",
      "11\t10\t4.6\t1\t0 10\n",
      "12\t11\t1.2\t1\t0 11\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 18824689 times.\n",
      "529373 times, it overlapped the 5' end of a read\n",
      "18295316 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1769\t18906.9\t0\t1769\n",
      "6\t893\t4726.7\t0\t893\n",
      "7\t1479\t1181.7\t0\t1479\n",
      "8\t915\t295.4\t0\t915\n",
      "9\t388\t73.9\t0\t388\n",
      "10\t1293\t18.5\t1\t195 1098\n",
      "11\t756\t4.6\t1\t141 615\n",
      "12\t382\t1.2\t1\t82 300\n",
      "13\t246\t0.3\t1\t45 201\n",
      "14\t177\t0.1\t1\t52 125\n",
      "15\t900\t0.0\t1\t109 791\n",
      "16\t356\t0.0\t1\t109 247\n",
      "17\t415\t0.0\t1\t127 288\n",
      "18\t733\t0.0\t1\t154 579\n",
      "19\t2699\t0.0\t1\t220 2479\n",
      "20\t38093\t0.0\t2\t362 8816 28915\n",
      "21\t94777\t0.0\t2\t146 23328 71303\n",
      "22\t215266\t0.0\t2\t91 57762 157413\n",
      "23\t92514\t0.0\t2\t64 30106 62344\n",
      "24\t36720\t0.0\t2\t28 14669 22023\n",
      "25\t15779\t0.0\t2\t34 5913 9832\n",
      "26\t4671\t0.0\t2\t12 1512 3147\n",
      "27\t1211\t0.0\t2\t12 375 824\n",
      "28\t992\t0.0\t2\t9 427 556\n",
      "29\t1083\t0.0\t2\t16 392 675\n",
      "30\t14866\t0.0\t3\t7404 4087 1508 1867\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t494581\t18906.9\t0\t494581\n",
      "6\t781151\t4726.7\t0\t781151\n",
      "7\t1186841\t1181.7\t0\t1186841\n",
      "8\t1073915\t295.4\t0\t1073915\n",
      "9\t907484\t73.9\t0\t907484\n",
      "10\t1680301\t18.5\t1\t1324427 355874\n",
      "11\t2398584\t4.6\t1\t2084845 313739\n",
      "12\t2261114\t1.2\t1\t1890598 370516\n",
      "13\t2235620\t0.3\t1\t1753672 481948\n",
      "14\t1401186\t0.1\t1\t1068713 332473\n",
      "15\t1056578\t0.0\t1\t716965 339613\n",
      "16\t823751\t0.0\t1\t508509 315242\n",
      "17\t544924\t0.0\t1\t308681 236243\n",
      "18\t369814\t0.0\t1\t176919 192895\n",
      "19\t215133\t0.0\t1\t79570 135563\n",
      "20\t281509\t0.0\t2\t39227 108173 134109\n",
      "21\t133004\t0.0\t2\t11924 47813 73267\n",
      "22\t96165\t0.0\t2\t8876 36107 51182\n",
      "23\t55536\t0.0\t2\t4370 18604 32562\n",
      "24\t39869\t0.0\t2\t3671 11369 24829\n",
      "25\t33939\t0.0\t2\t5657 12540 15742\n",
      "26\t25340\t0.0\t2\t2942 9645 12753\n",
      "27\t23084\t0.0\t2\t4125 9209 9750\n",
      "28\t18665\t0.0\t2\t6494 8946 3225\n",
      "29\t40025\t0.0\t2\t15619 18079 6327\n",
      "30\t26604\t0.0\t3\t11110 9424 3449 2621\n",
      "31\t26731\t0.0\t3\t14418 7694 2574 2045\n",
      "32\t26368\t0.0\t3\t17257 7066 1208 837\n",
      "33\t26788\t0.0\t3\t17045 6246 1766 1731\n",
      "34\t10712\t0.0\t3\t6934 2117 792 869\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1223698_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223698_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 2094.89 s (82 us/read; 0.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              25,664,498\n",
      "Reads with adapters:                25,236,179 (98.3%)\n",
      "Reads that were too short:           6,397,114 (24.9%)\n",
      "Reads written (passing filters):    19,267,384 (75.1%)\n",
      "\n",
      "Total basepairs processed: 1,129,228,167 bp\n",
      "Total written (filtered):    566,978,985 bp (50.2%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 147001 times.\n",
      "243 times, it overlapped the 5' end of a read\n",
      "146758 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t70\t25063.0\t0\t70\n",
      "6\t73\t6265.7\t0\t73\n",
      "7\t1\t1566.4\t0\t1\n",
      "8\t1\t391.6\t0\t1\n",
      "10\t1\t24.5\t1\t0 1\n",
      "16\t3\t0.0\t1\t3\n",
      "17\t10\t0.0\t1\t10\n",
      "20\t4\t0.0\t2\t4\n",
      "21\t5\t0.0\t2\t5\n",
      "22\t2\t0.0\t2\t1 1\n",
      "23\t13\t0.0\t2\t5 4 4\n",
      "24\t14\t0.0\t2\t0 8 6\n",
      "25\t20\t0.0\t2\t0 8 12\n",
      "26\t8\t0.0\t2\t0 0 8\n",
      "27\t11\t0.0\t2\t0 1 10\n",
      "28\t5\t0.0\t2\t0 0 5\n",
      "29\t2\t0.0\t2\t0 0 1 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t42684\t25063.0\t0\t42684\n",
      "6\t26595\t6265.7\t0\t26595\n",
      "7\t17747\t1566.4\t0\t17747\n",
      "8\t13213\t391.6\t0\t13213\n",
      "9\t9664\t97.9\t0\t9595 69\n",
      "10\t16708\t24.5\t1\t7196 9512\n",
      "11\t9839\t6.1\t1\t4144 5695\n",
      "12\t5997\t1.5\t1\t2433 3564\n",
      "13\t4148\t0.4\t1\t1830 2318\n",
      "14\t99\t0.1\t1\t0 99\n",
      "15\t64\t0.0\t1\t0 64\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 5512 times.\n",
      "5511 times, it overlapped the 5' end of a read\n",
      "1 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t284\t25063.0\t0\t284\n",
      "6\t48\t6265.7\t0\t48\n",
      "7\t5167\t1566.4\t0\t5167\n",
      "8\t12\t391.6\t0\t12\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 34 times.\n",
      "34 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t30\t25063.0\t0\t30\n",
      "6\t1\t6265.7\t0\t1\n",
      "8\t3\t391.6\t0\t3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 7021 times.\n",
      "169 times, it overlapped the 5' end of a read\n",
      "6852 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t6\t25063.0\t0\t6\n",
      "6\t3\t6265.7\t0\t3\n",
      "7\t2\t1566.4\t0\t2\n",
      "10\t1\t24.5\t1\t0 1\n",
      "11\t4\t6.1\t1\t0 4\n",
      "12\t2\t1.5\t1\t0 2\n",
      "25\t2\t0.0\t2\t2\n",
      "26\t17\t0.0\t2\t17\n",
      "27\t7\t0.0\t2\t7\n",
      "28\t11\t0.0\t2\t10 1\n",
      "29\t14\t0.0\t2\t10 4\n",
      "30\t19\t0.0\t3\t16 3\n",
      "31\t21\t0.0\t3\t20 1\n",
      "32\t36\t0.0\t3\t36\n",
      "33\t9\t0.0\t3\t7 1 0 1\n",
      "34\t15\t0.0\t3\t14 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t20\t0.4\t1\t0 20\n",
      "14\t2114\t0.1\t1\t904 1210\n",
      "15\t1451\t0.0\t1\t647 804\n",
      "16\t683\t0.0\t1\t265 418\n",
      "17\t469\t0.0\t1\t197 272\n",
      "18\t257\t0.0\t1\t111 145 1\n",
      "19\t171\t0.0\t1\t78 90 3\n",
      "20\t211\t0.0\t2\t38 82 91\n",
      "21\t172\t0.0\t2\t51 43 78\n",
      "22\t172\t0.0\t2\t67 65 40\n",
      "23\t487\t0.0\t2\t189 194 104\n",
      "24\t150\t0.0\t2\t45 60 45\n",
      "25\t131\t0.0\t2\t60 40 31\n",
      "26\t49\t0.0\t2\t19 18 12\n",
      "27\t14\t0.0\t2\t7 5 2\n",
      "28\t9\t0.0\t2\t4 3 2\n",
      "29\t6\t0.0\t2\t5 1\n",
      "30\t9\t0.0\t3\t5 3 0 1\n",
      "31\t7\t0.0\t3\t4 2 1\n",
      "32\t17\t0.0\t3\t11 1 4 1\n",
      "33\t22\t0.0\t3\t18 4\n",
      "34\t27\t0.0\t3\t24 3\n",
      "35\t28\t0.0\t3\t24 3 1\n",
      "36\t32\t0.0\t3\t28 2 1 1\n",
      "37\t29\t0.0\t3\t27 2\n",
      "38\t23\t0.0\t3\t22 1\n",
      "39\t27\t0.0\t3\t26 0 0 1\n",
      "40\t15\t0.0\t3\t12 2 1\n",
      "41\t23\t0.0\t3\t20 3\n",
      "42\t11\t0.0\t3\t11\n",
      "43\t16\t0.0\t3\t15 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 117 times.\n",
      "117 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t38\t25063.0\t0\t38\n",
      "6\t71\t6265.7\t0\t71\n",
      "7\t1\t1566.4\t0\t1\n",
      "8\t2\t391.6\t0\t2\n",
      "9\t2\t97.9\t0\t0 2\n",
      "10\t3\t24.5\t1\t0 3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 25076494 times.\n",
      "7374 times, it overlapped the 5' end of a read\n",
      "25069120 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t820\t25063.0\t0\t820\n",
      "6\t116\t6265.7\t0\t116\n",
      "7\t65\t1566.4\t0\t65\n",
      "8\t17\t391.6\t0\t17\n",
      "9\t6\t97.9\t0\t6\n",
      "10\t9\t24.5\t1\t4 5\n",
      "11\t7\t6.1\t1\t6 1\n",
      "12\t3\t1.5\t1\t2 1\n",
      "13\t5\t0.4\t1\t1 4\n",
      "14\t3\t0.1\t1\t1 2\n",
      "15\t2\t0.0\t1\t0 2\n",
      "16\t1\t0.0\t1\t1\n",
      "17\t1\t0.0\t1\t0 1\n",
      "20\t25\t0.0\t2\t1 1 23\n",
      "21\t46\t0.0\t2\t0 9 37\n",
      "22\t253\t0.0\t2\t5 31 217\n",
      "23\t306\t0.0\t2\t3 50 253\n",
      "24\t829\t0.0\t2\t3 216 610\n",
      "25\t851\t0.0\t2\t7 231 613\n",
      "26\t388\t0.0\t2\t2 127 259\n",
      "27\t169\t0.0\t2\t1 51 117\n",
      "28\t76\t0.0\t2\t5 26 45\n",
      "29\t71\t0.0\t2\t3 30 38\n",
      "30\t3305\t0.0\t3\t2878 249 95 83\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2114\t25063.0\t0\t2114\n",
      "6\t4024\t6265.7\t0\t4024\n",
      "7\t14298\t1566.4\t0\t14298\n",
      "8\t52830\t391.6\t0\t52830\n",
      "9\t191597\t97.9\t0\t191597\n",
      "10\t441221\t24.5\t1\t431571 9650\n",
      "11\t676938\t6.1\t1\t658789 18149\n",
      "12\t1028352\t1.5\t1\t983654 44698\n",
      "13\t1851706\t0.4\t1\t1698679 153027\n",
      "14\t2814694\t0.1\t1\t2658495 156199\n",
      "15\t3537276\t0.0\t1\t3321653 215623\n",
      "16\t2831192\t0.0\t1\t2489651 341541\n",
      "17\t2966992\t0.0\t1\t2091224 875768\n",
      "18\t2267980\t0.0\t1\t1767726 500254\n",
      "19\t1529766\t0.0\t1\t1121150 408616\n",
      "20\t1838510\t0.0\t2\t651654 520609 666247\n",
      "21\t979391\t0.0\t2\t181740 414548 383103\n",
      "22\t899565\t0.0\t2\t67421 410422 421722\n",
      "23\t404991\t0.0\t2\t18818 118453 267720\n",
      "24\t310589\t0.0\t2\t4494 66970 239125\n",
      "25\t89139\t0.0\t2\t1786 19659 67694\n",
      "26\t84411\t0.0\t2\t2098 17480 64833\n",
      "27\t33902\t0.0\t2\t3011 15319 15572\n",
      "28\t25865\t0.0\t2\t3592 15553 6720\n",
      "29\t40273\t0.0\t2\t5778 22476 12019\n",
      "30\t35317\t0.0\t3\t4990 15273 5702 9352\n",
      "31\t28776\t0.0\t3\t5920 14640 3838 4378\n",
      "32\t21730\t0.0\t3\t5926 11573 1808 2423\n",
      "33\t15623\t0.0\t3\t5191 7399 1360 1673\n",
      "34\t16172\t0.0\t3\t6866 6727 1133 1446\n",
      "35\t9925\t0.0\t3\t4796 3396 584 1149\n",
      "36\t7709\t0.0\t3\t4131 2556 408 614\n",
      "37\t4644\t0.0\t3\t2452 889 442 861\n",
      "38\t3273\t0.0\t3\t2192 582 155 344\n",
      "39\t2206\t0.0\t3\t1641 317 83 165\n",
      "40\t2123\t0.0\t3\t1715 220 70 118\n",
      "41\t1477\t0.0\t3\t1201 131 54 91\n",
      "42\t1557\t0.0\t3\t1126 124 75 232\n",
      "43\t972\t0.0\t3\t847 75 19 31\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1223699_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1223699_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 3790.74 s (82 us/read; 0.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              46,043,180\n",
      "Reads with adapters:                45,485,475 (98.8%)\n",
      "Reads that were too short:          10,624,591 (23.1%)\n",
      "Reads written (passing filters):    35,418,589 (76.9%)\n",
      "\n",
      "Total basepairs processed: 1,979,851,994 bp\n",
      "Total written (filtered):  1,050,884,558 bp (53.1%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 18472 times.\n",
      "755 times, it overlapped the 5' end of a read\n",
      "17717 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t264\t44964.0\t0\t264\n",
      "6\t383\t11241.0\t0\t383\n",
      "7\t9\t2810.3\t0\t9\n",
      "8\t2\t702.6\t0\t2\n",
      "9\t9\t175.6\t0\t0 9\n",
      "10\t23\t43.9\t1\t0 23\n",
      "11\t3\t11.0\t1\t0 3\n",
      "16\t1\t0.0\t1\t0 1\n",
      "17\t7\t0.0\t1\t7\n",
      "18\t1\t0.0\t1\t0 1\n",
      "20\t1\t0.0\t2\t1\n",
      "21\t3\t0.0\t2\t3\n",
      "22\t4\t0.0\t2\t0 4\n",
      "23\t7\t0.0\t2\t2 3 2\n",
      "24\t12\t0.0\t2\t0 7 5\n",
      "25\t11\t0.0\t2\t0 2 9\n",
      "26\t9\t0.0\t2\t0 1 8\n",
      "27\t5\t0.0\t2\t0 1 4\n",
      "30\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t4440\t44964.0\t0\t4440\n",
      "6\t2888\t11241.0\t0\t2888\n",
      "7\t1695\t2810.3\t0\t1695\n",
      "8\t1358\t702.6\t0\t1358\n",
      "9\t1209\t175.6\t0\t1201 8\n",
      "10\t2877\t43.9\t1\t662 2215\n",
      "11\t2262\t11.0\t1\t527 1735\n",
      "12\t499\t2.7\t1\t99 400\n",
      "13\t455\t0.7\t1\t100 355\n",
      "14\t6\t0.2\t1\t0 6\n",
      "15\t26\t0.0\t1\t0 26\n",
      "17\t1\t0.0\t1\t0 1\n",
      "35\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 2634 times.\n",
      "2634 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t403\t44964.0\t0\t403\n",
      "6\t67\t11241.0\t0\t67\n",
      "7\t2154\t2810.3\t0\t2154\n",
      "8\t8\t702.6\t0\t8\n",
      "9\t2\t175.6\t0\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 116 times.\n",
      "112 times, it overlapped the 5' end of a read\n",
      "4 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t47\t44964.0\t0\t47\n",
      "6\t18\t11241.0\t0\t18\n",
      "7\t23\t2810.3\t0\t23\n",
      "9\t3\t175.6\t0\t0 3\n",
      "10\t9\t43.9\t1\t0 9\n",
      "11\t1\t11.0\t1\t0 1\n",
      "13\t4\t0.7\t1\t0 4\n",
      "31\t1\t0.0\t3\t0 1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t4\t0.0\t3\t3 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "36\t1\t0.0\t3\t1\n",
      "38\t1\t0.0\t3\t1\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 2014 times.\n",
      "296 times, it overlapped the 5' end of a read\n",
      "1718 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t65\t44964.0\t0\t65\n",
      "6\t43\t11241.0\t0\t43\n",
      "10\t15\t43.9\t1\t0 15\n",
      "11\t5\t11.0\t1\t0 5\n",
      "12\t1\t2.7\t1\t0 1\n",
      "21\t2\t0.0\t2\t2\n",
      "23\t2\t0.0\t2\t1 1\n",
      "24\t4\t0.0\t2\t3 1\n",
      "25\t8\t0.0\t2\t7 1\n",
      "26\t24\t0.0\t2\t23 1\n",
      "27\t21\t0.0\t2\t20 1\n",
      "28\t18\t0.0\t2\t16 2\n",
      "29\t15\t0.0\t2\t12 2 0 1\n",
      "30\t13\t0.0\t3\t10 3\n",
      "31\t19\t0.0\t3\t14 3 1 1\n",
      "32\t29\t0.0\t3\t28 1\n",
      "33\t5\t0.0\t3\t5\n",
      "34\t7\t0.0\t3\t7\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.7\t1\t0 1\n",
      "14\t255\t0.2\t1\t76 179\n",
      "15\t270\t0.0\t1\t116 154\n",
      "16\t172\t0.0\t1\t66 106\n",
      "17\t163\t0.0\t1\t66 97\n",
      "18\t122\t0.0\t1\t59 62 1\n",
      "19\t71\t0.0\t1\t39 30 2\n",
      "20\t88\t0.0\t2\t24 34 30\n",
      "21\t68\t0.0\t2\t14 26 28\n",
      "22\t78\t0.0\t2\t27 24 27\n",
      "23\t45\t0.0\t2\t12 12 21\n",
      "24\t45\t0.0\t2\t6 13 26\n",
      "25\t27\t0.0\t2\t8 9 10\n",
      "26\t21\t0.0\t2\t10 5 6\n",
      "27\t10\t0.0\t2\t4 6\n",
      "28\t5\t0.0\t2\t0 4 1\n",
      "29\t7\t0.0\t2\t4 0 2 1\n",
      "30\t8\t0.0\t3\t2 2 3 1\n",
      "31\t7\t0.0\t3\t2 3 2\n",
      "32\t15\t0.0\t3\t9 1 3 2\n",
      "33\t16\t0.0\t3\t12 3 1\n",
      "34\t25\t0.0\t3\t19 4 1 1\n",
      "35\t39\t0.0\t3\t31 7 0 1\n",
      "36\t34\t0.0\t3\t32 2\n",
      "37\t34\t0.0\t3\t28 6\n",
      "38\t23\t0.0\t3\t21 1 1\n",
      "39\t25\t0.0\t3\t18 4 1 2\n",
      "40\t23\t0.0\t3\t17 4 0 2\n",
      "41\t10\t0.0\t3\t7 3\n",
      "42\t11\t0.0\t3\t9 0 2\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 139 times.\n",
      "139 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t68\t44964.0\t0\t68\n",
      "6\t54\t11241.0\t0\t54\n",
      "7\t1\t2810.3\t0\t1\n",
      "8\t1\t702.6\t0\t1\n",
      "9\t7\t175.6\t0\t0 7\n",
      "10\t3\t43.9\t1\t0 3\n",
      "11\t5\t11.0\t1\t0 5\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 45462100 times.\n",
      "66424 times, it overlapped the 5' end of a read\n",
      "45395676 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1376\t44964.0\t0\t1376\n",
      "6\t183\t11241.0\t0\t183\n",
      "7\t99\t2810.3\t0\t99\n",
      "8\t7\t702.6\t0\t7\n",
      "9\t8\t175.6\t0\t8\n",
      "10\t10\t43.9\t1\t4 6\n",
      "11\t17\t11.0\t1\t12 5\n",
      "12\t14\t2.7\t1\t7 7\n",
      "13\t14\t0.7\t1\t7 7\n",
      "14\t23\t0.2\t1\t5 18\n",
      "15\t15\t0.0\t1\t2 13\n",
      "16\t8\t0.0\t1\t3 5\n",
      "17\t2\t0.0\t1\t0 2\n",
      "18\t5\t0.0\t1\t2 3\n",
      "19\t3\t0.0\t1\t0 3\n",
      "20\t77\t0.0\t2\t8 5 64\n",
      "21\t144\t0.0\t2\t13 28 103\n",
      "22\t562\t0.0\t2\t19 79 464\n",
      "23\t675\t0.0\t2\t16 142 517\n",
      "24\t1548\t0.0\t2\t29 430 1089\n",
      "25\t1136\t0.0\t2\t36 331 769\n",
      "26\t505\t0.0\t2\t22 137 346\n",
      "27\t189\t0.0\t2\t47 46 96\n",
      "28\t205\t0.0\t2\t75 71 59\n",
      "29\t306\t0.0\t2\t113 107 86\n",
      "30\t59293\t0.0\t3\t57008 1611 393 281\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t50287\t44964.0\t0\t50287\n",
      "6\t104252\t11241.0\t0\t104252\n",
      "7\t310595\t2810.3\t0\t310595\n",
      "8\t811355\t702.6\t0\t811355\n",
      "9\t1364564\t175.6\t0\t1364564\n",
      "10\t1697227\t43.9\t1\t1625527 71700\n",
      "11\t2310210\t11.0\t1\t2150608 159602\n",
      "12\t3858594\t2.7\t1\t3350110 508484\n",
      "13\t5062947\t0.7\t1\t4641957 420990\n",
      "14\t5742778\t0.2\t1\t5295158 447620\n",
      "15\t4719283\t0.0\t1\t4070514 648769\n",
      "16\t5272720\t0.0\t1\t3876367 1396353\n",
      "17\t3532380\t0.0\t1\t2722311 810069\n",
      "18\t2351175\t0.0\t1\t1849146 502029\n",
      "19\t1718296\t0.0\t1\t1184255 534041\n",
      "20\t1450039\t0.0\t2\t478241 434136 537662\n",
      "21\t1575321\t0.0\t2\t256493 676148 642680\n",
      "22\t733699\t0.0\t2\t129112 239276 365311\n",
      "23\t693077\t0.0\t2\t57729 217765 417583\n",
      "24\t274759\t0.0\t2\t22779 79955 172025\n",
      "25\t376812\t0.0\t2\t28376 71923 276513\n",
      "26\t161531\t0.0\t2\t25957 59033 76541\n",
      "27\t111826\t0.0\t2\t31879 43541 36406\n",
      "28\t158581\t0.0\t2\t48852 67775 41954\n",
      "29\t145341\t0.0\t2\t53273 59262 32806\n",
      "30\t194256\t0.0\t3\t60151 60035 32257 41813\n",
      "31\t157198\t0.0\t3\t59062 50169 24000 23967\n",
      "32\t85655\t0.0\t3\t46507 23902 5683 9563\n",
      "33\t95037\t0.0\t3\t60141 22520 4813 7563\n",
      "34\t57951\t0.0\t3\t39997 11275 2413 4266\n",
      "35\t47247\t0.0\t3\t35548 7840 1588 2271\n",
      "36\t29610\t0.0\t3\t22453 3468 1268 2421\n",
      "37\t27061\t0.0\t3\t23000 2376 663 1022\n",
      "38\t20546\t0.0\t3\t18158 1392 401 595\n",
      "39\t27273\t0.0\t3\t24816 1423 414 620\n",
      "40\t21643\t0.0\t3\t19771 984 337 551\n",
      "41\t25073\t0.0\t3\t21760 1142 787 1384\n",
      "42\t19477\t0.0\t3\t18597 569 182 129\n",
      "\n",
      "get trimmed reads of third group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288391_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288391_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1110.45 s (62 us/read; 0.96 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,778,707\n",
      "Reads with adapters:                 6,952,729 (39.1%)\n",
      "Reads that were too short:           6,952,729 (39.1%)\n",
      "Reads written (passing filters):    10,825,978 (60.9%)\n",
      "\n",
      "Total basepairs processed:   515,547,570 bp\n",
      "Total written (filtered):    313,931,327 bp (60.9%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 19641 times.\n",
      "10463 times, it overlapped the 5' end of a read\n",
      "9178 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t3097\t17362.0\t0\t3097\n",
      "6\t2984\t4340.5\t0\t2984\n",
      "7\t52\t1085.1\t0\t52\n",
      "8\t91\t271.3\t0\t91\n",
      "9\t61\t67.8\t0\t12 49\n",
      "10\t205\t17.0\t1\t2 203\n",
      "11\t127\t4.2\t1\t2 125\n",
      "12\t28\t1.1\t1\t1 27\n",
      "15\t43\t0.0\t1\t37 6\n",
      "16\t83\t0.0\t1\t71 12\n",
      "17\t117\t0.0\t1\t35 82\n",
      "18\t289\t0.0\t1\t106 183\n",
      "19\t1868\t0.0\t1\t1726 138 4\n",
      "20\t245\t0.0\t2\t72 99 74\n",
      "21\t471\t0.0\t2\t160 81 230\n",
      "22\t162\t0.0\t2\t12 101 49\n",
      "23\t121\t0.0\t2\t20 30 71\n",
      "24\t291\t0.0\t2\t5 80 206\n",
      "25\t71\t0.0\t2\t0 12 59\n",
      "26\t35\t0.0\t2\t0 4 31\n",
      "27\t12\t0.0\t2\t0 2 10\n",
      "28\t2\t0.0\t2\t0 1 1\n",
      "29\t8\t0.0\t2\t0 0 5 3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1726\t17362.0\t0\t1726\n",
      "6\t1043\t4340.5\t0\t1043\n",
      "7\t157\t1085.1\t0\t157\n",
      "8\t78\t271.3\t0\t78\n",
      "9\t93\t67.8\t0\t64 29\n",
      "10\t136\t17.0\t1\t1 135\n",
      "11\t77\t4.2\t1\t1 76\n",
      "12\t7\t1.1\t1\t0 7\n",
      "13\t2\t0.3\t1\t0 2\n",
      "18\t2\t0.0\t1\t0 2\n",
      "19\t6\t0.0\t1\t5 1\n",
      "20\t5774\t0.0\t2\t5388 340 46\n",
      "21\t68\t0.0\t2\t49 16 3\n",
      "22\t6\t0.0\t2\t2 2 2\n",
      "23\t1\t0.0\t2\t0 1\n",
      "24\t1\t0.0\t2\t1\n",
      "25\t1\t0.0\t2\t1\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 3283 times.\n",
      "3283 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1894\t17362.0\t0\t1894\n",
      "6\t538\t4340.5\t0\t538\n",
      "7\t497\t1085.1\t0\t497\n",
      "8\t63\t271.3\t0\t63\n",
      "9\t167\t67.8\t0\t6 161\n",
      "10\t98\t17.0\t1\t1 97\n",
      "11\t23\t4.2\t1\t0 23\n",
      "12\t2\t1.1\t1\t0 2\n",
      "13\t1\t0.3\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 1143 times.\n",
      "1143 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t859\t17362.0\t0\t859\n",
      "6\t136\t4340.5\t0\t136\n",
      "7\t35\t1085.1\t0\t35\n",
      "8\t4\t271.3\t0\t4\n",
      "9\t52\t67.8\t0\t4 48\n",
      "10\t37\t17.0\t1\t0 37\n",
      "11\t13\t4.2\t1\t0 13\n",
      "12\t7\t1.1\t1\t0 7\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 11148 times.\n",
      "11146 times, it overlapped the 5' end of a read\n",
      "2 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t9686\t17362.0\t0\t9686\n",
      "6\t403\t4340.5\t0\t403\n",
      "7\t235\t1085.1\t0\t235\n",
      "8\t69\t271.3\t0\t69\n",
      "9\t227\t67.8\t0\t198 29\n",
      "10\t137\t17.0\t1\t0 137\n",
      "11\t352\t4.2\t1\t0 352\n",
      "12\t37\t1.1\t1\t0 37\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.3\t1\t0 1\n",
      "20\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 4402 times.\n",
      "4402 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2885\t17362.0\t0\t2885\n",
      "6\t392\t4340.5\t0\t392\n",
      "7\t82\t1085.1\t0\t82\n",
      "8\t14\t271.3\t0\t14\n",
      "9\t644\t67.8\t0\t8 636\n",
      "10\t288\t17.0\t1\t4 284\n",
      "11\t82\t4.2\t1\t0 82\n",
      "12\t11\t1.1\t1\t0 11\n",
      "13\t3\t0.3\t1\t0 3\n",
      "14\t1\t0.1\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 6913112 times.\n",
      "617516 times, it overlapped the 5' end of a read\n",
      "6295596 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t72417\t17362.0\t0\t72417\n",
      "6\t36924\t4340.5\t0\t36924\n",
      "7\t35918\t1085.1\t0\t35918\n",
      "8\t28483\t271.3\t0\t28483\n",
      "9\t11961\t67.8\t0\t11961\n",
      "10\t5692\t17.0\t1\t3204 2488\n",
      "11\t2438\t4.2\t1\t1214 1224\n",
      "12\t1222\t1.1\t1\t566 656\n",
      "13\t645\t0.3\t1\t249 396\n",
      "14\t407\t0.1\t1\t175 232\n",
      "15\t243\t0.0\t1\t141 102\n",
      "16\t256\t0.0\t1\t154 102\n",
      "17\t499\t0.0\t1\t194 305\n",
      "18\t1585\t0.0\t1\t301 1284\n",
      "19\t5462\t0.0\t1\t488 4974\n",
      "20\t34419\t0.0\t2\t472 7978 25969\n",
      "21\t60754\t0.0\t2\t155 15421 45178\n",
      "22\t116603\t0.0\t2\t64 31108 85431\n",
      "23\t53535\t0.0\t2\t36 16088 37411\n",
      "24\t25358\t0.0\t2\t29 7643 17686\n",
      "25\t13593\t0.0\t2\t22 3965 9606\n",
      "26\t6285\t0.0\t2\t19 1661 4605\n",
      "27\t3555\t0.0\t2\t12 713 2830\n",
      "28\t1632\t0.0\t2\t86 635 911\n",
      "29\t97630\t0.0\t2\t20716 23728 53186\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1447119\t17362.0\t0\t1447119\n",
      "6\t1268437\t4340.5\t0\t1268437\n",
      "7\t718530\t1085.1\t0\t718530\n",
      "8\t342962\t271.3\t0\t342962\n",
      "9\t185498\t67.8\t0\t185498\n",
      "10\t411284\t17.0\t1\t117673 293611\n",
      "11\t298898\t4.2\t1\t69655 229243\n",
      "12\t152911\t1.1\t1\t30155 122756\n",
      "13\t92330\t0.3\t1\t13209 79121\n",
      "14\t89860\t0.1\t1\t7745 82115\n",
      "15\t75036\t0.0\t1\t6387 68649\n",
      "16\t58946\t0.0\t1\t6345 52601\n",
      "17\t44258\t0.0\t1\t8016 36242\n",
      "18\t28632\t0.0\t1\t8004 20628\n",
      "19\t39601\t0.0\t1\t12991 26610\n",
      "20\t131535\t0.0\t2\t10951 21248 99336\n",
      "21\t116222\t0.0\t2\t10906 19011 86305\n",
      "22\t125363\t0.0\t2\t17900 24836 82627\n",
      "23\t137886\t0.0\t2\t22471 23839 91576\n",
      "24\t82967\t0.0\t2\t19821 18509 44637\n",
      "25\t136487\t0.0\t2\t32275 21280 82932\n",
      "26\t110242\t0.0\t2\t32313 21359 56570\n",
      "27\t94080\t0.0\t2\t33602 17457 43021\n",
      "28\t106512\t0.0\t2\t21777 35489 49246\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288392_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288392_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 306.90 s (77 us/read; 0.78 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                 3,727,112 (93.2%)\n",
      "Reads that were too short:           1,892,679 (47.3%)\n",
      "Reads written (passing filters):     2,107,321 (52.7%)\n",
      "\n",
      "Total basepairs processed:   180,000,000 bp\n",
      "Total written (filtered):     65,207,221 bp (36.2%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 138505 times.\n",
      "241 times, it overlapped the 5' end of a read\n",
      "138264 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t171\t3906.2\t0\t171\n",
      "6\t50\t976.6\t0\t50\n",
      "7\t1\t244.1\t0\t1\n",
      "9\t3\t15.3\t0\t0 3\n",
      "17\t3\t0.0\t1\t1 2\n",
      "18\t1\t0.0\t1\t0 1\n",
      "20\t3\t0.0\t2\t2 0 1\n",
      "22\t1\t0.0\t2\t0 1\n",
      "23\t3\t0.0\t2\t3\n",
      "24\t1\t0.0\t2\t0 0 1\n",
      "25\t2\t0.0\t2\t0 1 1\n",
      "28\t1\t0.0\t2\t0 0 1\n",
      "30\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t51896\t3906.2\t0\t51896\n",
      "6\t33877\t976.6\t0\t33877\n",
      "7\t19030\t244.1\t0\t19030\n",
      "8\t10126\t61.0\t0\t10126\n",
      "9\t5715\t15.3\t0\t5682 33\n",
      "10\t7323\t3.8\t1\t3549 3774\n",
      "11\t4652\t1.0\t1\t2289 2363\n",
      "12\t3429\t0.2\t1\t1710 1719\n",
      "13\t2176\t0.1\t1\t1045 1131\n",
      "14\t27\t0.0\t1\t0 27\n",
      "15\t13\t0.0\t1\t0 13\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 1532 times.\n",
      "1532 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t120\t3906.2\t0\t120\n",
      "6\t40\t976.6\t0\t40\n",
      "7\t1370\t244.1\t0\t1370\n",
      "9\t1\t15.3\t0\t0 1\n",
      "10\t1\t3.8\t1\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 38 times.\n",
      "38 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t36\t3906.2\t0\t36\n",
      "6\t2\t976.6\t0\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 4149 times.\n",
      "60 times, it overlapped the 5' end of a read\n",
      "4089 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t6\t3906.2\t0\t6\n",
      "6\t2\t976.6\t0\t2\n",
      "10\t1\t3.8\t1\t0 1\n",
      "11\t3\t1.0\t1\t0 3\n",
      "12\t1\t0.2\t1\t0 1\n",
      "25\t1\t0.0\t2\t1\n",
      "26\t1\t0.0\t2\t1\n",
      "28\t1\t0.0\t2\t1\n",
      "29\t6\t0.0\t2\t6\n",
      "30\t4\t0.0\t3\t4\n",
      "31\t10\t0.0\t3\t9 1\n",
      "32\t18\t0.0\t3\t17 1\n",
      "33\t4\t0.0\t3\t1 3\n",
      "34\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t32\t0.1\t1\t0 32\n",
      "14\t1376\t0.0\t1\t556 820\n",
      "15\t875\t0.0\t1\t372 503\n",
      "16\t419\t0.0\t1\t168 251\n",
      "17\t259\t0.0\t1\t118 141\n",
      "18\t90\t0.0\t1\t46 44\n",
      "19\t57\t0.0\t1\t21 30 6\n",
      "20\t58\t0.0\t2\t16 21 21\n",
      "21\t61\t0.0\t2\t17 18 26\n",
      "22\t61\t0.0\t2\t11 29 21\n",
      "23\t115\t0.0\t2\t38 43 34\n",
      "24\t431\t0.0\t2\t108 172 151\n",
      "25\t94\t0.0\t2\t40 25 29\n",
      "26\t58\t0.0\t2\t22 22 14\n",
      "27\t21\t0.0\t2\t7 8 6\n",
      "28\t11\t0.0\t2\t7 3 1\n",
      "29\t3\t0.0\t2\t1 2\n",
      "30\t4\t0.0\t3\t2 1 0 1\n",
      "31\t2\t0.0\t3\t0 2\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t3\t0.0\t3\t2 0 0 1\n",
      "34\t2\t0.0\t3\t2\n",
      "35\t5\t0.0\t3\t4 1\n",
      "36\t5\t0.0\t3\t5\n",
      "37\t10\t0.0\t3\t8 2\n",
      "38\t9\t0.0\t3\t9\n",
      "39\t9\t0.0\t3\t9\n",
      "40\t6\t0.0\t3\t5 1\n",
      "41\t2\t0.0\t3\t2\n",
      "42\t2\t0.0\t3\t2\n",
      "43\t4\t0.0\t3\t4\n",
      "44\t4\t0.0\t3\t4\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 129 times.\n",
      "129 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t37\t3906.2\t0\t37\n",
      "6\t86\t976.6\t0\t86\n",
      "8\t1\t61.0\t0\t1\n",
      "9\t2\t15.3\t0\t0 2\n",
      "10\t2\t3.8\t1\t0 2\n",
      "11\t1\t1.0\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 3582759 times.\n",
      "2663 times, it overlapped the 5' end of a read\n",
      "3580096 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t418\t3906.2\t0\t418\n",
      "6\t48\t976.6\t0\t48\n",
      "7\t23\t244.1\t0\t23\n",
      "8\t1\t61.0\t0\t1\n",
      "10\t3\t3.8\t1\t0 3\n",
      "11\t1\t1.0\t1\t0 1\n",
      "12\t1\t0.2\t1\t0 1\n",
      "20\t13\t0.0\t2\t0 1 12\n",
      "21\t23\t0.0\t2\t0 4 19\n",
      "22\t155\t0.0\t2\t0 19 136\n",
      "23\t184\t0.0\t2\t1 43 140\n",
      "24\t748\t0.0\t2\t0 237 511\n",
      "25\t644\t0.0\t2\t0 174 470\n",
      "26\t209\t0.0\t2\t1 70 138\n",
      "27\t80\t0.0\t2\t0 28 52\n",
      "28\t34\t0.0\t2\t0 14 20\n",
      "29\t23\t0.0\t2\t1 8 14\n",
      "30\t55\t0.0\t3\t30 5 8 12\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t543\t3906.2\t0\t543\n",
      "6\t172\t976.6\t0\t172\n",
      "7\t171\t244.1\t0\t171\n",
      "8\t293\t61.0\t0\t293\n",
      "9\t538\t15.3\t0\t538\n",
      "10\t1570\t3.8\t1\t1452 118\n",
      "11\t4104\t1.0\t1\t3993 111\n",
      "12\t9856\t0.2\t1\t9605 251\n",
      "13\t29624\t0.1\t1\t29149 475\n",
      "14\t93873\t0.0\t1\t91967 1906\n",
      "15\t209611\t0.0\t1\t204247 5364\n",
      "16\t260163\t0.0\t1\t249531 10632\n",
      "17\t337540\t0.0\t1\t308387 29153\n",
      "18\t406857\t0.0\t1\t326111 80746\n",
      "19\t335710\t0.0\t1\t269184 66526\n",
      "20\t413834\t0.0\t2\t248853 99681 65300\n",
      "21\t450174\t0.0\t2\t187136 154535 108503\n",
      "22\t386329\t0.0\t2\t109661 143518 133150\n",
      "23\t304916\t0.0\t2\t35019 105324 164573\n",
      "24\t179518\t0.0\t2\t6810 46409 126299\n",
      "25\t84685\t0.0\t2\t1321 14567 68797\n",
      "26\t32823\t0.0\t2\t424 3819 28580\n",
      "27\t11725\t0.0\t2\t368 1163 10194\n",
      "28\t4447\t0.0\t2\t370 850 3227\n",
      "29\t2664\t0.0\t2\t422 660 1582\n",
      "30\t5914\t0.0\t3\t413 537 1024 3940\n",
      "31\t3529\t0.0\t3\t317 386 609 2217\n",
      "32\t3013\t0.0\t3\t219 278 421 2095\n",
      "33\t2341\t0.0\t3\t230 253 334 1524\n",
      "34\t1245\t0.0\t3\t116 180 231 718\n",
      "35\t874\t0.0\t3\t77 74 96 627\n",
      "36\t449\t0.0\t3\t50 46 65 288\n",
      "37\t158\t0.0\t3\t25 22 30 81\n",
      "38\t173\t0.0\t3\t29 16 21 107\n",
      "39\t82\t0.0\t3\t18 8 8 48\n",
      "40\t79\t0.0\t3\t22 11 10 36\n",
      "41\t108\t0.0\t3\t41 8 11 48\n",
      "42\t75\t0.0\t3\t17 12 9 37\n",
      "43\t285\t0.0\t3\t30 22 55 178\n",
      "44\t31\t0.0\t3\t13 6 6 6\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288393_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288393_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 304.52 s (76 us/read; 0.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                 3,725,135 (93.1%)\n",
      "Reads that were too short:           1,870,480 (46.8%)\n",
      "Reads written (passing filters):     2,129,520 (53.2%)\n",
      "\n",
      "Total basepairs processed:   179,998,947 bp\n",
      "Total written (filtered):     66,093,571 bp (36.7%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 160952 times.\n",
      "232 times, it overlapped the 5' end of a read\n",
      "160720 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t177\t3906.2\t0\t177\n",
      "6\t37\t976.6\t0\t37\n",
      "17\t5\t0.0\t1\t4 1\n",
      "23\t5\t0.0\t2\t4 1\n",
      "24\t3\t0.0\t2\t0 2 1\n",
      "25\t5\t0.0\t2\t0 1 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t61301\t3906.2\t0\t61301\n",
      "6\t39868\t976.6\t0\t39868\n",
      "7\t21976\t244.1\t0\t21976\n",
      "8\t11485\t61.0\t0\t11485\n",
      "9\t6435\t15.3\t0\t6409 26\n",
      "10\t8068\t3.8\t1\t4343 3725\n",
      "11\t5180\t1.0\t1\t2706 2474\n",
      "12\t3830\t0.2\t1\t2029 1801\n",
      "13\t2542\t0.1\t1\t1274 1268\n",
      "14\t24\t0.0\t1\t0 24\n",
      "15\t11\t0.0\t1\t0 11\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 1513 times.\n",
      "1513 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t134\t3906.2\t0\t134\n",
      "6\t38\t976.6\t0\t38\n",
      "7\t1339\t244.1\t0\t1339\n",
      "8\t1\t61.0\t0\t1\n",
      "11\t1\t1.0\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 54 times.\n",
      "54 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t48\t3906.2\t0\t48\n",
      "6\t1\t976.6\t0\t1\n",
      "7\t2\t244.1\t0\t2\n",
      "10\t2\t3.8\t1\t0 2\n",
      "12\t1\t0.2\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 4967 times.\n",
      "74 times, it overlapped the 5' end of a read\n",
      "4893 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t12\t3906.2\t0\t12\n",
      "6\t4\t976.6\t0\t4\n",
      "7\t1\t244.1\t0\t1\n",
      "11\t1\t1.0\t1\t0 1\n",
      "12\t2\t0.2\t1\t0 2\n",
      "26\t2\t0.0\t2\t2\n",
      "27\t2\t0.0\t2\t2\n",
      "28\t1\t0.0\t2\t0 1\n",
      "29\t8\t0.0\t2\t7 1\n",
      "30\t2\t0.0\t3\t1 1\n",
      "31\t14\t0.0\t3\t14\n",
      "32\t23\t0.0\t3\t22 1\n",
      "33\t1\t0.0\t3\t0 1\n",
      "34\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t37\t0.1\t1\t0 37\n",
      "14\t1644\t0.0\t1\t707 937\n",
      "15\t1023\t0.0\t1\t467 556\n",
      "16\t530\t0.0\t1\t244 286\n",
      "17\t286\t0.0\t1\t145 141\n",
      "18\t124\t0.0\t1\t69 55\n",
      "19\t55\t0.0\t1\t21 31 3\n",
      "20\t64\t0.0\t2\t18 19 27\n",
      "21\t67\t0.0\t2\t17 30 20\n",
      "22\t86\t0.0\t2\t20 38 28\n",
      "23\t142\t0.0\t2\t56 45 41\n",
      "24\t513\t0.0\t2\t144 210 159\n",
      "25\t110\t0.0\t2\t46 32 32\n",
      "26\t91\t0.0\t2\t28 31 32\n",
      "27\t33\t0.0\t2\t12 16 5\n",
      "28\t17\t0.0\t2\t8 7 2\n",
      "29\t3\t0.0\t2\t2 1\n",
      "30\t1\t0.0\t3\t0 1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t5\t0.0\t3\t5\n",
      "33\t5\t0.0\t3\t3 2\n",
      "34\t5\t0.0\t3\t5\n",
      "35\t7\t0.0\t3\t7\n",
      "36\t8\t0.0\t3\t8\n",
      "37\t5\t0.0\t3\t5\n",
      "38\t9\t0.0\t3\t8 1\n",
      "39\t5\t0.0\t3\t4 1\n",
      "40\t4\t0.0\t3\t4\n",
      "41\t6\t0.0\t3\t6\n",
      "43\t1\t0.0\t3\t1\n",
      "44\t6\t0.0\t3\t6\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 166 times.\n",
      "166 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t56\t3906.2\t0\t56\n",
      "6\t104\t976.6\t0\t104\n",
      "9\t3\t15.3\t0\t0 3\n",
      "10\t1\t3.8\t1\t0 1\n",
      "11\t2\t1.0\t1\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 3557483 times.\n",
      "2766 times, it overlapped the 5' end of a read\n",
      "3554717 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t449\t3906.2\t0\t449\n",
      "6\t65\t976.6\t0\t65\n",
      "7\t11\t244.1\t0\t11\n",
      "8\t2\t61.0\t0\t2\n",
      "10\t4\t3.8\t1\t0 4\n",
      "20\t10\t0.0\t2\t0 1 9\n",
      "21\t26\t0.0\t2\t1 0 25\n",
      "22\t162\t0.0\t2\t0 22 140\n",
      "23\t181\t0.0\t2\t1 47 133\n",
      "24\t710\t0.0\t2\t4 234 472\n",
      "25\t707\t0.0\t2\t2 214 491\n",
      "26\t236\t0.0\t2\t0 77 159\n",
      "27\t65\t0.0\t2\t0 18 47\n",
      "28\t47\t0.0\t2\t0 21 26\n",
      "29\t17\t0.0\t2\t1 11 5\n",
      "30\t74\t0.0\t3\t33 10 14 17\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t422\t3906.2\t0\t422\n",
      "6\t172\t976.6\t0\t172\n",
      "7\t176\t244.1\t0\t176\n",
      "8\t272\t61.0\t0\t272\n",
      "9\t491\t15.3\t0\t491\n",
      "10\t1464\t3.8\t1\t1364 100\n",
      "11\t4155\t1.0\t1\t4030 125\n",
      "12\t9719\t0.2\t1\t9464 255\n",
      "13\t29355\t0.1\t1\t28891 464\n",
      "14\t93671\t0.0\t1\t91770 1901\n",
      "15\t208920\t0.0\t1\t203506 5414\n",
      "16\t259482\t0.0\t1\t248852 10630\n",
      "17\t336105\t0.0\t1\t307104 29001\n",
      "18\t406737\t0.0\t1\t324753 81984\n",
      "19\t336674\t0.0\t1\t267725 68949\n",
      "20\t414966\t0.0\t2\t242554 106765 65647\n",
      "21\t452444\t0.0\t2\t175037 167118 110289\n",
      "22\t389322\t0.0\t2\t100512 150249 138561\n",
      "23\t299505\t0.0\t2\t31573 99705 168227\n",
      "24\t170642\t0.0\t2\t5952 42329 122361\n",
      "25\t77990\t0.0\t2\t1092 12877 64021\n",
      "26\t29767\t0.0\t2\t344 3382 26041\n",
      "27\t10313\t0.0\t2\t268 970 9075\n",
      "28\t3780\t0.0\t2\t278 636 2866\n",
      "29\t2162\t0.0\t2\t247 544 1371\n",
      "30\t4953\t0.0\t3\t265 441 887 3360\n",
      "31\t3088\t0.0\t3\t215 298 561 2014\n",
      "32\t2798\t0.0\t3\t192 233 378 1995\n",
      "33\t2060\t0.0\t3\t159 217 287 1397\n",
      "34\t981\t0.0\t3\t106 123 177 575\n",
      "35\t715\t0.0\t3\t60 59 78 518\n",
      "36\t397\t0.0\t3\t38 44 59 256\n",
      "37\t197\t0.0\t3\t32 28 29 108\n",
      "38\t176\t0.0\t3\t21 11 19 125\n",
      "39\t112\t0.0\t3\t28 19 7 58\n",
      "40\t77\t0.0\t3\t28 9 10 30\n",
      "41\t103\t0.0\t3\t41 18 8 36\n",
      "42\t80\t0.0\t3\t15 9 4 52\n",
      "43\t243\t0.0\t3\t31 25 45 142\n",
      "44\t31\t0.0\t3\t9 7 4 11\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288394_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288394_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 305.74 s (76 us/read; 0.78 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                 3,702,224 (92.6%)\n",
      "Reads that were too short:           1,834,517 (45.9%)\n",
      "Reads written (passing filters):     2,165,483 (54.1%)\n",
      "\n",
      "Total basepairs processed:   179,998,517 bp\n",
      "Total written (filtered):     67,675,045 bp (37.6%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 178097 times.\n",
      "256 times, it overlapped the 5' end of a read\n",
      "177841 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t161\t3906.2\t0\t161\n",
      "6\t65\t976.6\t0\t65\n",
      "7\t2\t244.1\t0\t2\n",
      "11\t1\t1.0\t1\t0 1\n",
      "16\t2\t0.0\t1\t1 1\n",
      "17\t4\t0.0\t1\t4\n",
      "18\t1\t0.0\t1\t0 1\n",
      "20\t1\t0.0\t2\t1\n",
      "22\t2\t0.0\t2\t0 2\n",
      "23\t4\t0.0\t2\t1 2 1\n",
      "24\t3\t0.0\t2\t0 2 1\n",
      "25\t4\t0.0\t2\t0 1 3\n",
      "26\t4\t0.0\t2\t0 0 4\n",
      "27\t1\t0.0\t2\t0 0 1\n",
      "29\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t69687\t3906.2\t0\t69687\n",
      "6\t44044\t976.6\t0\t44044\n",
      "7\t24003\t244.1\t0\t24003\n",
      "8\t12544\t61.0\t0\t12544\n",
      "9\t7132\t15.3\t0\t7088 44\n",
      "10\t8598\t3.8\t1\t4671 3927\n",
      "11\t5439\t1.0\t1\t2803 2636\n",
      "12\t3786\t0.2\t1\t2014 1772\n",
      "13\t2569\t0.1\t1\t1322 1247\n",
      "14\t20\t0.0\t1\t0 20\n",
      "15\t19\t0.0\t1\t0 19\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 1663 times.\n",
      "1662 times, it overlapped the 5' end of a read\n",
      "1 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t147\t3906.2\t0\t147\n",
      "6\t41\t976.6\t0\t41\n",
      "7\t1473\t244.1\t0\t1473\n",
      "8\t1\t61.0\t0\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.1\t1\t0 1\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 45 times.\n",
      "45 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t44\t3906.2\t0\t44\n",
      "6\t1\t976.6\t0\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 4801 times.\n",
      "59 times, it overlapped the 5' end of a read\n",
      "4742 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t9\t3906.2\t0\t9\n",
      "6\t2\t976.6\t0\t2\n",
      "10\t1\t3.8\t1\t0 1\n",
      "11\t2\t1.0\t1\t0 2\n",
      "23\t1\t0.0\t2\t1\n",
      "25\t1\t0.0\t2\t1\n",
      "27\t3\t0.0\t2\t2 1\n",
      "28\t6\t0.0\t2\t5 0 0 1\n",
      "29\t3\t0.0\t2\t3\n",
      "30\t2\t0.0\t3\t1 0 1\n",
      "31\t4\t0.0\t3\t4\n",
      "32\t20\t0.0\t3\t19 1\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t3\t0.0\t3\t3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t44\t0.1\t1\t0 44\n",
      "14\t1691\t0.0\t1\t792 899\n",
      "15\t1019\t0.0\t1\t459 560\n",
      "16\t472\t0.0\t1\t195 277\n",
      "17\t231\t0.0\t1\t94 137\n",
      "18\t117\t0.0\t1\t50 67\n",
      "19\t68\t0.0\t1\t32 34 2\n",
      "20\t53\t0.0\t2\t15 17 21\n",
      "21\t49\t0.0\t2\t12 24 13\n",
      "22\t82\t0.0\t2\t23 27 32\n",
      "23\t151\t0.0\t2\t48 61 42\n",
      "24\t515\t0.0\t2\t144 229 142\n",
      "25\t88\t0.0\t2\t31 34 23\n",
      "26\t72\t0.0\t2\t31 23 18\n",
      "27\t24\t0.0\t2\t13 7 4\n",
      "28\t10\t0.0\t2\t7 2 1\n",
      "29\t2\t0.0\t2\t1 1\n",
      "31\t2\t0.0\t3\t2\n",
      "34\t4\t0.0\t3\t4\n",
      "35\t8\t0.0\t3\t7 1\n",
      "36\t10\t0.0\t3\t10\n",
      "37\t10\t0.0\t3\t9 1\n",
      "38\t5\t0.0\t3\t5\n",
      "39\t3\t0.0\t3\t3\n",
      "40\t4\t0.0\t3\t4\n",
      "41\t4\t0.0\t3\t4\n",
      "42\t2\t0.0\t3\t2\n",
      "43\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 159 times.\n",
      "159 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t49\t3906.2\t0\t49\n",
      "6\t106\t976.6\t0\t106\n",
      "9\t1\t15.3\t0\t0 1\n",
      "10\t2\t3.8\t1\t1 1\n",
      "11\t1\t1.0\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 3517459 times.\n",
      "2683 times, it overlapped the 5' end of a read\n",
      "3514776 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t420\t3906.2\t0\t420\n",
      "6\t60\t976.6\t0\t60\n",
      "7\t27\t244.1\t0\t27\n",
      "8\t3\t61.0\t0\t3\n",
      "9\t1\t15.3\t0\t1\n",
      "10\t1\t3.8\t1\t0 1\n",
      "11\t2\t1.0\t1\t0 2\n",
      "19\t2\t0.0\t1\t0 2\n",
      "20\t21\t0.0\t2\t0 2 19\n",
      "21\t27\t0.0\t2\t1 3 23\n",
      "22\t166\t0.0\t2\t2 21 143\n",
      "23\t199\t0.0\t2\t2 44 153\n",
      "24\t652\t0.0\t2\t1 226 425\n",
      "25\t657\t0.0\t2\t1 187 469\n",
      "26\t224\t0.0\t2\t0 63 161\n",
      "27\t88\t0.0\t2\t2 17 69\n",
      "28\t47\t0.0\t2\t0 16 31\n",
      "29\t29\t0.0\t2\t1 8 20\n",
      "30\t57\t0.0\t3\t21 6 15 15\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t379\t3906.2\t0\t379\n",
      "6\t157\t976.6\t0\t157\n",
      "7\t163\t244.1\t0\t163\n",
      "8\t299\t61.0\t0\t299\n",
      "9\t482\t15.3\t0\t482\n",
      "10\t1561\t3.8\t1\t1444 117\n",
      "11\t4010\t1.0\t1\t3872 138\n",
      "12\t9776\t0.2\t1\t9540 236\n",
      "13\t29098\t0.1\t1\t28625 473\n",
      "14\t93271\t0.0\t1\t91364 1907\n",
      "15\t208025\t0.0\t1\t202465 5560\n",
      "16\t258799\t0.0\t1\t247989 10810\n",
      "17\t336190\t0.0\t1\t306532 29658\n",
      "18\t404984\t0.0\t1\t322625 82359\n",
      "19\t336520\t0.0\t1\t262798 73722\n",
      "20\t418103\t0.0\t2\t230476 119811 67816\n",
      "21\t456930\t0.0\t2\t150416 190183 116331\n",
      "22\t390215\t0.0\t2\t81169 159273 149773\n",
      "23\t290639\t0.0\t2\t24775 88814 177050\n",
      "24\t154246\t0.0\t2\t4854 35233 114159\n",
      "25\t67313\t0.0\t2\t932 10727 55654\n",
      "26\t25415\t0.0\t2\t291 2859 22265\n",
      "27\t8751\t0.0\t2\t241 864 7646\n",
      "28\t3394\t0.0\t2\t268 637 2489\n",
      "29\t1915\t0.0\t2\t285 477 1153\n",
      "30\t4362\t0.0\t3\t253 406 771 2932\n",
      "31\t2616\t0.0\t3\t217 277 460 1662\n",
      "32\t2300\t0.0\t3\t171 227 337 1565\n",
      "33\t1815\t0.0\t3\t164 183 280 1188\n",
      "34\t1001\t0.0\t3\t104 149 195 553\n",
      "35\t719\t0.0\t3\t69 66 94 490\n",
      "36\t377\t0.0\t3\t45 41 52 239\n",
      "37\t169\t0.0\t3\t28 22 29 90\n",
      "38\t169\t0.0\t3\t23 23 20 103\n",
      "39\t83\t0.0\t3\t34 11 4 34\n",
      "40\t66\t0.0\t3\t18 10 7 31\n",
      "41\t97\t0.0\t3\t44 17 17 19\n",
      "42\t81\t0.0\t3\t24 6 16 35\n",
      "43\t254\t0.0\t3\t28 29 46 151\n",
      "44\t32\t0.0\t3\t14 4 5 9\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288395_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288395_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 310.36 s (78 us/read; 0.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                 3,706,309 (92.7%)\n",
      "Reads that were too short:           1,834,730 (45.9%)\n",
      "Reads written (passing filters):     2,165,270 (54.1%)\n",
      "\n",
      "Total basepairs processed:   179,998,753 bp\n",
      "Total written (filtered):     67,576,642 bp (37.5%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 175933 times.\n",
      "220 times, it overlapped the 5' end of a read\n",
      "175713 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t134\t3906.2\t0\t134\n",
      "6\t71\t976.6\t0\t71\n",
      "7\t1\t244.1\t0\t1\n",
      "17\t2\t0.0\t1\t1 1\n",
      "20\t2\t0.0\t2\t2\n",
      "21\t1\t0.0\t2\t1\n",
      "22\t2\t0.0\t2\t1 1\n",
      "23\t4\t0.0\t2\t4\n",
      "25\t2\t0.0\t2\t0 0 2\n",
      "26\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t68728\t3906.2\t0\t68728\n",
      "6\t42989\t976.6\t0\t42989\n",
      "7\t23550\t244.1\t0\t23550\n",
      "8\t12139\t61.0\t0\t12139\n",
      "9\t7241\t15.3\t0\t7191 50\n",
      "10\t8927\t3.8\t1\t4740 4187\n",
      "11\t5536\t1.0\t1\t2903 2633\n",
      "12\t3965\t0.2\t1\t1998 1967\n",
      "13\t2601\t0.1\t1\t1303 1298\n",
      "14\t19\t0.0\t1\t0 19\n",
      "15\t18\t0.0\t1\t0 18\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 1666 times.\n",
      "1666 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t146\t3906.2\t0\t146\n",
      "6\t56\t976.6\t0\t56\n",
      "7\t1462\t244.1\t0\t1462\n",
      "8\t2\t61.0\t0\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 48 times.\n",
      "48 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t47\t3906.2\t0\t47\n",
      "10\t1\t3.8\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 5064 times.\n",
      "46 times, it overlapped the 5' end of a read\n",
      "5018 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t9\t3906.2\t0\t9\n",
      "6\t2\t976.6\t0\t2\n",
      "9\t1\t15.3\t0\t0 1\n",
      "10\t3\t3.8\t1\t0 3\n",
      "11\t1\t1.0\t1\t0 1\n",
      "24\t1\t0.0\t2\t1\n",
      "25\t1\t0.0\t2\t1\n",
      "26\t1\t0.0\t2\t1\n",
      "27\t1\t0.0\t2\t1\n",
      "28\t1\t0.0\t2\t1\n",
      "29\t5\t0.0\t2\t5\n",
      "31\t4\t0.0\t3\t3 1\n",
      "32\t13\t0.0\t3\t12 0 1\n",
      "33\t1\t0.0\t3\t0 0 0 1\n",
      "34\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t39\t0.1\t1\t0 39\n",
      "14\t1705\t0.0\t1\t740 965\n",
      "15\t1058\t0.0\t1\t518 540\n",
      "16\t520\t0.0\t1\t212 308\n",
      "17\t269\t0.0\t1\t126 143\n",
      "18\t124\t0.0\t1\t49 74 1\n",
      "19\t66\t0.0\t1\t24 42\n",
      "20\t60\t0.0\t2\t15 24 21\n",
      "21\t73\t0.0\t2\t18 29 26\n",
      "22\t73\t0.0\t2\t23 25 25\n",
      "23\t150\t0.0\t2\t46 56 48\n",
      "24\t550\t0.0\t2\t152 241 157\n",
      "25\t125\t0.0\t2\t38 56 31\n",
      "26\t85\t0.0\t2\t31 23 31\n",
      "27\t24\t0.0\t2\t5 10 9\n",
      "28\t22\t0.0\t2\t15 4 3\n",
      "29\t3\t0.0\t2\t1 1 1\n",
      "30\t3\t0.0\t3\t3\n",
      "31\t1\t0.0\t3\t0 0 0 1\n",
      "32\t6\t0.0\t3\t5 1\n",
      "33\t5\t0.0\t3\t4 1\n",
      "34\t9\t0.0\t3\t6 3\n",
      "35\t4\t0.0\t3\t4\n",
      "36\t14\t0.0\t3\t13 1\n",
      "37\t6\t0.0\t3\t5 1\n",
      "38\t6\t0.0\t3\t6\n",
      "39\t6\t0.0\t3\t5 1\n",
      "40\t6\t0.0\t3\t4 1 0 1\n",
      "41\t2\t0.0\t3\t2\n",
      "42\t2\t0.0\t3\t2\n",
      "43\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 173 times.\n",
      "173 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t59\t3906.2\t0\t59\n",
      "6\t106\t976.6\t0\t106\n",
      "8\t1\t61.0\t0\t1\n",
      "9\t1\t15.3\t0\t0 1\n",
      "10\t2\t3.8\t1\t1 1\n",
      "11\t4\t1.0\t1\t0 4\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 3523425 times.\n",
      "2712 times, it overlapped the 5' end of a read\n",
      "3520713 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t411\t3906.2\t0\t411\n",
      "6\t60\t976.6\t0\t60\n",
      "7\t14\t244.1\t0\t14\n",
      "8\t2\t61.0\t0\t2\n",
      "10\t3\t3.8\t1\t0 3\n",
      "11\t2\t1.0\t1\t0 2\n",
      "15\t1\t0.0\t1\t0 1\n",
      "20\t14\t0.0\t2\t1 0 13\n",
      "21\t25\t0.0\t2\t0 2 23\n",
      "22\t168\t0.0\t2\t1 33 134\n",
      "23\t184\t0.0\t2\t2 45 137\n",
      "24\t717\t0.0\t2\t0 251 466\n",
      "25\t675\t0.0\t2\t2 198 475\n",
      "26\t263\t0.0\t2\t1 79 183\n",
      "27\t57\t0.0\t2\t0 14 43\n",
      "28\t30\t0.0\t2\t1 14 15\n",
      "29\t25\t0.0\t2\t1 9 15\n",
      "30\t61\t0.0\t3\t23 14 13 11\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t329\t3906.2\t0\t329\n",
      "6\t163\t976.6\t0\t163\n",
      "7\t195\t244.1\t0\t195\n",
      "8\t275\t61.0\t0\t275\n",
      "9\t521\t15.3\t0\t521\n",
      "10\t1501\t3.8\t1\t1411 90\n",
      "11\t4263\t1.0\t1\t4145 118\n",
      "12\t9910\t0.2\t1\t9630 280\n",
      "13\t29050\t0.1\t1\t28569 481\n",
      "14\t94083\t0.0\t1\t92167 1916\n",
      "15\t208738\t0.0\t1\t203409 5329\n",
      "16\t259905\t0.0\t1\t249239 10666\n",
      "17\t336773\t0.0\t1\t307686 29087\n",
      "18\t406773\t0.0\t1\t324581 82192\n",
      "19\t337118\t0.0\t1\t265637 71481\n",
      "20\t415570\t0.0\t2\t234634 113896 67040\n",
      "21\t453890\t0.0\t2\t159266 180943 113681\n",
      "22\t387902\t0.0\t2\t87159 155478 145265\n",
      "23\t293519\t0.0\t2\t26039 93790 173690\n",
      "24\t157752\t0.0\t2\t4663 36701 116388\n",
      "25\t69836\t0.0\t2\t787 11057 57992\n",
      "26\t25715\t0.0\t2\t272 2740 22703\n",
      "27\t8705\t0.0\t2\t196 789 7720\n",
      "28\t3189\t0.0\t2\t196 587 2406\n",
      "29\t1744\t0.0\t2\t216 427 1101\n",
      "30\t4046\t0.0\t3\t219 354 705 2768\n",
      "31\t2465\t0.0\t3\t173 276 370 1646\n",
      "32\t2142\t0.0\t3\t149 199 287 1507\n",
      "33\t1698\t0.0\t3\t121 170 239 1168\n",
      "34\t939\t0.0\t3\t89 124 156 570\n",
      "35\t703\t0.0\t3\t54 58 86 505\n",
      "36\t350\t0.0\t3\t36 36 61 217\n",
      "37\t164\t0.0\t3\t24 32 27 81\n",
      "38\t154\t0.0\t3\t30 13 13 98\n",
      "39\t96\t0.0\t3\t29 12 10 45\n",
      "40\t79\t0.0\t3\t28 14 11 26\n",
      "41\t96\t0.0\t3\t37 19 11 29\n",
      "42\t75\t0.0\t3\t22 8 15 30\n",
      "43\t250\t0.0\t3\t24 21 40 165\n",
      "44\t37\t0.0\t3\t15 10 3 9\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288396_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288396_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 179.22 s (78 us/read; 0.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               2,300,685\n",
      "Reads with adapters:                 2,135,089 (92.8%)\n",
      "Reads that were too short:           1,052,596 (45.8%)\n",
      "Reads written (passing filters):     1,248,089 (54.2%)\n",
      "\n",
      "Total basepairs processed:   103,526,659 bp\n",
      "Total written (filtered):     38,988,829 bp (37.7%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 111146 times.\n",
      "141 times, it overlapped the 5' end of a read\n",
      "111005 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t96\t2246.8\t0\t96\n",
      "6\t28\t561.7\t0\t28\n",
      "7\t1\t140.4\t0\t1\n",
      "11\t1\t0.5\t1\t0 1\n",
      "17\t3\t0.0\t1\t2 1\n",
      "20\t1\t0.0\t2\t0 0 1\n",
      "23\t6\t0.0\t2\t6\n",
      "25\t2\t0.0\t2\t0 1 1\n",
      "26\t2\t0.0\t2\t0 1 1\n",
      "28\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t44031\t2246.8\t0\t44031\n",
      "6\t27465\t561.7\t0\t27465\n",
      "7\t14908\t140.4\t0\t14908\n",
      "8\t7778\t35.1\t0\t7778\n",
      "9\t4349\t8.8\t0\t4312 37\n",
      "10\t5310\t2.2\t1\t2990 2320\n",
      "11\t3148\t0.5\t1\t1692 1456\n",
      "12\t2412\t0.1\t1\t1351 1061\n",
      "13\t1592\t0.0\t1\t826 766\n",
      "14\t9\t0.0\t1\t0 9\n",
      "15\t3\t0.0\t1\t0 3\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 955 times.\n",
      "955 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t67\t2246.8\t0\t67\n",
      "6\t25\t561.7\t0\t25\n",
      "7\t861\t140.4\t0\t861\n",
      "8\t1\t35.1\t0\t1\n",
      "9\t1\t8.8\t0\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 27 times.\n",
      "27 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t27\t2246.8\t0\t27\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 3064 times.\n",
      "37 times, it overlapped the 5' end of a read\n",
      "3027 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t3\t2246.8\t0\t3\n",
      "6\t2\t561.7\t0\t2\n",
      "7\t1\t140.4\t0\t1\n",
      "10\t3\t2.2\t1\t0 3\n",
      "11\t2\t0.5\t1\t0 2\n",
      "26\t1\t0.0\t2\t1\n",
      "27\t1\t0.0\t2\t1\n",
      "28\t2\t0.0\t2\t2\n",
      "29\t5\t0.0\t2\t5\n",
      "30\t4\t0.0\t3\t2 2\n",
      "31\t4\t0.0\t3\t4\n",
      "32\t9\t0.0\t3\t8 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t27\t0.0\t1\t0 27\n",
      "14\t973\t0.0\t1\t484 489\n",
      "15\t630\t0.0\t1\t308 322\n",
      "16\t314\t0.0\t1\t158 156\n",
      "17\t164\t0.0\t1\t71 93\n",
      "18\t77\t0.0\t1\t32 45\n",
      "19\t44\t0.0\t1\t18 25 1\n",
      "20\t32\t0.0\t2\t14 9 9\n",
      "21\t45\t0.0\t2\t14 18 13\n",
      "22\t46\t0.0\t2\t20 10 16\n",
      "23\t106\t0.0\t2\t41 35 30\n",
      "24\t362\t0.0\t2\t103 152 107\n",
      "25\t74\t0.0\t2\t33 22 19\n",
      "26\t57\t0.0\t2\t25 18 14\n",
      "27\t19\t0.0\t2\t7 7 5\n",
      "28\t7\t0.0\t2\t4 1 2\n",
      "29\t2\t0.0\t2\t2\n",
      "31\t2\t0.0\t3\t2\n",
      "32\t2\t0.0\t3\t1 0 0 1\n",
      "33\t2\t0.0\t3\t1 1\n",
      "34\t1\t0.0\t3\t1\n",
      "35\t9\t0.0\t3\t8 1\n",
      "36\t5\t0.0\t3\t3 1 1\n",
      "37\t9\t0.0\t3\t7 1 0 1\n",
      "38\t6\t0.0\t3\t6\n",
      "39\t5\t0.0\t3\t4 0 1\n",
      "40\t4\t0.0\t3\t4\n",
      "42\t1\t0.0\t3\t1\n",
      "44\t2\t0.0\t3\t2\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 111 times.\n",
      "111 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t35\t2246.8\t0\t35\n",
      "6\t73\t561.7\t0\t73\n",
      "8\t1\t35.1\t0\t1\n",
      "9\t2\t8.8\t0\t0 2\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 2019786 times.\n",
      "1560 times, it overlapped the 5' end of a read\n",
      "2018226 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t259\t2246.8\t0\t259\n",
      "6\t34\t561.7\t0\t34\n",
      "7\t6\t140.4\t0\t6\n",
      "9\t1\t8.8\t0\t1\n",
      "10\t2\t2.2\t1\t0 2\n",
      "11\t1\t0.5\t1\t0 1\n",
      "12\t1\t0.1\t1\t0 1\n",
      "14\t1\t0.0\t1\t0 1\n",
      "20\t10\t0.0\t2\t0 0 10\n",
      "21\t22\t0.0\t2\t0 2 20\n",
      "22\t99\t0.0\t2\t0 9 90\n",
      "23\t117\t0.0\t2\t0 29 88\n",
      "24\t416\t0.0\t2\t2 138 276\n",
      "25\t362\t0.0\t2\t0 95 267\n",
      "26\t114\t0.0\t2\t0 36 78\n",
      "27\t30\t0.0\t2\t0 6 24\n",
      "28\t25\t0.0\t2\t0 12 13\n",
      "29\t19\t0.0\t2\t0 5 14\n",
      "30\t41\t0.0\t3\t13 3 14 11\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t182\t2246.8\t0\t182\n",
      "6\t75\t561.7\t0\t75\n",
      "7\t95\t140.4\t0\t95\n",
      "8\t166\t35.1\t0\t166\n",
      "9\t302\t8.8\t0\t302\n",
      "10\t877\t2.2\t1\t809 68\n",
      "11\t2292\t0.5\t1\t2219 73\n",
      "12\t5608\t0.1\t1\t5453 155\n",
      "13\t16793\t0.0\t1\t16511 282\n",
      "14\t53034\t0.0\t1\t51977 1057\n",
      "15\t119680\t0.0\t1\t116555 3125\n",
      "16\t148724\t0.0\t1\t142575 6149\n",
      "17\t193104\t0.0\t1\t176220 16884\n",
      "18\t232650\t0.0\t1\t185326 47324\n",
      "19\t194525\t0.0\t1\t151947 42578\n",
      "20\t238804\t0.0\t2\t130497 69010 39297\n",
      "21\t262393\t0.0\t2\t84681 110835 66877\n",
      "22\t223371\t0.0\t2\t45490 91607 86274\n",
      "23\t168001\t0.0\t2\t14524 50995 102482\n",
      "24\t87808\t0.0\t2\t2890 19993 64925\n",
      "25\t38341\t0.0\t2\t523 6288 31530\n",
      "26\t14566\t0.0\t2\t188 1724 12654\n",
      "27\t5234\t0.0\t2\t144 538 4552\n",
      "28\t2093\t0.0\t2\t174 438 1481\n",
      "29\t1228\t0.0\t2\t177 291 760\n",
      "30\t2584\t0.0\t3\t173 253 465 1693\n",
      "31\t1583\t0.0\t3\t113 204 289 977\n",
      "32\t1312\t0.0\t3\t108 137 223 844\n",
      "33\t1067\t0.0\t3\t117 122 167 661\n",
      "34\t637\t0.0\t3\t76 100 121 340\n",
      "35\t404\t0.0\t3\t52 48 66 238\n",
      "36\t193\t0.0\t3\t24 21 32 116\n",
      "37\t84\t0.0\t3\t15 13 10 46\n",
      "38\t84\t0.0\t3\t15 8 4 57\n",
      "39\t50\t0.0\t3\t9 4 5 32\n",
      "40\t43\t0.0\t3\t17 5 4 17\n",
      "41\t59\t0.0\t3\t30 11 3 15\n",
      "42\t35\t0.0\t3\t5 7 7 16\n",
      "43\t131\t0.0\t3\t14 14 25 78\n",
      "44\t14\t0.0\t3\t7 2 3 2\n",
      "\n",
      "get trimmed reads of fourth group of Riboseq reads to  /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288400_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288400_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1228.20 s (60 us/read; 0.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              20,351,396\n",
      "Reads with adapters:                12,875,193 (63.3%)\n",
      "Reads that were too short:          12,875,193 (63.3%)\n",
      "Reads written (passing filters):     7,476,203 (36.7%)\n",
      "\n",
      "Total basepairs processed:   589,971,891 bp\n",
      "Total written (filtered):    216,727,717 bp (36.7%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 25460 times.\n",
      "20828 times, it overlapped the 5' end of a read\n",
      "4632 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1966\t19874.4\t0\t1966\n",
      "6\t1743\t4968.6\t0\t1743\n",
      "7\t46\t1242.2\t0\t46\n",
      "8\t57\t310.5\t0\t57\n",
      "9\t27\t77.6\t0\t2 25\n",
      "10\t64\t19.4\t1\t0 64\n",
      "11\t52\t4.9\t1\t1 51\n",
      "12\t8\t1.2\t1\t1 7\n",
      "14\t1\t0.1\t1\t1\n",
      "15\t36\t0.0\t1\t25 11\n",
      "16\t116\t0.0\t1\t95 21\n",
      "17\t261\t0.0\t1\t95 166\n",
      "18\t739\t0.0\t1\t466 272 1\n",
      "19\t13498\t0.0\t1\t12872 607 19\n",
      "20\t795\t0.0\t2\t195 342 258\n",
      "21\t587\t0.0\t2\t390 79 118\n",
      "22\t278\t0.0\t2\t13 191 74\n",
      "23\t175\t0.0\t2\t20 29 126\n",
      "24\t259\t0.0\t2\t3 59 197\n",
      "25\t76\t0.0\t2\t0 16 60\n",
      "26\t25\t0.0\t2\t0 5 20\n",
      "27\t11\t0.0\t2\t0 3 8\n",
      "28\t3\t0.0\t2\t0 0 2 1\n",
      "29\t5\t0.0\t2\t0 0 2 3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t516\t19874.4\t0\t516\n",
      "6\t209\t4968.6\t0\t209\n",
      "7\t54\t1242.2\t0\t54\n",
      "8\t25\t310.5\t0\t25\n",
      "9\t32\t77.6\t0\t19 13\n",
      "10\t50\t19.4\t1\t1 49\n",
      "11\t25\t4.9\t1\t0 25\n",
      "12\t4\t1.2\t1\t0 4\n",
      "13\t2\t0.3\t1\t0 2\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t3\t0.0\t1\t3\n",
      "20\t3651\t0.0\t2\t3463 164 24\n",
      "21\t50\t0.0\t2\t46 4\n",
      "22\t4\t0.0\t2\t1 1 2\n",
      "23\t1\t0.0\t2\t1\n",
      "24\t3\t0.0\t2\t1 1 1\n",
      "25\t2\t0.0\t2\t2\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 2850 times.\n",
      "2850 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1244\t19874.4\t0\t1244\n",
      "6\t337\t4968.6\t0\t337\n",
      "7\t943\t1242.2\t0\t943\n",
      "8\t117\t310.5\t0\t117\n",
      "9\t111\t77.6\t0\t3 108\n",
      "10\t70\t19.4\t1\t0 70\n",
      "11\t27\t4.9\t1\t0 27\n",
      "12\t1\t1.2\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 801 times.\n",
      "800 times, it overlapped the 5' end of a read\n",
      "1 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t554\t19874.4\t0\t554\n",
      "6\t114\t4968.6\t0\t114\n",
      "7\t42\t1242.2\t0\t42\n",
      "8\t6\t310.5\t0\t6\n",
      "9\t35\t77.6\t0\t4 31\n",
      "10\t29\t19.4\t1\t0 29\n",
      "11\t15\t4.9\t1\t0 15\n",
      "12\t5\t1.2\t1\t0 5\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t1\t0.0\t2\t1\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 9758 times.\n",
      "9758 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t9226\t19874.4\t0\t9226\n",
      "6\t258\t4968.6\t0\t258\n",
      "7\t72\t1242.2\t0\t72\n",
      "8\t12\t310.5\t0\t12\n",
      "9\t14\t77.6\t0\t8 6\n",
      "10\t39\t19.4\t1\t0 39\n",
      "11\t125\t4.9\t1\t1 124\n",
      "12\t11\t1.2\t1\t0 11\n",
      "15\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 4374 times.\n",
      "4374 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2572\t19874.4\t0\t2572\n",
      "6\t320\t4968.6\t0\t320\n",
      "7\t41\t1242.2\t0\t41\n",
      "8\t7\t310.5\t0\t7\n",
      "9\t1249\t77.6\t0\t5 1244\n",
      "10\t128\t19.4\t1\t4 124\n",
      "11\t49\t4.9\t1\t0 49\n",
      "12\t8\t1.2\t1\t0 8\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 12831950 times.\n",
      "713978 times, it overlapped the 5' end of a read\n",
      "12117972 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t30444\t19874.4\t0\t30444\n",
      "6\t14007\t4968.6\t0\t14007\n",
      "7\t12656\t1242.2\t0\t12656\n",
      "8\t8309\t310.5\t0\t8309\n",
      "9\t3558\t77.6\t0\t3558\n",
      "10\t3232\t19.4\t1\t1086 2146\n",
      "11\t1490\t4.9\t1\t560 930\n",
      "12\t911\t1.2\t1\t491 420\n",
      "13\t895\t0.3\t1\t605 290\n",
      "14\t883\t0.1\t1\t714 169\n",
      "15\t918\t0.0\t1\t803 115\n",
      "16\t970\t0.0\t1\t765 205\n",
      "17\t1235\t0.0\t1\t766 469\n",
      "18\t2145\t0.0\t1\t683 1462\n",
      "19\t5528\t0.0\t1\t764 4764\n",
      "20\t55844\t0.0\t2\t663 11301 43880\n",
      "21\t108997\t0.0\t2\t227 23303 85467\n",
      "22\t227601\t0.0\t2\t106 51596 175899\n",
      "23\t101818\t0.0\t2\t64 27495 74259\n",
      "24\t46302\t0.0\t2\t50 13741 32511\n",
      "25\t21261\t0.0\t2\t32 6280 14949\n",
      "26\t7548\t0.0\t2\t21 2170 5357\n",
      "27\t4889\t0.0\t2\t16 1225 3648\n",
      "28\t2373\t0.0\t2\t162 1134 1077\n",
      "29\t50164\t0.0\t2\t7261 12298 30605\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t1794491\t19874.4\t0\t1794491\n",
      "6\t2185879\t4968.6\t0\t2185879\n",
      "7\t1917771\t1242.2\t0\t1917771\n",
      "8\t1380237\t310.5\t0\t1380237\n",
      "9\t658185\t77.6\t0\t658185\n",
      "10\t935697\t19.4\t1\t405878 529819\n",
      "11\t815657\t4.9\t1\t365851 449806\n",
      "12\t526608\t1.2\t1\t257796 268812\n",
      "13\t317017\t0.3\t1\t131769 185248\n",
      "14\t276376\t0.1\t1\t64867 211509\n",
      "15\t157824\t0.0\t1\t37252 120572\n",
      "16\t89820\t0.0\t1\t15033 74787\n",
      "17\t80388\t0.0\t1\t16688 63700\n",
      "18\t54269\t0.0\t1\t14745 39524\n",
      "19\t67948\t0.0\t1\t18446 49502\n",
      "20\t106845\t0.0\t2\t13223 34408 59214\n",
      "21\t93737\t0.0\t2\t12084 31937 49716\n",
      "22\t110878\t0.0\t2\t17476 42399 51003\n",
      "23\t118312\t0.0\t2\t22069 38298 57945\n",
      "24\t69852\t0.0\t2\t14830 21218 33804\n",
      "25\t102997\t0.0\t2\t23540 23418 56039\n",
      "26\t104815\t0.0\t2\t31231 30488 43096\n",
      "27\t87928\t0.0\t2\t28911 22020 36997\n",
      "28\t64441\t0.0\t2\t10170 20277 33994\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288401_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288401_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 310.51 s (78 us/read; 0.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                 3,925,853 (98.1%)\n",
      "Reads that were too short:           1,588,512 (39.7%)\n",
      "Reads written (passing filters):     2,411,488 (60.3%)\n",
      "\n",
      "Total basepairs processed:   180,000,000 bp\n",
      "Total written (filtered):     71,045,688 bp (39.5%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 41125 times.\n",
      "47 times, it overlapped the 5' end of a read\n",
      "41078 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t12\t3906.2\t0\t12\n",
      "6\t18\t976.6\t0\t18\n",
      "17\t1\t0.0\t1\t1\n",
      "20\t1\t0.0\t2\t0 0 1\n",
      "22\t1\t0.0\t2\t1\n",
      "23\t4\t0.0\t2\t2 1 1\n",
      "24\t2\t0.0\t2\t0 2\n",
      "25\t5\t0.0\t2\t0 2 3\n",
      "26\t1\t0.0\t2\t0 0 1\n",
      "27\t1\t0.0\t2\t0 0 1\n",
      "30\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t22411\t3906.2\t0\t22411\n",
      "6\t10241\t976.6\t0\t10241\n",
      "7\t4071\t244.1\t0\t4071\n",
      "8\t1718\t61.0\t0\t1718\n",
      "9\t782\t15.3\t0\t778 4\n",
      "10\t856\t3.8\t1\t424 432\n",
      "11\t503\t1.0\t1\t257 246\n",
      "12\t296\t0.2\t1\t149 147\n",
      "13\t193\t0.1\t1\t87 106\n",
      "14\t4\t0.0\t1\t0 4\n",
      "15\t3\t0.0\t1\t0 3\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 452 times.\n",
      "452 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t16\t3906.2\t0\t16\n",
      "6\t1\t976.6\t0\t1\n",
      "7\t434\t244.1\t0\t434\n",
      "8\t1\t61.0\t0\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 6 times.\n",
      "6 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t6\t3906.2\t0\t6\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 777 times.\n",
      "90 times, it overlapped the 5' end of a read\n",
      "687 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2\t3906.2\t0\t2\n",
      "10\t1\t3.8\t1\t0 1\n",
      "11\t1\t1.0\t1\t0 1\n",
      "12\t2\t0.2\t1\t0 2\n",
      "24\t1\t0.0\t2\t1\n",
      "26\t2\t0.0\t2\t2\n",
      "27\t3\t0.0\t2\t3\n",
      "28\t4\t0.0\t2\t4\n",
      "29\t10\t0.0\t2\t9 0 0 1\n",
      "30\t5\t0.0\t3\t5\n",
      "31\t21\t0.0\t3\t19 2\n",
      "32\t30\t0.0\t3\t30\n",
      "33\t2\t0.0\t3\t1 0 1\n",
      "34\t6\t0.0\t3\t5 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t4\t0.1\t1\t0 4\n",
      "14\t124\t0.0\t1\t57 67\n",
      "15\t99\t0.0\t1\t36 63\n",
      "16\t64\t0.0\t1\t23 41\n",
      "17\t47\t0.0\t1\t18 29\n",
      "18\t30\t0.0\t1\t14 16\n",
      "19\t26\t0.0\t1\t6 18 2\n",
      "20\t22\t0.0\t2\t4 9 9\n",
      "21\t20\t0.0\t2\t5 5 10\n",
      "22\t27\t0.0\t2\t9 9 9\n",
      "23\t35\t0.0\t2\t10 9 16\n",
      "24\t48\t0.0\t2\t10 24 14\n",
      "25\t11\t0.0\t2\t4 5 2\n",
      "26\t7\t0.0\t2\t2 4 1\n",
      "27\t4\t0.0\t2\t0 3 1\n",
      "31\t2\t0.0\t3\t1 0 0 1\n",
      "32\t3\t0.0\t3\t2 1\n",
      "33\t8\t0.0\t3\t5 3\n",
      "34\t7\t0.0\t3\t5 2\n",
      "35\t11\t0.0\t3\t8 1 2\n",
      "36\t15\t0.0\t3\t14 0 1\n",
      "37\t16\t0.0\t3\t16\n",
      "38\t12\t0.0\t3\t11 0 1\n",
      "39\t14\t0.0\t3\t12 1 1\n",
      "40\t15\t0.0\t3\t11 4\n",
      "41\t5\t0.0\t3\t4 0 1\n",
      "42\t3\t0.0\t3\t3\n",
      "43\t5\t0.0\t3\t4 1\n",
      "44\t3\t0.0\t3\t3\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 20 times.\n",
      "20 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t8\t3906.2\t0\t8\n",
      "6\t11\t976.6\t0\t11\n",
      "7\t1\t244.1\t0\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 3883473 times.\n",
      "533 times, it overlapped the 5' end of a read\n",
      "3882940 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t105\t3906.2\t0\t105\n",
      "6\t35\t976.6\t0\t35\n",
      "7\t8\t244.1\t0\t8\n",
      "8\t2\t61.0\t0\t2\n",
      "10\t3\t3.8\t1\t0 3\n",
      "11\t3\t1.0\t1\t0 3\n",
      "12\t1\t0.2\t1\t1\n",
      "20\t2\t0.0\t2\t1 0 1\n",
      "21\t2\t0.0\t2\t1 0 1\n",
      "22\t23\t0.0\t2\t0 4 19\n",
      "23\t29\t0.0\t2\t0 11 18\n",
      "24\t87\t0.0\t2\t2 32 53\n",
      "25\t86\t0.0\t2\t0 27 59\n",
      "26\t47\t0.0\t2\t0 12 35\n",
      "27\t13\t0.0\t2\t0 7 6\n",
      "28\t5\t0.0\t2\t0 1 4\n",
      "29\t3\t0.0\t2\t0 2 1\n",
      "30\t79\t0.0\t3\t65 4 3 7\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t475\t3906.2\t0\t475\n",
      "6\t303\t976.6\t0\t303\n",
      "7\t691\t244.1\t0\t691\n",
      "8\t2027\t61.0\t0\t2027\n",
      "9\t4274\t15.3\t0\t4274\n",
      "10\t14995\t3.8\t1\t14574 421\n",
      "11\t41935\t1.0\t1\t41141 794\n",
      "12\t77658\t0.2\t1\t75622 2036\n",
      "13\t98409\t0.1\t1\t95157 3252\n",
      "14\t195205\t0.0\t1\t175833 19372\n",
      "15\t380774\t0.0\t1\t298430 82344\n",
      "16\t348022\t0.0\t1\t323373 24649\n",
      "17\t370026\t0.0\t1\t332873 37153\n",
      "18\t419381\t0.0\t1\t326580 92801\n",
      "19\t341022\t0.0\t1\t268962 72060\n",
      "20\t449553\t0.0\t2\t245854 105365 98334\n",
      "21\t386155\t0.0\t2\t148836 131897 105422\n",
      "22\t283387\t0.0\t2\t66912 99270 117205\n",
      "23\t268140\t0.0\t2\t14837 97760 155543\n",
      "24\t94069\t0.0\t2\t1603 18911 73555\n",
      "25\t56904\t0.0\t2\t228 6539 50137\n",
      "26\t15623\t0.0\t2\t72 1615 13936\n",
      "27\t5820\t0.0\t2\t117 811 4892\n",
      "28\t2289\t0.0\t2\t167 841 1281\n",
      "29\t2068\t0.0\t2\t163 882 1023\n",
      "30\t14676\t0.0\t3\t205 867 710 12894\n",
      "31\t2343\t0.0\t3\t135 493 311 1404\n",
      "32\t3209\t0.0\t3\t156 494 182 2377\n",
      "33\t1293\t0.0\t3\t173 434 172 514\n",
      "34\t809\t0.0\t3\t177 358 94 180\n",
      "35\t577\t0.0\t3\t153 198 82 144\n",
      "36\t314\t0.0\t3\t108 82 29 95\n",
      "37\t184\t0.0\t3\t80 57 16 31\n",
      "38\t98\t0.0\t3\t30 15 9 44\n",
      "39\t56\t0.0\t3\t38 7 5 6\n",
      "40\t32\t0.0\t3\t23 5 1 3\n",
      "41\t49\t0.0\t3\t32 6 3 8\n",
      "42\t35\t0.0\t3\t18 3 4 10\n",
      "43\t49\t0.0\t3\t29 3 3 14\n",
      "44\t11\t0.0\t3\t9 1 0 1\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288402_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288402_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 310.81 s (78 us/read; 0.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,000,000\n",
      "Reads with adapters:                 3,919,721 (98.0%)\n",
      "Reads that were too short:           1,575,245 (39.4%)\n",
      "Reads written (passing filters):     2,424,755 (60.6%)\n",
      "\n",
      "Total basepairs processed:   179,998,277 bp\n",
      "Total written (filtered):     71,607,638 bp (39.8%)\n",
      "\n",
      "=== Adapter TruSeq2_SE ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG; Type: variable 5'/3'; Length: 34; Trimmed: 49038 times.\n",
      "56 times, it overlapped the 5' end of a read\n",
      "48982 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t15\t3906.2\t0\t15\n",
      "6\t24\t976.6\t0\t24\n",
      "17\t1\t0.0\t1\t1\n",
      "19\t1\t0.0\t1\t0 1\n",
      "20\t1\t0.0\t2\t0 0 1\n",
      "23\t7\t0.0\t2\t4 3\n",
      "25\t3\t0.0\t2\t0 0 3\n",
      "26\t2\t0.0\t2\t0 0 2\n",
      "27\t1\t0.0\t2\t0 0 1\n",
      "32\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t27009\t3906.2\t0\t27009\n",
      "6\t12234\t976.6\t0\t12234\n",
      "7\t4765\t244.1\t0\t4765\n",
      "8\t1887\t61.0\t0\t1887\n",
      "9\t928\t15.3\t0\t920 8\n",
      "10\t975\t3.8\t1\t539 436\n",
      "11\t577\t1.0\t1\t303 274\n",
      "12\t372\t0.2\t1\t179 193\n",
      "13\t227\t0.1\t1\t112 115\n",
      "14\t5\t0.0\t1\t0 5\n",
      "15\t3\t0.0\t1\t0 3\n",
      "\n",
      "=== Adapter TruSeq2_PE_f ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT; Type: variable 5'/3'; Length: 33; Trimmed: 489 times.\n",
      "489 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t13\t3906.2\t0\t13\n",
      "6\t2\t976.6\t0\t2\n",
      "7\t474\t244.1\t0\t474\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq2_PE_r ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG; Type: variable 5'/3'; Length: 33; Trimmed: 13 times.\n",
      "13 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-33 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t11\t3906.2\t0\t11\n",
      "7\t1\t244.1\t0\t1\n",
      "12\t1\t0.2\t1\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter TruSeq3_IndexedAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC; Type: variable 5'/3'; Length: 34; Trimmed: 821 times.\n",
      "109 times, it overlapped the 5' end of a read\n",
      "712 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t2\t3906.2\t0\t2\n",
      "10\t1\t3.8\t1\t0 1\n",
      "24\t1\t0.0\t2\t1\n",
      "26\t2\t0.0\t2\t2\n",
      "27\t1\t0.0\t2\t1\n",
      "28\t5\t0.0\t2\t4 1\n",
      "29\t9\t0.0\t2\t9\n",
      "30\t6\t0.0\t3\t5 1\n",
      "31\t27\t0.0\t3\t25 2\n",
      "32\t48\t0.0\t3\t48\n",
      "33\t6\t0.0\t3\t6\n",
      "34\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t4\t0.1\t1\t0 4\n",
      "14\t142\t0.0\t1\t59 83\n",
      "15\t120\t0.0\t1\t53 67\n",
      "16\t54\t0.0\t1\t25 29\n",
      "17\t31\t0.0\t1\t16 15\n",
      "18\t30\t0.0\t1\t14 16\n",
      "19\t22\t0.0\t1\t13 8 1\n",
      "20\t16\t0.0\t2\t4 6 6\n",
      "21\t28\t0.0\t2\t7 8 13\n",
      "22\t25\t0.0\t2\t7 6 12\n",
      "23\t30\t0.0\t2\t10 10 10\n",
      "24\t70\t0.0\t2\t18 30 22\n",
      "25\t17\t0.0\t2\t8 7 2\n",
      "26\t7\t0.0\t2\t3 4\n",
      "27\t3\t0.0\t2\t0 2 1\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t5\t0.0\t3\t4 1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t11\t0.0\t3\t10 1\n",
      "34\t3\t0.0\t3\t3\n",
      "35\t9\t0.0\t3\t8 1\n",
      "36\t11\t0.0\t3\t10 1\n",
      "37\t13\t0.0\t3\t12 1\n",
      "38\t16\t0.0\t3\t13 2 1\n",
      "39\t15\t0.0\t3\t14 1\n",
      "40\t8\t0.0\t3\t8\n",
      "41\t7\t0.0\t3\t5 2\n",
      "42\t4\t0.0\t3\t4\n",
      "43\t4\t0.0\t3\t4\n",
      "44\t4\t0.0\t3\t3 1\n",
      "\n",
      "=== Adapter TruSeq3_UniversalAdapter ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA; Type: variable 5'/3'; Length: 34; Trimmed: 25 times.\n",
      "25 times, it overlapped the 5' end of a read\n",
      "0 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-34 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t13\t3906.2\t0\t13\n",
      "6\t11\t976.6\t0\t11\n",
      "9\t1\t15.3\t0\t0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "\n",
      "=== Adapter polyA_adapter ===\n",
      "\n",
      "Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA; Type: variable 5'/3'; Length: 30; Trimmed: 3869335 times.\n",
      "571 times, it overlapped the 5' end of a read\n",
      "3868764 times, it overlapped the 3' end or was within the read\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30 bp: 3\n",
      "\n",
      "Overview of removed sequences (5')\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t138\t3906.2\t0\t138\n",
      "6\t45\t976.6\t0\t45\n",
      "7\t13\t244.1\t0\t13\n",
      "8\t6\t61.0\t0\t6\n",
      "10\t3\t3.8\t1\t0 3\n",
      "11\t1\t1.0\t1\t0 1\n",
      "13\t3\t0.1\t1\t0 3\n",
      "20\t2\t0.0\t2\t0 0 2\n",
      "21\t3\t0.0\t2\t1 1 1\n",
      "22\t16\t0.0\t2\t1 1 14\n",
      "23\t24\t0.0\t2\t0 7 17\n",
      "24\t96\t0.0\t2\t0 28 68\n",
      "25\t79\t0.0\t2\t0 26 53\n",
      "26\t40\t0.0\t2\t0 14 26\n",
      "27\t13\t0.0\t2\t0 3 10\n",
      "28\t11\t0.0\t2\t1 4 6\n",
      "29\t6\t0.0\t2\t0 2 4\n",
      "30\t72\t0.0\t3\t57 6 6 3\n",
      "\n",
      "\n",
      "Overview of removed sequences (3' or within)\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t437\t3906.2\t0\t437\n",
      "6\t302\t976.6\t0\t302\n",
      "7\t674\t244.1\t0\t674\n",
      "8\t2063\t61.0\t0\t2063\n",
      "9\t4311\t15.3\t0\t4311\n",
      "10\t14842\t3.8\t1\t14411 431\n",
      "11\t41792\t1.0\t1\t41015 777\n",
      "12\t78098\t0.2\t1\t76023 2075\n",
      "13\t99160\t0.1\t1\t95755 3405\n",
      "14\t194912\t0.0\t1\t175561 19351\n",
      "15\t379700\t0.0\t1\t297669 82031\n",
      "16\t347308\t0.0\t1\t322287 25021\n",
      "17\t369172\t0.0\t1\t331691 37481\n",
      "18\t419498\t0.0\t1\t325958 93540\n",
      "19\t342186\t0.0\t1\t266563 75623\n",
      "20\t450082\t0.0\t2\t236096 114494 99492\n",
      "21\t388381\t0.0\t2\t133389 146777 108215\n",
      "22\t285125\t0.0\t2\t57999 104090 123036\n",
      "23\t264009\t0.0\t2\t12618 92052 159339\n",
      "24\t88718\t0.0\t2\t1280 16819 70619\n",
      "25\t53777\t0.0\t2\t198 5857 47722\n",
      "26\t14272\t0.0\t2\t52 1470 12750\n",
      "27\t5149\t0.0\t2\t80 717 4352\n",
      "28\t2035\t0.0\t2\t105 854 1076\n",
      "29\t1812\t0.0\t2\t151 817 844\n",
      "30\t12512\t0.0\t3\t168 803 686 10855\n",
      "31\t2171\t0.0\t3\t131 485 295 1260\n",
      "32\t2892\t0.0\t3\t124 471 201 2096\n",
      "33\t1293\t0.0\t3\t161 462 159 511\n",
      "34\t805\t0.0\t3\t192 345 106 162\n",
      "35\t494\t0.0\t3\t111 208 60 115\n",
      "36\t261\t0.0\t3\t75 95 20 71\n",
      "37\t169\t0.0\t3\t68 48 19 34\n",
      "38\t101\t0.0\t3\t45 14 12 30\n",
      "39\t64\t0.0\t3\t50 8 3 3\n",
      "40\t35\t0.0\t3\t19 6 3 7\n",
      "41\t50\t0.0\t3\t32 8 3 7\n",
      "42\t43\t0.0\t3\t27 3 1 12\n",
      "43\t40\t0.0\t3\t23 2 4 11\n",
      "44\t19\t0.0\t3\t14 1 2 2\n",
      "\n",
      "This is cutadapt 1.16 with Python 3.6.3\n",
      "Command line parameters: -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/SRR1288403_adapters_trimming.fastq /projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/SRR1288403_first_trimming.fastq\n",
      "Running on 1 core\n",
      "Trimming 6 adapter"
     ]
    },
    {
     "data": {
      "text/html": [
       "<b>limit_output extension: Maximum message size of 100000 exceeded with 111768 characters</b>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_first_trimming/\"\n",
    "extention=\".fastq\"\n",
    "label1=\"_first_trimming\"\n",
    "label=\"_adapters_trimming\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "# First we trim the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get trimmed reads of first group of Riboseq reads to  '$output_directory\n",
    "END=688\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$label1$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the second group of reads that go from SRR1223697 to SRR1223699\n",
    "echo 'get trimmed reads of second group of Riboseq reads to  '$output_directory\n",
    "END=699\n",
    "for ((i=697;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base1$i$label$extention $path_fastq_files$accession_base1$i$label1$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the third group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get trimmed reads of third group of Riboseq reads to  '$output_directory\n",
    "END=396\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$label1$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Trim the fourth group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get trimmed reads of fourth group of Riboseq reads to  '$output_directory\n",
    "END=404\n",
    "for ((i=400;i<=END;i++))\n",
    "do\n",
    "cutadapt -O 5 -m 26 -b file:/home/simone/Downloads/Trimmomatic-0.36/adapters/adapters.fa -o $output_directory$accession_base2$i$label$extention $path_fastq_files$accession_base2$i$label1$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FastQC reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T16:23:04.437015Z",
     "start_time": "2018-02-27T15:40:37.672622Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get fastQC report of first group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_final_trimming/\n",
      "Analysis complete for SRR1223686_adapters_trimming.fastq\n",
      "Analysis complete for SRR1223687_adapters_trimming.fastq\n",
      "Analysis complete for SRR1223688_adapters_trimming.fastq\n",
      "get fastQC report of second group of reads to  /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_final_trimming/\n",
      "Analysis complete for SRR1223697_adapters_trimming.fastq\n",
      "Analysis complete for SRR1223698_adapters_trimming.fastq\n",
      "Analysis complete for SRR1223699_adapters_trimming.fastq\n",
      "get fastQC report of third group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_final_trimming/\n",
      "Analysis complete for SRR1288391_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288392_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288393_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288394_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288395_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288396_adapters_trimming.fastq\n",
      "get fastQC report of third group of reads to /projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_final_trimming/\n",
      "Analysis complete for SRR1288400_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288401_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288402_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288403_adapters_trimming.fastq\n",
      "Analysis complete for SRR1288404_adapters_trimming.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Unknown option: i\n",
      "Started analysis of SRR1223686_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1223686_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1223686_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223687_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1223687_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1223687_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223688_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1223688_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1223688_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223697_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1223697_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1223697_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223698_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1223698_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1223698_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1223699_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1223699_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1223699_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288391_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288391_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288391_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288392_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288392_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288392_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288393_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288393_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288393_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288394_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288394_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288394_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288395_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288395_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288395_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288396_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288396_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288396_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288400_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288400_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288400_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288401_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288401_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288401_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288402_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288402_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288402_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288403_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288403_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288403_adapters_trimming.fastq\n",
      "Unknown option: i\n",
      "Started analysis of SRR1288404_adapters_trimming.fastq\n",
      "Approx 5% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 10% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 15% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 20% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 25% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 30% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 35% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 40% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 45% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 50% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 55% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 60% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 65% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 70% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 75% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 80% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 85% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 90% complete for SRR1288404_adapters_trimming.fastq\n",
      "Approx 95% complete for SRR1288404_adapters_trimming.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_adapters_trimming\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/fastqc_results/Riboseq_final_trimming/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get fastQC report of first group of reads to  '$output_directory\n",
    "END=688\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223697 to SRR1223699\n",
    "\n",
    "echo 'get fastQC report of second group of reads to  '$output_directory\n",
    "END=699\n",
    "for ((i=697;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base1$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get fastQC report of third group of reads to '$output_directory\n",
    "END=396\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$label$extention\n",
    "done\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get fastQC report of third group of reads to '$output_directory\n",
    "END=404\n",
    "for ((i=400;i<=END;i++))\n",
    "do\n",
    "fastqc -I -O $output_directory $path_fastq_files$accession_base2$i$label$extention\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Map reads to ribosomal RNA of *S. cerevisiae* using Bowtie 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Map Riboseq reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-27T17:30:49.224630Z",
     "start_time": "2018-02-27T16:40:15.300444Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get filtered file of reads not mapping to S. cerevisiae rRNA of first group of reads to  /projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\n",
      "get filtered file of reads not mapping to S. cerevisiae rRNA of second group of reads to  /projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\n",
      "get filtered file of reads not mapping to S. cerevisiae rRNA of third group of reads to /projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\n",
      "get filtered file of reads not mapping to S. cerevisiae rRNA of fourth group of reads to /projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/’: File exists\n",
      "8925942 reads; of these:\n",
      "  8925942 (100.00%) were unpaired; of these:\n",
      "    8784944 (98.42%) aligned 0 times\n",
      "    94734 (1.06%) aligned exactly 1 time\n",
      "    46264 (0.52%) aligned >1 times\n",
      "1.58% overall alignment rate\n",
      "8634427 reads; of these:\n",
      "  8634427 (100.00%) were unpaired; of these:\n",
      "    8630785 (99.96%) aligned 0 times\n",
      "    2508 (0.03%) aligned exactly 1 time\n",
      "    1134 (0.01%) aligned >1 times\n",
      "0.04% overall alignment rate\n",
      "19563607 reads; of these:\n",
      "  19563607 (100.00%) were unpaired; of these:\n",
      "    19551489 (99.94%) aligned 0 times\n",
      "    8393 (0.04%) aligned exactly 1 time\n",
      "    3725 (0.02%) aligned >1 times\n",
      "0.06% overall alignment rate\n",
      "4979428 reads; of these:\n",
      "  4979428 (100.00%) were unpaired; of these:\n",
      "    4952698 (99.46%) aligned 0 times\n",
      "    19529 (0.39%) aligned exactly 1 time\n",
      "    7201 (0.14%) aligned >1 times\n",
      "0.54% overall alignment rate\n",
      "19267384 reads; of these:\n",
      "  19267384 (100.00%) were unpaired; of these:\n",
      "    19256198 (99.94%) aligned 0 times\n",
      "    8038 (0.04%) aligned exactly 1 time\n",
      "    3148 (0.02%) aligned >1 times\n",
      "0.06% overall alignment rate\n",
      "35418589 reads; of these:\n",
      "  35418589 (100.00%) were unpaired; of these:\n",
      "    35397788 (99.94%) aligned 0 times\n",
      "    14993 (0.04%) aligned exactly 1 time\n",
      "    5808 (0.02%) aligned >1 times\n",
      "0.06% overall alignment rate\n",
      "10825978 reads; of these:\n",
      "  10825978 (100.00%) were unpaired; of these:\n",
      "    10635905 (98.24%) aligned 0 times\n",
      "    140537 (1.30%) aligned exactly 1 time\n",
      "    49536 (0.46%) aligned >1 times\n",
      "1.76% overall alignment rate\n",
      "2107321 reads; of these:\n",
      "  2107321 (100.00%) were unpaired; of these:\n",
      "    2103721 (99.83%) aligned 0 times\n",
      "    2340 (0.11%) aligned exactly 1 time\n",
      "    1260 (0.06%) aligned >1 times\n",
      "0.17% overall alignment rate\n",
      "2129520 reads; of these:\n",
      "  2129520 (100.00%) were unpaired; of these:\n",
      "    2125856 (99.83%) aligned 0 times\n",
      "    2395 (0.11%) aligned exactly 1 time\n",
      "    1269 (0.06%) aligned >1 times\n",
      "0.17% overall alignment rate\n",
      "2165483 reads; of these:\n",
      "  2165483 (100.00%) were unpaired; of these:\n",
      "    2161853 (99.83%) aligned 0 times\n",
      "    2349 (0.11%) aligned exactly 1 time\n",
      "    1281 (0.06%) aligned >1 times\n",
      "0.17% overall alignment rate\n",
      "2165270 reads; of these:\n",
      "  2165270 (100.00%) were unpaired; of these:\n",
      "    2161794 (99.84%) aligned 0 times\n",
      "    2287 (0.11%) aligned exactly 1 time\n",
      "    1189 (0.05%) aligned >1 times\n",
      "0.16% overall alignment rate\n",
      "1248089 reads; of these:\n",
      "  1248089 (100.00%) were unpaired; of these:\n",
      "    1246074 (99.84%) aligned 0 times\n",
      "    1310 (0.10%) aligned exactly 1 time\n",
      "    705 (0.06%) aligned >1 times\n",
      "0.16% overall alignment rate\n",
      "7476203 reads; of these:\n",
      "  7476203 (100.00%) were unpaired; of these:\n",
      "    7459285 (99.77%) aligned 0 times\n",
      "    12809 (0.17%) aligned exactly 1 time\n",
      "    4109 (0.05%) aligned >1 times\n",
      "0.23% overall alignment rate\n",
      "2411488 reads; of these:\n",
      "  2411488 (100.00%) were unpaired; of these:\n",
      "    2405108 (99.74%) aligned 0 times\n",
      "    4305 (0.18%) aligned exactly 1 time\n",
      "    2075 (0.09%) aligned >1 times\n",
      "0.26% overall alignment rate\n",
      "2424755 reads; of these:\n",
      "  2424755 (100.00%) were unpaired; of these:\n",
      "    2418355 (99.74%) aligned 0 times\n",
      "    4312 (0.18%) aligned exactly 1 time\n",
      "    2088 (0.09%) aligned >1 times\n",
      "0.26% overall alignment rate\n",
      "2434474 reads; of these:\n",
      "  2434474 (100.00%) were unpaired; of these:\n",
      "    2428044 (99.74%) aligned 0 times\n",
      "    4337 (0.18%) aligned exactly 1 time\n",
      "    2093 (0.09%) aligned >1 times\n",
      "0.26% overall alignment rate\n",
      "997121 reads; of these:\n",
      "  997121 (100.00%) were unpaired; of these:\n",
      "    994383 (99.73%) aligned 0 times\n",
      "    1824 (0.18%) aligned exactly 1 time\n",
      "    914 (0.09%) aligned >1 times\n",
      "0.27% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/Riboseq_adapters_trimming/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_adapters_trimming\"\n",
    "label2=\"_filtered\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "# create index file\n",
    "# bowtie2-build /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae.fa  rRNA_s_cerevisiae\n",
    "# export BOWTIE2_INDEXES='.'\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of first group of reads to  '$output_directory\n",
    "END=688\n",
    "for ((i=686;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base1$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base1$i$label$extention>/dev/null \n",
    "done\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223697 to SRR1223699\n",
    "\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of second group of reads to  '$output_directory\n",
    "END=699\n",
    "for ((i=697;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base1$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base1$i$label$extention>/dev/null \n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of third group of reads to '$output_directory\n",
    "END=396\n",
    "for ((i=391;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base2$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base2$i$label$extention>/dev/null \n",
    "done\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of fourth group of reads to '$output_directory\n",
    "END=404\n",
    "for ((i=400;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base2$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base2$i$label$extention>/dev/null \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Map RNAseq reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-28T09:29:47.716566Z",
     "start_time": "2018-02-28T08:34:42.646104Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "get filtered file of reads not mapping to S. cerevisiae rRNA of first group of reads to  /projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\n",
      "get filtered file of reads not mapping to S. cerevisiae rRNA of second group of reads to  /projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\n",
      "get filtered file of reads not mapping to S. cerevisiae rRNA of third group of reads to /projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\n",
      "get filtered file of reads not mapping to S. cerevisiae rRNA of fourth group of reads to /projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/’: File exists\n",
      "11169626 reads; of these:\n",
      "  11169626 (100.00%) were unpaired; of these:\n",
      "    11156993 (99.89%) aligned 0 times\n",
      "    8180 (0.07%) aligned exactly 1 time\n",
      "    4453 (0.04%) aligned >1 times\n",
      "0.11% overall alignment rate\n",
      "7394961 reads; of these:\n",
      "  7394961 (100.00%) were unpaired; of these:\n",
      "    7386705 (99.89%) aligned 0 times\n",
      "    5383 (0.07%) aligned exactly 1 time\n",
      "    2873 (0.04%) aligned >1 times\n",
      "0.11% overall alignment rate\n",
      "5995186 reads; of these:\n",
      "  5995186 (100.00%) were unpaired; of these:\n",
      "    5866093 (97.85%) aligned 0 times\n",
      "    86521 (1.44%) aligned exactly 1 time\n",
      "    42572 (0.71%) aligned >1 times\n",
      "2.15% overall alignment rate\n",
      "10596801 reads; of these:\n",
      "  10596801 (100.00%) were unpaired; of these:\n",
      "    10581812 (99.86%) aligned 0 times\n",
      "    9762 (0.09%) aligned exactly 1 time\n",
      "    5227 (0.05%) aligned >1 times\n",
      "0.14% overall alignment rate\n",
      "6374697 reads; of these:\n",
      "  6374697 (100.00%) were unpaired; of these:\n",
      "    6365725 (99.86%) aligned 0 times\n",
      "    5781 (0.09%) aligned exactly 1 time\n",
      "    3191 (0.05%) aligned >1 times\n",
      "0.14% overall alignment rate\n",
      "8456526 reads; of these:\n",
      "  8456526 (100.00%) were unpaired; of these:\n",
      "    8427257 (99.65%) aligned 0 times\n",
      "    19246 (0.23%) aligned exactly 1 time\n",
      "    10023 (0.12%) aligned >1 times\n",
      "0.35% overall alignment rate\n",
      "3717623 reads; of these:\n",
      "  3717623 (100.00%) were unpaired; of these:\n",
      "    3704738 (99.65%) aligned 0 times\n",
      "    8493 (0.23%) aligned exactly 1 time\n",
      "    4392 (0.12%) aligned >1 times\n",
      "0.35% overall alignment rate\n",
      "3842147 reads; of these:\n",
      "  3842147 (100.00%) were unpaired; of these:\n",
      "    3818435 (99.38%) aligned 0 times\n",
      "    14990 (0.39%) aligned exactly 1 time\n",
      "    8722 (0.23%) aligned >1 times\n",
      "0.62% overall alignment rate\n",
      "28081555 reads; of these:\n",
      "  28081555 (100.00%) were unpaired; of these:\n",
      "    28063800 (99.94%) aligned 0 times\n",
      "    11269 (0.04%) aligned exactly 1 time\n",
      "    6486 (0.02%) aligned >1 times\n",
      "0.06% overall alignment rate\n",
      "3593755 reads; of these:\n",
      "  3593755 (100.00%) were unpaired; of these:\n",
      "    3536680 (98.41%) aligned 0 times\n",
      "    37123 (1.03%) aligned exactly 1 time\n",
      "    19952 (0.56%) aligned >1 times\n",
      "1.59% overall alignment rate\n",
      "4879838 reads; of these:\n",
      "  4879838 (100.00%) were unpaired; of these:\n",
      "    4877085 (99.94%) aligned 0 times\n",
      "    1745 (0.04%) aligned exactly 1 time\n",
      "    1008 (0.02%) aligned >1 times\n",
      "0.06% overall alignment rate\n",
      "2236755 reads; of these:\n",
      "  2236755 (100.00%) were unpaired; of these:\n",
      "    2235426 (99.94%) aligned 0 times\n",
      "    823 (0.04%) aligned exactly 1 time\n",
      "    506 (0.02%) aligned >1 times\n",
      "0.06% overall alignment rate\n",
      "4922213 reads; of these:\n",
      "  4922213 (100.00%) were unpaired; of these:\n",
      "    4914512 (99.84%) aligned 0 times\n",
      "    5126 (0.10%) aligned exactly 1 time\n",
      "    2575 (0.05%) aligned >1 times\n",
      "0.16% overall alignment rate\n",
      "4303430 reads; of these:\n",
      "  4303430 (100.00%) were unpaired; of these:\n",
      "    4296731 (99.84%) aligned 0 times\n",
      "    4496 (0.10%) aligned exactly 1 time\n",
      "    2203 (0.05%) aligned >1 times\n",
      "0.16% overall alignment rate\n",
      "7173036 reads; of these:\n",
      "  7173036 (100.00%) were unpaired; of these:\n",
      "    7132743 (99.44%) aligned 0 times\n",
      "    24813 (0.35%) aligned exactly 1 time\n",
      "    15480 (0.22%) aligned >1 times\n",
      "0.56% overall alignment rate\n",
      "8362706 reads; of these:\n",
      "  8362706 (100.00%) were unpaired; of these:\n",
      "    8231075 (98.43%) aligned 0 times\n",
      "    89823 (1.07%) aligned exactly 1 time\n",
      "    41808 (0.50%) aligned >1 times\n",
      "1.57% overall alignment rate\n",
      "3099545 reads; of these:\n",
      "  3099545 (100.00%) were unpaired; of these:\n",
      "    3072677 (99.13%) aligned 0 times\n",
      "    16870 (0.54%) aligned exactly 1 time\n",
      "    9998 (0.32%) aligned >1 times\n",
      "0.87% overall alignment rate\n",
      "2375844 reads; of these:\n",
      "  2375844 (100.00%) were unpaired; of these:\n",
      "    2355315 (99.14%) aligned 0 times\n",
      "    13070 (0.55%) aligned exactly 1 time\n",
      "    7459 (0.31%) aligned >1 times\n",
      "0.86% overall alignment rate\n",
      "9629781 reads; of these:\n",
      "  9629781 (100.00%) were unpaired; of these:\n",
      "    9563354 (99.31%) aligned 0 times\n",
      "    47114 (0.49%) aligned exactly 1 time\n",
      "    19313 (0.20%) aligned >1 times\n",
      "0.69% overall alignment rate\n",
      "2686751 reads; of these:\n",
      "  2686751 (100.00%) were unpaired; of these:\n",
      "    2672514 (99.47%) aligned 0 times\n",
      "    8590 (0.32%) aligned exactly 1 time\n",
      "    5647 (0.21%) aligned >1 times\n",
      "0.53% overall alignment rate\n",
      "2687435 reads; of these:\n",
      "  2687435 (100.00%) were unpaired; of these:\n",
      "    2672797 (99.46%) aligned 0 times\n",
      "    8829 (0.33%) aligned exactly 1 time\n",
      "    5809 (0.22%) aligned >1 times\n",
      "0.54% overall alignment rate\n",
      "1622541 reads; of these:\n",
      "  1622541 (100.00%) were unpaired; of these:\n",
      "    1613764 (99.46%) aligned 0 times\n",
      "    5290 (0.33%) aligned exactly 1 time\n",
      "    3487 (0.21%) aligned >1 times\n",
      "0.54% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/cutadapt/RNAseq_afterNtrim/\"\n",
    "extention=\".fastq\"\n",
    "label=\"_final_trim\"\n",
    "label2=\"_filtered\"\n",
    "output_directory=\"/projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\"\n",
    "mkdir $output_directory\n",
    "\n",
    "\n",
    "# create index file\n",
    "# bowtie2-build /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae.fa  rRNA_s_cerevisiae\n",
    "# export BOWTIE2_INDEXES='.'\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223686 to SRR1223688\n",
    "accession_base1=\"SRR1223\"\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of first group of reads to  '$output_directory\n",
    "END=696\n",
    "for ((i=689;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base1$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base1$i$label$extention>/dev/null \n",
    "done\n",
    "\n",
    "# Analyse the group of reads that goes from SRR1223697 to SRR1223699\n",
    "\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of second group of reads to  '$output_directory\n",
    "END=706\n",
    "for ((i=700;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base1$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base1$i$label$extention>/dev/null \n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288391 to SRR1288396\n",
    "accession_base2=\"SRR1288\"\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of third group of reads to '$output_directory\n",
    "END=399\n",
    "for ((i=397;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base2$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base2$i$label$extention>/dev/null \n",
    "done\n",
    "\n",
    "\n",
    "# Analyse the group of reads that go from SRR1288400 to SRR1288404\n",
    "echo 'get filtered file of reads not mapping to S. cerevisiae rRNA of fourth group of reads to '$output_directory\n",
    "END=408\n",
    "for ((i=405;i<=END;i++))\n",
    "do\n",
    "bowtie2 --seedlen=25 --un $output_directory$accession_base2$i$label2$extention -x /projects_eg/projects/simone/glu_paper/Bowtie/index_file/rRNA_s_cerevisiae $path_fastq_files$accession_base2$i$label$extention>/dev/null \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mapping filtered reads to reference genome with Bowtie 2 (obtain SAM files)\n",
    "\n",
    "From this point we consider only **Riboseq** and **RNAseq-polyA** data for the two strains  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# create index file\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/Bowtie/genome_index_file\n",
    "\n",
    "bowtie2-build /projects_eg/projects/simone/reference_files/s_cerevisiae_reference_genome.fa  s_cerevisiae_reference_genome\n",
    "\n",
    "export BOWTIE2_INDEXES='.'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Riboseq data for BY4741"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-01T16:05:13.894163Z",
     "start_time": "2018-03-01T15:35:27.819025Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/sam_files/’: File exists\n",
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/bam_files/’: File exists\n",
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/bam_files/BY_Ribo/’: File exists\n",
      "8784944 reads; of these:\n",
      "  8784944 (100.00%) were unpaired; of these:\n",
      "    622807 (7.09%) aligned 0 times\n",
      "    2532746 (28.83%) aligned exactly 1 time\n",
      "    5629391 (64.08%) aligned >1 times\n",
      "92.91% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "10635905 reads; of these:\n",
      "  10635905 (100.00%) were unpaired; of these:\n",
      "    2763006 (25.98%) aligned 0 times\n",
      "    3427429 (32.23%) aligned exactly 1 time\n",
      "    4445470 (41.80%) aligned >1 times\n",
      "74.02% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "4952698 reads; of these:\n",
      "  4952698 (100.00%) were unpaired; of these:\n",
      "    831401 (16.79%) aligned 0 times\n",
      "    969468 (19.57%) aligned exactly 1 time\n",
      "    3151829 (63.64%) aligned >1 times\n",
      "83.21% overall alignment rate\n",
      "7459285 reads; of these:\n",
      "  7459285 (100.00%) were unpaired; of these:\n",
      "    2903234 (38.92%) aligned 0 times\n",
      "    1231145 (16.50%) aligned exactly 1 time\n",
      "    3324906 (44.57%) aligned >1 times\n",
      "61.08% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load SAMtools\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\"\n",
    "label_extension=\"_filtered.fastq\"\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/glu_paper/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "mkdir $sam_output_directory\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "mkdir $bam_output_directory\n",
    "\n",
    "BY_Ribo_bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/BY_Ribo/\"\n",
    "mkdir $BY_Ribo_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "\n",
    "\n",
    "list='SRR1223686 SRR1288391 SRR1223697 SRR1288400'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/glu_paper/Bowtie/genome_index_file/s_cerevisiae_reference_genome -U $path_fastq_files$i$label_extension -S $sam_output_directory$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$i$sam_ext > $BY_Ribo_bam_output_directory$i$bam_ext\n",
    "samtools sort $BY_Ribo_bam_output_directory$i$bam_ext -o $BY_Ribo_bam_output_directory$i$sorted_bam_ext -O BAM\n",
    "done\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RNAseq data for BY4741"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-01T16:33:22.208755Z",
     "start_time": "2018-03-01T16:05:13.897306Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/bam_files/BY_RNA/’: File exists\n",
      "5866093 reads; of these:\n",
      "  5866093 (100.00%) were unpaired; of these:\n",
      "    458837 (7.82%) aligned 0 times\n",
      "    2342595 (39.93%) aligned exactly 1 time\n",
      "    3064661 (52.24%) aligned >1 times\n",
      "92.18% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "8231075 reads; of these:\n",
      "  8231075 (100.00%) were unpaired; of these:\n",
      "    931212 (11.31%) aligned 0 times\n",
      "    2916899 (35.44%) aligned exactly 1 time\n",
      "    4382964 (53.25%) aligned >1 times\n",
      "88.69% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "3536680 reads; of these:\n",
      "  3536680 (100.00%) were unpaired; of these:\n",
      "    332403 (9.40%) aligned 0 times\n",
      "    1231268 (34.81%) aligned exactly 1 time\n",
      "    1973009 (55.79%) aligned >1 times\n",
      "90.60% overall alignment rate\n",
      "9563354 reads; of these:\n",
      "  9563354 (100.00%) were unpaired; of these:\n",
      "    1083698 (11.33%) aligned 0 times\n",
      "    2801975 (29.30%) aligned exactly 1 time\n",
      "    5677681 (59.37%) aligned >1 times\n",
      "88.67% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load SAMtools\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\"\n",
    "label_extension=\"_filtered.fastq\"\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/glu_paper/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "\n",
    "\n",
    "BY_RNA_bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/BY_RNA/\"\n",
    "mkdir $BY_RNA_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "\n",
    "\n",
    "list='SRR1223691 SRR1288397 SRR1223701 SRR1288405'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/glu_paper/Bowtie/genome_index_file/s_cerevisiae_reference_genome -U $path_fastq_files$i$label_extension -S $sam_output_directory$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$i$sam_ext > $BY_RNA_bam_output_directory$i$bam_ext\n",
    "samtools sort $BY_RNA_bam_output_directory$i$bam_ext -o $BY_RNA_bam_output_directory$i$sorted_bam_ext -O BAM\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Riboseq data for EY0690"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-02-28T18:57:40.250211Z",
     "start_time": "2018-02-28T18:07:01.354427Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# concatenate files containing reads for the same biological replicate\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping \n",
    "\n",
    "cat SRR1223687_filtered.fastq SRR1223688_filtered.fastq > EY-Ribo_logphase_Replicate1_filtered.fastq\n",
    "cat SRR1288392_filtered.fastq SRR1288393_filtered.fastq SRR1288394_filtered.fastq SRR1288395_filtered.fastq SRR1288396_filtered.fastq > EY-Ribo_logphase_Replicate2_filtered.fastq\n",
    "\n",
    "cat SRR1223698_filtered.fastq SRR1223699_filtered.fastq > EY-Ribo_-Glu15_Replicate1_filtered.fastq\n",
    "cat SRR1288401_filtered.fastq SRR1288402_filtered.fastq SRR1288403_filtered.fastq SRR1288404_filtered.fastq > EY-Ribo_-Glu15_Replicate2_filtered.fastq\n",
    "\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping\n",
    "\n",
    "cat SRR1288398_filtered.fastq SRR1288399_filtered.fastq > EY-RNA_logphase_polyA_Replicate2_filtered.fastq\n",
    "\n",
    "cat SRR1288406_filtered.fastq SRR1288407_filtered.fastq SRR1288408_filtered.fastq > EY-RNA_-Glu15_polyA_Replicate2_filtered.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-01T17:51:12.598193Z",
     "start_time": "2018-03-01T16:33:22.212101Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/’: File exists\n",
      "17415729 reads; of these:\n",
      "  17415729 (100.00%) were unpaired; of these:\n",
      "    3678453 (21.12%) aligned 0 times\n",
      "    2634350 (15.13%) aligned exactly 1 time\n",
      "    11102926 (63.75%) aligned >1 times\n",
      "78.88% overall alignment rate\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "9799298 reads; of these:\n",
      "  9799298 (100.00%) were unpaired; of these:\n",
      "    2157132 (22.01%) aligned 0 times\n",
      "    404944 (4.13%) aligned exactly 1 time\n",
      "    7237222 (73.85%) aligned >1 times\n",
      "77.99% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "54653986 reads; of these:\n",
      "  54653986 (100.00%) were unpaired; of these:\n",
      "    2429144 (4.44%) aligned 0 times\n",
      "    1447801 (2.65%) aligned exactly 1 time\n",
      "    50777041 (92.91%) aligned >1 times\n",
      "95.56% overall alignment rate\n",
      "[bam_sort_core] merging from 11 files and 1 in-memory blocks...\n",
      "8245890 reads; of these:\n",
      "  8245890 (100.00%) were unpaired; of these:\n",
      "    549161 (6.66%) aligned 0 times\n",
      "    850623 (10.32%) aligned exactly 1 time\n",
      "    6846106 (83.02%) aligned >1 times\n",
      "93.34% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load SAMtools\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\"\n",
    "label_extension=\"_filtered.fastq\"\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/glu_paper/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "\n",
    "\n",
    "EY_Ribo_bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/\"\n",
    "mkdir $EY_Ribo_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "\n",
    "\n",
    "list='EY-Ribo_logphase_Replicate1 EY-Ribo_logphase_Replicate2 EY-Ribo_-Glu15_Replicate1 EY-Ribo_-Glu15_Replicate2'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/glu_paper/Bowtie/genome_index_file/s_cerevisiae_reference_genome -U $path_fastq_files$i$label_extension -S $sam_output_directory$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$i$sam_ext > $EY_Ribo_bam_output_directory$i$bam_ext\n",
    "samtools sort $EY_Ribo_bam_output_directory$i$bam_ext -o $EY_Ribo_bam_output_directory$i$sorted_bam_ext -O BAM\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T16:29:53.575073Z",
     "start_time": "2018-03-05T16:03:47.202795Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "28182274 reads; of these:\n",
      "  28182274 (100.00%) were unpaired; of these:\n",
      "    6499998 (23.06%) aligned 0 times\n",
      "    449949 (1.60%) aligned exactly 1 time\n",
      "    21232327 (75.34%) aligned >1 times\n",
      "76.94% overall alignment rate\n",
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load SAMtools\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/Bowtie/Riboseq_rRNA_mapping/\"\n",
    "label_extension=\"_filtered.fastq\"\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/glu_paper/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "\n",
    "\n",
    "EY_Ribo_bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/\"\n",
    "#mkdir $EY_Ribo_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "\n",
    "\n",
    "list='EY-Ribo_logphase_Replicate1'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/glu_paper/Bowtie/genome_index_file/s_cerevisiae_reference_genome -U $path_fastq_files$i$label_extension -S $sam_output_directory$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$i$sam_ext > $EY_Ribo_bam_output_directory$i$bam_ext\n",
    "samtools sort $EY_Ribo_bam_output_directory$i$bam_ext -o $EY_Ribo_bam_output_directory$i$sorted_bam_ext -O BAM\n",
    "done\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/\n",
    "sorted=\"_sorted.bam\"\n",
    "sorted_bam_bai=\"_sorted.bam.bai\"\n",
    "list='EY-Ribo_logphase_Replicate1'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "samtools index -b $i$sorted  $i$sorted_bam_bai\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RNAseq data for EY0690"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-01T18:12:39.545513Z",
     "start_time": "2018-03-01T17:51:12.601185Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/projects_eg/projects/simone/glu_paper/bam_files/EY_RNA/’: File exists\n",
      "3818435 reads; of these:\n",
      "  3818435 (100.00%) were unpaired; of these:\n",
      "    250949 (6.57%) aligned 0 times\n",
      "    417196 (10.93%) aligned exactly 1 time\n",
      "    3150290 (82.50%) aligned >1 times\n",
      "93.43% overall alignment rate\n",
      "5427992 reads; of these:\n",
      "  5427992 (100.00%) were unpaired; of these:\n",
      "    96578 (1.78%) aligned 0 times\n",
      "    1053146 (19.40%) aligned exactly 1 time\n",
      "    4278268 (78.82%) aligned >1 times\n",
      "98.22% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "7132743 reads; of these:\n",
      "  7132743 (100.00%) were unpaired; of these:\n",
      "    287482 (4.03%) aligned 0 times\n",
      "    1175952 (16.49%) aligned exactly 1 time\n",
      "    5669309 (79.48%) aligned >1 times\n",
      "95.97% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "6959075 reads; of these:\n",
      "  6959075 (100.00%) were unpaired; of these:\n",
      "    96324 (1.38%) aligned 0 times\n",
      "    1009850 (14.51%) aligned exactly 1 time\n",
      "    5852901 (84.10%) aligned >1 times\n",
      "98.62% overall alignment rate\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load SAMtools\n",
    "path_fastq_files=\"/projects_eg/projects/simone/glu_paper/Bowtie/RNAseq_rRNA_mapping/\"\n",
    "label_extension=\"_filtered.fastq\"\n",
    "\n",
    "sam_output_directory=\"/projects_eg/projects/simone/glu_paper/sam_files/\"\n",
    "sam_ext=\".sam\"\n",
    "\n",
    "\n",
    "bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/\"\n",
    "bam_ext=\".bam\"\n",
    "\n",
    "\n",
    "EY_RNA_bam_output_directory=\"/projects_eg/projects/simone/glu_paper/bam_files/EY_RNA/\"\n",
    "mkdir $EY_RNA_bam_output_directory\n",
    "sorted_bam_ext=\"_sorted.bam\"\n",
    "\n",
    "\n",
    "list='SRR1223696 EY-RNA_logphase_polyA_Replicate2 SRR1223706 EY-RNA_-Glu15_polyA_Replicate2'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "bowtie2 --no-unal -x /projects_eg/projects/simone/glu_paper/Bowtie/genome_index_file/s_cerevisiae_reference_genome -U $path_fastq_files$i$label_extension -S $sam_output_directory$i$sam_ext\n",
    "samtools view -S -b $sam_output_directory$i$sam_ext > $EY_RNA_bam_output_directory$i$bam_ext\n",
    "samtools sort $EY_RNA_bam_output_directory$i$bam_ext -o $EY_RNA_bam_output_directory$i$sorted_bam_ext -O BAM\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Creating *.bam.bai* files from *sorted.bam* files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-02T10:27:34.589517Z",
     "start_time": "2018-03-02T10:25:33.054064Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load SAMtools\n",
    "\n",
    "\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/bam_files/BY_Ribo/\n",
    "sorted=\"_sorted.bam\"\n",
    "sorted_bam_bai=\"_sorted.bam.bai\"\n",
    "list='SRR1223686 SRR1288391 SRR1223697 SRR1288400'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "samtools index -b $i$sorted  $i$sorted_bam_bai\n",
    "done\n",
    "\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/bam_files/BY_RNA/\n",
    "sorted=\"_sorted.bam\"\n",
    "bam_bai=\".bam.bai\"\n",
    "list='SRR1223691 SRR1288397 SRR1223701 SRR1288405'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "samtools index -b $i$sorted  $i$sorted_bam_bai\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/\n",
    "sorted=\"_sorted.bam\"\n",
    "bam_bai=\".bam.bai\"\n",
    "list='EY-Ribo_logphase_Replicate1 EY-Ribo_logphase_Replicate2 EY-Ribo_-Glu15_Replicate1 EY-Ribo_-Glu15_Replicate2'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "samtools index -b $i$sorted  $i$sorted_bam_bai\n",
    "done\n",
    "\n",
    "\n",
    "\n",
    "cd /projects_eg/projects/simone/glu_paper/bam_files/EY_RNA/\n",
    "sorted=\"_sorted.bam\"\n",
    "bam_bai=\".bam.bai\"\n",
    "list='SRR1223696 EY-RNA_logphase_polyA_Replicate2 SRR1223706 EY-RNA_-Glu15_polyA_Replicate2'\n",
    "\n",
    "for i in $list\n",
    "do \n",
    "samtools index -b $i$sorted  $i$sorted_bam_bai\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**In IGV, load file_sorted.bam and genome annotation file (.gff)**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# DGE analysis (EY0690)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Riboseq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T10:19:03.194558Z",
     "start_time": "2018-03-05T10:19:03.189718Z"
    }
   },
   "source": [
    "### Loading required packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T15:33:40.641097Z",
     "start_time": "2018-03-05T15:33:39.620225Z"
    }
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T14:19:31.771739Z",
     "start_time": "2018-03-05T14:19:00.753874Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is terminated.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#conda install conda-build\n",
    "conda skeleton cran ggrepel\n",
    "conda build r-ggrepel\n",
    "conda install -c local r-ggrepel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T13:58:44.013919Z",
     "start_time": "2018-03-05T13:55:24.327494Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "source(\"https://bioconductor.org/biocLite.R\")\n",
    "biocLite(\"rtracklayer\")\n",
    "library(rtracklayer)\n",
    "biocLite(\"GenomicAlignments\")\n",
    "library(GenomicAlignments)\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "biocLite(\"calibrate\")\n",
    "library(calibrate) \n",
    "biocLite(\"dplyr\")\n",
    "library(dplyr) \n",
    "library(ggplot2)\n",
    "biocLite(\"ggrepel\")\n",
    "library(ggrepel) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T15:33:43.534765Z",
     "start_time": "2018-03-05T15:33:43.311814Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error in file(filename, \"r\", encoding = encoding) : \n",
      "  cannot open the connection to 'https://bioconductor.org/biocLite.R'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/simone/anaconda3/lib/python3.6/site-packages/rpy2/rinterface/__init__.py:145: RRuntimeWarning: Error in file(filename, \"r\", encoding = encoding) : \n",
      "  cannot open the connection to 'https://bioconductor.org/biocLite.R'\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/simone/anaconda3/lib/python3.6/site-packages/rpy2/rinterface/__init__.py:145: RRuntimeWarning: In addition: \n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/simone/anaconda3/lib/python3.6/site-packages/rpy2/rinterface/__init__.py:145: RRuntimeWarning: Warning message:\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/simone/anaconda3/lib/python3.6/site-packages/rpy2/rinterface/__init__.py:145: RRuntimeWarning: In file(filename, \"r\", encoding = encoding) :\n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/simone/anaconda3/lib/python3.6/site-packages/rpy2/rinterface/__init__.py:145: RRuntimeWarning: \n",
      " \n",
      "  warnings.warn(x, RRuntimeWarning)\n",
      "/home/simone/anaconda3/lib/python3.6/site-packages/rpy2/rinterface/__init__.py:145: RRuntimeWarning:  URL 'https://bioconductor.org/biocLite.R': status was 'Problem with the SSL CA cert (path? access rights?)'\n",
      "\n",
      "  warnings.warn(x, RRuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "source(\"https://bioconductor.org/biocLite.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load BAM files as list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-05T13:53:49.951293Z",
     "start_time": "2018-03-05T13:53:49.941029Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error in BamFile(\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/EY-Ribo_logphase_Replicate1_sorted.bam\",  : \n",
      "  could not find function \"BamFile\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# https://bioconductor.org/packages/3.7/bioc/manuals/GenomicAlignments/man/GenomicAlignments.pdf  (pag. 38)\n",
    "\n",
    "## When 'file' is a BamFile, 'asMates' must be TRUE. If FALSE,\n",
    "## the data are treated as single-end and each list element of the\n",
    "## GAlignmentsList will be of length 1. For single-end data\n",
    "## use readGAlignments() instead of readGAlignmentsList().\n",
    "\n",
    "\n",
    "# load BAM files\n",
    "\n",
    "bam_Ribo_C1 <- BamFile(\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/EY-Ribo_logphase_Replicate1_sorted.bam\", asMates = FALSE)\n",
    "bam_Ribo_C2 <- BamFile(\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/EY-Ribo_logphase_Replicate2_sorted.bam\", asMates = FALSE)\n",
    "bam_Ribo_T1 <- BamFile(\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/EY-Ribo_-Glu15_Replicate1_sorted.bam\", asMates = FALSE)\n",
    "bam_Ribo_T2 <- BamFile(\"/projects_eg/projects/simone/glu_paper/bam_files/EY_Ribo/EY-Ribo_-Glu15_Replicate2_sorted.bam\", asMates = FALSE)\n",
    "\n",
    "\n",
    "# create GAlignments-class object for each sample\n",
    "\n",
    "GA_Ribo_C1 <- readGAlignments(bam_Ribo_C1, use.names = TRUE)\n",
    "GA_Ribo_C2 <- readGAlignments(bam_Ribo_C2, use.names = TRUE)\n",
    "GA_Ribo_T1 <- readGAlignments(bam_Ribo_T1, use.names = TRUE)\n",
    "GA_Ribo_T2 <- readGAlignments(bam_Ribo_T2, use.names = TRUE)\n",
    "\n",
    "length(GA_Ribo_C1)\n",
    "length(GA_Ribo_C2)\n",
    "length(GA_Ribo_T1)\n",
    "length(GA_Ribo_T2)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "As it is not possible to load R packages here the analysis continues on RStudio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
